diff --git a/kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/utils/KiePMMLModelUtilsTest.java b/kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/utils/KiePMMLModelUtilsTest.java
index a663d0f5b4..3a738daca3 100644
--- a/kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/utils/KiePMMLModelUtilsTest.java
+++ b/kie-pmml-trusty/kie-pmml-commons/src/test/java/org/kie/pmml/commons/utils/KiePMMLModelUtilsTest.java
@@ -39,7 +39,6 @@ public class KiePMMLModelUtilsTest {
         packageNameMap.put("AnUpperCasedMame", "anuppercasedmame");
         packageNameMap.put("a_Mixed -name", "amixedname");
         packageNameMap.put("C:\\w-ind_ow Path", "cwindowpath");
-        packageNameMap.put("a.Dotted.pA th", "a.dotted.path");
 
         classNameMap = new HashMap<>();
         classNameMap.put("a-dashed-name", "Adashedname");
