diff --git a/src/test/java/com/alibaba/mtc/MtContextCallableTest.java b/src/test/java/com/alibaba/mtc/MtContextCallableTest.java
index dd51d77..005a7a5 100644
--- a/src/test/java/com/alibaba/mtc/MtContextCallableTest.java
+++ b/src/test/java/com/alibaba/mtc/MtContextCallableTest.java
@@ -65,6 +65,7 @@ public class MtContextCallableTest {
         assertMtContext(call.copied,
                 PARENT_UNMODIFIED_IN_CHILD, PARENT_UNMODIFIED_IN_CHILD,
                 PARENT_MODIFIED_IN_CHILD + 1, PARENT_MODIFIED_IN_CHILD,
+                PARENT_AFTER_CREATE_MTC_TASK, PARENT_AFTER_CREATE_MTC_TASK, // same thread, parent is available from task!
                 CHILD + 1, CHILD + 1
         );
 
@@ -72,7 +73,8 @@ public class MtContextCallableTest {
         assertMtContext(copied(mtContexts),
                 PARENT_UNMODIFIED_IN_CHILD, PARENT_UNMODIFIED_IN_CHILD,
                 PARENT_MODIFIED_IN_CHILD, PARENT_MODIFIED_IN_CHILD, // restored after call!
-                PARENT_AFTER_CREATE_MTC_TASK, PARENT_AFTER_CREATE_MTC_TASK
+                PARENT_AFTER_CREATE_MTC_TASK, PARENT_AFTER_CREATE_MTC_TASK,
+                CHILD + 1, CHILD + 1 // same thread, task set is available from parent
         );
     }
 
diff --git a/src/test/java/com/alibaba/mtc/MtContextRunnableTest.java b/src/test/java/com/alibaba/mtc/MtContextRunnableTest.java
index 371aeea..9fc8190 100644
--- a/src/test/java/com/alibaba/mtc/MtContextRunnableTest.java
+++ b/src/test/java/com/alibaba/mtc/MtContextRunnableTest.java
@@ -68,6 +68,7 @@ public class MtContextRunnableTest {
         assertMtContext(task.copied,
                 PARENT_UNMODIFIED_IN_CHILD, PARENT_UNMODIFIED_IN_CHILD,
                 PARENT_MODIFIED_IN_CHILD + "1", PARENT_MODIFIED_IN_CHILD,
+                PARENT_AFTER_CREATE_MTC_TASK, PARENT_AFTER_CREATE_MTC_TASK, // same thread, parent is available from task
                 CHILD + "1", CHILD + "1"
         );
 
@@ -75,7 +76,8 @@ public class MtContextRunnableTest {
         assertMtContext(copied(mtContexts),
                 PARENT_UNMODIFIED_IN_CHILD, PARENT_UNMODIFIED_IN_CHILD,
                 PARENT_MODIFIED_IN_CHILD, PARENT_MODIFIED_IN_CHILD, // restored after call!
-                PARENT_AFTER_CREATE_MTC_TASK, PARENT_AFTER_CREATE_MTC_TASK
+                PARENT_AFTER_CREATE_MTC_TASK, PARENT_AFTER_CREATE_MTC_TASK,
+                CHILD + "1", CHILD + "1" // same thread, task set is available from parent
         );
     }
 
