diff --git a/src/main/java/com/jcabi/log/PreFormatter.java b/src/main/java/com/jcabi/log/PreFormatter.java
index e40f12d..0e39feb 100644
--- a/src/main/java/com/jcabi/log/PreFormatter.java
+++ b/src/main/java/com/jcabi/log/PreFormatter.java
@@ -110,25 +110,18 @@ final class PreFormatter {
         final Matcher matcher = PATTERN.matcher(fmt);
         int pos = 0;
         while (matcher.find()) {
-            final String group = matcher.group();
-            if (NO_ARG_SPECIFIERS.contains(group)) {
+            final String decor = matcher.group(2);
+            if (decor == null) {
                 matcher.appendReplacement(
                     buf,
-                    Matcher.quoteReplacement(group)
+                    Matcher.quoteReplacement(matcher.group())
                 );
+                this.arguments.add(args[pos]);
             } else {
-                final String decor = matcher.group(2);
-                if (decor == null) {
-                    matcher.appendReplacement(
-                        buf,
-                        Matcher.quoteReplacement(group)
-                    );
-                    this.arguments.add(args[pos]);
-                } else {
                     matcher.appendReplacement(
                         buf,
                         Matcher.quoteReplacement(
-                            group.replace(matcher.group(1), "")
+                            matcher.group().replace(matcher.group(1), "")
                         )
                     );
                     try {
@@ -143,7 +136,6 @@ final class PreFormatter {
                 }
                 pos += 1;
             }
-        }
         if (pos < args.length) {
             throw new IllegalArgumentException(
                 String.format(
