diff --git a/src/test/java/com/jcabi/log/VerboseProcessTest.java b/src/test/java/com/jcabi/log/VerboseProcessTest.java
index a5a09b2..f9eb41b 100644
--- a/src/test/java/com/jcabi/log/VerboseProcessTest.java
+++ b/src/test/java/com/jcabi/log/VerboseProcessTest.java
@@ -178,24 +178,6 @@ public final class VerboseProcessTest {
         new VerboseProcess(builder);
     }
 
-    /**
-     * VerboseProcess can reject ALL stdout level.
-     * @throws Exception If something goes wrong
-     */
-    @Test(expected = IllegalArgumentException.class)
-    public void rejectsStdoutWithLevelAll() throws Exception {
-        new VerboseProcess(Mockito.mock(Process.class), Level.ALL, Level.INFO);
-    }
-
-    /**
-     * VerboseProcess can reject ALL stderr level.
-     * @throws Exception If something goes wrong
-     */
-    @Test(expected = IllegalArgumentException.class)
-    public void rejectsStderrWithLevelAll() throws Exception {
-        new VerboseProcess(Mockito.mock(Process.class), Level.INFO, Level.ALL);
-    }
-
     /**
      * VerboseProcess can quietly terminate a long-running process.
      * @throws Exception If something goes wrong
@@ -249,7 +231,7 @@ public final class VerboseProcessTest {
             );
         }
         final VerboseProcess process = new VerboseProcess(
-            builder, Level.OFF, Level.WARNING
+            builder, Level.OFF, Level.ALL
         );
         process.stdoutQuietly();
         MatcherAssert.assertThat(
@@ -279,8 +261,8 @@ public final class VerboseProcessTest {
             .when(prc).getErrorStream();
         final VerboseProcess verboseProcess = new VerboseProcess(
             prc,
-            Level.FINEST,
-            Level.FINEST
+            Level.ALL,
+            Level.ALL
         );
         Logger.debug(
             this,
