diff --git a/src/test/java/com/jcabi/log/VerboseProcessTest.java b/src/test/java/com/jcabi/log/VerboseProcessTest.java
index ed47600..15942d3 100644
--- a/src/test/java/com/jcabi/log/VerboseProcessTest.java
+++ b/src/test/java/com/jcabi/log/VerboseProcessTest.java
@@ -192,12 +192,7 @@ public final class VerboseProcessTest {
         } catch (final IllegalArgumentException ex) {
             MatcherAssert.assertThat(
                 ex.getMessage(),
-                Matchers.equalTo(
-                    new StringBuilder("stdout LEVEL can't be set to ALL ")
-                        .append("because it is intended only for ")
-                        .append("internal configuration")
-                        .toString()
-                )
+                Matchers.equalTo("stdout LEVEL can't be set to ALL")
             );
         }
     }
@@ -216,12 +211,7 @@ public final class VerboseProcessTest {
         } catch (final IllegalArgumentException ex) {
             MatcherAssert.assertThat(
                 ex.getMessage(),
-                Matchers.equalTo(
-                    new StringBuilder("stderr LEVEL can't be set to ALL ")
-                        .append("because it is intended only for ")
-                        .append("internal configuration")
-                        .toString()
-                )
+                Matchers.equalTo("stderr LEVEL can't be set to ALL")
             );
         }
     }
