diff --git a/src/main/java/com/jcabi/log/PreFormatter.java b/src/main/java/com/jcabi/log/PreFormatter.java
index 71da09a..d3f475b 100644
--- a/src/main/java/com/jcabi/log/PreFormatter.java
+++ b/src/main/java/com/jcabi/log/PreFormatter.java
@@ -55,7 +55,7 @@ final class PreFormatter {
      */
     private static final Pattern PATTERN = Pattern.compile(
         // @checkstyle LineLength (1 line)
-        "%(\\d+\\$)?(\\[([A-Za-z\\-\\.0-9]+)\\])?[\\+\\-]?(?:\\d*(?:\\.\\d+)?)?[a-zA-Z%]"
+        "%(?:\\d+\\$)?(\\[([A-Za-z\\-\\.0-9]+)\\])?[\\+\\-]?(?:\\d*(?:\\.\\d+)?)?[a-zA-Z%]"
     );
 
     /**
@@ -106,24 +106,20 @@ final class PreFormatter {
      * @checkstyle ExecutableStatementCountCheck (100 lines)
      */
     @SuppressWarnings("PMD.ConfusingTernary")
-    private void process(final CharSequence fmt, final Object... args) {
+    private void process(final String fmt, final Object... args) {
         this.arguments = new CopyOnWriteArrayList<Object>();
-        final StringBuffer buf = new StringBuffer(fmt.length());
-        final Matcher matcher = PreFormatter.PATTERN.matcher(fmt);
+        final StringBuffer buf = new StringBuffer();
+        final Matcher matcher = PATTERN.matcher(fmt);
         int pos = 0;
         while (matcher.find()) {
             final String group = matcher.group();
-            if (PreFormatter.NO_ARG_SPECIFIERS.contains(group)) {
+            if (NO_ARG_SPECIFIERS.contains(group)) {
                 matcher.appendReplacement(
                     buf, Matcher.quoteReplacement(group)
                 );
             } else {
-                final String decor = matcher.group(3);
-                if (matcher.group(1) != null) {
-                    matcher.appendReplacement(
-                        buf, Matcher.quoteReplacement(group)
-                    );
-                } else if (decor == null) {
+                final String decor = matcher.group(2);
+                if (decor == null) {
                     matcher.appendReplacement(
                         buf, Matcher.quoteReplacement(group)
                     );
@@ -132,7 +128,7 @@ final class PreFormatter {
                     matcher.appendReplacement(
                         buf,
                         Matcher.quoteReplacement(
-                            group.replace(matcher.group(2), "")
+                            group.replace(matcher.group(1), "")
                         )
                     );
                     try {
