diff --git a/src/test/java/com/jcabi/log/LoggerTest.java b/src/test/java/com/jcabi/log/LoggerTest.java
index c5afe30..f903611 100644
--- a/src/test/java/com/jcabi/log/LoggerTest.java
+++ b/src/test/java/com/jcabi/log/LoggerTest.java
@@ -160,13 +160,10 @@ public final class LoggerTest {
      */
     @Test
     public void findsArgsByPositions() {
-        final String first = "xyz";
-        final String second = "ddd";
+        final String val = "xyz";
         MatcherAssert.assertThat(
-            Logger.format("first: %s, first again: %1$s %s", first, second),
-            Matchers.endsWith(
-                String.format(": %s, first again: %1$s %s", first, second)
-            )
+            Logger.format("first: %s, first again: %1$s", val),
+            Matchers.endsWith(String.format(": %s, first again: %1$s", val))
         );
     }
 
diff --git a/src/test/java/com/jcabi/log/PreFormatterTest.java b/src/test/java/com/jcabi/log/PreFormatterTest.java
index 46d9de0..b5706a8 100644
--- a/src/test/java/com/jcabi/log/PreFormatterTest.java
+++ b/src/test/java/com/jcabi/log/PreFormatterTest.java
@@ -49,7 +49,8 @@ public final class PreFormatterTest {
     public void decoratesArguments() {
         final PreFormatter pre = new PreFormatter(
             "%[com.jcabi.log.DecorMocker]-5.2f and %1$+.6f",
-            1.0d
+            1d,
+            "some text"
         );
         MatcherAssert.assertThat(
             pre.getFormat(),
