diff --git a/src/main/java/org/apache/commons/digester3/xmlrules/ObjectParamRule.java b/src/main/java/org/apache/commons/digester3/xmlrules/ObjectParamRule.java
deleted file mode 100644
index b2f17e82..00000000
--- a/src/main/java/org/apache/commons/digester3/xmlrules/ObjectParamRule.java
+++ /dev/null
@@ -1,81 +0,0 @@
-package org.apache.commons.digester3.xmlrules;
-
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-import static java.lang.Integer.parseInt;
-import static org.apache.commons.beanutils.ConvertUtils.convert;
-
-import org.apache.commons.digester3.binder.LinkedRuleBuilder;
-import org.apache.commons.digester3.binder.ObjectParamBuilder;
-import org.apache.commons.digester3.binder.RulesBinder;
-import org.xml.sax.Attributes;
-
-/**
- * @since 3.2
- */
-final class ObjectParamRule
-    extends AbstractXmlRule
-{
-
-    /**
-     * @param targetRulesBinder
-     * @param patternStack
-     */
-    public ObjectParamRule( RulesBinder targetRulesBinder, PatternStack patternStack )
-    {
-        super( targetRulesBinder, patternStack );
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    protected void bindRule( LinkedRuleBuilder linkedRuleBuilder, Attributes attributes )
-        throws Exception
-    {
-        // create callparamrule
-        String paramNumber = attributes.getValue( "paramnumber" );
-        String attributeName = attributes.getValue( "attrname" );
-        String type = attributes.getValue( "type" );
-        String value = attributes.getValue( "value" );
-
-        int paramIndex = parseInt( paramNumber );
-
-        // create object instance
-        Class<?> clazz = getDigester().getClassLoader().loadClass( type );
-        Object param;
-        if ( value != null )
-        {
-            param = convert( value, clazz );
-        }
-        else
-        {
-            param = clazz.newInstance();
-        }
-
-        ObjectParamBuilder<?> builder = linkedRuleBuilder.objectParam( param ).ofIndex( paramIndex );
-
-        if ( attributeName != null )
-        {
-            builder.matchingAttribute( attributeName );
-        }
-    }
-
-}
diff --git a/src/main/java/org/apache/commons/digester3/xmlrules/XmlRulesModule.java b/src/main/java/org/apache/commons/digester3/xmlrules/XmlRulesModule.java
index bdcfbc05..8d572efe 100644
--- a/src/main/java/org/apache/commons/digester3/xmlrules/XmlRulesModule.java
+++ b/src/main/java/org/apache/commons/digester3/xmlrules/XmlRulesModule.java
@@ -89,7 +89,6 @@ final class XmlRulesModule
             forPattern( "*/factory-create-rule" ).addRule( new FactoryCreateRule( targetRulesBinder, patternStack ) );
             forPattern( "*/node-create-rule" ).addRule( new NodeCreateRule( targetRulesBinder, patternStack ) );
             forPattern( "*/object-create-rule" ).addRule( new ObjectCreateRule( targetRulesBinder, patternStack ) );
-            forPattern( "*/object-param-rule" ).addRule( new ObjectParamRule( targetRulesBinder, patternStack ) );
 
             forPattern( "*/set-properties-rule" ).addRule( new SetPropertiesRule( targetRulesBinder, patternStack ) );
             forPattern( "*/set-properties-rule/alias" )
