diff --git a/oak-core/src/test/java/org/apache/jackrabbit/oak/spi/state/MoveDetectorTest.java b/oak-core/src/test/java/org/apache/jackrabbit/oak/spi/state/MoveDetectorTest.java
index 7a8fd251bd..512e03413d 100644
--- a/oak-core/src/test/java/org/apache/jackrabbit/oak/spi/state/MoveDetectorTest.java
+++ b/oak-core/src/test/java/org/apache/jackrabbit/oak/spi/state/MoveDetectorTest.java
@@ -32,6 +32,7 @@ import org.apache.jackrabbit.oak.plugins.memory.EmptyNodeState;
 import org.apache.jackrabbit.oak.spi.commit.DefaultMoveValidator;
 import org.apache.jackrabbit.oak.spi.commit.EditorDiff;
 import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Test;
 
 public class MoveDetectorTest {
@@ -55,6 +56,7 @@ public class MoveDetectorTest {
      * @throws CommitFailedException
      */
     @Test
+    @Ignore("OAK-1297")  // FIXME OAK-1297
     public void simpleMove() throws CommitFailedException {
         NodeState moved1 = move(root.builder(), "/test/x", "/test/y/xx").getNodeState();
         MoveExpectation moveExpectation1 = new MoveExpectation(
