diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/mongomk/MongoNodeStore.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/mongomk/MongoNodeStore.java
index 6fee0d657d..ecd48c61a0 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/mongomk/MongoNodeStore.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/mongomk/MongoNodeStore.java
@@ -1135,15 +1135,12 @@ public final class MongoNodeStore
             Revision r = e.getValue();
             Revision last = lastKnownRevision.get(machineId);
             if (last == null || r.compareRevisionTime(last) > 0) {
-                if (!hasNewRevisions) {
                     // publish our revision once before any foreign revision
 
                     // the latest revisions of the current cluster node
                     // happened before the latest revisions of other cluster nodes
-                    revisionComparator.add(Revision.newRevision(clusterId), headSeen);
-                }
-                hasNewRevisions = true;
                 lastKnownRevision.put(machineId, r);
+                hasNewRevisions = true;
                 revisionComparator.add(r, otherSeen);
             }
         }
@@ -1151,6 +1144,8 @@ public final class MongoNodeStore
             store.invalidateCache();
             // TODO only invalidate affected items
             docChildrenCache.invalidateAll();
+            Revision r = Revision.newRevision(clusterId);
+            revisionComparator.add(r, headSeen);
             // the head revision is after other revisions
             setHeadRevision(Revision.newRevision(clusterId));
         }
