diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/security/authorization/accesscontrol/ACL.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/security/authorization/accesscontrol/ACL.java
index 9b0afaad4a..a868b62890 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/security/authorization/accesscontrol/ACL.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/security/authorization/accesscontrol/ACL.java
@@ -190,7 +190,7 @@ abstract class ACL extends AbstractAccessControlList {
         List<ACE> subList = Lists.newArrayList(Iterables.filter(entries, new Predicate<ACE>() {
             @Override
             public boolean apply(@Nullable ACE ace) {
-                return (ace != null) && ace.getPrincipal().getName().equals(principal.getName());
+                return (ace != null) && ace.getPrincipal().equals(principal);
             }
         }));
 
