diff --git a/src/test/java/com/jcabi/w3c/DefaultCssValidatorITCase.java b/src/test/java/com/jcabi/w3c/DefaultCssValidatorITCase.java
index fe5fb81..3a5a540 100644
--- a/src/test/java/com/jcabi/w3c/DefaultCssValidatorITCase.java
+++ b/src/test/java/com/jcabi/w3c/DefaultCssValidatorITCase.java
@@ -1,5 +1,5 @@
-/*
- * Copyright (c) 2011-2022, jcabi.com
+/**
+ * Copyright (c) 2011-2017, jcabi.com
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -36,6 +36,8 @@ import org.junit.Test;
 
 /**
  * Integration case for {@link DefaultCssValidator}.
+ * @author Yegor Bugayenko (yegor@tpc2.com)
+ * @version $Id$
  * @since 0.8
  */
 public final class DefaultCssValidatorITCase {
diff --git a/src/test/java/com/jcabi/w3c/DefaultCssValidatorTest.java b/src/test/java/com/jcabi/w3c/DefaultCssValidatorTest.java
index 0535e6d..3ac40be 100644
--- a/src/test/java/com/jcabi/w3c/DefaultCssValidatorTest.java
+++ b/src/test/java/com/jcabi/w3c/DefaultCssValidatorTest.java
@@ -1,5 +1,5 @@
-/*
- * Copyright (c) 2011-2022, jcabi.com
+/**
+ * Copyright (c) 2011-2017, jcabi.com
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -44,6 +44,8 @@ import org.junit.Test;
 
 /**
  * Test case for {@link DefaultCssValidator}.
+ * @author Yegor Bugayenko (yegor@tpc2.com)
+ * @version $Id$
  * @since 0.1
  */
 public final class DefaultCssValidatorTest {
@@ -116,7 +118,7 @@ public final class DefaultCssValidatorTest {
         MatcherAssert.assertThat(
             caught,
             Matchers.containsInAnyOrder(
-                responses.toArray(new Integer[0])
+                responses.toArray(new Integer[responses.size()])
             )
         );
     }
@@ -165,12 +167,8 @@ public final class DefaultCssValidatorTest {
      * @return Document with JIGSAW IGNORE.
      */
     private String documentWithIgnore() {
-        return String.join(
-            "",
-            "/",
-            "* hey */\n\n/",
-            "* JIGSAW IGNORE: .. */\n\n* { abc: cde }\n"
-        );
+        // @checkstyle RegexpSingleline (1 line)
+        return "/* hey */\n\n/* JIGSAW IGNORE: .. */\n\n* { abc: cde }\n";
     }
 
     /**
diff --git a/src/test/java/com/jcabi/w3c/DefaultHtmlValidatorITCase.java b/src/test/java/com/jcabi/w3c/DefaultHtmlValidatorITCase.java
index a04d41a..e91a064 100644
--- a/src/test/java/com/jcabi/w3c/DefaultHtmlValidatorITCase.java
+++ b/src/test/java/com/jcabi/w3c/DefaultHtmlValidatorITCase.java
@@ -1,5 +1,5 @@
-/*
- * Copyright (c) 2011-2022, jcabi.com
+/**
+ * Copyright (c) 2011-2017, jcabi.com
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -38,6 +38,8 @@ import org.junit.Test;
 /**
  * Integration case for {@link DefaultHtmlValidator}.
  *
+ * @author Yegor Bugayenko (yegor@tpc2.com)
+ * @version $Id$
  * @since 0.8
  */
 public final class DefaultHtmlValidatorITCase {
diff --git a/src/test/java/com/jcabi/w3c/DefaultHtmlValidatorTest.java b/src/test/java/com/jcabi/w3c/DefaultHtmlValidatorTest.java
index 00b4e01..e4f95be 100644
--- a/src/test/java/com/jcabi/w3c/DefaultHtmlValidatorTest.java
+++ b/src/test/java/com/jcabi/w3c/DefaultHtmlValidatorTest.java
@@ -1,5 +1,5 @@
-/*
- * Copyright (c) 2011-2022, jcabi.com
+/**
+ * Copyright (c) 2011-2017, jcabi.com
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -50,6 +50,8 @@ import org.junit.Test;
 /**
  * Test case for {@link DefaultHtmlValidator}.
  *
+ * @author Yegor Bugayenko (yegor@tpc2.com)
+ * @version $Id$
  * @since 0.1
  */
 public final class DefaultHtmlValidatorTest {
@@ -143,7 +145,7 @@ public final class DefaultHtmlValidatorTest {
                 container.stop();
             }
         }
-        final Integer[] data = responses.toArray(new Integer[0]);
+        final Integer[] data = responses.toArray(new Integer[responses.size()]);
         MatcherAssert.assertThat(caught, Matchers.containsInAnyOrder(data));
     }
 
diff --git a/src/test/java/com/jcabi/w3c/DefectTest.java b/src/test/java/com/jcabi/w3c/DefectTest.java
index 73dfd3a..3121323 100644
--- a/src/test/java/com/jcabi/w3c/DefectTest.java
+++ b/src/test/java/com/jcabi/w3c/DefectTest.java
@@ -1,5 +1,5 @@
-/*
- * Copyright (c) 2011-2022, jcabi.com
+/**
+ * Copyright (c) 2011-2017, jcabi.com
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -35,6 +35,8 @@ import org.junit.Test;
 
 /**
  * Test case for {@link Defect}.
+ * @author Yegor Bugayenko (yegor@tpc2.com)
+ * @version $Id$
  * @since 0.1
  */
 public final class DefectTest {
diff --git a/src/test/java/com/jcabi/w3c/package-info.java b/src/test/java/com/jcabi/w3c/package-info.java
index e00ac40..75a72a2 100644
--- a/src/test/java/com/jcabi/w3c/package-info.java
+++ b/src/test/java/com/jcabi/w3c/package-info.java
@@ -1,5 +1,5 @@
-/*
- * Copyright (c) 2011-2022, jcabi.com
+/**
+ * Copyright (c) 2011-2017, jcabi.com
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -31,6 +31,8 @@
 /**
  * W3C validators.
  *
+ * @author Yegor Bugayenko (yegor@tpc2.com)
+ * @version $Id$
  * @since 1.0
  */
 package com.jcabi.w3c;
