diff --git a/datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.java b/datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.java
index 98276d8..5881fef 100644
--- a/datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.java
+++ b/datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.java
@@ -187,13 +187,15 @@ public class InstantDeserializer<T extends Temporal>
     {
         InstantDeserializer<T> deserializer =
                 (InstantDeserializer<T>)super.createContextual(ctxt, property);
+        if (deserializer != this) {
         JsonFormat.Value val = findFormatOverrides(ctxt, property, handledType());
         if (val != null) {
             deserializer = new InstantDeserializer<>(deserializer, val.getFeature(JsonFormat.Feature.ADJUST_DATES_TO_CONTEXT_TIME_ZONE));
             if (val.hasLenient()) {
                 Boolean leniency = val.getLenient();
                 if (leniency != null) {
                     deserializer = deserializer.withLeniency(leniency);
+                    }
                 }
             }
         }
