diff --git a/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java b/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
index 50b73bf..59a631e 100644
--- a/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
+++ b/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
@@ -541,7 +541,8 @@ public class GenericObjectPool<T> extends BaseGenericObjectPool<T>
                 // Make sure object is not being reclaimed
                 synchronized(p) {
                     final PooledObjectState state = p.getState();
-                    if (state != PooledObjectState.ALLOCATED) {
+                    if (state == PooledObjectState.ABANDONED ||
+                            state == PooledObjectState.INVALID) {
                         return;
                     } else {
                         p.markReturning(); // Keep from being marked abandoned
