diff --git a/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java b/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
index 5e1ed6b8..62f04955 100644
--- a/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
+++ b/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
@@ -2107,32 +2107,6 @@ public class TestGenericObjectPool extends TestBaseObjectPool {
         assertEquals(2, swallowedExceptions.size());
     }
 
-    @Test
-    public void testMultipleReturnOfSameObject() throws Exception {
-        final GenericObjectPool<String> pool = new GenericObjectPool<String>(
-                factory, new GenericObjectPoolConfig(),
-                new AbandonedConfig());

-        Assert.assertEquals(0, pool.getNumActive());
-        Assert.assertEquals(0, pool.getNumIdle());

-        String obj = pool.borrowObject();

-        Assert.assertEquals(1, pool.getNumActive());
-        Assert.assertEquals(0, pool.getNumIdle());

-        pool.returnObject(obj);

-        Assert.assertEquals(0, pool.getNumActive());
-        Assert.assertEquals(1, pool.getNumIdle());

-        pool.returnObject(obj);

-        Assert.assertEquals(0, pool.getNumActive());
-        Assert.assertEquals(1, pool.getNumIdle());
-    }

     private static final class DummyFactory
             extends BasePooledObjectFactory<Object> {
         @Override
