diff --git a/src/test/java/org/apache/commons/pool2/impl/TestGenericKeyedObjectPool.java b/src/test/java/org/apache/commons/pool2/impl/TestGenericKeyedObjectPool.java
index 8d7dcecf..e4e4b4af 100644
--- a/src/test/java/org/apache/commons/pool2/impl/TestGenericKeyedObjectPool.java
+++ b/src/test/java/org/apache/commons/pool2/impl/TestGenericKeyedObjectPool.java
@@ -413,28 +413,6 @@ public class TestGenericKeyedObjectPool extends TestKeyedObjectPool {
             assertFalse(pool.getBlockWhenExhausted());
         }
     }

-    @Test
-    public void testContructorEvictionConfig() throws Exception {
-        GenericKeyedObjectPoolConfig config = new GenericKeyedObjectPoolConfig();
-        config.setTimeBetweenEvictionRunsMillis(500);
-        config.setMinEvictableIdleTimeMillis(50);
-        config.setNumTestsPerEvictionRun(5);
-        GenericKeyedObjectPool p = new GenericKeyedObjectPool<String, String>(factory, config);
-        for(int i=0;i<5;i++) {
-            p.addObject("one");
-        }
-        try { Thread.sleep(100); } catch(InterruptedException e) { }
-        assertEquals(5, p.getNumIdle("one"));
-        try { Thread.sleep(500); } catch(InterruptedException e) { }
-        assertEquals(0, p.getNumIdle("one"));
-        p.close();
-    }
    
 
     @Test(timeout=60000)
     public void testEviction() throws Exception {
