diff --git a/src/test/java/org/apache/commons/pool2/impl/TestAbandonedObjectPool.java b/src/test/java/org/apache/commons/pool2/impl/TestAbandonedObjectPool.java
index 8415d8ec..1c0ea1ef 100644
--- a/src/test/java/org/apache/commons/pool2/impl/TestAbandonedObjectPool.java
+++ b/src/test/java/org/apache/commons/pool2/impl/TestAbandonedObjectPool.java
@@ -17,9 +17,6 @@
 
 package org.apache.commons.pool2.impl;
 
-import java.io.BufferedOutputStream;
-import java.io.ByteArrayOutputStream;
-import java.io.PrintWriter;
 import java.lang.management.ManagementFactory;
 import java.util.ArrayList;
 import java.util.Set;
@@ -264,25 +261,6 @@ public class TestAbandonedObjectPool {
 
         Assert.assertTrue (end - start < 5000);
     }
-    @Test
-    public void testStackTrace() throws Exception {
-        abandonedConfig.setRemoveAbandonedOnMaintenance(true);
-        abandonedConfig.setLogAbandoned(true);
-        abandonedConfig.setRemoveAbandonedTimeout(1);
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        PrintWriter pw = new PrintWriter(new BufferedOutputStream(baos));
-        abandonedConfig.setLogWriter(pw);
-        pool.setAbandonedConfig(abandonedConfig);
-        pool.setTimeBetweenEvictionRunsMillis(100);
-        PooledTestObject o1 = pool.borrowObject();
-        Thread.sleep(2000);
-        Assert.assertTrue(o1.isDestroyed());
-        Assert.assertTrue(baos.toString().indexOf("Pooled object") >= 0);
-    }
 
     class ConcurrentBorrower extends Thread {
         private final ArrayList<PooledTestObject> _borrowed;
