diff --git a/pool_21_fix/src/main/java/org/apache/commons/pool2/impl/GenericKeyedObjectPool.java b/pool_21_buggy/src/main/java/org/apache/commons/pool2/impl/GenericKeyedObjectPool.java
index 697ae05..eb335d6 100644
--- a/pool_21_fix/src/main/java/org/apache/commons/pool2/impl/GenericKeyedObjectPool.java
+++ b/pool_21_buggy/src/main/java/org/apache/commons/pool2/impl/GenericKeyedObjectPool.java
@@ -1014,9 +1012,6 @@ public class GenericKeyedObjectPool<K,T> extends BaseGenericObjectPool<T>
             if (maxTotal > -1 && newNumTotal > maxTotal) {
                 numTotal.decrementAndGet();
                 if (getNumIdle() == 0) {
-                    objectDeque.idleObjects.interuptTakeWaiters();
                     return null;
                 }
                 clearOldest();
@@ -1025,6 +1020,7 @@ public class GenericKeyedObjectPool<K,T> extends BaseGenericObjectPool<T>
             }
         }
 
+        final ObjectDeque<T> objectDeque = poolMap.get(key);
         final long newCreateCount = objectDeque.getCreateCount().incrementAndGet();
 
         // Check against the per key limit
@@ -1032,9 +1028,6 @@ public class GenericKeyedObjectPool<K,T> extends BaseGenericObjectPool<T>
                 newCreateCount > Integer.MAX_VALUE) {
             numTotal.decrementAndGet();
             objectDeque.getCreateCount().decrementAndGet();
-            objectDeque.idleObjects.interuptTakeWaiters();
             return null;
         }
 
@@ -1045,9 +1038,6 @@ public class GenericKeyedObjectPool<K,T> extends BaseGenericObjectPool<T>
         } catch (final Exception e) {
             numTotal.decrementAndGet();
             objectDeque.getCreateCount().decrementAndGet();
-            objectDeque.idleObjects.interuptTakeWaiters();
             throw e;
         }
 
diff --git a/pool_21_fix/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java b/pool_21_buggy/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
index b5ef9bd..6c8dc71 100644
--- a/pool_21_fix/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
+++ b/pool_21_buggy/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
@@ -856,9 +856,6 @@ public class GenericObjectPool<T> extends BaseGenericObjectPool<T>
         if (localMaxTotal > -1 && newCreateCount > localMaxTotal ||
                 newCreateCount > Integer.MAX_VALUE) {
             createCount.decrementAndGet();
-            idleObjects.interuptTakeWaiters();
             return null;
         }
 
@@ -867,9 +864,6 @@ public class GenericObjectPool<T> extends BaseGenericObjectPool<T>
             p = factory.makeObject();
         } catch (final Exception e) {
             createCount.decrementAndGet();
-            idleObjects.interuptTakeWaiters();
             throw e;
         }
 
