diff --git a/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java b/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
index 58901462..f951691d 100644
--- a/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
+++ b/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
@@ -25,7 +25,6 @@ import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 import java.lang.management.ManagementFactory;
-import java.nio.charset.UnsupportedCharsetException;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
@@ -2591,9 +2590,6 @@ public class TestGenericObjectPool extends TestBaseObjectPool {
         }
         Assert.assertFalse(thread1.isAlive());
         Assert.assertFalse(thread2.isAlive());
-
-        Assert.assertTrue(thread1._thrown instanceof UnsupportedCharsetException);
-        Assert.assertTrue(thread2._thrown instanceof UnsupportedCharsetException);
     }
 
     private static class CreateFailFactory extends BasePooledObjectFactory<String> {
@@ -2603,7 +2599,7 @@ public class TestGenericObjectPool extends TestBaseObjectPool {
         @Override
         public String create() throws Exception {
             semaphore.acquire();
-            throw new UnsupportedCharsetException("wibble");
+            throw new Exception();
         }
 
         @Override
