diff --git a/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java b/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
index 0575f7e9..abdf7e18 100644
--- a/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
+++ b/src/main/java/org/apache/commons/pool2/impl/GenericObjectPool.java
@@ -914,15 +914,11 @@ public class GenericObjectPool<T> extends BaseGenericObjectPool<T>
             destroyedCount.incrementAndGet();
             createCount.decrementAndGet();
         }

-        if (idleObjects.isEmpty() && idleObjects.hasTakeWaiters()) {
             // POOL-356.
             // In case there are already threads waiting on something in the pool
             // (e.g. idleObjects.takeFirst(); then we need to provide them a fresh instance.
             // Otherwise they will be stuck forever (or until timeout)
-            PooledObject<T> freshPooled = create();
-            idleObjects.put(freshPooled);
-        }
     }
 
     @Override
