diff --git a/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java b/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
index 0b4509e3..a1539f76 100644
--- a/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
+++ b/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
@@ -2126,27 +2126,6 @@ public class TestGenericObjectPool extends TestBaseObjectPool {
         }
     }
 
-    @Test(timeout = 60000)
-    public void testNoInvalidateNPE() throws Exception {
-        genericObjectPool.setMaxTotal(1);
-        genericObjectPool.setTestOnCreate(true);
-        genericObjectPool.setMaxWaitMillis(-1);
-        String obj = genericObjectPool.borrowObject();
-        simpleFactory.setValid(false);
-        final WaitingTestThread wtt = new WaitingTestThread(genericObjectPool, 200);
-        wtt.start();
-        Thread.sleep(200);
-        genericObjectPool.invalidateObject(obj);
-        simpleFactory.setValid(true);
-    }

     @Test(timeout = 60000)
     public void testMaxTotal() throws Exception {
         genericObjectPool.setMaxTotal(3);
