diff --git a/src/test/org/apache/commons/pool/impl/TestGenericObjectPool.java b/src/test/org/apache/commons/pool/impl/TestGenericObjectPool.java
index dac91aab..4008f1ee 100644
--- a/src/test/org/apache/commons/pool/impl/TestGenericObjectPool.java
+++ b/src/test/org/apache/commons/pool/impl/TestGenericObjectPool.java
@@ -110,46 +110,6 @@ public class TestGenericObjectPool extends TestBaseObjectPool {
         pool.close();
     }
 
-    public void testWhenExhaustedBlockInterupt() throws Exception {
-        pool.setMaxActive(1);
-        pool.setWhenExhaustedAction(GenericObjectPool.WHEN_EXHAUSTED_BLOCK);
-        pool.setMaxWait(0);
-        Object obj1 = pool.borrowObject();

-        assertNotNull(obj1);
-        WaitingTestThread wtt = new WaitingTestThread(pool, 200);
-        wtt.start();
-        Thread.sleep(200);
-        wtt.interrupt();

-        Thread.sleep(200);
-        assertTrue(wtt._thrown instanceof InterruptedException);
-        pool.returnObject(obj1);
-        pool.setMaxWait(10L);
-        Object obj2 = null;
-        try {
-             obj2 = pool.borrowObject();
-            assertNotNull(obj2);
-        } catch(NoSuchElementException e) {
-            fail("NoSuchElementException not expected");
-        }
-        pool.returnObject(obj2);
-        pool.close(); 
-    }
     public void testEvictWhileEmpty() throws Exception {
         pool.evict();
         pool.evict();
