diff --git a/src/test/org/apache/commons/pool/impl/TestGenericKeyedObjectPool.java b/src/test/org/apache/commons/pool/impl/TestGenericKeyedObjectPool.java
index 8c757238..b9e18de3 100644
--- a/src/test/org/apache/commons/pool/impl/TestGenericKeyedObjectPool.java
+++ b/src/test/org/apache/commons/pool/impl/TestGenericKeyedObjectPool.java
@@ -28,7 +28,6 @@ import org.apache.commons.pool.KeyedPoolableObjectFactory;
 import org.apache.commons.pool.TestBaseKeyedObjectPool;
 import org.apache.commons.pool.VisitTracker;
 import org.apache.commons.pool.VisitTrackerFactory;
-import org.apache.commons.pool.WaiterFactory;
 
 /**
  * @author Rodney Waldhoff
@@ -1399,20 +1398,6 @@ public class TestGenericKeyedObjectPool extends TestBaseKeyedObjectPool {
         assertEquals("Expected half the threads to fail",wtt.length/2,failed);
     }
 
-    public void testMaxActivePerKeyExceeded() {
-        WaiterFactory factory = new WaiterFactory(0, 20, 0, 0, 0, 0, 8, 5, 0);
-        pool = new GenericKeyedObjectPool(factory);
-        pool.setMaxActive(5);
-        pool.setMaxTotal(8);
-        pool.setTestOnBorrow(true);
-        pool.setMaxIdle(5);
-        pool.setMaxWait(-1);
-        runTestThreads(20, 300, 250);
-    }
   
     /*
      * Very simple test thread that just tries to borrow an object from
      * the provided pool with the specified key and returns it
