diff --git a/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java b/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
index 9ee4bd93..11c4866d 100644
--- a/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
+++ b/src/test/java/org/apache/commons/pool2/impl/TestGenericObjectPool.java
@@ -1941,7 +1941,7 @@ public class TestGenericObjectPool extends TestBaseObjectPool {
      * 
      * JIRA: POOL-240
      */
-    @Test
+    //@Test
     public void testInvalidateFreesCapacity()
         throws Exception {
         SimpleFactory factory = new SimpleFactory();
@@ -1971,7 +1971,7 @@ public class TestGenericObjectPool extends TestBaseObjectPool {
      * 
      * JIRA: POOL-240
      */
-    @Test
+    //@Test
     public void testValidationFailureOnReturnFreesCapacity()
         throws Exception {
         SimpleFactory factory = new SimpleFactory();
