diff --git a/compress_53_fix/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java b/compress_53_buggy/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java
index 96bb3ee..cdbc80f 100644
--- a/compress_53_fix/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java
+++ b/compress_53_buggy/src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java
@@ -728,7 +728,8 @@ public class TarArchiveEntry implements TarConstants, ArchiveEntry {
      * @return true if this is a long name extension provided by GNU tar
      */
     public boolean isGNULongLinkEntry() {
-        return linkFlag == LF_GNUTYPE_LONGLINK;
+        return linkFlag == LF_GNUTYPE_LONGLINK
+            && name.equals(GNU_LONGLINK);
     }
 
     /**
@@ -737,7 +738,8 @@ public class TarArchiveEntry implements TarConstants, ArchiveEntry {
      * @return true if this is a long name extension provided by GNU tar
      */
     public boolean isGNULongNameEntry() {
-        return linkFlag == LF_GNUTYPE_LONGNAME;
+        return linkFlag == LF_GNUTYPE_LONGNAME
+            && name.equals(GNU_LONGLINK);
     }
 
     /**
