diff --git a/src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java b/src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java
index b41dbb6..14691c4 100644
--- a/src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java
+++ b/src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java
@@ -175,17 +175,12 @@ public class ExtraFieldUtils {
             }
             try {
                 final ZipExtraField ze = createExtraField(headerId);
-                try {
                 if (local) {
                     ze.parseFromLocalFileData(data, start + WORD, length);
                 } else {
                     ze.parseFromCentralDirectoryData(data, start + WORD,
                                                      length);
                 }
-                } catch (ArrayIndexOutOfBoundsException aiobe) {
-                    throw (ZipException) new ZipException("Failed to parse corrupt ZIP extra field of type "
-                        + Integer.toHexString(headerId.getValue())).initCause(aiobe);
-                }
                 v.add(ze);
             } catch (final InstantiationException | IllegalAccessException ie) {
                 throw (ZipException) new ZipException(ie.getMessage()).initCause(ie);
