diff --git a/src/main/java/org/apache/maven/plugins/shade/DefaultShader.java b/src/main/java/org/apache/maven/plugins/shade/DefaultShader.java
index 7255647..26c65f9 100644
--- a/src/main/java/org/apache/maven/plugins/shade/DefaultShader.java
+++ b/src/main/java/org/apache/maven/plugins/shade/DefaultShader.java
@@ -274,47 +274,25 @@ public class DefaultShader
 
         public Object mapValue( Object object )
         {
-            if ( object instanceof String )
-            {
-                String name = (String) object;
-                String value = name;
-                for ( Iterator i = relocators.iterator(); i.hasNext(); )
-                {
-                    Relocator r = (Relocator) i.next();
 
-                    if ( r.canRelocatePath( name ) )
-                    {
-                        value = r.relocatePath( name );
-                        break;
-                    }
 
-                    if ( r.canRelocateClass( name ) )
-                    {
-                        value = r.relocateClass( name );
-                        break;
-                    }
-                }
 
-                return value;
-            }
             return object;
         }
 
         public String map( String name )
         {
-            String value = name;
             for ( Iterator i = relocators.iterator(); i.hasNext(); )
             {
                 Relocator r = (Relocator) i.next();
 
-                if ( r.canRelocatePath( name ) )
+                if ( r.canRelocate( name ) )
                 {
-                    value = r.relocatePath( name );
-                    break;
+                    return r.relocate( name );
                 }
             }
 
-            return value;
+            return name;
         }
     }
 }
diff --git a/src/main/java/org/apache/maven/plugins/shade/relocation/Relocator.java b/src/main/java/org/apache/maven/plugins/shade/relocation/Relocator.java
index f162fcb..d0d7a9c 100644
--- a/src/main/java/org/apache/maven/plugins/shade/relocation/Relocator.java
+++ b/src/main/java/org/apache/maven/plugins/shade/relocation/Relocator.java
@@ -24,11 +24,9 @@ public interface Relocator
 {
     String ROLE = Relocator.class.getName();
 
-    boolean canRelocatePath( String clazz );
+    boolean canRelocate( String clazz );
 
-    String relocatePath( String clazz );
 
-    boolean canRelocateClass( String clazz );
 
-    String relocateClass( String clazz );
+    String relocate( String clazz );
 }
diff --git a/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java b/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
index d4e2b36..4f1322e 100644
--- a/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
+++ b/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
@@ -32,27 +32,22 @@ public class SimpleRelocator
 {
     private String pattern;
 
-    private String pathPattern;
 
     private String shadedPattern;
 
-    private String shadedPathPattern;
 
     private List excludes;
 
     public SimpleRelocator(String patt, String shadedPattern, List excludes)
     {
-        this.pattern = patt;
-        this.pathPattern = patt.replace('.', '/');
+        this.pattern = patt.replace('.', '/');
 
         if ( shadedPattern != null )
         {
-            this.shadedPattern = shadedPattern;
-            this.shadedPathPattern = shadedPattern.replace('.', '/');
+            this.shadedPattern = shadedPattern.replace('.', '/');
         } else
         {
-            this.shadedPattern = "hidden." + this.pattern;
-            this.shadedPathPattern = "hidden/" + this.pathPattern;
+            this.shadedPattern = "hidden/" + this.pattern;
         }
 
         if (excludes != null)
@@ -68,7 +61,7 @@ public class SimpleRelocator
         }
     }
 
-    public boolean canRelocatePath( String clazz )
+    public boolean canRelocate( String clazz )
     {
         if ( excludes != null )
         {
@@ -88,39 +81,15 @@ public class SimpleRelocator
             }
         }
 
-        return clazz.startsWith( pathPattern );
-    }
 
-    public boolean canRelocateClass( String clazz )
-    {
-        if ( excludes != null )
-        {
-            for ( Iterator i = excludes.iterator(); i.hasNext(); )
-            {
-                String exclude = (String) i.next();
 
-                exclude = exclude.replace( '/', '.' );
-                if ( exclude.endsWith( ".*" ) && clazz.startsWith( exclude.substring( 0, exclude.length() - 2 ) ) )
-                {
-                    return false;
-                }
-                else if ( clazz.equals( exclude ) )
-                {
-                    return false;
-                }
-            }
-        }
 
         return clazz.startsWith( pattern );
     }
 
-    public String relocatePath( String clazz )
-    {
-        return clazz.replaceFirst(pathPattern, shadedPathPattern);
-    }
 
-    public String relocateClass( String clazz )
+    public String relocate( String clazz )
     {
         return clazz.replaceFirst(pattern, shadedPattern);
     }
 }
