diff --git a/src/test/java/org/apache/maven/plugins/shade/DefaultShaderTest.java b/src/test/java/org/apache/maven/plugins/shade/DefaultShaderTest.java
index 386fc9b..535ef79 100644
--- a/src/test/java/org/apache/maven/plugins/shade/DefaultShaderTest.java
+++ b/src/test/java/org/apache/maven/plugins/shade/DefaultShaderTest.java
@@ -25,8 +25,6 @@ import java.util.Arrays;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
-import java.net.URLClassLoader;
-import java.net.URL;
 
 import junit.framework.TestCase;
 
@@ -50,32 +48,6 @@ public class DefaultShaderTest
         shaderWithPattern( null, new File( "target/foo-default.jar" ), EXCLUDES );
     }
 
-    public void testShaderWithStaticInitializedClass()
-        throws Exception
-    {
-        Shader s = new DefaultShader();
-
-        Set set = new HashSet();
-
-        set.add( new File( "src/test/jars/test-artifact-1.0-SNAPSHOT.jar" ) );
-
-        List relocators = new ArrayList();
-
-        relocators.add( new SimpleRelocator( "org.apache.maven.plugins.shade", null, null ) );
-
-        List resourceTransformers = new ArrayList();
-
-        List filters = new ArrayList();
-
-        File file = new File( "target/testShaderWithStaticInitializedClass.jar" );
-        s.shade( set, file, filters, relocators, resourceTransformers );
-
-        URLClassLoader cl = new URLClassLoader( new URL[]{file.toURL()} );
-        Class c = cl.loadClass( "hidden.org.apache.maven.plugins.shade.Lib" );
-        Object o = c.newInstance();
-        assertEquals( "foo.bar/baz", c.getDeclaredField( "CONSTANT" ).get( o ) );
-    }
-
     public void testShaderWithCustomShadedPattern()
         throws Exception
     {
