diff --git a/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java b/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
index 61a4ccd..729bb18 100644
--- a/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
+++ b/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
@@ -19,10 +19,9 @@ package org.apache.maven.plugins.shade.relocation;
  * under the License.
  */
 
+import java.util.ArrayList;
 import java.util.Iterator;
-import java.util.LinkedHashSet;
 import java.util.List;
-import java.util.Set;
 
 import org.codehaus.plexus.util.SelectorUtils;
 
@@ -41,7 +40,7 @@ public class SimpleRelocator
 
     private String shadedPathPattern;
 
-    private Set excludes;
+    private List excludes;
 
     public SimpleRelocator(String patt, String shadedPattern, List excludes)
     {
@@ -60,20 +59,15 @@ public class SimpleRelocator
 
         if ( excludes != null && !excludes.isEmpty() )
         {
-            this.excludes = new LinkedHashSet();
+            this.excludes = new ArrayList();
 
             for (Iterator i = excludes.iterator(); i.hasNext();)
             {
                 String e = (String) i.next();
 
                 String classExclude = e.replace( '.', '/' );
-                this.excludes.add( classExclude );
 
-                if ( classExclude.endsWith( "/*" ) )
-                {
-                    String packageExclude = classExclude.substring( 0, classExclude.lastIndexOf( '/' ) );
-                    this.excludes.add( packageExclude );
-                }
+                this.excludes.add(classExclude);
             }
         }
     }
