diff --git a/src/test/java/org/apache/maven/plugins/shade/filter/SimpleFilterTest.java b/src/test/java/org/apache/maven/plugins/shade/filter/SimpleFilterTest.java
deleted file mode 100644
index ccf05fe..0000000
--- a/src/test/java/org/apache/maven/plugins/shade/filter/SimpleFilterTest.java
+++ /dev/null
@@ -1,77 +0,0 @@
-package org.apache.maven.plugins.shade.filter;

-import java.util.Collections;
-import junit.framework.TestCase;

-public class SimpleFilterTest
-    extends TestCase
-{

-    public void testIsFiltered()
-    {
-        SimpleFilter filter;

-        filter = new SimpleFilter( null, null, null );
-        assertFalse( filter.isFiltered( "a.properties" ) );
-        assertFalse( filter.isFiltered( "org/Test.class" ) );

-        filter = new SimpleFilter( null, Collections.EMPTY_SET, Collections.EMPTY_SET );
-        assertFalse( filter.isFiltered( "a.properties" ) );
-        assertFalse( filter.isFiltered( "org/Test.class" ) );

-        filter = new SimpleFilter( null, Collections.singleton( "org/Test.class" ), Collections.EMPTY_SET );
-        assertTrue( filter.isFiltered( "a.properties" ) );
-        assertFalse( filter.isFiltered( "org/Test.class" ) );
-        assertTrue( filter.isFiltered( "org/Test.properties" ) );

-        filter = new SimpleFilter( null, Collections.EMPTY_SET, Collections.singleton( "org/Test.class" ) );
-        assertFalse( filter.isFiltered( "a.properties" ) );
-        assertTrue( filter.isFiltered( "org/Test.class" ) );
-        assertFalse( filter.isFiltered( "org/Test.properties" ) );

-        filter = new SimpleFilter( null, Collections.singleton( "**/a.properties" ), Collections.EMPTY_SET );
-        assertFalse( filter.isFiltered( "a.properties" ) );
-        assertFalse( filter.isFiltered( "org/a.properties" ) );
-        assertFalse( filter.isFiltered( "org/maven/a.properties" ) );
-        assertTrue( filter.isFiltered( "org/maven/a.class" ) );

-        filter = new SimpleFilter( null, Collections.EMPTY_SET, Collections.singleton( "org/*" ) );
-        assertFalse( filter.isFiltered( "Test.class" ) );
-        assertTrue( filter.isFiltered( "org/Test.class" ) );
-        assertFalse( filter.isFiltered( "org/apache/Test.class" ) );

-        filter = new SimpleFilter( null, Collections.EMPTY_SET, Collections.singleton( "org/**" ) );
-        assertFalse( filter.isFiltered( "Test.class" ) );
-        assertTrue( filter.isFiltered( "org/Test.class" ) );
-        assertTrue( filter.isFiltered( "org/apache/Test.class" ) );

-        filter = new SimpleFilter( null, Collections.EMPTY_SET, Collections.singleton( "org/" ) );
-        assertFalse( filter.isFiltered( "Test.class" ) );
-        assertTrue( filter.isFiltered( "org/Test.class" ) );
-        assertTrue( filter.isFiltered( "org/apache/Test.class" ) );
-    }
-}
