diff --git a/mshade_7_fix/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java b/mshade_7_buggy/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
index 198eeed..4028bf1 100644
--- a/mshade_7_fix/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
+++ b/mshade_7_buggy/src/main/java/org/apache/maven/plugins/shade/relocation/SimpleRelocator.java
@@ -69,16 +69,8 @@ public class SimpleRelocator
         }
         else
         {
-            if ( patt == null )
-            {
-                this.pattern = "";
-                this.pathPattern = "";
-            }
-            else
-            {
                 this.pattern = patt.replace( '/', '.' );
                 this.pathPattern = patt.replace( '.', '/' );
-            }
 
             if ( shadedPattern != null )
             {
