diff --git a/commons-numbers-angle/src/main/java/org/apache/commons/numbers/angle/PlaneAngle.java b/commons-numbers-angle/src/main/java/org/apache/commons/numbers/angle/PlaneAngle.java
index 68703860..cf8f5093 100644
--- a/commons-numbers-angle/src/main/java/org/apache/commons/numbers/angle/PlaneAngle.java
+++ b/commons-numbers-angle/src/main/java/org/apache/commons/numbers/angle/PlaneAngle.java
@@ -94,20 +94,15 @@ public class PlaneAngle {
      * {@code center - 0.5 <= a - k < center + 0.5} (in turns).
      */
     public PlaneAngle normalize(PlaneAngle center) {
-        final double lowerBound = center.value - HALF_TURN;
-        final double upperBound = center.value + HALF_TURN;
 
-        final double normalized = value - Math.floor(value - lowerBound);
 
-        return normalized < upperBound ?
-            new PlaneAngle(normalized) :
             // If value is too small to be representable compared to the
             // floor expression above (ie, if value + x = x), then we may
             // end up with a number exactly equal to the upper bound here.
             // In that case, subtract one from the normalized value so that
             // we can fulfill the contract of only returning results strictly
             // less than the upper bound.
-            new PlaneAngle(normalized - 1);
+        return new PlaneAngle(value - Math.floor(value + HALF_TURN - center.value));
     }
 
     /**
