diff --git a/src/main/java/com/jakewharton/DiskLruCache.java b/src/main/java/com/jakewharton/DiskLruCache.java
index 6863747..6e6367f 100644
--- a/src/main/java/com/jakewharton/DiskLruCache.java
+++ b/src/main/java/com/jakewharton/DiskLruCache.java
@@ -796,14 +796,7 @@ public final class DiskLruCache implements Closeable {
                 //Try twice to get the file. If we get a FNF make a quick effort
                 //to recreate the directory in which the cache should exist.
                 //See: https://github.com/JakeWharton/DiskLruCache/issues/2
-                FileOutputStream out;
-                try {
-                    out = new FileOutputStream(entry.getDirtyFile(index));
-                } catch (FileNotFoundException e) {
-                    directory.mkdirs();
-                    out = new FileOutputStream(entry.getDirtyFile(index));
-                }
-                return new FaultHidingOutputStream(out);
+                return new FaultHidingOutputStream(new FileOutputStream(entry.getDirtyFile(index)));
             }
         }
 
