diff --git a/src/test/java/com/jakewharton/DiskLruCacheTest.java b/src/test/java/com/jakewharton/DiskLruCacheTest.java
index 3786d29..18b7132 100644
--- a/src/test/java/com/jakewharton/DiskLruCacheTest.java
+++ b/src/test/java/com/jakewharton/DiskLruCacheTest.java
@@ -27,7 +27,6 @@ import java.io.Writer;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
-import org.apache.commons.io.FileUtils;
 import junit.framework.TestCase;
 import static com.jakewharton.DiskLruCache.JOURNAL_FILE;
 import static com.jakewharton.DiskLruCache.MAGIC;
@@ -654,29 +653,6 @@ public final class DiskLruCacheTest extends TestCase {
         assertNull(snapshot.edit());
     }
 
-    /** @see <a href="https://github.com/JakeWharton/DiskLruCache/issues/2}">Issue #2</a> */
-    public void testAggressiveClearingHandledOnWrite() throws Exception {
-        FileUtils.deleteDirectory(cacheDir);
-        set("A", "a", "a");
-        //assertValue("A", "a", "a");
-    }
-
-    /** @see <a href="https://github.com/JakeWharton/DiskLruCache/issues/2}">Issue #2</a> */
-    public void testAggressiveClearingHandledOnRead() throws Exception {
-        set("A", "a", "a");
-        FileUtils.deleteDirectory(cacheDir);
-        assertNull(cache.get("A"));
-    }
-
-    /** @see <a href="https://github.com/JakeWharton/DiskLruCache/issues/2}">Issue #2</a> */
-    public void testAggressiveClearingHandledOnEdit() throws Exception {
-        set("A", "a", "a");
-        DiskLruCache.Editor a = cache.get("A").edit();
-        FileUtils.deleteDirectory(cacheDir);
-        a.set(1, "a2");
-        a.commit();
-    }
-
     private void assertJournalEquals(String... expectedBodyLines) throws Exception {
         List<String> expectedLines = new ArrayList<String>();
         expectedLines.add(MAGIC);
@@ -827,4 +803,4 @@ public final class DiskLruCacheTest extends TestCase {
         assertTrue(getCleanFile(key, 1).exists());
         snapshot.close();
     }
-}
+}
\ No newline at end of file
