diff --git a/src/main/java/com/jakewharton/DiskLruCache.java b/src/main/java/com/jakewharton/DiskLruCache.java
index 90c32bb..c910177 100644
--- a/src/main/java/com/jakewharton/DiskLruCache.java
+++ b/src/main/java/com/jakewharton/DiskLruCache.java
@@ -741,20 +741,15 @@ public final class DiskLruCache implements Closeable {
                 File dirtyFile = entry.getDirtyFile(index);
                 FileOutputStream outputStream;
                 try {
-                    outputStream = new FileOutputStream(dirtyFile);
+                    return new FaultHidingOutputStream(new FileOutputStream(entry.getDirtyFile(index)));
                 } catch (FileNotFoundException e) {
                     // Attempt to recreate the cache directory and allow reading and writing.
                     directory.mkdirs();
                     directory.setReadable(true, false);
                     directory.setWritable(true, false);
-                    try {
-                      outputStream = new FileOutputStream(dirtyFile);
-                    } catch (FileNotFoundException e2) {
                       // We are unable to recover. Silently eat the writes.
                       return NULL_OUTPUT_STREAM;
-                    }
                 }
-                return new FaultHidingOutputStream(outputStream);
             }
         }
 
