diff --git a/src/test/java/com/jakewharton/disklrucache/DiskLruCacheTest.java b/src/test/java/com/jakewharton/disklrucache/DiskLruCacheTest.java
index f1540ac..ef0cc8a 100644
--- a/src/test/java/com/jakewharton/disklrucache/DiskLruCacheTest.java
+++ b/src/test/java/com/jakewharton/disklrucache/DiskLruCacheTest.java
@@ -852,12 +852,6 @@ public final class DiskLruCacheTest {
     a.commit();
   }
 
-  @Test public void removeHandlesMissingFile() throws Exception {
-    set("a", "a", "a");
-    getCleanFile("a", 0).delete();
-    cache.remove("a");
-  }
-
   /** @see <a href="https://github.com/JakeWharton/DiskLruCache/issues/2">Issue #2</a> */
   @Test public void aggressiveClearingHandlesPartialEdit() throws Exception {
     set("a", "a", "a");
