diff --git a/cayenne-xmpp/src/test/java/org/apache/cayenne/event/XMPPModuleIT.java b/cayenne-xmpp/src/test/java/org/apache/cayenne/event/XMPPModuleIT.java
index 1eaa82ca0..45173bbf1 100644
--- a/cayenne-xmpp/src/test/java/org/apache/cayenne/event/XMPPModuleIT.java
+++ b/cayenne-xmpp/src/test/java/org/apache/cayenne/event/XMPPModuleIT.java
@@ -28,7 +28,6 @@ import org.apache.cayenne.di.Module;
 import org.junit.Test;
 
 import static org.hamcrest.CoreMatchers.instanceOf;
-import static org.junit.Assert.assertNotSame;
 import static org.junit.Assert.assertThat;
 
 /**
@@ -46,13 +45,9 @@ public class XMPPModuleIT {
         };
 
         Injector injector = DIBootstrap.createInjector(new ServerModule(), new XMPPModule(), configModule);
-
         EventBridge bridge = injector.getInstance(EventBridge.class);
-        assertThat(bridge, instanceOf(XMPPBridge.class));
 
-        EventBridge bridge2 = injector.getInstance(EventBridge.class);
-        assertThat(bridge2, instanceOf(XMPPBridge.class));
-        assertNotSame(bridge, bridge2);
+        assertThat(bridge, instanceOf(XMPPBridge.class));
     }
 
 }
