diff --git a/src/main/java/org/apache/maven/plugin/deploy/DeployFileMojo.java b/src/main/java/org/apache/maven/plugin/deploy/DeployFileMojo.java
index 0b5f767..88a5b65 100644
--- a/src/main/java/org/apache/maven/plugin/deploy/DeployFileMojo.java
+++ b/src/main/java/org/apache/maven/plugin/deploy/DeployFileMojo.java
@@ -268,13 +268,9 @@ public class DeployFileMojo
         {
             this.artifactId = model.getArtifactId();
         }
-        if ( this.version == null )
+        if ( this.version == null && model.getVersion() != null )
         {
-        	this.version = model.getVersion();
-        	if ( this.version == null && parent != null )
-        	{
-        		this.version = parent.getVersion();
-        	}
+            this.version = model.getVersion();
         }
         if ( this.packaging == null && model.getPackaging() != null )
         {
