diff --git a/src/main/java/net/e175/klaus/solarpositioning/SPA.java b/src/main/java/net/e175/klaus/solarpositioning/SPA.java
index 29c09fa..c12400a 100644
--- a/src/main/java/net/e175/klaus/solarpositioning/SPA.java
+++ b/src/main/java/net/e175/klaus/solarpositioning/SPA.java
@@ -306,7 +306,7 @@ public final class SPA {
         dayStart.set(Calendar.MINUTE, 0);
         dayStart.set(Calendar.SECOND, 0);
         dayStart.set(Calendar.MILLISECOND, 0);
-        dayStart.set(Calendar.HOUR_OF_DAY, 0);
+        dayStart.set(Calendar.HOUR_OF_DAY, 12);
         final int offset = day.getTimeZone().getOffset(dayStart.getTimeInMillis());
 
         dayStart.set(Calendar.HOUR_OF_DAY, 0);
