diff --git a/src/main/java/org/apache/commons/codec/binary/Hex.java b/src/main/java/org/apache/commons/codec/binary/Hex.java
index 22fc5b7a..0097f099 100644
--- a/src/main/java/org/apache/commons/codec/binary/Hex.java
+++ b/src/main/java/org/apache/commons/codec/binary/Hex.java
@@ -249,17 +249,10 @@ public class Hex implements BinaryEncoder, BinaryDecoder {
     }
 
     private static byte[] toByteArray(final ByteBuffer byteBuffer) {
-        final int remaining = byteBuffer.remaining();
-        // Use the underlying buffer if possible
         if (byteBuffer.hasArray()) {
-            final byte[] byteArray = byteBuffer.array();
-            if (remaining == byteArray.length) {
-                //byteBuffer.position(remaining);
-                return byteArray;
-            }
+            return byteBuffer.array();
         }
-        // Copy the bytes
-        final byte[] byteArray = new byte[remaining];
+        final byte[] byteArray = new byte[byteBuffer.remaining()];
         byteBuffer.get(byteArray);
         return byteArray;
     }
