diff --git a/src/test/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParserTest.java b/src/test/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParserTest.java
index bb294adb..04e5a900 100644
--- a/src/test/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParserTest.java
+++ b/src/test/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParserTest.java
@@ -138,19 +138,6 @@ public class UnixFTPEntryParserTest extends FTPParseTestFramework {
 	protected FTPFileEntryParser getParser() {
 		return (new UnixFTPEntryParser());
 	}
	
-	public void testOwnerNameWithSpaces() {
-		FTPFile f = getParser().parseFTPEntry("drwxr-xr-x   2 john smith     group         4096 Mar  2 15:13 zxbox");
-		assertNotNull(f);
-		assertEquals("john smith", f.getUser());
-	}
	
-	public void testOwnerANdGroupNameWithSpaces() {
-		FTPFile f = getParser().parseFTPEntry("drwxr-xr-x   2 john smith     test group         4096 Mar  2 15:13 zxbox");
-		assertNotNull(f);
-		assertEquals("john smith", f.getUser());
-		assertEquals("test group", f.getGroup());
-	}
 
 	/**
 	 * @see org.apache.commons.net.ftp.parser.FTPParseTestFramework#testParseFieldsOnDirectory()
