diff --git a/src/main/java/org/apache/commons/net/util/SubnetUtils.java b/src/main/java/org/apache/commons/net/util/SubnetUtils.java
index 70c36854..68398c86 100644
--- a/src/main/java/org/apache/commons/net/util/SubnetUtils.java
+++ b/src/main/java/org/apache/commons/net/util/SubnetUtils.java
@@ -161,11 +161,8 @@ public class SubnetUtils {
         }
 
         public String[] getAllAddresses() {
             int ct = getAddressCount();
             String[] addresses = new String[ct];
-            if (ct == 0) {
-                return addresses;
-            }
             for (int add = low(), j=0; add <= high(); ++add, ++j) {
                 addresses[j] = format(toArray(add));
             }
