diff --git a/net_10_fix/src/main/java/org/apache/commons/net/util/SubnetUtils.java b/net_10_buggy/src/main/java/org/apache/commons/net/util/SubnetUtils.java
index 70c3685..81f1c68 100644
--- a/net_10_fix/src/main/java/org/apache/commons/net/util/SubnetUtils.java
+++ b/net_10_buggy/src/main/java/org/apache/commons/net/util/SubnetUtils.java
@@ -163,9 +163,6 @@ public class SubnetUtils {
         public String[] getAllAddresses() {
             int ct = getAddressCount();
             String[] addresses = new String[ct];
-            if (ct == 0) {
-                return addresses;
-            }
             for (int add = low(), j=0; add <= high(); ++add, ++j) {
                 addresses[j] = format(toArray(add));
             }
