diff --git a/src/test/java/org/apache/commons/net/ftp/parser/FTPTimestampParserImplTest.java b/src/test/java/org/apache/commons/net/ftp/parser/FTPTimestampParserImplTest.java
index 87cf7b52..090a3d60 100644
--- a/src/test/java/org/apache/commons/net/ftp/parser/FTPTimestampParserImplTest.java
+++ b/src/test/java/org/apache/commons/net/ftp/parser/FTPTimestampParserImplTest.java
@@ -97,25 +97,6 @@ public class FTPTimestampParserImplTest extends TestCase {
         }
     }
 
-    public void testNET444() throws Exception {
-        FTPTimestampParserImpl parser = new FTPTimestampParserImpl();
-        parser.setLenientFutureDates(true);
-        SimpleDateFormat sdf = new SimpleDateFormat(parser.getRecentDateFormatString());
-        GregorianCalendar now = new GregorianCalendar(2012, Calendar.FEBRUARY, 28, 12, 0);
-
-        GregorianCalendar nowplus1 = new GregorianCalendar(2012, Calendar.FEBRUARY, 28, 13, 0);
-        // Create a suitable short date
-        String future1 = sdf.format(nowplus1.getTime());
-        Calendar parsed1 = parser.parseTimestamp(future1, now);
-        assertEquals(nowplus1.get(Calendar.YEAR), parsed1.get(Calendar.YEAR));
-
-        GregorianCalendar nowplus25 = new GregorianCalendar(2012, Calendar.FEBRUARY, 29, 13, 0);
-        // Create a suitable short date
-        String future25 = sdf.format(nowplus25.getTime());
-        Calendar parsed25 = parser.parseTimestamp(future25, now);
-        assertEquals(nowplus25.get(Calendar.YEAR) - 1, parsed25.get(Calendar.YEAR));
-    }
-
     public void testParseTimestampAcrossTimeZones() {
 
 
@@ -193,8 +174,8 @@ public class FTPTimestampParserImplTest extends TestCase {
                 fail("failed.to.parse.default");
             }
             try {
-                Calendar c = parser.parseTimestamp("f\u00e9v 22 2002");
-                fail("should.have.failed.to.parse.default, but was: "+c.getTime().toString());
+                parser.parseTimestamp("f\u00e9v 22 2002");
+                fail("should.have.failed.to.parse.default");
             } catch (ParseException e) {
                 // this is the success case
             }
