diff --git a/net_12_fix/src/main/java/org/apache/commons/net/util/SubnetUtils.java b/net_12_buggy/src/main/java/org/apache/commons/net/util/SubnetUtils.java
index 8e0d580..f7efa2b 100644
--- a/net_12_fix/src/main/java/org/apache/commons/net/util/SubnetUtils.java
+++ b/net_12_buggy/src/main/java/org/apache/commons/net/util/SubnetUtils.java
@@ -226,10 +226,7 @@ public class SubnetUtils {
             address = matchAddress(matcher);
 
             /* Create a binary netmask from the number of bits specification /x */
-            int cidrPart = rangeCheck(Integer.parseInt(matcher.group(5)), 0, NBITS);
-            if (cidrPart == 0 && address != 0) {
-                throw new IllegalArgumentException("Cannot have /0 cidr with non-zero address");
-            }
+            int cidrPart = rangeCheck(Integer.parseInt(matcher.group(5)), 1, NBITS);
             for (int j = 0; j < cidrPart; ++j) {
                 netmask |= (1 << 31 - j);
             }
