diff --git a/src/main/java/org/apache/commons/net/util/SubnetUtils.java b/src/main/java/org/apache/commons/net/util/SubnetUtils.java
index 8e0d5806..540874b7 100644
--- a/src/main/java/org/apache/commons/net/util/SubnetUtils.java
+++ b/src/main/java/org/apache/commons/net/util/SubnetUtils.java
@@ -226,12 +226,9 @@ public class SubnetUtils {
             address = matchAddress(matcher);
 
             /* Create a binary netmask from the number of bits specification /x */
-            int cidrPart = rangeCheck(Integer.parseInt(matcher.group(5)), 0, NBITS);
-            if (cidrPart == 0 && address != 0) {
-                throw new IllegalArgumentException("Cannot have /0 cidr with non-zero address");
-            }
+            int cidrPart = rangeCheck(Integer.parseInt(matcher.group(5)), 1, NBITS);
             for (int j = 0; j < cidrPart; ++j) {
-                netmask |= (1 << 31 - j);
+                netmask |= (1 << 31-j);
             }
 
             /* Calculate base network address */
