diff --git a/src/test/java/org/apache/commons/net/SubnetUtilsTest.java b/src/test/java/org/apache/commons/net/SubnetUtilsTest.java
index 0b5fa3ee..0e502e26 100644
--- a/src/test/java/org/apache/commons/net/SubnetUtilsTest.java
+++ b/src/test/java/org/apache/commons/net/SubnetUtilsTest.java
@@ -298,10 +298,6 @@ public class SubnetUtilsTest extends TestCase {
          }
      }
 
-    public void testZeroAddressAndCidr() {
-        new SubnetUtils("0.0.0.0/0");
-    }
     public void testZeroNetmaskBits() {
         try {
             new SubnetUtils("192.168.0.1/0");
