diff --git a/src/main/java/org/apache/commons/net/nntp/Threader.java b/src/main/java/org/apache/commons/net/nntp/Threader.java
index bf934517..037374d7 100644
--- a/src/main/java/org/apache/commons/net/nntp/Threader.java
+++ b/src/main/java/org/apache/commons/net/nntp/Threader.java
@@ -27,7 +27,6 @@ package org.apache.commons.net.nntp;
  *
  */
 
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
@@ -66,10 +65,7 @@ public class Threader {
             }
         }
 
-        if (idTable.isEmpty()) {
-            return null;
-        }
 
         ThreadContainer root = findRootSet(idTable);
         idTable.clear();
         idTable = null;
@@ -451,16 +446,13 @@ public class Threader {
     /**
      * The client passes in an array of Threadable objects, and
      * the Threader constructs a connected 'graph' of messages
      * @param messages array of messages to thread, must not be empty
      * @return null if messages == null or root.child == null or messages array is empty
      * @deprecated (2.2) prefer {@link #thread(List)}
      */
     @Deprecated
     public Threadable thread(Threadable[] messages) {
-        if (messages == null) {
-            return null;
-        }
-        return thread(Arrays.asList(messages));
+        return thread(java.util.Arrays.asList(messages));
     }
 
 }
