diff --git a/src/main/java/org/apache/commons/net/nntp/Threader.java b/src/main/java/org/apache/commons/net/nntp/Threader.java
index bf93451..037374d 100644
--- a/src/main/java/org/apache/commons/net/nntp/Threader.java
+++ b/src/main/java/org/apache/commons/net/nntp/Threader.java
@@ -27,7 +27,6 @@ package org.apache.commons.net.nntp;
  *
  */
 
-import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;

@@ -66,10 +65,6 @@ public class Threader {
             }
         }
 
-        if (idTable.isEmpty()) {
-            return null;
-        }

         ThreadContainer root = findRootSet(idTable);
         idTable.clear();
         idTable = null;
@@ -451,16 +446,13 @@ public class Threader {
     @Deprecated
     public Threadable thread(Threadable[] messages) {
-        if (messages == null) {
-            return null;
-        }
-        return thread(Arrays.asList(messages));
+        return thread(java.util.Arrays.asList(messages));
     }
 
 }
