diff --git a/src/main/java/org/apache/commons/net/io/Util.java b/src/main/java/org/apache/commons/net/io/Util.java
index d18b075f..fbb5f2c6 100644
--- a/src/main/java/org/apache/commons/net/io/Util.java
+++ b/src/main/java/org/apache/commons/net/io/Util.java
@@ -285,7 +285,7 @@ public final class Util
                     dest.flush();
                     ++total;
                     if (listener != null) {
-                        listener.bytesTransferred(total, 1, streamSize);
+                        listener.bytesTransferred(total, chars, streamSize);
                     }
                     continue;
                 }
