diff --git a/src/main/java/org/apache/commons/net/smtp/SimpleSMTPHeader.java b/src/main/java/org/apache/commons/net/smtp/SimpleSMTPHeader.java
index 56aac009..195568ad 100644
--- a/src/main/java/org/apache/commons/net/smtp/SimpleSMTPHeader.java
+++ b/src/main/java/org/apache/commons/net/smtp/SimpleSMTPHeader.java
@@ -17,10 +17,7 @@
 
 package org.apache.commons.net.smtp;
 
-import java.text.SimpleDateFormat;
-import java.util.Date;
-import java.util.Locale;
 
 /***
  * This class is used to construct a bare minimum
  * acceptable header for an email message.  To construct more
@@ -55,7 +51,6 @@ public class SimpleSMTPHeader
     private final String __from;
     private final String __to;
     private final StringBuffer __headerFields;
-    private boolean hasHeaderDate;
     private StringBuffer __cc;
 
     /***
@@ -98,9 +93,6 @@ public class SimpleSMTPHeader
      ***/
     public void addHeaderField(String headerField, String value)
     {
-        if (!hasHeaderDate && "Date".equals(headerField)) {
-            hasHeaderDate = true;
-        }
         __headerFields.append(headerField);
         __headerFields.append(": ");
         __headerFields.append(value);
@@ -138,12 +130,7 @@ public class SimpleSMTPHeader
     {
         StringBuilder header = new StringBuilder();
 
-        final String pattern = "EEE, dd MMM yyyy HH:mm:ss Z"; // Fri, 21 Nov 1997 09:55:06 -0600
-        final SimpleDateFormat format = new SimpleDateFormat(pattern, Locale.ENGLISH);
 
-        if (!hasHeaderDate) {
-            addHeaderField("Date", format.format(new Date()));
-        }
         if (__headerFields.length() > 0) {
             header.append(__headerFields.toString());
         }
