diff --git a/src/main/java/org/apache/commons/net/util/SubnetUtils.java b/src/main/java/org/apache/commons/net/util/SubnetUtils.java
index c6b4c1e..2f33d58 100644
--- a/src/main/java/org/apache/commons/net/util/SubnetUtils.java
+++ b/src/main/java/org/apache/commons/net/util/SubnetUtils.java
@@ -354,25 +354,11 @@ public class SubnetUtils {
         return x & 0x0000003F;
     }
 
     private String toCidrNotation(String addr, String mask) {
-        int maskInt = toInteger(mask);

-        if ((maskInt & -maskInt) - 1 != ~maskInt) {
-            throw new IllegalArgumentException("Could not parse [" + mask + "]");
-        }

-        return addr + "/" + pop(maskInt);
+        return addr + "/" + pop(toInteger(mask));
     }
 }
