diff --git a/net_26_fix/src/main/java/org/apache/commons/net/util/SubnetUtils.java b/net_26_buggy/src/main/java/org/apache/commons/net/util/SubnetUtils.java
index f18f2c6..4de394d 100644
--- a/net_26_fix/src/main/java/org/apache/commons/net/util/SubnetUtils.java
+++ b/net_26_buggy/src/main/java/org/apache/commons/net/util/SubnetUtils.java
@@ -157,17 +157,12 @@ public class SubnetUtils {
         }
 
         public boolean isInRange(int address) {
-            if (address == 0) { // cannot ever be in range; rejecting now avoids problems with CIDR/31,32
-                return false;
-            }
             long addLong = address & UNSIGNED_INT_MASK;
             long lowLong = low() & UNSIGNED_INT_MASK;
             long highLong = high() & UNSIGNED_INT_MASK;
