diff --git a/src/main/java/org/apache/commons/net/ftp/FTPClientConfig.java b/src/main/java/org/apache/commons/net/ftp/FTPClientConfig.java
index eb8f6ca..5730f87 100644
--- a/src/main/java/org/apache/commons/net/ftp/FTPClientConfig.java
+++ b/src/main/java/org/apache/commons/net/ftp/FTPClientConfig.java
@@ -190,12 +190,9 @@ public class FTPClientConfig

-    public static final String SYST_L8 = "TYPE: L8";
+    public static final String SYST_L8 = "Type: L8";
     
     /**
      * Identifier by which an Netware-based ftp server is known throughout
