diff --git a/src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java b/src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java
index e1b19190..408d247b 100644
--- a/src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java
+++ b/src/main/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParser.java
@@ -84,24 +84,21 @@ public class UnixFTPEntryParser extends ConfigurableFTPFileEntryParserImpl
     private static final String REGEX =
         "([bcdelfmpSs-])"
         +"(((r|-)(w|-)([xsStTL-]))((r|-)(w|-)([xsStTL-]))((r|-)(w|-)([xsStTL-])))\\+?\\s*"
-        + "(\\d+)\\s+"                                  // link count
+        + "(\\d+)\\s+"
         + "(?:(\\S+(?:\\s\\S+)*?)\\s+)?"                // owner name (optional spaces)
         + "(?:(\\S+(?:\\s\\S+)*)\\s+)?"                 // group name (optional spaces)
-        + "(\\d+(?:,\\s*\\d+)?)\\s+"                    // size or n,m
+        + "(\\d+(?:,\\s*\\d+)?)\\s+"
         /*
-         * numeric or standard format date:
-         *   yyyy-mm-dd (expecting hh:mm to follow)
-         *   MMMM [d]d
-         *   [d]d MMM
+          numeric or standard format date
         */
-        + "((?:\\d+[-/]\\d+[-/]\\d+)|(?:[a-zA-Z]{3}\\s+\\d{1,2})|(?:\\d{1,2}\\s+[a-zA-Z]{3}))\\s+"
+        + "((?:\\d+[-/]\\d+[-/]\\d+)|(?:[a-zA-Z0-9]+\\s+\\S+))\\s+"
         /* 
-           year (for non-recent standard format) - yyyy
-           or time (for numeric or recent standard format) [h]h:mm  
+           year (for non-recent standard format) 
+           or time (for numeric or recent standard format  
         */
         + "(\\d+(?::\\d+)?)\\s+"
         
-        + "(\\S*)(\\s*.*)"; // the rest
+        + "(\\S*)(\\s*.*)";
 
 
     /**
@@ -189,7 +186,6 @@ public class UnixFTPEntryParser extends ConfigurableFTPFileEntryParserImpl
             case 'c':
                 isDevice = true;
                 // break; - fall through
-                //$FALL-THROUGH$ TODO change this if DEVICE_TYPE implemented
             case 'f':
             case '-':
                 type = FTPFile.FILE_TYPE;
