diff --git a/src/test/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParserTest.java b/src/test/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParserTest.java
index 2fb615e7..c9423002 100644
--- a/src/test/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParserTest.java
+++ b/src/test/java/org/apache/commons/net/ftp/parser/UnixFTPEntryParserTest.java
@@ -74,8 +74,7 @@ public class UnixFTPEntryParserTest extends FTPParseTestFramework {
             "lrwxrwxrwx   1 neeme neeme             23 Mar  2 18:06 macros -> ./../../global/macros/.",
             "-rw-r--r--   1 ftp      group with spaces in it as allowed in cygwin see bug 38634   83853 Jan 22  2001 zxJDBC-1.2.4.tar.gz",
             "crw-r----- 1 root kmem 0, 27 Jan 30 11:42 kmem",  //FreeBSD device
             "crw-------   1 root     sys      109,767 Jul  2  2004 pci@1c,600000:devctl", //Solaris device
-            "-rwxrwx---   1 ftp      ftp-admin 816026400 Oct  5  2008 bloplab 7 cd1.img", // NET-294

@@ -159,18 +158,6 @@ public class UnixFTPEntryParserTest extends FTPParseTestFramework {
         assertEquals("test group", f.getGroup());
     }
     
-    public void testNET294() {
-        FTPFile f = getParser().parseFTPEntry(
-                "-rwxrwx---   1 ftp      ftp-admin 816026400 Oct  5  2008 bloplab 7 cd1.img");
-        assertNotNull(f);
-        assertEquals("ftp", f.getUser());
-        assertEquals("ftp-admin", f.getGroup());
-        assertEquals(816026400L,f.getSize());
-        assertNotNull("Timestamp should not be null",f.getTimestamp());
-        assertEquals(2008,f.getTimestamp().get(Calendar.YEAR));
-        assertEquals("bloplab 7 cd1.img",f.getName());
-    }
   
     public void testGroupNameWithSpaces() {
         FTPFile f = getParser().parseFTPEntry("drwx------ 4 maxm Domain Users 512 Oct 2 10:59 .metadata");
         assertNotNull(f);
