diff --git a/src/main/java/org/apache/commons/net/ftp/parser/NTFTPEntryParser.java b/src/main/java/org/apache/commons/net/ftp/parser/NTFTPEntryParser.java
index 13f01d07..24065b96 100644
--- a/src/main/java/org/apache/commons/net/ftp/parser/NTFTPEntryParser.java
+++ b/src/main/java/org/apache/commons/net/ftp/parser/NTFTPEntryParser.java
@@ -18,7 +18,6 @@
 package org.apache.commons.net.ftp.parser;
 import java.text.ParseException;
 
-import org.apache.commons.net.ftp.Configurable;
 import org.apache.commons.net.ftp.FTPClientConfig;
 import org.apache.commons.net.ftp.FTPFile;
 
@@ -36,10 +35,6 @@ public class NTFTPEntryParser extends ConfigurableFTPFileEntryParserImpl
     private static final String DEFAULT_DATE_FORMAT 
         = "MM-dd-yy hh:mma"; //11-09-01 12:30PM
 
-    private static final String DEFAULT_DATE_FORMAT2
-        = "MM-dd-yy kk:mm"; //11-09-01 18:30
-
-    private FTPTimestampParser timestampParser;
 
     /**
      * this is the regular expression used by this parser.
@@ -78,13 +73,6 @@ public class NTFTPEntryParser extends ConfigurableFTPFileEntryParserImpl
     {
         super(REGEX);
         configure(config);
-        FTPClientConfig config2 = new FTPClientConfig(
-                FTPClientConfig.SYST_NT,
-                DEFAULT_DATE_FORMAT2,
-                null, null, null, null);
-        config2.setDefaultDateFormatStr(DEFAULT_DATE_FORMAT2);
-        this.timestampParser = new FTPTimestampParserImpl();
-        ((Configurable)this.timestampParser).configure(config2);
     }
 
     /**
@@ -114,15 +102,7 @@ public class NTFTPEntryParser extends ConfigurableFTPFileEntryParserImpl
             }
             catch (ParseException e)
             {
-                // parsing fails, try the other date format
-                try
-                {
-                    f.setTimestamp(timestampParser.parseTimestamp(datestr));
-                }
-                catch (ParseException e2)
-                {
-                    // intentionally do nothing
-                }
+                // intentionally do nothing
             }
 
             if (null == name || name.equals(".") || name.equals(".."))
