diff --git a/net_9_fix/src/main/java/org/apache/commons/net/util/SubnetUtils.java b/net_9_buggy/src/main/java/org/apache/commons/net/util/SubnetUtils.java
index 69676ca..58f3e92 100644
--- a/net_9_fix/src/main/java/org/apache/commons/net/util/SubnetUtils.java
+++ b/net_9_buggy/src/main/java/org/apache/commons/net/util/SubnetUtils.java
@@ -96,14 +96,11 @@ public class SubnetUtils {
         private int network()       { return network; }
         private int address()       { return address; }
         private int broadcast()     { return broadcast; }

         private int low() {
-            return (isInclusiveHostCount() ? network() :
-                broadcast() - network() > 1 ? network() + 1 : 0); 
+            return network() + (isInclusiveHostCount() ? 0 : 1);
         }
         private int high() { 
-            return (isInclusiveHostCount() ? broadcast() :
-                broadcast() - network() > 1 ? broadcast() -1  : 0); 
+            return broadcast() - (isInclusiveHostCount() ? 0 : 1);
         }
 
         /**
@@ -124,31 +121,10 @@ public class SubnetUtils {
         public String getNetworkAddress()           { return format(toArray(network())); }
         public String getNetmask()                  { return format(toArray(netmask())); }
         public String getAddress()                  { return format(toArray(address())); }

         public String getHighAddress()              { return format(toArray(high())); }

         public int getAddressCount()                { 
-            int count = broadcast() - network() + (isInclusiveHostCount() ? 1 : -1);
-            return count < 0 ? 0 : count;
+            return (broadcast() - low() + (isInclusiveHostCount() ? 1 : 0));
         }
 
         public int asInteger(String address)        { return toInteger(address); }
