diff --git a/src/main/java/com/fasterxml/aalto/async/AsyncByteArrayScanner.java b/src/main/java/com/fasterxml/aalto/async/AsyncByteArrayScanner.java
index f66b894..6d21357 100644
--- a/src/main/java/com/fasterxml/aalto/async/AsyncByteArrayScanner.java
+++ b/src/main/java/com/fasterxml/aalto/async/AsyncByteArrayScanner.java
@@ -2393,7 +2393,7 @@ public class AsyncByteArrayScanner
                         && _inputBuffer[ptr+2] == BYTE_t
                         && _inputBuffer[ptr+3] == BYTE_SEMICOLON) {
                     _inputPtr = ptr + 4;
-                    return INT_QUOTE;
+                    return INT_APOS;
                 }
             }
         }
@@ -2949,7 +2949,7 @@ public class AsyncByteArrayScanner
                         && _inputBuffer[ptr+2] == BYTE_t
                         && _inputBuffer[ptr+3] == BYTE_SEMICOLON) {
                     _inputPtr = ptr + 4;
-                    return INT_QUOTE;
+                    return INT_APOS;
                 }
             }
         }
diff --git a/src/main/java/com/fasterxml/aalto/async/AsyncByteBufferScanner.java b/src/main/java/com/fasterxml/aalto/async/AsyncByteBufferScanner.java
index 135135a..7047b34 100644
--- a/src/main/java/com/fasterxml/aalto/async/AsyncByteBufferScanner.java
+++ b/src/main/java/com/fasterxml/aalto/async/AsyncByteBufferScanner.java
@@ -2398,7 +2398,7 @@ public class AsyncByteBufferScanner
                         && _inputBuffer.get(ptr+2) == BYTE_t
                         && _inputBuffer.get(ptr+3) == BYTE_SEMICOLON) {
                     _inputPtr = ptr + 4;
-                    return INT_QUOTE;
+                    return INT_APOS;
                 }
             }
         }
@@ -2954,7 +2954,7 @@ public class AsyncByteBufferScanner
                         && _inputBuffer.get(ptr+2) == BYTE_t
                         && _inputBuffer.get(ptr+3) == BYTE_SEMICOLON) {
                     _inputPtr = ptr + 4;
-                    return INT_QUOTE;
+                    return INT_APOS;
                 }
             }
         }
diff --git a/src/main/java/com/fasterxml/aalto/async/AsyncByteScanner.java b/src/main/java/com/fasterxml/aalto/async/AsyncByteScanner.java
index 74a3ab8..642b4a5 100644
--- a/src/main/java/com/fasterxml/aalto/async/AsyncByteScanner.java
+++ b/src/main/java/com/fasterxml/aalto/async/AsyncByteScanner.java
@@ -372,7 +372,7 @@ public abstract class AsyncByteScanner
     protected void _releaseBuffers()
     {
         super._releaseBuffers();
-        if ((_symbols != null) && _symbols.maybeDirty()) {
+        if (_symbols.maybeDirty()) {
             _config.updateBBSymbols(_symbols);
         }
     }
