diff --git a/src/main/java/com/fasterxml/aalto/stax/InputFactoryImpl.java b/src/main/java/com/fasterxml/aalto/stax/InputFactoryImpl.java
index 7b90a74..fec9b3c 100644
--- a/src/main/java/com/fasterxml/aalto/stax/InputFactoryImpl.java
+++ b/src/main/java/com/fasterxml/aalto/stax/InputFactoryImpl.java
@@ -352,7 +352,10 @@ public final class InputFactoryImpl
     @Override
     public AsyncXMLStreamReader<AsyncByteArrayFeeder> createAsyncFor(byte[] input) throws XMLStreamException
     {
-        return createAsyncFor(input, 0, input.length);
+        ReaderConfig cfg = getNonSharedConfig(null, null, null, false, false);
+        cfg.setActualEncoding("UTF-8");
+        AsyncByteArrayScanner scanner = new AsyncByteArrayScanner(cfg);
+        return new AsyncStreamReaderImpl<AsyncByteArrayFeeder>(scanner);
     }
 
     @Override
