diff --git a/src/main/java/com/amihaiemil/docker/RtContainers.java b/src/main/java/com/amihaiemil/docker/RtContainers.java
index ae00267..8ac801f 100644
--- a/src/main/java/com/amihaiemil/docker/RtContainers.java
+++ b/src/main/java/com/amihaiemil/docker/RtContainers.java
@@ -101,9 +101,9 @@ final class RtContainers implements Containers {
     ) throws IOException {
         final String uri;
         if(!name.isEmpty()) {
-            uri = this.baseUri.toString() + "/containers/create?name=" + name;
+            uri = this.baseUri.toString() + "/create?name=" + name;
         } else {
-            uri = this.baseUri.toString() + "/containers/create";
+            uri = this.baseUri.toString() + "/create";
         }
         final HttpPost post = new HttpPost(uri);
         post.setEntity(new StringEntity(container.toString()));
