diff --git a/src/main/java/com/amihaiemil/docker/RtPlugin.java b/src/main/java/com/amihaiemil/docker/RtPlugin.java
index 5ba9aa4..477b45d 100644
--- a/src/main/java/com/amihaiemil/docker/RtPlugin.java
+++ b/src/main/java/com/amihaiemil/docker/RtPlugin.java
@@ -30,9 +30,7 @@ import java.net.URI;
 import java.util.Map;
 import javax.json.JsonArray;
 import javax.json.JsonObject;
-import org.apache.http.HttpStatus;
 import org.apache.http.client.HttpClient;
-import org.apache.http.client.methods.HttpPost;
 
 /**
  * Runtime {@link Plugin}.
@@ -83,40 +81,24 @@ final class RtPlugin extends JsonResource implements Plugin {
 
     @Override
     public void enable() throws IOException, UnexpectedResponseException {
-        final HttpPost enable =
-            new HttpPost(
-                String.format("%s/%s", this.uri.toString(), "enable")
-            );
-        try {
-            this.client.execute(
-                enable,
-                new MatchStatus(
-                    enable.getURI(),
-                    HttpStatus.SC_OK
-                )
-            );
-        } finally {
-            enable.releaseConnection();
-        }
+        throw new UnsupportedOperationException(
+            String.join(" ",
+                "RtPlugin.enable() is not yet implemented.",
+                "If you can contribute please",
+                "do it here: https://www.github.com/amihaiemil/docker-java-api"
+            )
+        );
     }
 
     @Override
     public void disable() throws IOException, UnexpectedResponseException {
-        final HttpPost disable =
-            new HttpPost(
-                String.format("%s/%s", this.uri.toString(), "disable")
-            );
-        try {
-            this.client.execute(
-                disable,
-                new MatchStatus(
-                    disable.getURI(),
-                    HttpStatus.SC_OK
-                )
-            );
-        } finally {
-            disable.releaseConnection();
-        }
+        throw new UnsupportedOperationException(
+            String.join(" ",
+                "RtPlugin.disable() is not yet implemented.",
+                "If you can contribute please",
+                "do it here: https://www.github.com/amihaiemil/docker-java-api"
+            )
+        );
     }
 
     @Override
