diff --git a/wicket-request/src/test/java/org/apache/wicket/request/UrlTest.java b/wicket-request/src/test/java/org/apache/wicket/request/UrlTest.java
index eaebe14e15..91a4914bb9 100644
--- a/wicket-request/src/test/java/org/apache/wicket/request/UrlTest.java
+++ b/wicket-request/src/test/java/org/apache/wicket/request/UrlTest.java
@@ -911,20 +911,4 @@ public class UrlTest extends Assert
 
 		url.removeLeadingSegments(3);
 	}
-
-    @Test
-    public void wicket_5114_allowtoStringFullWhenContainingTwoDots()
-    {
-        Url url = Url.parse("/mountPoint/whatever.../");
-        url.setHost("wicketHost");
-        assertEquals("//wicketHost/mountPoint/whatever.../", url.toString(StringMode.FULL));
-    }
-
-    @Test(expected = IllegalStateException.class)
-    public void wicket_5114_throwExceptionWhenToStringFullContainsRelativePathSegment()
-    {
-        Url url = Url.parse("/mountPoint/../whatever/");
-        url.setHost("wicketHost");
-        url.toString(StringMode.FULL);
-    }
 }
