diff --git a/wicket-request/src/main/java/org/apache/wicket/request/Url.java b/wicket-request/src/main/java/org/apache/wicket/request/Url.java
index 5297857c17..6aa6e2565d 100755
--- a/wicket-request/src/main/java/org/apache/wicket/request/Url.java
+++ b/wicket-request/src/main/java/org/apache/wicket/request/Url.java
@@ -337,16 +337,16 @@ public class Url implements Serializable
 	 */
 	private static QueryParameter parseQueryParameter(final String qp, final Charset charset)
 	{
 		int idxOfEquals = qp.indexOf('=');
 		if (idxOfEquals == -1)
 		{
 			// name => empty value
 			return new QueryParameter(decodeParameter(qp, charset), "");
 		}
 
-		String parameterName = qp.substring(0, idxOfEquals);
-		String parameterValue = qp.substring(idxOfEquals + 1);
-		return new QueryParameter(decodeParameter(parameterName, charset), decodeParameter(parameterValue, charset));
+		String parts[] = Strings.split(qp, '=');
+		return new QueryParameter(decodeParameter(parts[0], charset), decodeParameter(parts[1],
+			charset));
 	}
 
 	/**
