diff --git a/wicket-request/src/test/java/org/apache/wicket/request/UrlTest.java b/wicket-request/src/test/java/org/apache/wicket/request/UrlTest.java
index b047bbf53e..17082d36bb 100644
--- a/wicket-request/src/test/java/org/apache/wicket/request/UrlTest.java
+++ b/wicket-request/src/test/java/org/apache/wicket/request/UrlTest.java
@@ -260,31 +260,6 @@ public class UrlTest extends Assert
 		checkSegments(url, ";jsessionid=8kxeo3reannw1qjtxgkju8yiu");
 	}
 
-	/**
-	 * WICKET-5259
-	 */
-	@Test
-	public void parse17()
-	{
-		String s = "http://me:secret@localhost";
-		Url url = Url.parse(s);
-		assertEquals("http", url.getProtocol());
-		assertEquals("me:secret@localhost", url.getHost());
-	}
-
-	/**
-	 * WICKET-5259
-	 */
-	@Test
-	public void parse18()
-	{
-		String s = "http://me:secret@localhost:8080";
-		Url url = Url.parse(s);
-		assertEquals("http", url.getProtocol());
-		assertEquals("me:secret@localhost", url.getHost());
-		assertEquals(Integer.valueOf(8080), url.getPort());
-	}
-
 	/**
 	 * 
 	 */
