diff --git a/wicket-request/src/test/java/org/apache/wicket/request/mapper/parameter/PageParametersEncoderTest.java b/wicket-request/src/test/java/org/apache/wicket/request/mapper/parameter/PageParametersEncoderTest.java
index e58a920b78..0490d0c4dc 100644
--- a/wicket-request/src/test/java/org/apache/wicket/request/mapper/parameter/PageParametersEncoderTest.java
+++ b/wicket-request/src/test/java/org/apache/wicket/request/mapper/parameter/PageParametersEncoderTest.java
@@ -16,8 +16,6 @@
  */
 package org.apache.wicket.request.mapper.parameter;
 
-import static org.hamcrest.CoreMatchers.is;
-
 import org.apache.wicket.request.Url;
 import org.junit.Assert;
 import org.junit.Test;
@@ -37,11 +35,9 @@ public class PageParametersEncoderTest extends Assert
 	{
 		PageParametersEncoder encoder = new PageParametersEncoder();
 
-		Url url = Url.parse("idx1/idx2?named1=value1&=&named2=value2");
+		Url url = Url.parse("idx1/idx2?named1=value1&named2=value2");
 
 		PageParameters pageParameters = encoder.decodePageParameters(url);
-		assertThat(pageParameters.getIndexedCount(), is(2));
-		assertThat(pageParameters.getNamedKeys().size(), is(2));
 		assertEquals("idx1", pageParameters.get(0).toOptionalString());
 		assertEquals("idx2", pageParameters.get(1).toOptionalString());
 		assertEquals("value1", pageParameters.get("named1").toOptionalString());
