diff --git a/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java b/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java
index 6b266f3..47a0406 100644
--- a/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java
+++ b/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java
@@ -252,16 +252,15 @@ public class YAMLGenerator extends GeneratorBase
             // if not, MUST specify flow styles
             opt.setDefaultFlowStyle(FlowStyle.BLOCK);
         }
         // split-lines for text blocks?
         opt.setSplitLines(Feature.SPLIT_LINES.enabledIn(_formatFeatures));
         // array indentation?
         if (Feature.INDENT_ARRAYS.enabledIn(_formatFeatures)) {
             // But, wrt [dataformats-text#34]: need to set both to diff values to work around bug
             // (otherwise indentation level is "invisible". Note that this should NOT be necessary
             // but is needed up to at least SnakeYAML 1.18.
             // Also looks like all kinds of values do work, except for both being 2... weird.
-            opt.setIndicatorIndent(1);
-            opt.setIndent(2);
+            opt.setIndicatorIndent(2);
         }
         return opt;
     }
