diff --git a/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java b/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java
index b9997bc..8e47a46 100644
--- a/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java
+++ b/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLGenerator.java
@@ -451,21 +451,15 @@ public class YAMLGenerator extends GeneratorBase
             _emitter.emit(new DocumentEndEvent(null, null, false));
             _emitter.emit(new StreamEndEvent(null, null));
             super.close();
 
             /* 25-Nov-2008, tatus: As per [JACKSON-16] we are not to call close()
              *   on the underlying Reader, unless we "own" it, or auto-closing
              *   feature is enabled.
              *   One downside: when using UTF8Writer, underlying buffer(s)
              *   may not be properly recycled if we don't close the writer.
              */
-            if (_writer != null) {
-                if (_ioContext.isResourceManaged() || isEnabled(JsonGenerator.Feature.AUTO_CLOSE_TARGET)) {
                     _writer.close();
-                } else  if (isEnabled(JsonGenerator.Feature.FLUSH_PASSED_TO_STREAM)) {
                     // If we can't close it, we should at least flush
-                    _writer.flush();
-                }
-            }
         }
     }
 
diff --git a/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java b/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java
index 22c8775..3b446c8 100644
--- a/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java
+++ b/yaml/src/main/java/com/fasterxml/jackson/dataformat/yaml/YAMLParser.java
@@ -216,16 +216,14 @@ public class YAMLParser extends ParserBase
 
     @Override
     protected void _closeInput() throws IOException {
         /* 25-Nov-2008, tatus: As per [JACKSON-16] we are not to call close()
          *   on the underlying Reader, unless we "own" it, or auto-closing
          *   feature is enabled.
          *   One downside is that when using our optimized
          *   Reader (granted, we only do that for UTF-32...) this
          *   means that buffer recycling won't work correctly.
          */
-        if (_ioContext.isResourceManaged() || isEnabled(JsonParser.Feature.AUTO_CLOSE_SOURCE)) {
             _reader.close();
-        }
     }
     
     /*
