diff --git a/yaml/src/test/java/com/fasterxml/jackson/dataformat/yaml/deser/StreamingParseTest.java b/yaml/src/test/java/com/fasterxml/jackson/dataformat/yaml/deser/StreamingParseTest.java
index 8fa71a8..17c2c20 100644
--- a/yaml/src/test/java/com/fasterxml/jackson/dataformat/yaml/deser/StreamingParseTest.java
+++ b/yaml/src/test/java/com/fasterxml/jackson/dataformat/yaml/deser/StreamingParseTest.java
@@ -423,6 +423,7 @@ public class StreamingParseTest extends ModuleTestBase
         yp.close();
     }
 
+    // [Issue#10]
     // Scalars should not be parsed when not in the plain flow style.
     public void testQuotedStyles() throws Exception
     {
@@ -434,10 +435,8 @@ public class StreamingParseTest extends ModuleTestBase
         assertEquals("strings", p.getCurrentName());
         assertToken(JsonToken.START_ARRAY, p.nextToken());
         assertToken(JsonToken.VALUE_STRING, p.nextToken());
-        assertEquals(0, p.getParsingContext().getCurrentIndex());
         assertEquals("true", p.getText());
         assertToken(JsonToken.VALUE_STRING, p.nextToken());
-        assertEquals(1, p.getParsingContext().getCurrentIndex());
         assertEquals("false", p.getText());
         assertToken(JsonToken.END_ARRAY, p.nextToken());
         assertToken(JsonToken.END_OBJECT, p.nextToken());
@@ -508,10 +507,8 @@ public class StreamingParseTest extends ModuleTestBase
 
         assertToken(JsonToken.START_ARRAY, p.nextToken());
         assertToken(JsonToken.VALUE_STRING, p.nextToken());
-        assertEquals(0, p.getParsingContext().getCurrentIndex());
         assertEquals("Foo Bar", p.getText());
         assertToken(JsonToken.VALUE_STRING, p.nextToken());
-        assertEquals(1, p.getParsingContext().getCurrentIndex());
         assertEquals("Max Power", p.getText());
         assertToken(JsonToken.END_ARRAY, p.nextToken());
 
