diff --git a/commons-geometry-core/src/test/java/org/apache/commons/geometry/core/precision/EpsilonDoublePrecisionContextTest.java b/commons-geometry-core/src/test/java/org/apache/commons/geometry/core/precision/EpsilonDoublePrecisionContextTest.java
index 224a2a8..ff75ee8 100644
--- a/commons-geometry-core/src/test/java/org/apache/commons/geometry/core/precision/EpsilonDoublePrecisionContextTest.java
+++ b/commons-geometry-core/src/test/java/org/apache/commons/geometry/core/precision/EpsilonDoublePrecisionContextTest.java
@@ -16,7 +16,6 @@
  */
 package org.apache.commons.geometry.core.precision;
 
-import org.apache.commons.geometry.core.GeometryTestUtils;
 import org.apache.commons.geometry.core.precision.EpsilonDoublePrecisionContext;
 import org.junit.Assert;
 import org.junit.Test;
@@ -36,26 +35,6 @@ public class EpsilonDoublePrecisionContextTest {
         Assert.assertEquals(ctx.getMaxZero(), eps, 0.0);
     }
 
-    @Test
-    public void testInvalidEpsilonValues() {
-        // act/assert
-        GeometryTestUtils.assertThrows(() -> {
-            new EpsilonDoublePrecisionContext(-1.0);
-        }, IllegalArgumentException.class);
-
-        GeometryTestUtils.assertThrows(() -> {
-            new EpsilonDoublePrecisionContext(Double.NaN);
-        }, IllegalArgumentException.class, "Invalid epsilon value: NaN");
-
-        GeometryTestUtils.assertThrows(() -> {
-            new EpsilonDoublePrecisionContext(Double.POSITIVE_INFINITY);
-        }, IllegalArgumentException.class, "Invalid epsilon value: Infinity");
-
-        GeometryTestUtils.assertThrows(() -> {
-            new EpsilonDoublePrecisionContext(Double.NEGATIVE_INFINITY);
-        }, IllegalArgumentException.class, "Invalid epsilon value: -Infinity");
-    }
-
     @Test
     public void testSign() {
         // arrange
