diff --git a/src/main/java/net/lingala/zip4j/util/FileUtils.java b/src/main/java/net/lingala/zip4j/util/FileUtils.java
index 014d7f3..a0145a0 100644
--- a/src/main/java/net/lingala/zip4j/util/FileUtils.java
+++ b/src/main/java/net/lingala/zip4j/util/FileUtils.java
@@ -72,7 +72,7 @@ public class FileUtils {
 
   public static byte[] getFileAttributes(File file) {
     try {
-      if (file == null || (!Files.isSymbolicLink(file.toPath()) && !file.exists())) {
+      if (file == null || !file.exists()) {
         return new byte[4];
       }
 
@@ -484,8 +484,7 @@ public class FileUtils {
     byte[] fileAttributes = new byte[4];
 
     try {
-      PosixFileAttributeView posixFileAttributeView = Files.getFileAttributeView(file, PosixFileAttributeView.class,
-          LinkOption.NOFOLLOW_LINKS);
+      PosixFileAttributeView posixFileAttributeView = Files.getFileAttributeView(file, PosixFileAttributeView.class);
       Set<PosixFilePermission> posixFilePermissions = posixFileAttributeView.readAttributes().permissions();
 
       fileAttributes[3] = setBitIfApplicable(Files.isRegularFile(file), fileAttributes[3], 7);
