diff --git a/src/main/java/net/lingala/zip4j/tasks/AbstractExtractFileTask.java b/src/main/java/net/lingala/zip4j/tasks/AbstractExtractFileTask.java
index 6d34d29..60793de 100644
--- a/src/main/java/net/lingala/zip4j/tasks/AbstractExtractFileTask.java
+++ b/src/main/java/net/lingala/zip4j/tasks/AbstractExtractFileTask.java
@@ -128,11 +128,7 @@ public abstract class AbstractExtractFileTask<T> extends AsyncZipTask<T> {
   }
 
   private void verifyNextEntry(ZipInputStream zipInputStream, FileHeader fileHeader) throws IOException {
-    if (BitUtils.isBitSet(fileHeader.getGeneralPurposeFlag()[0], 6)) {
-      throw new ZipException("Entry with name " + fileHeader.getFileName() + " is encrypted with Strong Encryption. " +
-          "Zip4j does not support Strong Encryption, as this is patented.");
-    }
 
     LocalFileHeader localFileHeader = zipInputStream.getNextEntry(fileHeader);
 
     if (localFileHeader == null) {
