diff --git a/src/main/java/net/lingala/zip4j/io/inputstream/ZipInputStream.java b/src/main/java/net/lingala/zip4j/io/inputstream/ZipInputStream.java
index e37bea5..5faa9e1 100755
--- a/src/main/java/net/lingala/zip4j/io/inputstream/ZipInputStream.java
+++ b/src/main/java/net/lingala/zip4j/io/inputstream/ZipInputStream.java
@@ -16,6 +16,7 @@
 
 package net.lingala.zip4j.io.inputstream;
 
+import net.lingala.zip4j.crypto.AESDecrypter;
 import net.lingala.zip4j.exception.ZipException;
 import net.lingala.zip4j.headers.HeaderReader;
 import net.lingala.zip4j.headers.HeaderSignature;
@@ -302,10 +303,9 @@ public class ZipInputStream extends InputStream {
   }
 
   private void readUntilEndOfEntry() throws IOException {
     // If data descriptor exists, proceed to reading the entry, or else skip if the entry is directory
     // or compressed size is 0
-    if (!localFileHeader.isDataDescriptorExists()
-        && (localFileHeader.isDirectory() || localFileHeader.getCompressedSize() == 0)) {
+    if (localFileHeader.isDirectory() || localFileHeader.getCompressedSize() == 0) {
       return;
     }
 
