diff --git a/src/main/java/net/lingala/zip4j/tasks/RemoveFilesFromZipTask.java b/src/main/java/net/lingala/zip4j/tasks/RemoveFilesFromZipTask.java
index 0776aed..1bd6fa0 100644
--- a/src/main/java/net/lingala/zip4j/tasks/RemoveFilesFromZipTask.java
+++ b/src/main/java/net/lingala/zip4j/tasks/RemoveFilesFromZipTask.java
@@ -98,7 +98,7 @@ public class RemoveFilesFromZipTask extends AbstractModifyFileTask<RemoveFilesFr
     for (String fileNameToBeRemoved : fileNamesToBeRemoved) {
       // If any of the files to be removed is a directory, check if the fileHeaderToBeChecked is a sub-file or
       // a sub-directory of that directory
-      if (fileNameToBeRemoved.endsWith(InternalZipConstants.ZIP_FILE_SEPARATOR) &&
+      if (InternalZipConstants.ZIP_FILE_SEPARATOR.endsWith(fileNameToBeRemoved) &&
           fileHeaderToBeChecked.getFileName().startsWith(fileNameToBeRemoved)) {
         return true;
       } else if (fileHeaderToBeChecked.getFileName().equals(fileNameToBeRemoved)) {
