diff --git a/src/test/java/net/lingala/zip4j/AddFilesToZipIT.java b/src/test/java/net/lingala/zip4j/AddFilesToZipIT.java
index 9a4dc8d..9a7aa83 100644
--- a/src/test/java/net/lingala/zip4j/AddFilesToZipIT.java
+++ b/src/test/java/net/lingala/zip4j/AddFilesToZipIT.java
@@ -195,19 +195,6 @@ public class AddFilesToZipIT extends AbstractIT {
         EncryptionMethod.AES, AesKeyStrength.KEY_STRENGTH_256);
   }
 
-  @Test
-  public void testAddFileWithAfterDeflateRemainingBytesTestFile() throws IOException {
-    ZipParameters zipParameters = createZipParameters(EncryptionMethod.AES, AesKeyStrength.KEY_STRENGTH_256);
-    zipParameters.setAesVersion(AesVersion.TWO);
-    ZipFile zipFile = new ZipFile(generatedZipFile, PASSWORD);
-
-    zipFile.addFile(TestUtils.getTestFileFromResources("after_deflate_remaining_bytes.bin"), zipParameters);
-
-    ZipFileVerifier.verifyZipFileByExtractingAllFiles(generatedZipFile, PASSWORD, outputFolder, 1);
-    verifyZipFileContainsFiles(generatedZipFile, singletonList("after_deflate_remaining_bytes.bin"),
-        CompressionMethod.DEFLATE, EncryptionMethod.AES, AesKeyStrength.KEY_STRENGTH_256);
-  }
-
   @Test
   public void testAddFileProgressMonitorThrowsExceptionWhenPerformingActionInBusyState() throws ZipException {
     expectedException.expectMessage("invalid operation - Zip4j is in busy state");
