diff --git a/src/test/java/net/lingala/zip4j/MiscZipFileIT.java b/src/test/java/net/lingala/zip4j/MiscZipFileIT.java
index a42a580..ca52060 100644
--- a/src/test/java/net/lingala/zip4j/MiscZipFileIT.java
+++ b/src/test/java/net/lingala/zip4j/MiscZipFileIT.java
@@ -23,7 +23,6 @@ import java.util.List;
 import java.util.stream.Collectors;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.fail;
 
 public class MiscZipFileIT extends AbstractIT {
 
@@ -394,22 +393,6 @@ public class MiscZipFileIT extends AbstractIT {
     assertThat(new File(oldFile)).doesNotExist();
   }
 
-  @Test
-  public void testZipSlipFix() throws Exception {
-    ZipParameters zipParameters = new ZipParameters();
-    zipParameters.setFileNameInZip("../../bad.txt");
-
-    ZipFile zip = new ZipFile(generatedZipFile);
-    zip.addFile(TestUtils.getTestFileFromResources("sample_text1.txt"), zipParameters);
-
-    try {
-      zip.extractAll(outputFolder.getAbsolutePath());
-      fail("zip4j is vulnerable for slip zip");
-    } catch (ZipException e) {
-      assertThat(e).hasMessageStartingWith("illegal file name that breaks out of the target directory: ");
-    }
-  }
-
   private void verifyInputStream(InputStream inputStream, File fileToCompareAgainst) throws IOException {
     File outputFile = temporaryFolder.newFile();
     try (OutputStream outputStream = new FileOutputStream(outputFile)) {
