diff --git a/src/main/java/net/lingala/zip4j/headers/HeaderReader.java b/src/main/java/net/lingala/zip4j/headers/HeaderReader.java
index 803a751..f3ee899 100755
--- a/src/main/java/net/lingala/zip4j/headers/HeaderReader.java
+++ b/src/main/java/net/lingala/zip4j/headers/HeaderReader.java
@@ -280,12 +280,8 @@ public class HeaderReader {
     byte[] extraFieldBuf = new byte[extraFieldLength];
     zip4jRaf.read(extraFieldBuf);
 
-    try {
-      return parseExtraDataRecords(extraFieldBuf, extraFieldLength);
-    } catch (Exception e) {
       // Ignore any errors when parsing extra data records
-      return Collections.emptyList();
-    }
+    return parseExtraDataRecords(extraFieldBuf, extraFieldLength);
   }
 
   private List<ExtraDataRecord> readExtraDataRecords(InputStream inputStream, int extraFieldLength)
@@ -302,12 +298,8 @@ public class HeaderReader {
     byte[] extraFieldBuf = new byte[extraFieldLength];
     inputStream.read(extraFieldBuf);
 
-    try {
-      return parseExtraDataRecords(extraFieldBuf, extraFieldLength);
-    } catch (Exception e) {
       // Ignore any errors when parsing extra data records
-      return Collections.emptyList();
-    }
+    return parseExtraDataRecords(extraFieldBuf, extraFieldLength);
   }
 
   private List<ExtraDataRecord> parseExtraDataRecords(byte[] extraFieldBuf, int extraFieldLength) {
