diff --git a/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java b/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
index 40e8e54..61da977 100644
--- a/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
+++ b/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
@@ -475,12 +475,6 @@ public class ExtractZipFileIT extends AbstractIT {
     ZipFileVerifier.verifyFileContent(TestUtils.getTestFileFromResources("öüäöäö/asöäööl"), outputFile);
   }
 
-  @Test
-  public void testExtractJarFile() throws IOException {
-    ZipFile zipFile = new ZipFile(TestUtils.getTestArchiveFromResources("zip4j-for-testing.jar"));
-    zipFile.extractAll(outputFolder.getPath());
-  }
-
   private void addFileToZip(ZipFile zipFile, String fileName, EncryptionMethod encryptionMethod, String password) throws ZipException {
     ZipParameters zipParameters = new ZipParameters();
     zipParameters.setEncryptFiles(encryptionMethod != null);
diff --git a/src/test/java/net/lingala/zip4j/io/inputstream/ZipInputStreamIT.java b/src/test/java/net/lingala/zip4j/io/inputstream/ZipInputStreamIT.java
index 8f948b5..7bdd28b 100644
--- a/src/test/java/net/lingala/zip4j/io/inputstream/ZipInputStreamIT.java
+++ b/src/test/java/net/lingala/zip4j/io/inputstream/ZipInputStreamIT.java
@@ -188,17 +188,6 @@ public class ZipInputStreamIT extends AbstractIT {
     assertThat(filenameSet.contains(expactedFileName)).isTrue();
   }
 
-  @Test
-  public void testExtractJarFile() throws IOException {
-    byte[] b = new byte[4096];
-    File jarFile = getTestArchiveFromResources("zip4j-for-testing.jar");
-    try (ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(jarFile))) {
-      while (zipInputStream.getNextEntry() != null) {
-        zipInputStream.read(b);
-      }
-    }
-  }
-
   private void extractZipFileWithInputStreams(File zipFile, char[] password) throws IOException {
     extractZipFileWithInputStreams(zipFile, password, 4096, AesVersion.TWO);
   }
