diff --git a/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java b/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
index 6e150ff..40e8e54 100644
--- a/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
+++ b/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
@@ -476,13 +476,9 @@ public class ExtractZipFileIT extends AbstractIT {
   }
 
   @Test
-  public void testExtractJarFileWithFileHeaderCompressedSize2() throws IOException {
-    extractFile(TestUtils.getTestArchiveFromResources("jar-dir-fh-entry-size-2.jar"));
-  }
-
-  @Test
-  public void testExtractJarFileWithOnlyFileAndLocalFileHeaderCompressedSize2() throws IOException {
-    extractFile(TestUtils.getTestArchiveFromResources("jar-dir-lfh-and-fh-entry-size-2.jar"));
+  public void testExtractJarFile() throws IOException {
+    ZipFile zipFile = new ZipFile(TestUtils.getTestArchiveFromResources("zip4j-for-testing.jar"));
+    zipFile.extractAll(outputFolder.getPath());
   }
 
   private void addFileToZip(ZipFile zipFile, String fileName, EncryptionMethod encryptionMethod, String password) throws ZipException {
@@ -560,9 +556,7 @@ public class ExtractZipFileIT extends AbstractIT {
     return file.get();
   }
 
-  private File createZipFileAndSplit(List<File> filesToAddToZip, long splitLength, boolean encrypt,
-                                     EncryptionMethod encryptionMethod) throws IOException {
-
+  private File createZipFileAndSplit(List<File> filesToAddToZip, long splitLength, boolean encrypt, EncryptionMethod encryptionMethod) throws IOException {
     ZipFile zipFile = new ZipFile(generatedZipFile, PASSWORD);
     ZipParameters zipParameters = new ZipParameters();
     zipParameters.setEncryptFiles(encrypt);
@@ -572,8 +566,4 @@ public class ExtractZipFileIT extends AbstractIT {
     return TestUtils.splitFileWith7ZipFormat(zipFile.getFile(), temporaryFolder.getRoot(), splitLength);
   }
 
-  private void extractFile(File fileToExtract) throws IOException {
-    ZipFile zipFile = new ZipFile(fileToExtract);
-    zipFile.extractAll(outputFolder.getPath());
-  }
 }
diff --git a/src/test/java/net/lingala/zip4j/io/inputstream/ZipInputStreamIT.java b/src/test/java/net/lingala/zip4j/io/inputstream/ZipInputStreamIT.java
index d24bd42..8f948b5 100644
--- a/src/test/java/net/lingala/zip4j/io/inputstream/ZipInputStreamIT.java
+++ b/src/test/java/net/lingala/zip4j/io/inputstream/ZipInputStreamIT.java
@@ -191,7 +191,7 @@ public class ZipInputStreamIT extends AbstractIT {
   @Test
   public void testExtractJarFile() throws IOException {
     byte[] b = new byte[4096];
-    File jarFile = getTestArchiveFromResources("jar-dir-fh-entry-size-2.jar");
+    File jarFile = getTestArchiveFromResources("zip4j-for-testing.jar");
     try (ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(jarFile))) {
       while (zipInputStream.getNextEntry() != null) {
         zipInputStream.read(b);
