diff --git a/src/test/java/net/lingala/zip4j/util/Zip4jUtilTest.java b/src/test/java/net/lingala/zip4j/util/Zip4jUtilTest.java
index 863125c..ff6f5dc 100644
--- a/src/test/java/net/lingala/zip4j/util/Zip4jUtilTest.java
+++ b/src/test/java/net/lingala/zip4j/util/Zip4jUtilTest.java
@@ -127,25 +127,6 @@ public class Zip4jUtilTest {
     assertThat(byteArray[8]).isEqualTo((byte)'y');
   }
 
-  @Test
-  public void testConvertCharArrayToByteArrayChineseChars() {
-    char[] charArray = "你好".toCharArray();
-
-    byte[] byteArray = Zip4jUtil.convertCharArrayToByteArray(charArray);
-
-    try {
-      // Make sure that StandardCharsets exists on the classpath
-      Class.forName("java.nio.charset.StandardCharsets");
-      assertThat(byteArray.length).isEqualTo(6);
-      assertThat(byteArray).isEqualTo(new byte[]{-28, -67, -96, -27, -91, -67});
-    } catch (ClassNotFoundException e) {
-      // In some test environments (old Android SDK), StandardCharset class does not exist, in this case
-      // the method under test falls back to converting char to its byte representation
-      assertThat(byteArray.length).isEqualTo(2);
-      assertThat(byteArray).isEqualTo(new byte[]{96, 125});
-    }
-  }
-
   @Test
   public void testGetCompressionMethodForNonAesReturnsAsIs() {
     LocalFileHeader localFileHeader = new LocalFileHeader();
