diff --git a/src/test/java/net/lingala/zip4j/RemoveFilesFromZipIT.java b/src/test/java/net/lingala/zip4j/RemoveFilesFromZipIT.java
index 909e5bd..a81ff23 100644
--- a/src/test/java/net/lingala/zip4j/RemoveFilesFromZipIT.java
+++ b/src/test/java/net/lingala/zip4j/RemoveFilesFromZipIT.java
@@ -234,18 +234,6 @@ public class RemoveFilesFromZipIT extends AbstractIT {
     assertZipFileDoesNotContainsFileByName(new ZipFile(zipFileUnderTest), fileNameToRemove);
   }
 
-  @Test
-  public void testRemoveEntryFromAZipFileWithDuplicateEntriesRemovesSuccessfully() throws IOException {
-    TestUtils.copyFile(TestUtils.getTestArchiveFromResources("zip_with_duplicate_entries.zip"), generatedZipFile);
-    ZipFile zipFile = new ZipFile(generatedZipFile);
-    int actualNumberOfEntries = zipFile.getFileHeaders().size();
-    zipFile.removeFile("sample.pdf");
-
-    zipFile = new ZipFile(generatedZipFile);
-    assertThat(zipFile.getFileHeaders().size()).isEqualTo(actualNumberOfEntries - 1);
-    assertZipFileDoesNotContainsFileByName(zipFile, "sample.pdf");
-  }
-
   private void testRemoveEntryFromZipWhichHasCentralDirEntriesInDifferentOrderThanLocalEntries(
       String fileNameToRemove) throws IOException {
     TestUtils.copyFile(TestUtils.getTestArchiveFromResources("cen_dir_entries_diff_order_as_local_entries.zip"),
