diff --git a/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java b/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
index 041e5f5..edc4c84 100644
--- a/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
+++ b/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
@@ -366,16 +366,6 @@ public class ExtractZipFileIT extends AbstractIT {
     testExtractNestedZipFileWithEncrpytion(EncryptionMethod.AES, EncryptionMethod.ZIP_STANDARD);
   }
 
-  @Test
-  public void testExtractZipFileLessThanMinimumExpectedZipFileSizeThrowsException() throws IOException {
-    expectedException.expect(ZipException.class);
-    expectedException.expectMessage("Zip file size less than minimum expected zip file size. " +
-        "Probably not a zip file or a corrupted zip file");
-
-    ZipFile zipFile = new ZipFile(getTestArchiveFromResources("invalid_zip_file_size_less_than_22kb.zip"));
-    zipFile.extractAll(temporaryFolder.toString());
-  }
-
   private void testExtractNestedZipFileWithEncrpytion(EncryptionMethod innerZipEncryption,
                                                        EncryptionMethod outerZipEncryption) throws IOException {
     File innerZipFile = temporaryFolder.newFile("inner.zip");
