diff --git a/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java b/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
index ada86e1..041e5f5 100644
--- a/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
+++ b/src/test/java/net/lingala/zip4j/ExtractZipFileIT.java
@@ -376,13 +376,6 @@ public class ExtractZipFileIT extends AbstractIT {
     zipFile.extractAll(temporaryFolder.toString());
   }
 
-  @Test
-  public void testExtractZipFileEmptyZipFileExtractsNone() throws IOException {
-    ZipFile zipFile = new ZipFile(getTestArchiveFromResources("valid_empty_zip_file.zip"));
-    zipFile.extractAll(outputFolder.getPath());
-    assertThat(outputFolder.listFiles()).isEmpty();
-  }
-
   private void testExtractNestedZipFileWithEncrpytion(EncryptionMethod innerZipEncryption,
                                                        EncryptionMethod outerZipEncryption) throws IOException {
     File innerZipFile = temporaryFolder.newFile("inner.zip");
