diff --git a/src/main/java/net/lingala/zip4j/util/RawIO.java b/src/main/java/net/lingala/zip4j/util/RawIO.java
index 78a4fa3..39f0c3d 100755
--- a/src/main/java/net/lingala/zip4j/util/RawIO.java
+++ b/src/main/java/net/lingala/zip4j/util/RawIO.java
@@ -52,10 +52,10 @@ public class RawIO {
   }
 
   public long readLongLittleEndian(byte[] array, int pos) {
-    if (array.length - pos < 8) {
+    if (array.length < 8) {
       resetBytes(longBuff);
     }
-    System.arraycopy(array, pos, longBuff, 0, array.length < 8 ? array.length - pos : 8);
+    System.arraycopy(array, pos, longBuff, 0, array.length < 8 ? array.length : 8);
 
     long temp = 0;
     temp |= longBuff[7] & 0xff;
