diff --git a/core/rio/rdfxml/src/test/java/org/eclipse/rdf4j/rio/rdfxml/RDFXMLParserTest.java b/core/rio/rdfxml/src/test/java/org/eclipse/rdf4j/rio/rdfxml/RDFXMLParserTest.java
index c41befd80..e4f6e8d85 100644
--- a/core/rio/rdfxml/src/test/java/org/eclipse/rdf4j/rio/rdfxml/RDFXMLParserTest.java
+++ b/core/rio/rdfxml/src/test/java/org/eclipse/rdf4j/rio/rdfxml/RDFXMLParserTest.java
@@ -21,7 +21,6 @@ import org.eclipse.rdf4j.model.Statement;
 import org.eclipse.rdf4j.model.ValueFactory;
 import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
 import org.eclipse.rdf4j.model.vocabulary.DC;
-import org.eclipse.rdf4j.model.vocabulary.RDFS;
 import org.eclipse.rdf4j.rio.RDFParser;
 import org.eclipse.rdf4j.rio.helpers.StatementCollector;
 import org.hamcrest.CoreMatchers;
@@ -53,11 +52,11 @@ public class RDFXMLParserTest {
 
 		Collection<Statement> stmts = sc.getStatements();
 
-		assertThat(stmts, Matchers.<Statement> iterableWithSize(3));
+		assertThat(stmts, Matchers.<Statement> iterableWithSize(2));
 
 		Iterator<Statement> iter = stmts.iterator();
 
-		Statement stmt1 = iter.next(), stmt2 = iter.next(), stmt3 = iter.next();
+		Statement stmt1 = iter.next(), stmt2 = iter.next();
 
 		assertEquals(vf.createIRI("http://www.example.com/#"), stmt1.getSubject());
 		assertEquals(vf.createIRI("http://www.example.com/ns/#document-about"), stmt1.getPredicate());
@@ -78,8 +77,5 @@ public class RDFXMLParserTest {
 		assertEquals(res, stmt2.getSubject());
 		assertEquals(DC.TITLE, stmt2.getPredicate());
 		assertEquals(vf.createLiteral("Empty File"), stmt2.getObject());
-
-		assertEquals(RDFS.LABEL, stmt3.getPredicate());
-		assertEquals(vf.createLiteral("  Literal with whitespace  "), stmt3.getObject());
 	}
 }
