diff --git a/src/test/java/com/spotify/sparkey/SparkeyTest.java b/src/test/java/com/spotify/sparkey/SparkeyTest.java
deleted file mode 100644
index 6326e65..0000000
--- a/src/test/java/com/spotify/sparkey/SparkeyTest.java
+++ /dev/null
@@ -1,41 +0,0 @@
-/*
- * Copyright (c) 2015 Spotify AB
- *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not
- * use this file except in compliance with the License. You may obtain a copy of
- * the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations under
- * the License.
- */
-package com.spotify.sparkey;
-
-import org.junit.Test;
-
-import java.io.File;
-
-import static org.junit.Assert.*;
-
-public class SparkeyTest {
-
-  @Test
-  public void testFilenames() throws Exception {
-    assertEquals(new File("foo.bar.spi"), Sparkey.getIndexFile(new File("foo.bar")));
-    assertEquals(new File("foo.bar.spi"), Sparkey.getIndexFile(new File("foo.bar.")));
-    assertEquals(new File("foo.bar.spi"), Sparkey.getIndexFile(new File("foo.bar.spi")));
-    assertEquals(new File("foo.bar.spi"), Sparkey.getIndexFile(new File("foo.bar.spl")));
-    assertEquals(new File("foo.bar.baz.spi"), Sparkey.getIndexFile(new File("foo.bar.baz")));
-
-    assertEquals(new File("foo.bar.spl"), Sparkey.getLogFile(new File("foo.bar")));
-    assertEquals(new File("foo.bar.spl"), Sparkey.getLogFile(new File("foo.bar.")));
-    assertEquals(new File("foo.bar.spl"), Sparkey.getLogFile(new File("foo.bar.spi")));
-    assertEquals(new File("foo.bar.spl"), Sparkey.getLogFile(new File("foo.bar.spl")));
-    assertEquals(new File("foo.bar.baz.spl"), Sparkey.getLogFile(new File("foo.bar.baz")));
-
-  }
-}
