diff --git a/src/test/java/net/kotek/jdbm/DBTest.java b/src/test/java/net/kotek/jdbm/DBTest.java
index 47ee2e3..fe0a446 100644
--- a/src/test/java/net/kotek/jdbm/DBTest.java
+++ b/src/test/java/net/kotek/jdbm/DBTest.java
@@ -437,21 +437,6 @@ public class DBTest extends TestCaseWithTestFile {
 
     }
 
-    public void testDeleteAndPutCollection() throws IOException {
-        DB db = newDBNoCache();
-        db = DBMakerTest.newDBCache();
-        Map toAdd = new HashMap() ;
-        toAdd.put("description","test");
-        toAdd.put("descriptio1","test");
-        Map<String,Object> map = db.createHashMap("test");
-        map.putAll(toAdd);
-        db.commit();
-        db.deleteCollection("test");
-        map = db.getHashMap("test");
-        assertNull(map);
-
-    }
-
 
 }
 
