diff --git a/src/test/java/org/apache/jdbm/TestIssues.java b/src/test/java/org/apache/jdbm/TestIssues.java
index 651795e..81d7bba 100644
--- a/src/test/java/org/apache/jdbm/TestIssues.java
+++ b/src/test/java/org/apache/jdbm/TestIssues.java
@@ -69,13 +69,4 @@ public class TestIssues extends TestCaseWithTestFile {
         thread.join();
         db.close();
     }
-
-    public void test_issue_84_reopen_after_close(){
-        String f = newTestFile();
-        DB db = DBMaker.openFile(f).make();
-        db.close();
-
-        db = DBMaker.openFile(f).readonly().make();
-        db.close();
-    }
 }
