diff --git a/src/test/java/org/apache/commons/math4/distribution/HypergeometricDistributionTest.java b/src/test/java/org/apache/commons/math4/distribution/HypergeometricDistributionTest.java
index 7e7ce8cee..e12c5b42c 100644
--- a/src/test/java/org/apache/commons/math4/distribution/HypergeometricDistributionTest.java
+++ b/src/test/java/org/apache/commons/math4/distribution/HypergeometricDistributionTest.java
@@ -311,32 +311,6 @@ public class HypergeometricDistributionTest extends IntegerDistributionAbstractT
         Assert.assertTrue(Precision.compareTo(1.0, upper, 1) == 0);
     }
 
-    @Test
-    public void testZeroTrial() {
-        final int n = 11; // population
-        final int m = 4;  // successes in population
-        final int s = 0;  // number of trials
-
-        final HypergeometricDistribution dist = new HypergeometricDistribution(n, m, 0);
-
-        for (int i = 1; i <= n; i++) {
-            final double p = dist.probability(i);
-            Assert.assertEquals("p=" + p, 0, p, 0d);
-        }
-    }
-
-    @Test
-    public void testMath1356() {
-        final int n = 11;  // population
-        final int m = 11;  // successes in population
-
-        for (int s = 0; s <= n; s++) {
-            final HypergeometricDistribution dist = new HypergeometricDistribution(n, m, s);
-            final double p = dist.probability(s);
-            Assert.assertEquals("p=" + p, 1, p, 0d);
-        }
-    }
-
     @Test
     public void testMath1021() {
         final int N = 43130568;
