diff --git a/src/main/java/org/apache/commons/math4/geometry/euclidean/oned/Interval.java b/src/main/java/org/apache/commons/math4/geometry/euclidean/oned/Interval.java
index 87dbba16c..97857761b 100644
--- a/src/main/java/org/apache/commons/math4/geometry/euclidean/oned/Interval.java
+++ b/src/main/java/org/apache/commons/math4/geometry/euclidean/oned/Interval.java
@@ -17,8 +17,6 @@
 package org.apache.commons.math4.geometry.euclidean.oned;
 
 import org.apache.commons.math4.geometry.partitioning.Region.Location;
-import org.apache.commons.math4.exception.NumberIsTooSmallException;
-import org.apache.commons.math4.exception.util.LocalizedFormats;
 
 
 /** This class represents a 1D interval.
@@ -38,10 +36,6 @@ public class Interval {
      * @param upper upper bound of the interval
      */
     public Interval(final double lower, final double upper) {
-        if (upper < lower) {
-            throw new NumberIsTooSmallException(LocalizedFormats.ENDPOINTS_NOT_AN_INTERVAL,
-                                                upper, lower, true);
-        }
         this.lower = lower;
         this.upper = upper;
     }
