diff --git a/src/test/java/org/apache/commons/math4/geometry/euclidean/oned/IntervalTest.java b/src/test/java/org/apache/commons/math4/geometry/euclidean/oned/IntervalTest.java
index 3c854348d..2ebd847f2 100644
--- a/src/test/java/org/apache/commons/math4/geometry/euclidean/oned/IntervalTest.java
+++ b/src/test/java/org/apache/commons/math4/geometry/euclidean/oned/IntervalTest.java
@@ -20,7 +20,6 @@ import org.apache.commons.math4.geometry.euclidean.oned.Interval;
 import org.apache.commons.math4.geometry.partitioning.Region;
 import org.apache.commons.math4.util.FastMath;
 import org.apache.commons.math4.util.Precision;
-import org.apache.commons.math4.exception.NumberIsTooSmallException;
 import org.junit.Assert;
 import org.junit.Test;
 
@@ -73,9 +72,4 @@ public class IntervalTest {
         Assert.assertEquals(1.0, interval.getBarycenter(), Precision.EPSILON);
     }
 
-    // MATH-1256
-    @Test(expected=NumberIsTooSmallException.class)
-    public void testStrictOrdering() {
-        new Interval(0, -1);
-    }
 }
