diff --git a/src/test/java/org/apache/commons/math4/util/FastMathTest.java b/src/test/java/org/apache/commons/math4/util/FastMathTest.java
index 1e93148f4..ec0d91ce1 100644
--- a/src/test/java/org/apache/commons/math4/util/FastMathTest.java
+++ b/src/test/java/org/apache/commons/math4/util/FastMathTest.java
@@ -217,15 +217,6 @@ public class FastMathTest {
         }
         Assert.assertEquals(0, maxErr, 3);
     }
-    
-    @Test
-    public void testMath1269() {
-        final double arg = 709.8125;
-        final double vM = Math.exp(arg);
-        final double vFM = FastMath.exp(arg);
-        Assert.assertTrue("exp(" + arg + ") is " + vFM + " instead of " + vM,
-                          Precision.equalsIncludingNaN(vM, vFM));
-    }
 
     @Test
     public void testHyperbolicInverses() {
