diff --git a/src/test/java/org/apache/commons/math4/special/GammaTest.java b/src/test/java/org/apache/commons/math4/special/GammaTest.java
index 51982a3a9..35da9fe5e 100644
--- a/src/test/java/org/apache/commons/math4/special/GammaTest.java
+++ b/src/test/java/org/apache/commons/math4/special/GammaTest.java
@@ -974,21 +974,6 @@ public class GammaTest {
         }
     }
 
-    @Test
-    public void testGammaNegativeDouble() {
-        // check that the gamma function properly switches sign
-        // see: https://en.wikipedia.org/wiki/Gamma_function
-
-        double previousGamma = Gamma.gamma(-18.5);
-        for (double x = -19.5; x > -25; x -= 1.0) {
-            double gamma = Gamma.gamma(x);
-            Assert.assertEquals(  (int) FastMath.signum(previousGamma),
-                                - (int) FastMath.signum(gamma));
-
-            previousGamma = gamma;
-        }
-    }
-
     private void checkRelativeError(String msg, double expected, double actual,
                                     double tolerance) {
 
