diff --git a/src/test/java/org/apache/commons/math4/fraction/FractionTest.java b/src/test/java/org/apache/commons/math4/fraction/FractionTest.java
index 971e74089..aca7d05a4 100644
--- a/src/test/java/org/apache/commons/math4/fraction/FractionTest.java
+++ b/src/test/java/org/apache/commons/math4/fraction/FractionTest.java
@@ -262,17 +262,6 @@ public class FractionTest {
         Assert.assertEquals(percentage, frac.percentageValue(), Math.ulp(percentage));
     }
 
-    @Test
-    public void testMath1261() {
-        final Fraction a = new Fraction(Integer.MAX_VALUE, 2);
-        final Fraction b = a.multiply(2);
-        Assert.assertTrue(b.equals(new Fraction(Integer.MAX_VALUE)));
-
-        final Fraction c = new Fraction(2, Integer.MAX_VALUE);
-        final Fraction d = c.divide(2);
-        Assert.assertTrue(d.equals(new Fraction(1, Integer.MAX_VALUE)));
-    }
-
     @Test
     public void testReciprocal() {
         Fraction f = null;
