diff --git a/src/test/java/org/apache/commons/math4/util/ResizableDoubleArrayTest.java b/src/test/java/org/apache/commons/math4/util/ResizableDoubleArrayTest.java
index 88392e108..c7be6ea1f 100644
--- a/src/test/java/org/apache/commons/math4/util/ResizableDoubleArrayTest.java
+++ b/src/test/java/org/apache/commons/math4/util/ResizableDoubleArrayTest.java
@@ -124,13 +124,6 @@ public class ResizableDoubleArrayTest extends DoubleArrayAbstractTest {
         ResizableDoubleArray copyDa = new ResizableDoubleArray(testDa);
         Assert.assertEquals(copyDa, testDa);
         Assert.assertEquals(testDa, copyDa);
-        
-        // JIRA: MATH-1252
-        final double[] values = {1};
-        testDa = new ResizableDoubleArray(values);
-        Assert.assertArrayEquals(values, testDa.getElements(), 0);
-        Assert.assertEquals(1, testDa.getNumElements());
-        Assert.assertEquals(1, testDa.getElement(0), 0);
     }
 
     @Test
