diff --git a/src/main/java/org/apache/commons/math4/analysis/solvers/UnivariateSolverUtils.java b/src/main/java/org/apache/commons/math4/analysis/solvers/UnivariateSolverUtils.java
index 49742d84b..2521c9b5a 100644
--- a/src/main/java/org/apache/commons/math4/analysis/solvers/UnivariateSolverUtils.java
+++ b/src/main/java/org/apache/commons/math4/analysis/solvers/UnivariateSolverUtils.java
@@ -314,7 +314,7 @@ public class UnivariateSolverUtils {
         double delta = 0;
 
         for (int numIterations = 0;
-             (numIterations < maximumIterations) && (a > lowerBound || b < upperBound);
+             (numIterations < maximumIterations) && (a > lowerBound || b > upperBound);
              ++numIterations) {
 
             final double previousA  = a;
