diff --git a/src/main/java/org/apache/commons/math3/stat/inference/KolmogorovSmirnovTest.java b/src/main/java/org/apache/commons/math3/stat/inference/KolmogorovSmirnovTest.java
index 131d0c656..6154eeb7b 100644
--- a/src/main/java/org/apache/commons/math3/stat/inference/KolmogorovSmirnovTest.java
+++ b/src/main/java/org/apache/commons/math3/stat/inference/KolmogorovSmirnovTest.java
@@ -240,11 +240,10 @@ public class KolmogorovSmirnovTest {
      * @throws NullArgumentException if either {@code x} or {@code y} is null
      */
     public double kolmogorovSmirnovTest(double[] x, double[] y, boolean strict) {
-        final long lengthProduct = (long) x.length * y.length;
-        if (lengthProduct < SMALL_SAMPLE_PRODUCT) {
+        if (x.length * y.length < SMALL_SAMPLE_PRODUCT) {
             return exactP(kolmogorovSmirnovStatistic(x, y), x.length, y.length, strict);
         }
-        if (lengthProduct < LARGE_SAMPLE_PRODUCT) {
+        if (x.length * y.length < LARGE_SAMPLE_PRODUCT) {
             return monteCarloP(kolmogorovSmirnovStatistic(x, y), x.length, y.length, strict, MONTE_CARLO_ITERATIONS);
         }
         return approximateP(kolmogorovSmirnovStatistic(x, y), x.length, y.length);
