diff --git a/src/main/java/org/apache/commons/math4/ode/UnknownParameterException.java b/src/main/java/org/apache/commons/math4/ode/UnknownParameterException.java
index bbbaa4a16..75c621056 100644
--- a/src/main/java/org/apache/commons/math4/ode/UnknownParameterException.java
+++ b/src/main/java/org/apache/commons/math4/ode/UnknownParameterException.java
@@ -38,7 +38,7 @@ public class UnknownParameterException extends MathIllegalArgumentException {
      * @param name parameter name.
      */
     public UnknownParameterException(final String name) {
-        super(LocalizedFormats.UNKNOWN_PARAMETER, name);
+        super(LocalizedFormats.UNKNOWN_PARAMETER);
         this.name = name;
     }
 
