diff --git a/src/main/java/org/apache/commons/math3/geometry/euclidean/twod/hull/MonotoneChain.java b/src/main/java/org/apache/commons/math3/geometry/euclidean/twod/hull/MonotoneChain.java
index 50fd6b752..2ade7a69e 100644
--- a/src/main/java/org/apache/commons/math3/geometry/euclidean/twod/hull/MonotoneChain.java
+++ b/src/main/java/org/apache/commons/math3/geometry/euclidean/twod/hull/MonotoneChain.java
@@ -160,8 +160,8 @@ public class MonotoneChain extends AbstractConvexHullGenerator2D {
                 } else {
                     if (distanceToCurrent > distanceToLast) {
                         hull.remove(size - 1);
-                        hull.add(point);
                     }
+                    hull.add(point);
                 }
                 return;
             } else if (offset > 0) {
