diff --git a/src/main/java/org/apache/commons/math3/optim/linear/LinearConstraintSet.java b/src/main/java/org/apache/commons/math3/optim/linear/LinearConstraintSet.java
index b2a120996..cf5279a34 100644
--- a/src/main/java/org/apache/commons/math3/optim/linear/LinearConstraintSet.java
+++ b/src/main/java/org/apache/commons/math3/optim/linear/LinearConstraintSet.java
@@ -16,11 +16,11 @@
  */
 package org.apache.commons.math3.optim.linear;
 
-import java.util.LinkedHashSet;
 import java.util.Set;
+import java.util.HashSet;
 import java.util.Collection;
 import java.util.Collections;
 
 import org.apache.commons.math3.optim.OptimizationData;
 
 /**
@@ -31,7 +30,8 @@ import org.apache.commons.math3.optim.OptimizationData;
  */
 public class LinearConstraintSet implements OptimizationData {
     /** Set of constraints. */
-    private final Set<LinearConstraint> linearConstraints = new LinkedHashSet<LinearConstraint>();
+    private final Set<LinearConstraint> linearConstraints
+        = new HashSet<LinearConstraint>();
 
     /**
      * Creates a set containing the given constraints.
