diff --git a/src/main/java/org/apache/commons/math4/distribution/NormalDistribution.java b/src/main/java/org/apache/commons/math4/distribution/NormalDistribution.java
index 52168672a..6313ef04d 100644
--- a/src/main/java/org/apache/commons/math4/distribution/NormalDistribution.java
+++ b/src/main/java/org/apache/commons/math4/distribution/NormalDistribution.java
@@ -193,7 +193,7 @@ public class NormalDistribution extends AbstractRealDistribution {
         if (FastMath.abs(dev) > 40 * standardDeviation) {
             return dev < 0 ? 0.0d : 1.0d;
         }
-        return 0.5 * Erf.erfc(-dev / (standardDeviation * SQRT2));
+        return 0.5 * (1 + Erf.erf(dev / (standardDeviation * SQRT2)));
     }
 
     /** {@inheritDoc}
