diff --git a/src/test/java/org/apache/commons/math4/distribution/NormalDistributionTest.java b/src/test/java/org/apache/commons/math4/distribution/NormalDistributionTest.java
index 4e00f5662..3ced97c90 100644
--- a/src/test/java/org/apache/commons/math4/distribution/NormalDistributionTest.java
+++ b/src/test/java/org/apache/commons/math4/distribution/NormalDistributionTest.java
@@ -111,17 +111,6 @@ public class NormalDistributionTest extends RealDistributionAbstractTest {
         verifyInverseCumulativeProbabilities();
     }
 
-    // MATH-1257
-    @Test
-    public void testCumulativeProbability() {
-        final RealDistribution dist = new NormalDistribution(0, 1);
-        double x = -10;
-        double expected = 7.61985e-24;
-        double v = dist.cumulativeProbability(x);
-        double tol = 1e-5;
-        Assert.assertEquals(1, v / expected, 1e-5);
-    }
-
     @Test
     public void testGetMean() {
         NormalDistribution distribution = (NormalDistribution) getDistribution();
