diff --git a/core/src/test/java/dagger/ModuleIncludesTest.java b/core/src/test/java/dagger/ModuleIncludesTest.java
index b72f2e96..2ed6ea71 100644
--- a/core/src/test/java/dagger/ModuleIncludesTest.java
+++ b/core/src/test/java/dagger/ModuleIncludesTest.java
@@ -154,27 +154,4 @@ public final class ModuleIncludesTest {
     objectGraph.inject(entryPoint);
     assertThat(entryPoint.s).isEqualTo("a");
   }
-
-  static class A {}
-
-  static class B { @Inject A a; }
-
-  @Module(entryPoints = A.class) public static class TestModuleA {
-    @Provides A a() { return new A(); }
-  }
-
-  @Module(includes = TestModuleA.class, entryPoints = B.class) public static class TestModuleB {}
-
-  @Test public void autoInstantiationOfModules() {
-    // Have to make these non-method-scoped or instantiation errors occur.
-    ObjectGraph objectGraph = ObjectGraph.create(TestModuleA.class);
-    assertThat(objectGraph.get(A.class)).isNotNull();
-  }
-
-  @Test public void autoInstantiationOfIncludedModules() {
-    // Have to make these non-method-scoped or instantiation errors occur.
-    ObjectGraph objectGraph = ObjectGraph.create(new TestModuleB()); // TestModuleA auto-created.
-    assertThat(objectGraph.get(A.class)).isNotNull();
-    assertThat(objectGraph.get(B.class).a).isNotNull();
-  }
 }
