diff --git a/core/src/test/java/dagger/InjectionTest.java b/core/src/test/java/dagger/InjectionTest.java
index 09df5c29..bbb671e0 100644
--- a/core/src/test/java/dagger/InjectionTest.java
+++ b/core/src/test/java/dagger/InjectionTest.java
@@ -435,24 +435,6 @@ public final class InjectionTest {
     }
   }
 
-  @Test public void objectGraphGetInterface() {
-    final Runnable runnable = new Runnable() {
-      @Override public void run() {
-      }
-    };
-
-    @Module(injects = Runnable.class)
-    class TestModule {
-      @Provides Runnable provideRunnable() {
-        return runnable;
-      }
-    }
-
-    ObjectGraph graph = ObjectGraph.create(new TestModule());
-    graph.validate();
-    assertThat(graph.get(Runnable.class)).isSameAs(runnable);
-  }
-
   @Test public void noProvideBindingsForAbstractClasses() {
     class TestEntryPoint {
       @Inject AbstractList<?> abstractList;
@@ -581,8 +563,8 @@ public final class InjectionTest {
     }
 
     ObjectGraph graph = ObjectGraph.create(new TestModule());
-    assertThat((int) graph.get(Integer.class)).isEqualTo(0);
-    assertThat((int) graph.get(Integer.class)).isEqualTo(1);
+    assertEquals(0, (int) graph.get(Integer.class));
+    assertEquals(1, (int) graph.get(Integer.class));
   }
 
   @Test public void getInstanceRequiresEntryPoint() {
