diff --git a/core/src/test/java/dagger/InjectionTest.java b/core/src/test/java/dagger/InjectionTest.java
index a12fb2c5..09df5c29 100644
--- a/core/src/test/java/dagger/InjectionTest.java
+++ b/core/src/test/java/dagger/InjectionTest.java
@@ -17,7 +17,6 @@
 package dagger;
 
 import java.util.AbstractList;
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.RandomAccess;
@@ -825,20 +824,4 @@ public final class InjectionTest {
       assertThat(e.getMessage()).contains("Can't inject private constructor: ");
     }
   }
-
-  /** https://github.com/square/dagger/issues/231 */
-  @Test public void atInjectAlwaysRequiredForConstruction() {
-    @Module(injects = ArrayList.class)
-    class TestModule {
-    }
-
-    ObjectGraph objectGraph = ObjectGraph.create(new TestModule());
-    objectGraph.validate();
-    try {
-      objectGraph.get(ArrayList.class);
-      fail();
-    } catch (IllegalStateException e) {
-      assertThat(e.getMessage()).contains("Unable to create binding for java.util.ArrayList");
-    }
-  }
 }
