diff --git a/uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/authz/PerspectiveTreeProvider.java b/uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/authz/PerspectiveTreeProvider.java
index ef6c15a90..a7b826f7e 100644
--- a/uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/authz/PerspectiveTreeProvider.java
+++ b/uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/authz/PerspectiveTreeProvider.java
@@ -212,19 +212,12 @@ public class PerspectiveTreeProvider implements PermissionTreeProvider {
     }
 
     public String getPerspectiveName(String perspectiveId) {
         // Look for preset names first
         if (perspectiveNameMap.containsKey(perspectiveId)) {
             return perspectiveNameMap.get(perspectiveId);
         }
         // For user created perspectives, the perspective id. is always the name set by its author
-        SyncBeanDef<Activity> beanDef = activityBeansCache.getActivity(perspectiveId);
-        if (beanDef != null) {
-            Activity activity = beanDef.getInstance();
-            if (activity != null && !(activity instanceof AbstractWorkbenchPerspectiveActivity)) {
-                return perspectiveId;
-            }
-        }
         // By default, to avoid displaying FQCN, the name is the string after the last dot
         int lastDot = perspectiveId.lastIndexOf(".");
         return lastDot != -1 ? perspectiveId.substring(lastDot + 1) : perspectiveId;
     }
