diff --git a/uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/authz/PerspectiveTreeProviderTest.java b/uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/authz/PerspectiveTreeProviderTest.java
index 76180ad79..dc34c0626 100644
--- a/uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/authz/PerspectiveTreeProviderTest.java
+++ b/uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/authz/PerspectiveTreeProviderTest.java
@@ -27,7 +27,6 @@ import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
-import org.uberfire.client.mvp.AbstractWorkbenchPerspectiveActivity;
 import org.uberfire.client.mvp.Activity;
 import org.uberfire.client.mvp.ActivityBeansCache;
 import org.uberfire.client.mvp.PerspectiveActivity;
@@ -63,28 +62,14 @@ public class PerspectiveTreeProviderTest {
         List<SyncBeanDef<Activity>> beanDefs = new ArrayList<>();
         SyncBeanDef<Activity> bean1 = mock(SyncBeanDef.class);
         SyncBeanDef<Activity> bean2 = mock(SyncBeanDef.class);
-        SyncBeanDef<Activity> bean3 = mock(SyncBeanDef.class);
-        SyncBeanDef<Activity> bean4 = mock(SyncBeanDef.class);
-        PerspectiveActivity perspective1 = mock(AbstractWorkbenchPerspectiveActivity.class);
+        PerspectiveActivity perspective1 = mock(PerspectiveActivity.class);
         PerspectiveActivity perspective2 = mock(PerspectiveActivity.class);
-        PerspectiveActivity perspective3 = mock(PerspectiveActivity.class);
-        PerspectiveActivity perspective4 = mock(AbstractWorkbenchPerspectiveActivity.class);
         when(bean1.getInstance()).thenReturn(perspective1);
         when(bean2.getInstance()).thenReturn(perspective2);
-        when(bean3.getInstance()).thenReturn(perspective3);
-        when(bean4.getInstance()).thenReturn(perspective4);
         when(perspective1.getIdentifier()).thenReturn("Perspective1");
         when(perspective2.getIdentifier()).thenReturn("Perspective2");
-        when(perspective3.getIdentifier()).thenReturn("org.Perspective3");
-        when(perspective4.getIdentifier()).thenReturn("org.Perspective4");
         beanDefs.add(bean1);
         beanDefs.add(bean2);
-        beanDefs.add(bean3);
-        beanDefs.add(bean4);
-        when(activityBeansCache.getActivity("Perspective1")).thenReturn(bean1);
-        when(activityBeansCache.getActivity("Perspective2")).thenReturn(bean2);
-        when(activityBeansCache.getActivity("org.Perspective3")).thenReturn(bean3);
-        when(activityBeansCache.getActivity("org.Perspective4")).thenReturn(bean4);
         when(activityBeansCache.getPerspectiveActivities()).thenReturn(beanDefs);
 
         permissionManager = new DefaultPermissionManager();
@@ -144,7 +129,7 @@ public class PerspectiveTreeProviderTest {
                               options,
                               children -> {
                                   assertEquals(children.size(),
-                                               3);
+                                               1);
                               });
     }
 
@@ -202,18 +187,6 @@ public class PerspectiveTreeProviderTest {
         });
     }
 
-    @Test
-    public void testPerspectiveName() {
-        String name = provider.getPerspectiveName("Perspective1");
-        assertEquals(name, "A nice perspective");
-        name = provider.getPerspectiveName("Perspective2");
-        assertEquals(name, "Another nice perspective");
-        name = provider.getPerspectiveName("org.Perspective3");
-        assertEquals(name, "org.Perspective3");
-        name = provider.getPerspectiveName("org.Perspective4");
-        assertEquals(name, "Perspective4");
-    }
-
     protected void checkDependencies(PermissionNode permissionNode) {
         for (Permission permission : permissionNode.getPermissionList()) {
             Collection<Permission> dependencies = permissionNode.getDependencies(permission);
diff --git a/uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/mvp/ActivityBeansCacheTest.java b/uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/mvp/ActivityBeansCacheTest.java
index a8021cd75..77223fbde 100644
--- a/uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/mvp/ActivityBeansCacheTest.java
+++ b/uberfire-workbench/uberfire-workbench-client/src/test/java/org/uberfire/client/mvp/ActivityBeansCacheTest.java
@@ -194,7 +194,7 @@ public class ActivityBeansCacheTest {
 
         SyncBeanDef mock2 = mock(SyncBeanDef.class);
         when(mock2.getName()).thenReturn("screen");
-        when(mock2.isAssignableTo(PerspectiveActivity.class)).thenReturn(false);
+        when(mock2.getType()).thenReturn(WorkbenchScreenActivity.class);
 
         cache.addNewPerspectiveActivity(mock1);
         cache.addNewPerspectiveActivity(mock2);
