diff --git a/uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayoutInfoImpl.java b/uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayoutInfoImpl.java
index aa8b257cc..37ae182f4 100644
--- a/uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayoutInfoImpl.java
+++ b/uberfire-workbench/uberfire-workbench-client/src/main/java/org/uberfire/client/workbench/WorkbenchLayoutInfoImpl.java
@@ -27,10 +27,6 @@ public class WorkbenchLayoutInfoImpl implements WorkbenchLayoutInfo {
 
     @Override
     public int getHeaderHeight() {
-        try {
             return workbenchLayout.getHeaderPanel().getBoundingClientRect().getHeight().intValue();
-        } catch (Exception e) {
-            return 0;
-        }
     }
 }
