diff --git a/src/main/java/org/nuxeo/tools/gatling/report/SimulationParserV3.java b/src/main/java/org/nuxeo/tools/gatling/report/SimulationParserV3.java
index a8a8e7a..1c1b2ed 100644
--- a/src/main/java/org/nuxeo/tools/gatling/report/SimulationParserV3.java
+++ b/src/main/java/org/nuxeo/tools/gatling/report/SimulationParserV3.java
@@ -26,16 +26,13 @@ import java.util.Map;
  */
 public class SimulationParserV3 extends SimulationParser {
 
-    protected final Map<String, String> userIdToScenarioMap;
 
     public SimulationParserV3(File file, Float apdexT) {
         super(file, apdexT);
-        this.userIdToScenarioMap = new HashMap();
     }
 
     public SimulationParserV3(File file) {
         super(file);
-        this.userIdToScenarioMap = new HashMap();
     }
 
     protected String getSimulationName(List<String> line) {
@@ -47,12 +44,7 @@ public class SimulationParserV3 extends SimulationParser {
     }
 
     protected String getScenario(List<String> line) {
-        if (getType(line).equals(REQUEST)) {
-            final String userId = line.get(1);
-            return this.userIdToScenarioMap.get(userId);
-        } else {
-            return line.get(1);
-        }
+        return line.get(1);
     }
 
     protected String getType(List<String> line) {
@@ -63,7 +55,6 @@ public class SimulationParserV3 extends SimulationParser {
         // In Gatling 3.0, the REQUEST line contains the userid, but no longer contains the scenario.
         // To determine a REQUEST's scenario, we need to look it up based on the user id, so save the mapping.
         // Ugly to do it here like this, but the alternative is to add new methods to SimulationParser that would be Gatling 3.0 specfic.
-        saveUserIdAndScenario(line);
         return line.get(3);
     }
 
@@ -76,18 +67,13 @@ public class SimulationParserV3 extends SimulationParser {
     }
 
     protected Long getRequestEnd(List<String> line) {
-        return Long.parseLong(line.get(5));
+        return Long.parseLong(line.get(6));
     }
 
     protected boolean getRequestSuccess(List<String> line) {
         return OK.equals(line.get(6));
     }
 
-    private void saveUserIdAndScenario(List<String> line) {
-        final String scenario = getScenario(line);
-        final String userId = line.get(2);
-        userIdToScenarioMap.put(userId, scenario);
-    }
 
 
 }
