diff --git a/src/main/java/spoon/reflect/visitor/filter/CompositeFilter.java b/src/main/java/spoon/reflect/visitor/filter/CompositeFilter.java
index b5e66dc5..2cc21f0c 100644
--- a/src/main/java/spoon/reflect/visitor/filter/CompositeFilter.java
+++ b/src/main/java/spoon/reflect/visitor/filter/CompositeFilter.java
@@ -36,13 +36,13 @@ public class CompositeFilter<T extends CtElement> implements Filter<T> {
 		switch (operator) {
 		case INTERSECTION:
 			for (Filter<T> f : filters) {
-				if (! hasMatch(f, element))
+				if (!f.matches(element))
 					return false;
 			}
 			return true;
 		case UNION:
 			for (Filter<T> f : filters) {
-				if (hasMatch(f, element))
+				if (f.matches(element))
 					return true;
 			}
 			return false;
@@ -91,4 +91,4 @@ public class CompositeFilter<T extends CtElement> implements Filter<T> {
 		return (Class<T>) CtElement.class;
 	}
 
-}
\ No newline at end of file
+}
