diff --git a/src/main/java/spoon/support/compiler/FileSystemFolder.java b/src/main/java/spoon/support/compiler/FileSystemFolder.java
index 28acbd35..419165ff 100644
--- a/src/main/java/spoon/support/compiler/FileSystemFolder.java
+++ b/src/main/java/spoon/support/compiler/FileSystemFolder.java
@@ -76,7 +76,7 @@ public class FileSystemFolder implements SpoonFolder {
 		List<SpoonFolder> subFolders;
 		subFolders = new ArrayList<SpoonFolder>();
 		for (File f : file.listFiles()) {
-			if (!(SpoonResourceHelper.isArchive(f) || f.isFile()))
+			if (!SpoonResourceHelper.isFile(f))
 				try {
 					subFolders.add(SpoonResourceHelper.createFolder(f));
 				} catch (FileNotFoundException e) {
