diff --git a/johnzon-mapper/src/test/java/org/apache/johnzon/mapper/ObjectConverterTest.java b/johnzon-mapper/src/test/java/org/apache/johnzon/mapper/ObjectConverterTest.java
index 20064be..f138a0b 100644
--- a/johnzon-mapper/src/test/java/org/apache/johnzon/mapper/ObjectConverterTest.java
+++ b/johnzon-mapper/src/test/java/org/apache/johnzon/mapper/ObjectConverterTest.java
@@ -24,9 +24,7 @@ import org.junit.Test;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Comparator;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 public class ObjectConverterTest {
 
@@ -35,8 +33,6 @@ public class ObjectConverterTest {
         Contact contact = new Contact();
         contact.linkedPersons.addAll(Arrays.asList(new Person("f1", "l1"), new Person("f2", "l2")));
         contact.linkedPersonsArray = new Person[] { new Person("f3", "l3"), new Person("f4", "l4") };
-        contact.personMap.put("cinq", new Person("f5", "l5"));
-        contact.personMap.put("six", new Person("f6", "l6"));
 
         MapperBuilder mapperBuilder = new MapperBuilder();
         mapperBuilder.addConverter(Person.class, new PersonConverter());
@@ -49,7 +45,7 @@ public class ObjectConverterTest {
 
         String s = mapper.writeObjectAsString(contact);
         Contact c = mapper.readObject(s, Contact.class);
-        String expected = "{\"linkedPersons\":[\"f1|l1\",\"f2|l2\"],\"linkedPersonsArray\":[\"f3|l3\",\"f4|l4\"],\"personMap\":{\"six\":\"f6|l6\",\"cinq\":\"f5|l5\"}}";
+        String expected = "{\"linkedPersons\":[\"f1|l1\",\"f2|l2\"],\"linkedPersonsArray\":[\"f3|l3\",\"f4|l4\"]}";
         Assert.assertEquals(expected, s);
         Assert.assertEquals(contact, c);
     }
@@ -84,9 +80,6 @@ public class ObjectConverterTest {
         @JohnzonConverter(PersonConverter.class)
         private Person[] linkedPersonsArray;
 
-        @JohnzonConverter(PersonConverter.class)
-        private Map<String, Person> personMap = new HashMap<String, Person>();
-
         @Override
         public boolean equals(Object o) {
             if (this == o) {
@@ -97,8 +90,7 @@ public class ObjectConverterTest {
             }
 
             final Contact contact = Contact.class.cast(o);
-            return linkedPersons.equals(contact.linkedPersons) && personMap.equals(contact.personMap)
-                && Arrays.equals(linkedPersonsArray, contact.linkedPersonsArray);
+            return linkedPersons.equals(contact.linkedPersons) && Arrays.equals(linkedPersonsArray, contact.linkedPersonsArray);
 
         }
 
