diff --git a/tape/src/main/java/com/squareup/tape/QueueFile.java b/tape/src/main/java/com/squareup/tape/QueueFile.java
index 1ddf8fc..567ea66 100644
--- a/tape/src/main/java/com/squareup/tape/QueueFile.java
+++ b/tape/src/main/java/com/squareup/tape/QueueFile.java
@@ -145,9 +145,9 @@ public class QueueFile implements Closeable {
     if (fileLength > raf.length()) {
       throw new IOException(
           "File is truncated. Expected length: " + fileLength + ", Actual length: " + raf.length());
-    } else if (fileLength <= 0) {
+    } else if (fileLength == 0) {
       throw new IOException(
-          "File is corrupt; length stored in header (" + fileLength + ") is invalid.");
+          "File is corrupt; length stored in header is 0.");
     }
     elementCount = readInt(buffer, 4);
     int firstOffset = readInt(buffer, 8);
