diff --git a/src/test/java/com/fasterxml/jackson/databind/deser/TestEnumDeserialization.java b/src/test/java/com/fasterxml/jackson/databind/deser/TestEnumDeserialization.java
index 3ca06ffa8..eecd4a8c8 100644
--- a/src/test/java/com/fasterxml/jackson/databind/deser/TestEnumDeserialization.java
+++ b/src/test/java/com/fasterxml/jackson/databind/deser/TestEnumDeserialization.java
@@ -109,16 +109,6 @@ public class TestEnumDeserialization
         }
     }
 
-    // [Issue#324]: exception from creator method
-    protected enum TestEnum324
-    {
-        A, B;
-        
-        @JsonCreator public static TestEnum324 creator(String arg) {
-            throw new RuntimeException("Foobar!");
-        }
-    }
-    
     /*
     /**********************************************************
     /* Tests
@@ -341,7 +331,8 @@ public class TestEnumDeserialization
     // [JACKSON-834]
     public void testEnumsFromInts() throws Exception
     {
-        TestEnumFor834 res = MAPPER.readValue("1 ", TestEnumFor834.class);
+        ObjectMapper mapper = new ObjectMapper();
+        TestEnumFor834 res = mapper.readValue("1 ", TestEnumFor834.class);
         assertSame(TestEnumFor834.ENUM_A, res);
     }
 
@@ -363,15 +354,4 @@ public class TestEnumDeserialization
        // not sure this is totally safe but...
        assertEquals(TestEnum.JACKSON, mapper.readValue(quote("jackson"), TestEnum.class));
     }
-    
-    // [Issue#324]
-    public void testExceptionFromCreator() throws Exception
-    {
-        try {
-            /*TestEnum324 e =*/ MAPPER.readValue(quote("xyz"), TestEnum324.class);
-            fail("Should throw exception");
-        } catch (JsonMappingException e) {
-            verifyException(e, "foobar");
-        }
-    }
 }
