diff --git a/src/main/java/com/fasterxml/jackson/databind/deser/std/EnumDeserializer.java b/src/main/java/com/fasterxml/jackson/databind/deser/std/EnumDeserializer.java
index dab3d00b0..d83b36448 100644
--- a/src/main/java/com/fasterxml/jackson/databind/deser/std/EnumDeserializer.java
+++ b/src/main/java/com/fasterxml/jackson/databind/deser/std/EnumDeserializer.java
@@ -161,12 +161,9 @@ public class EnumDeserializer
             try {
                 return _factory.invoke(_enumClass, value);
             } catch (Exception e) {
-                Throwable t = ClassUtil.getRootCause(e);
-                if (t instanceof IOException) {
-                    throw (IOException) t;
-                }
-                throw ctxt.instantiationException(_enumClass, t);
+                ClassUtil.unwrapAndThrowAsIAE(e);
             }
+            return null;
         }
     }
 }
