diff --git a/src/test/java/com/fasterxml/jackson/databind/node/TestObjectNode.java b/src/test/java/com/fasterxml/jackson/databind/node/TestObjectNode.java
index bcf7945fb..aa67b47f8 100644
--- a/src/test/java/com/fasterxml/jackson/databind/node/TestObjectNode.java
+++ b/src/test/java/com/fasterxml/jackson/databind/node/TestObjectNode.java
@@ -3,10 +3,7 @@ package com.fasterxml.jackson.databind.node;
 import java.math.BigDecimal;
 import java.util.*;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonValue;
 import com.fasterxml.jackson.databind.*;
-import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 
 /**
  * Additional tests for {@link ObjectNode} container class.
@@ -14,45 +11,8 @@ import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 public class TestObjectNode
     extends BaseMapTest
 {
-    @JsonDeserialize(as = DataImpl.class)
-    public interface Data {
-    }
-
-    public static class DataImpl implements Data
-    {
-        protected JsonNode root;
-
-        @JsonCreator
-        public DataImpl(JsonNode n) {
-            root = n;
-        }
-        
-        @JsonValue
-        public JsonNode value() { return root; }
-        
-        /*
-        public Wrapper(ObjectNode n) { root = n; }
-        
-        @JsonValue
-        public ObjectNode value() { return root; }
-        */
-    }
-    
-    /*
-    /**********************************************************
-    /* Test methods
-    /**********************************************************
-     */
-    
-    private final ObjectMapper MAPPER = objectMapper();
+    private final ObjectMapper MAPPER = new ObjectMapper();
 
-    // for [Issue#346]
-    public void testEmptyNodeAsValue() throws Exception
-    {
-        Data w = MAPPER.readValue("{}", Data.class);
-        assertNotNull(w);
-    }
-    
     public void testBasics()
     {
         ObjectNode n = new ObjectNode(JsonNodeFactory.instance);
