diff --git a/src/main/java/com/fasterxml/jackson/databind/deser/std/JsonNodeDeserializer.java b/src/main/java/com/fasterxml/jackson/databind/deser/std/JsonNodeDeserializer.java
index 11b76b748..748979f1d 100644
--- a/src/main/java/com/fasterxml/jackson/databind/deser/std/JsonNodeDeserializer.java
+++ b/src/main/java/com/fasterxml/jackson/databind/deser/std/JsonNodeDeserializer.java
@@ -274,7 +274,6 @@ abstract class BaseNodeDeserializer<T extends JsonNode>
     {
         switch (jp.getCurrentToken()) {
         case START_OBJECT:
-        case END_OBJECT: // for empty JSON Objects we may point to this
             return deserializeObject(jp, ctxt, nodeFactory);
 
         case START_ARRAY:
