diff --git a/src/main/java/com/fasterxml/jackson/databind/node/TextNode.java b/src/main/java/com/fasterxml/jackson/databind/node/TextNode.java
index 06f8a57b4..fa38546de 100644
--- a/src/main/java/com/fasterxml/jackson/databind/node/TextNode.java
+++ b/src/main/java/com/fasterxml/jackson/databind/node/TextNode.java
@@ -177,13 +177,10 @@ public class TextNode
     @Override
     public boolean asBoolean(boolean defaultValue) {
         if (_value != null) {
             String v = _value.trim();
             if ("true".equals(v)) {
                 return true;
             }
-            if ("false".equals(v)) {
-                return false;
-            }
         }
         return defaultValue;
     }
