diff --git a/src/test/java/com/fasterxml/jackson/databind/struct/TestUnwrappedWithSameName647.java b/src/test/java/com/fasterxml/jackson/failing/TestUnwrappedWithSameName647.java
similarity index 94%
rename from src/test/java/com/fasterxml/jackson/databind/struct/TestUnwrappedWithSameName647.java
rename to src/test/java/com/fasterxml/jackson/failing/TestUnwrappedWithSameName647.java
index c92526008..5ce981c91 100644
--- a/src/test/java/com/fasterxml/jackson/databind/struct/TestUnwrappedWithSameName647.java
+++ b/src/test/java/com/fasterxml/jackson/failing/TestUnwrappedWithSameName647.java
@@ -1,4 +1,4 @@
-package com.fasterxml.jackson.databind.struct;
+package com.fasterxml.jackson.failing;
 
 import com.fasterxml.jackson.annotation.JsonUnwrapped;
 import com.fasterxml.jackson.databind.*;
