diff --git a/src/main/java/com/fasterxml/jackson/databind/MappingJsonFactory.java b/src/main/java/com/fasterxml/jackson/databind/MappingJsonFactory.java
index c07ad20b9..13eae43ee 100644
--- a/src/main/java/com/fasterxml/jackson/databind/MappingJsonFactory.java
+++ b/src/main/java/com/fasterxml/jackson/databind/MappingJsonFactory.java
@@ -33,14 +33,7 @@ public class MappingJsonFactory
         }
     }
 
-    public MappingJsonFactory(JsonFactory src, ObjectMapper mapper)
-    {
-        super(src, mapper);
-        if (mapper == null) {
-          setCodec(new ObjectMapper(this));
-        }
-    }
 
     /**
      * We'll override the method to return more specific type; co-variance
      * helps here
@@ -54,7 +46,7 @@ public class MappingJsonFactory
     {
         _checkInvalidCopy(MappingJsonFactory.class);
         // note: as with base class, must NOT copy mapper reference
-        return new MappingJsonFactory(this, null);
+        return new MappingJsonFactory(null);
     }
     
     /*
