diff --git a/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java b/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java
index 88def8c54..fd111bc41 100644
--- a/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java
+++ b/src/main/java/com/fasterxml/jackson/databind/type/TypeFactory.java            
@@ -391,20 +390,10 @@ public final class TypeFactory
             if (newType == null) {
                 // But otherwise gets bit tricky, as we need to partially resolve the type hierarchy
                 // (hopefully passing null Class for root is ok)
-                TypeBindings tb = null;
 
                 // 14-Apr-2016, tatu: One possible short-cut; if type parameter counts
                 //   match, chances are they ought to match. Let's take our chances...
-                if (baseType.containedTypeCount() == typeParamCount) {
-                    if (typeParamCount == 1) {
-                        tb = TypeBindings.create(subclass, baseType.containedType(0));
-                    } else if (typeParamCount == 2) {
-                        tb = TypeBindings.create(subclass, baseType.containedType(0),
-                                baseType.containedType(1));
-                    }
-                }
-                newType = _fromClass(null, subclass,
-                        (tb == null) ? TypeBindings.emptyBindings() : tb);
+                newType = _fromClass(null, subclass, TypeBindings.emptyBindings());        
             }
         } while (false);
 
