diff --git a/src/test/java/com/fasterxml/jackson/databind/jsontype/PolymorphicViaRefTypeTest.java b/src/test/java/com/fasterxml/jackson/databind/jsontype/PolymorphicViaRefTypeTest.java
deleted file mode 100644
index dc6dc8ddd..000000000
--- a/src/test/java/com/fasterxml/jackson/databind/jsontype/PolymorphicViaRefTypeTest.java
+++ /dev/null
@@ -1,48 +0,0 @@
-package com.fasterxml.jackson.databind.jsontype;
-
-import java.util.concurrent.atomic.AtomicReference;
-
-import com.fasterxml.jackson.annotation.JsonSubTypes;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
-import com.fasterxml.jackson.databind.*;
-
-public class PolymorphicViaRefTypeTest extends BaseMapTest
-{
-    
-    @JsonSubTypes({
-        @JsonSubTypes.Type(name = "impl5", value = ImplForAtomic.class)
-    })
-    static class BaseForAtomic {
-    }
-
-    static class ImplForAtomic extends BaseForAtomic {
-        public int x;
-
-        protected ImplForAtomic() { }
-        public ImplForAtomic(int x) { this.x = x; }
-    }
-
-    static class TypeInfoAtomic {
-        @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "$type")
-        public AtomicReference<BaseForAtomic> value;
-    }
-
-    /*
-    /**********************************************************************
-    /* Test methods
-    /**********************************************************************
-     */
-
-    private final ObjectMapper MAPPER = objectMapper();
-
-    public void testOptionalWithAtomic() throws Exception
-    {
-        TypeInfoAtomic data = new TypeInfoAtomic();
-        data.value = new AtomicReference<BaseForAtomic>(new ImplForAtomic(42));
-        String json = MAPPER.writeValueAsString(data);
-System.err.println("JSON = "+json);
-        TypeInfoAtomic result = MAPPER.readValue(json, TypeInfoAtomic.class);
-        assertNotNull(result);
-    }
-
-}
diff --git a/src/test/java/com/fasterxml/jackson/databind/jsontype/TestAbstractContainers.java b/src/test/java/com/fasterxml/jackson/databind/jsontype/TestAbstractContainers.java
index 49925efc1..6e1fc085d 100644
--- a/src/test/java/com/fasterxml/jackson/databind/jsontype/TestAbstractContainers.java
+++ b/src/test/java/com/fasterxml/jackson/databind/jsontype/TestAbstractContainers.java
@@ -6,7 +6,7 @@ import com.fasterxml.jackson.annotation.*;
 import com.fasterxml.jackson.databind.*;
 
 /**
- * For [databind#292]
+ * For [Issue#292]
  */
 @SuppressWarnings("serial")
 public class TestAbstractContainers extends BaseMapTest
