diff --git a/src/main/java/com/fasterxml/jackson/databind/JsonMappingException.java b/src/main/java/com/fasterxml/jackson/databind/JsonMappingException.java
index d73b75c90..dc9000500 100644
--- a/src/main/java/com/fasterxml/jackson/databind/JsonMappingException.java
+++ b/src/main/java/com/fasterxml/jackson/databind/JsonMappingException.java
@@ -5,7 +5,6 @@ import java.io.IOException;
 import java.io.Serializable;
 import java.util.*;
 
-import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.core.*;
 import com.fasterxml.jackson.databind.util.ClassUtil;
 
@@ -455,7 +454,6 @@ public class JsonMappingException
      */
 
     @Override // since 2.7.5
-    @JsonIgnore // as per [databind#1368]
     public Object getProcessor() { return _processor; }
 
     @Override

