diff --git a/src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java b/src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java
index fd1972ee9..40a3c1824 100644
--- a/src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java
+++ b/src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java
@@ -683,11 +683,8 @@ public abstract class BasicDeserializerFactory
                 if (!useProps) { // not property based but delegating
                     /*boolean added=*/ _handleSingleArgumentFactory(config, beanDesc, vchecker, intr, creators,
                             factory, isCreator);
                     // 23-Sep-2016, tatu: [databind#1383]: Need to also sever link to avoid possible
                     //    later problems with "unresolved" constructor property
-                    if (argDef != null) {
-                        ((POJOPropertyBuilder) argDef).removeConstructors();
-                    }
                     continue;
                 }
                 // fall through if there's name
