diff --git a/src/test/java/com/fasterxml/jackson/databind/creators/SingleArgCreatorTest.java b/src/test/java/com/fasterxml/jackson/databind/creators/SingleArgCreatorTest.java
index 04cb86f9f..e075151ec 100644
--- a/src/test/java/com/fasterxml/jackson/databind/creators/SingleArgCreatorTest.java
+++ b/src/test/java/com/fasterxml/jackson/databind/creators/SingleArgCreatorTest.java
@@ -115,30 +115,7 @@ public class SingleArgCreatorTest extends BaseMapTest
 
         public String value() { return value; }
     }
-
-    static class XY {
-        public int x, y;
-    }
     
-    // [databind#1383]
-    static class SingleArgWithImplicit {
-        protected XY _value;
-
-        private SingleArgWithImplicit() {
-            throw new Error("Should not get called");
-        }
-        private SingleArgWithImplicit(XY v, boolean bogus) {
-            _value = v;
-        }
-
-        @JsonCreator
-        public static SingleArgWithImplicit from(XY v) {
-            return new SingleArgWithImplicit(v, true);
-        }
-
-        public XY getFoobar() { return _value; }
-    }
-
     /*
     /**********************************************************
     /* Test methods
@@ -170,7 +147,7 @@ public class SingleArgCreatorTest extends BaseMapTest
         StringyBeanWithProps bean = mapper.readValue("{\"value\":\"x\"}", StringyBeanWithProps.class);
         assertEquals("x", bean.getValue());
     }    
-
+    
     // [databind#714]
     public void testSingleExplicitlyNamedButDelegating() throws Exception
     {
@@ -194,18 +171,5 @@ public class SingleArgCreatorTest extends BaseMapTest
         assertNotNull(bean2);
         assertEquals("def", bean2.value());
     }
-
-    // [databind#1383]
-    public void testSingleImplicitDelegating() throws Exception
-    {
-        final ObjectMapper mapper = new ObjectMapper();
-        mapper.setAnnotationIntrospector(new MyParamIntrospector("value"));
-        SingleArgWithImplicit bean = mapper.readValue(aposToQuotes("{'x':1,'y':2}"),
-                SingleArgWithImplicit.class);
-        XY v = bean.getFoobar();
-        assertNotNull(v);
-        assertEquals(1, v.x);
-        assertEquals(2, v.y);
-    }
 }
 
