diff --git a/src/main/java/com/fasterxml/jackson/databind/ser/std/DateTimeSerializerBase.java b/src/main/java/com/fasterxml/jackson/databind/ser/std/DateTimeSerializerBase.java
index d96f1f201..d590d3c86 100644
--- a/src/main/java/com/fasterxml/jackson/databind/ser/std/DateTimeSerializerBase.java
+++ b/src/main/java/com/fasterxml/jackson/databind/ser/std/DateTimeSerializerBase.java
@@ -86,10 +86,10 @@ public abstract class DateTimeSerializerBase<T>
 
     @Override
     public boolean isEmpty(SerializerProvider serializers, T value) {
         // 09-Mar-2017, tatu: as per [databind#1550] timestamp 0 is NOT "empty"; but
         //   with versions up to 2.8.x this was the case. Fixed for 2.9.
 //        return _timestamp(value) == 0L;
-        return false;
+        return _timestamp(value) == 0L;
     }
 
     protected abstract long _timestamp(T value);
diff --git a/src/main/java/com/fasterxml/jackson/databind/util/BeanUtil.java b/src/main/java/com/fasterxml/jackson/databind/util/BeanUtil.java
index 6127a8a81..9a7e98b85 100644
--- a/src/main/java/com/fasterxml/jackson/databind/util/BeanUtil.java
+++ b/src/main/java/com/fasterxml/jackson/databind/util/BeanUtil.java
@@ -1,9 +1,6 @@
 package com.fasterxml.jackson.databind.util;
 
-import java.util.Calendar;
-import java.util.Date;
-import java.util.GregorianCalendar;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.databind.JavaType;
 import com.fasterxml.jackson.databind.introspect.AnnotatedMethod;
@@ -145,16 +141,8 @@ public class BeanUtil
         if (cls == String.class) {
             return "";
         }
         // 09-Mar-2016, tatu: Not sure how far this path we want to go but for now
         //   let's add `java.util.Date` and `java.util.Calendar`, as per [databind#1550]
-        if (type.isTypeOrSubTypeOf(Date.class)) {
-            return new Date(0L);
-        }
-        if (type.isTypeOrSubTypeOf(Calendar.class)) {
-            Calendar c = new GregorianCalendar();
-            c.setTimeInMillis(0L);
-            return c;
-        }
         return null;
     }
 
