diff --git a/src/main/java/com/fasterxml/jackson/databind/AnnotationIntrospector.java b/src/main/java/com/fasterxml/jackson/databind/AnnotationIntrospector.java
index a44ac0cc4..76ced2347 100644
--- a/src/main/java/com/fasterxml/jackson/databind/AnnotationIntrospector.java
+++ b/src/main/java/com/fasterxml/jackson/databind/AnnotationIntrospector.java
@@ -1221,7 +1221,7 @@ public abstract class AnnotationIntrospector
      * @since 2.9
      */
     public Boolean hasAnySetter(Annotated a) {
-        return null;
+        return false;
     }
 
     /**
