diff --git a/src/test/java/com/fasterxml/jackson/databind/ser/KeySerializers1679Test.java b/src/test/java/com/fasterxml/jackson/failing/KeySerializers1679Test.java
similarity index 82%
rename from src/test/java/com/fasterxml/jackson/databind/ser/KeySerializers1679Test.java
rename to src/test/java/com/fasterxml/jackson/failing/KeySerializers1679Test.java
index ea7e60000..095be8595 100644
--- a/src/test/java/com/fasterxml/jackson/databind/ser/KeySerializers1679Test.java
+++ b/src/test/java/com/fasterxml/jackson/failing/KeySerializers1679Test.java
@@ -1,4 +1,4 @@
-package com.fasterxml.jackson.databind.ser;
+package com.fasterxml.jackson.failing;
 
 import java.util.*;
 
@@ -18,8 +18,8 @@ public class KeySerializers1679Test extends BaseMapTest
     public void testRecursion1679() throws Exception
     {
         Map<Object, Object> objectMap = new HashMap<Object, Object>();
-        objectMap.put(new Object(), "foo");
+        objectMap.put(new Object(), new Object());
         String json = MAPPER.writeValueAsString(objectMap);
-        assertNotNull(json);
+        assertEquals("{}", json);
     }
 }
