diff --git a/src/test/java/com/fasterxml/jackson/databind/util/TestTokenBuffer.java b/src/test/java/com/fasterxml/jackson/databind/util/TestTokenBuffer.java
index 5f77017df..58b914a46 100644
--- a/src/test/java/com/fasterxml/jackson/databind/util/TestTokenBuffer.java
+++ b/src/test/java/com/fasterxml/jackson/databind/util/TestTokenBuffer.java
@@ -15,10 +15,6 @@ public class TestTokenBuffer extends BaseMapTest
 {
     private final ObjectMapper MAPPER = objectMapper();
 
-    static class Base1730 { }
-
-    static class Sub1730 extends Base1730 { }
-
     /*
     /**********************************************************
     /* Basic TokenBuffer tests
@@ -577,7 +573,7 @@ public class TestTokenBuffer extends BaseMapTest
         buf2.close();
         buf3.close();
         buf4.close();
-    }
+    }    
 
     // [databind#743]
     public void testRawValues() throws Exception
@@ -596,20 +592,4 @@ public class TestTokenBuffer extends BaseMapTest
         // then verify it would be serialized just fine
         assertEquals(RAW, MAPPER.writeValueAsString(buf));
     }
-
-    // [databind#1730]
-    public void testEmbeddedObjectCoerceCheck() throws Exception
-    {
-        TokenBuffer buf = new TokenBuffer(null, false);
-        Object inputPojo = new Sub1730();
-        buf.writeEmbeddedObject(inputPojo);
-
-        // first: raw value won't be transformed in any way:
-        JsonParser p = buf.asParser();
-        Base1730 out = MAPPER.readValue(p, Base1730.class);
-
-        assertSame(inputPojo, out);
-        p.close();
-        buf.close();
-    }
 }
