diff --git a/src/test/java/com/fasterxml/jackson/databind/exc/ExceptionDeserializationTest.java b/src/test/java/com/fasterxml/jackson/databind/exc/ExceptionDeserializationTest.java
index 536a5b428..bf3587d3b 100644
--- a/src/test/java/com/fasterxml/jackson/databind/exc/ExceptionDeserializationTest.java
+++ b/src/test/java/com/fasterxml/jackson/databind/exc/ExceptionDeserializationTest.java
@@ -170,15 +170,4 @@ public class ExceptionDeserializationTest
         ), IOException.class);
         assertNotNull(exc);
     }
-
-    // [databind#1842]:
-    public void testNullAsMessage() throws IOException
-    {
-        Exception exc = MAPPER.readValue(aposToQuotes(
-                "{'message':null, 'localizedMessage':null }"
-        ), IOException.class);
-        assertNotNull(exc);
-        assertNull(exc.getMessage());
-        assertNull(exc.getLocalizedMessage());
-    }
 }
