diff --git a/src/main/java/com/fasterxml/jackson/databind/ser/BasicSerializerFactory.java b/src/main/java/com/fasterxml/jackson/databind/ser/BasicSerializerFactory.java
index a3d422ffa..0c09ae156 100644
--- a/src/main/java/com/fasterxml/jackson/databind/ser/BasicSerializerFactory.java
+++ b/src/main/java/com/fasterxml/jackson/databind/ser/BasicSerializerFactory.java
@@ -863,10 +863,8 @@ public abstract class BasicSerializerFactory
     {
         // [databind#865]: Allow serialization "as POJO" -- note: to undo, declare
         //   serialization as `Shape.NATURAL` instead; that's JSON Object too.
-        JsonFormat.Value formatOverride = prov.getDefaultPropertyFormat(Map.Entry.class);
-        JsonFormat.Value formatFromAnnotation = beanDesc.findExpectedFormat(null);
-        JsonFormat.Value format = JsonFormat.Value.merge(formatFromAnnotation, formatOverride);
-        if (format.getShape() == JsonFormat.Shape.OBJECT) {
+        JsonFormat.Value format = beanDesc.findExpectedFormat(null);
+        if (format != null && format.getShape() == JsonFormat.Shape.OBJECT) {
             return null;
         }
         MapEntrySerializer ser = new MapEntrySerializer(valueType, keyType, valueType,
