diff --git a/src/test/java/com/fasterxml/jackson/databind/ser/TestIterable.java b/src/test/java/com/fasterxml/jackson/databind/ser/TestIterable.java
index 2d49d3a1f..aed6a0f0a 100644
--- a/src/test/java/com/fasterxml/jackson/databind/ser/TestIterable.java
+++ b/src/test/java/com/fasterxml/jackson/databind/ser/TestIterable.java
@@ -143,13 +143,6 @@ public class TestIterable extends BaseMapTest
     {
         assertEquals("{\"values\":[\"itValue\"]}",
                 STATIC_MAPPER.writeValueAsString(new BeanWithIterator()));
-
-        // [databind#1977]
-        ArrayList<Number> numbersList = new ArrayList<>();
-        numbersList.add(1);
-        numbersList.add(0.25);
-        String json = MAPPER.writeValueAsString(numbersList.iterator());
-        assertEquals("[1,0.25]", json);
     }
 
     // [databind#358]
