diff --git a/src/main/java/com/fasterxml/jackson/databind/deser/impl/CreatorCollector.java b/src/main/java/com/fasterxml/jackson/databind/deser/impl/CreatorCollector.java
index ae9dede3f..2ea0894bc 100644
--- a/src/main/java/com/fasterxml/jackson/databind/deser/impl/CreatorCollector.java
+++ b/src/main/java/com/fasterxml/jackson/databind/deser/impl/CreatorCollector.java
@@ -184,8 +184,8 @@ public class CreatorCollector {
                     Integer old = names.put(name, Integer.valueOf(i));
                     if (old != null) {
                         throw new IllegalArgumentException(String.format(
-                                "Duplicate creator property \"%s\" (index %s vs %d) for type %s ",
-                                name, old, i, ClassUtil.nameOf(_beanDesc.getBeanClass())));
+                                "Duplicate creator property \"%s\" (index %s vs %d)",
+                                name, old, i));
                     }
                 }
             }
