diff --git a/src/test/java/com/fasterxml/jackson/databind/module/SimpleModuleTest.java b/src/test/java/com/fasterxml/jackson/databind/module/SimpleModuleTest.java
index da9e7dc7e..40a189e69 100644
--- a/src/test/java/com/fasterxml/jackson/databind/module/SimpleModuleTest.java
+++ b/src/test/java/com/fasterxml/jackson/databind/module/SimpleModuleTest.java
@@ -319,7 +319,6 @@ public class SimpleModuleTest extends BaseMapTest
         AnotherSimpleModule mod2 = new AnotherSimpleModule("test2", Version.unknownVersion());
 
         ObjectMapper mapper = new ObjectMapper();
-
         mapper.registerModule(mod1);
         mapper.registerModule(mod2);
 
@@ -327,10 +326,6 @@ public class SimpleModuleTest extends BaseMapTest
         assertEquals(2, registeredModuleIds.size());
         assertTrue(registeredModuleIds.contains(mod1.getTypeId()));
         assertTrue(registeredModuleIds.contains(mod2.getTypeId()));
-
-        // 01-Jul-2019, [databind#2374]: verify empty list is fine
-        mapper = new ObjectMapper();
-        assertEquals(0, mapper.getRegisteredModuleIds().size());
     }
 
     /*
