diff --git a/src/test/java/com/fasterxml/jackson/databind/deser/jdk/EnumMapDeserializationTest.java b/src/test/java/com/fasterxml/jackson/databind/deser/jdk/EnumMapDeserializationTest.java
index d458c0670..8cdf9afbf 100644
--- a/src/test/java/com/fasterxml/jackson/databind/deser/jdk/EnumMapDeserializationTest.java
+++ b/src/test/java/com/fasterxml/jackson/databind/deser/jdk/EnumMapDeserializationTest.java
@@ -1,7 +1,6 @@
 package com.fasterxml.jackson.databind.deser.jdk;
 
 import java.util.EnumMap;
-import java.util.LinkedHashMap;
 import java.util.Map;
 
 import com.fasterxml.jackson.annotation.*;
@@ -78,22 +77,6 @@ public class EnumMapDeserializationTest extends BaseMapTest
         }
     }
 
-    // [databind#2457]
-    enum MyEnum2457 {
-        A,
-        B() {
-            // just to ensure subclass construction
-            @Override
-            public void foo() { }
-        };
-
-        // needed to force subclassing
-        public void foo() { }
-
-        @Override
-        public String toString() { return name() + " as string"; }
-    }
-
     /*
     /**********************************************************
     /* Test methods, basic
@@ -241,21 +224,4 @@ public class EnumMapDeserializationTest extends BaseMapTest
         assertEquals(1, value2.size());
         assertEquals("value", value2.get(null));
     }
-
-    // [databind#2457]
-    public void testCustomEnumAsRootMapKey() throws Exception
-    {
-        final ObjectMapper mapper = newJsonMapper();
-        final Map<MyEnum2457, String> map = new LinkedHashMap<>();
-        map.put(MyEnum2457.A, "1");
-        map.put(MyEnum2457.B, "2");
-        assertEquals(aposToQuotes("{'A':'1','B':'2'}"),
-                mapper.writeValueAsString(map));
-
-        // But should be able to override
-        assertEquals(aposToQuotes("{'"+MyEnum2457.A.toString()+"':'1','"+MyEnum2457.B.toString()+"':'2'}"),
-                mapper.writer()
-                    .with(SerializationFeature.WRITE_ENUMS_USING_TO_STRING)
-                    .writeValueAsString(map));
-    }
 }
