diff --git a/protobuf/src/test/java/com/fasterxml/jackson/dataformat/protobuf/failing/WriteLongString94Test.java b/protobuf/src/test/java/com/fasterxml/jackson/dataformat/protobuf/failing/WriteLongString94Test.java
index 28ccf2d0..37b543c4 100644
--- a/protobuf/src/test/java/com/fasterxml/jackson/dataformat/protobuf/failing/WriteLongString94Test.java
+++ b/protobuf/src/test/java/com/fasterxml/jackson/dataformat/protobuf/failing/WriteLongString94Test.java
@@ -31,7 +31,7 @@ public class WriteLongString94Test extends ProtobufTestBase
         ProtobufSchema schema = MAPPER.generateSchemaFor(p.getClass());
         byte[] proto = MAPPER.writer(schema)
                 .writeValueAsBytes(p);
-        assertEquals(2 * (7995 + 3), proto.length);
+        assertEquals(13, proto.length);
 
         TwoStrings result = MAPPER.readerFor(p.getClass())
                 .with(schema)
