diff --git a/src/main/java/com/jcabi/email/stamp/StBCC.java b/src/main/java/com/jcabi/email/stamp/StBCC.java
index b4f5da0..cc68f3b 100644
--- a/src/main/java/com/jcabi/email/stamp/StBCC.java
+++ b/src/main/java/com/jcabi/email/stamp/StBCC.java
@@ -74,7 +74,7 @@ public final class StBCC implements Stamp {
      */
     public StBCC(final String name, final String addr) {
         try {
-            this.email = new InternetAddress(addr, name, "UTF-8").toString();
+            this.email = new InternetAddress(name, addr, "UTF-8").toString();
         } catch (final UnsupportedEncodingException ex) {
             throw new IllegalStateException(ex);
         }
diff --git a/src/main/java/com/jcabi/email/stamp/StCC.java b/src/main/java/com/jcabi/email/stamp/StCC.java
index c111d6e..de35928 100644
--- a/src/main/java/com/jcabi/email/stamp/StCC.java
+++ b/src/main/java/com/jcabi/email/stamp/StCC.java
@@ -74,7 +74,7 @@ public final class StCC implements Stamp {
      */
     public StCC(final String name, final String addr) {
         try {
-            this.email = new InternetAddress(addr, name, "UTF-8").toString();
+            this.email = new InternetAddress(name, addr, "UTF-8").toString();
         } catch (final UnsupportedEncodingException ex) {
             throw new IllegalStateException(ex);
         }
diff --git a/src/main/java/com/jcabi/email/stamp/StRecipient.java b/src/main/java/com/jcabi/email/stamp/StRecipient.java
index 6ac0746..4435f1e 100644
--- a/src/main/java/com/jcabi/email/stamp/StRecipient.java
+++ b/src/main/java/com/jcabi/email/stamp/StRecipient.java
@@ -74,7 +74,7 @@ public final class StRecipient implements Stamp {
      */
     public StRecipient(final String name, final String addr) {
         try {
-            this.email = new InternetAddress(addr, name, "UTF-8").toString();
+            this.email = new InternetAddress(name, addr, "UTF-8").toString();
         } catch (final UnsupportedEncodingException ex) {
             throw new IllegalStateException(ex);
         }
diff --git a/src/main/java/com/jcabi/email/stamp/StSender.java b/src/main/java/com/jcabi/email/stamp/StSender.java
index a91baad..056dee7 100644
--- a/src/main/java/com/jcabi/email/stamp/StSender.java
+++ b/src/main/java/com/jcabi/email/stamp/StSender.java
@@ -74,7 +74,7 @@ public final class StSender implements Stamp {
      */
     public StSender(final String name, final String addr) {
         try {
-            this.email = new InternetAddress(addr, name, "UTF-8").toString();
+            this.email = new InternetAddress(name, addr, "UTF-8").toString();
         } catch (final UnsupportedEncodingException ex) {
             throw new IllegalStateException(ex);
         }
