diff --git a/src/test/java/com/jcabi/email/EnvelopeTest.java b/src/test/java/com/jcabi/email/EnvelopeTest.java
index 6ed3708..3d1b516 100644
--- a/src/test/java/com/jcabi/email/EnvelopeTest.java
+++ b/src/test/java/com/jcabi/email/EnvelopeTest.java
@@ -29,9 +29,6 @@
  */
 package com.jcabi.email;
 
-import java.util.Properties;
-import javax.mail.Session;
-import javax.mail.internet.MimeMessage;
 import org.junit.Test;
 import org.mockito.Mockito;
 
@@ -51,9 +48,6 @@ public final class EnvelopeTest {
     @Test
     public void cachesPreviousCall() throws Exception {
         final Envelope origin = Mockito.mock(Envelope.class);
-        Mockito.doReturn(
-            new MimeMessage(Session.getDefaultInstance(new Properties()))
-        ).when(origin).unwrap();
         final Envelope envelope = new Envelope.Constant(origin);
         envelope.unwrap();
         envelope.unwrap();
