diff --git a/src/main/java/com/jcabi/email/Envelope.java b/src/main/java/com/jcabi/email/Envelope.java
index e446c82..55d988c 100644
--- a/src/main/java/com/jcabi/email/Envelope.java
+++ b/src/main/java/com/jcabi/email/Envelope.java
@@ -170,7 +170,6 @@ public interface Envelope {
          */
         public Envelope.MIME with(final Stamp stamp) {
             return new Envelope.MIME(
-                this.origin,
                 this.stamps.with(stamp),
                 this.encs
             );
@@ -183,7 +182,6 @@ public interface Envelope {
          */
         public Envelope.MIME with(final Enclosure enc) {
             return new Envelope.MIME(
-                this.origin,
                 this.stamps,
                 this.encs.with(enc)
             );
