diff --git a/src/main/java/com/jcabi/email/Envelope.java b/src/main/java/com/jcabi/email/Envelope.java
index d38f41b..e446c82 100644
--- a/src/main/java/com/jcabi/email/Envelope.java
+++ b/src/main/java/com/jcabi/email/Envelope.java
@@ -94,6 +94,10 @@ public interface Envelope {
     @EqualsAndHashCode(of = "encs")
     @Loggable(Loggable.DEBUG)
     final class MIME implements Envelope {
+        /**
+         * Original envelope.
+         */
+        private final transient Envelope origin;
         /**
          * List of stamps.
          */
@@ -107,7 +111,7 @@ public interface Envelope {
          * @since 1.3
          */
         public MIME() {
-            this(new Array<Stamp>(), new Array<Enclosure>());
+            this(Envelope.EMPTY, new Array<Stamp>(), new Array<Enclosure>());
         }
         /**
          * Ctor.
@@ -115,24 +119,35 @@ public interface Envelope {
          * @since 1.5
          */
         public MIME(final Envelope env) {
+            this(env, new Array<Stamp>(), new Array<Enclosure>());
+        }
+        /**
+         * Ctor.
+         * @param stmps Stamps
+         * @param list List of enclosures
+         */
+        public MIME(final Iterable<Stamp> stmps,
+            final Iterable<Enclosure> list) {
             this(
-                Envelope.MIME.class.cast(env).stamps,
-                Envelope.MIME.class.cast(env).encs
+                Envelope.EMPTY, new Array<Stamp>(stmps),
+                new Array<Enclosure>(list)
             );
         }
         /**
          * Ctor.
+         * @param env Original envelope
          * @param stmps Stamps
          * @param list List of enclosures
          */
-        public MIME(final Iterable<Stamp> stmps,
+        public MIME(final Envelope env, final Iterable<Stamp> stmps,
             final Iterable<Enclosure> list) {
+            this.origin = env;
             this.stamps = new Array<Stamp>(stmps);
             this.encs = new Array<Enclosure>(list);
         }
         @Override
         public Message unwrap() throws IOException {
-            final Message msg = Envelope.EMPTY.unwrap();
+            final Message msg = this.origin.unwrap();
             final Multipart multi = new MimeMultipart("alternative");
             try {
                 for (final Enclosure enc : this.encs) {
@@ -155,6 +170,7 @@ public interface Envelope {
          */
         public Envelope.MIME with(final Stamp stamp) {
             return new Envelope.MIME(
+                this.origin,
                 this.stamps.with(stamp),
                 this.encs
             );
@@ -167,6 +183,7 @@ public interface Envelope {
          */
         public Envelope.MIME with(final Enclosure enc) {
             return new Envelope.MIME(
+                this.origin,
                 this.stamps,
                 this.encs.with(enc)
             );
