diff --git a/src/test/java/com/jcabi/email/EnvelopeTest.java b/src/test/java/com/jcabi/email/EnvelopeTest.java
index c069b0a..9150820 100644
--- a/src/test/java/com/jcabi/email/EnvelopeTest.java
+++ b/src/test/java/com/jcabi/email/EnvelopeTest.java
@@ -49,7 +49,6 @@ import org.mockito.Mockito;
  * @author Yegor Bugayenko (yegor@teamed.io)
  * @version $Id$
  * @since 1.4
- * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
 public final class EnvelopeTest {
 
