diff --git a/leshan-core/src/main/java/org/eclipse/leshan/tlv/TlvDecoder.java b/leshan-core/src/main/java/org/eclipse/leshan/tlv/TlvDecoder.java
index cccae075..a2c2f12f 100644
--- a/leshan-core/src/main/java/org/eclipse/leshan/tlv/TlvDecoder.java
+++ b/leshan-core/src/main/java/org/eclipse/leshan/tlv/TlvDecoder.java
@@ -191,8 +191,8 @@ public class TlvDecoder {
     public static ObjectLink decodeObjlnk(byte[] value) throws TlvException {
         ByteBuffer bff = ByteBuffer.allocate(4).order(ByteOrder.BIG_ENDIAN);
         bff.put(value);
-        int val1 = bff.getShort(0) & 0xFFFF;
-        int val2 = bff.getShort(2) & 0xFFFF;
+        int val1 = bff.getShort(0);
+        int val2 = bff.getShort(2);
         return new ObjectLink(val1, val2);
     }
 
