diff --git a/leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonDecoder.java b/leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonDecoder.java
index 1b8e9aaa..a6b42862 100644
--- a/leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonDecoder.java
+++ b/leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonDecoder.java
@@ -309,22 +309,13 @@ public class LwM2mNodeJsonDecoder {
                     multiResource = new HashMap<>();
                     multiResourceMap.put(resourcePath, multiResource);
                 }
-                JsonArrayEntry previousResInstance = multiResource.put(nodePath.getResourceInstanceId(), resourceElt);
-                if (previousResInstance != null) {
-                    throw new CodecException(
-                            "2 RESOURCE_INSTANCE nodes (%s,%s) with the same identifier %d for path %s",
-                            previousResInstance, resourceElt, nodePath.getResourceInstanceId(), nodePath);
-                }
+                multiResource.put(nodePath.getResourceInstanceId(), resourceElt);
             } else if (nodePath.isResource()) {
                 // Single resource
                 Type expectedType = getResourceType(nodePath, model, resourceElt);
                 LwM2mResource res = LwM2mSingleResource.newResource(nodePath.getResourceId(),
                         parseJsonValue(resourceElt.getResourceValue(), expectedType, nodePath), expectedType);
-                LwM2mResource previousRes = lwM2mResourceMap.put(nodePath.getResourceId(), res);
-                if (previousRes != null) {
-                    throw new CodecException("2 RESOURCE nodes (%s,%s) with the same identifier %d for path %s",
-                            previousRes, res, res.getId(), nodePath);
-                }
+                lwM2mResourceMap.put(nodePath.getResourceId(), res);
             } else {
                 throw new CodecException(
                         "Invalid path [%s] for resource, it should be a resource or a resource instance path",
@@ -347,11 +338,7 @@ public class LwM2mNodeJsonDecoder {
                 }
                 LwM2mResource resource = LwM2mMultipleResource.newResource(resourcePath.getResourceId(), values,
                         expectedType);
-                LwM2mResource previousRes = lwM2mResourceMap.put(resourcePath.getResourceId(), resource);
-                if (previousRes != null) {
-                    throw new CodecException("2 RESOURCE nodes (%s,%s) with the same identifier %d for path %s",
-                            previousRes, resource, resource.getId(), resourcePath);
-                }
+                lwM2mResourceMap.put(resourcePath.getResourceId(), resource);
             }
         }
 
