diff --git a/leshan-core/src/main/java/org/eclipse/leshan/Link.java b/leshan-core/src/main/java/org/eclipse/leshan/Link.java
index ef633ecc..611bcbfc 100644
--- a/leshan-core/src/main/java/org/eclipse/leshan/Link.java
+++ b/leshan-core/src/main/java/org/eclipse/leshan/Link.java
@@ -159,7 +159,7 @@ public class Link implements Serializable {
         return linksResult;
     }
 
-    private static final String TRAILER = ",";
+    private static final String TRAILER = ", ";
 
     /***
      * Serialize severals {@code Link} to {@code String} as defined in http://tools.ietf.org/html/rfc6690.
