diff --git a/src/test/java/org/apache/commons/lang3/time/DateUtilsFragmentTest.java b/src/test/java/org/apache/commons/lang3/time/DateUtilsFragmentTest.java
index de9d7c6f2..dfd2e2741 100644
--- a/src/test/java/org/apache/commons/lang3/time/DateUtilsFragmentTest.java
+++ b/src/test/java/org/apache/commons/lang3/time/DateUtilsFragmentTest.java
@@ -561,32 +561,4 @@ testResult);
                         / DateUtils.MILLIS_PER_HOUR,
                 testResult);
     }
-    
-    @Test
-    public void testDaysOfMonthWithCalendar() throws Exception {
-        final long testResult = DateUtils.getFragmentInDays(aCalendar, Calendar.MONTH);
-        assertEquals(days, testResult);
-    }
-    
-    @Test
-    public void testDaysOfMonthWithDate() throws Exception {
-        final long testResult = DateUtils.getFragmentInDays(aDate, Calendar.MONTH);
-        final Calendar cal = Calendar.getInstance();
-        cal.setTime(aDate);
-        assertEquals(cal.get(Calendar.DAY_OF_MONTH), testResult);
-    }    
-    
-    @Test
-    public void testDaysOfYearWithCalendar() throws Exception {
-        final long testResult = DateUtils.getFragmentInDays(aCalendar, Calendar.YEAR);
-        assertEquals(aCalendar.get(Calendar.DAY_OF_YEAR), testResult);
-    }
-    
-    @Test
-    public void testDaysOfYearWithDate() throws Exception {
-        final long testResult = DateUtils.getFragmentInDays(aDate, Calendar.YEAR);
-        final Calendar cal = Calendar.getInstance();
-        cal.setTime(aDate);
-        assertEquals(cal.get(Calendar.DAY_OF_YEAR), testResult);
-    }
 }
