diff --git a/src/main/java/org/apache/commons/lang3/text/WordUtils.java b/src/main/java/org/apache/commons/lang3/text/WordUtils.java
index ce21d92..6340ca4 100644
--- a/src/main/java/org/apache/commons/lang3/text/WordUtils.java
+++ b/src/main/java/org/apache/commons/lang3/text/WordUtils.java
@@ -183,15 +183,11 @@ public class WordUtils {
         int offset = 0;
         final StringBuilder wrappedLine = new StringBuilder(inputLineLength + 32);
         
-        while (offset < inputLineLength) {
+        while (inputLineLength - offset > wrapLength) {
             if (str.charAt(offset) == ' ') {
                 offset++;
                 continue;
             }

-            if(inputLineLength - offset <= wrapLength) {
-                break;
-            }
             int spaceToWrapAt = str.lastIndexOf(' ', wrapLength + offset);
 
             if (spaceToWrapAt >= offset) {
