diff --git a/src/test/java/org/apache/commons/lang3/time/DurationFormatUtilsTest.java b/src/test/java/org/apache/commons/lang3/time/DurationFormatUtilsTest.java
index 21e11dc65..ec6603fdd 100644
--- a/src/test/java/org/apache/commons/lang3/time/DurationFormatUtilsTest.java
+++ b/src/test/java/org/apache/commons/lang3/time/DurationFormatUtilsTest.java
@@ -157,11 +157,6 @@ public class DurationFormatUtilsTest {
         assertEquals("1 day 1 hour 1 minute 1 second", text);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testFormatNegativeDurationWords() throws Exception {
-        DurationFormatUtils.formatDurationWords(-5000, true, true);
-    }
-
     @Test
     public void testFormatDurationHMS() {
         long time = 0;
@@ -192,11 +187,6 @@ public class DurationFormatUtilsTest {
         assertEquals("1:02:12.789", DurationFormatUtils.formatDurationHMS(time));
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testFormatNegativeDurationHMS() throws Exception {
-        DurationFormatUtils.formatDurationHMS(-5000);
-    }
-
     @Test
     public void testFormatDurationISO() {
         assertEquals("P0Y0M0DT0H0M0.000S", DurationFormatUtils.formatDurationISO(0L));
@@ -206,11 +196,6 @@ public class DurationFormatUtilsTest {
         assertEquals("P0Y0M0DT0H1M15.321S", DurationFormatUtils.formatDurationISO(75321L));
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testFormatNegativeDurationISO() throws Exception {
-        DurationFormatUtils.formatDurationISO(-5000);
-    }
-
     @Test
     public void testFormatDuration() {
         long duration = 0;
@@ -249,11 +234,6 @@ public class DurationFormatUtilsTest {
         assertEquals("0 0 " + days, DurationFormatUtils.formatDuration(duration, "y M d"));
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testFormatNegativeDuration() throws Exception {
-        DurationFormatUtils.formatDuration(-5000, "S", true);
-    }
-
     @Test
     public void testFormatPeriodISO() {
         final TimeZone timeZone = TimeZone.getTimeZone("GMT-3");
@@ -282,11 +262,6 @@ public class DurationFormatUtilsTest {
         // assertEquals("P1Y2M3DT10H30M", text);
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testFormatPeriodISOStartGreaterEnd() throws Exception {
-        DurationFormatUtils.formatPeriodISO(5000, 2000);
-    }
-
     @Test
     public void testFormatPeriodISOMethod() {
         assertEquals("P0Y0M0DT0H0M0.000S", DurationFormatUtils.formatPeriodISO(0L, 0L));
@@ -348,11 +323,6 @@ public class DurationFormatUtilsTest {
         assertEquals("048", DurationFormatUtils.formatPeriod(time1970, time, "MMM"));
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testFormatPeriodeStartGreaterEnd() throws Exception {
-        DurationFormatUtils.formatPeriod(5000, 2500, "yy/MM");
-    }
-
     @Test
     public void testLexx() {
         // tests each constant
