diff --git a/lang_84_fix/src/main/java/org/apache/commons/lang3/time/DurationFormatUtils.java b/lang_84_buggy/src/main/java/org/apache/commons/lang3/time/DurationFormatUtils.java
index 554cef5..e9fd007 100644
--- a/lang_84_fix/src/main/java/org/apache/commons/lang3/time/DurationFormatUtils.java
+++ b/lang_84_buggy/src/main/java/org/apache/commons/lang3/time/DurationFormatUtils.java
@@ -72,14 +72,15 @@ public class DurationFormatUtils {

     public static String formatDurationHMS(final long durationMillis) {
-        return formatDuration(durationMillis, "HH:mm:ss.SSS");
+        return formatDuration(durationMillis, "H:mm:ss.SSS");
     }
 
     /**
