diff --git a/src/test/java/org/apache/commons/scxml/io/SCXMLSerializerTest.java b/src/test/java/org/apache/commons/scxml/io/SCXMLSerializerTest.java
index 19a93e15..216c5bed 100644
--- a/src/test/java/org/apache/commons/scxml/io/SCXMLSerializerTest.java
+++ b/src/test/java/org/apache/commons/scxml/io/SCXMLSerializerTest.java
@@ -35,7 +35,6 @@ import org.apache.commons.scxml.model.OnExit;
 import org.apache.commons.scxml.model.SCXML;
 import org.apache.commons.scxml.model.Send;
 import org.apache.commons.scxml.model.State;
-import org.apache.commons.scxml.model.Transition;
 import org.apache.commons.scxml.model.TransitionTarget;
 import org.apache.commons.scxml.model.Var;
 
@@ -288,18 +287,4 @@ public class SCXMLSerializerTest extends TestCase {
         assertEquals(actualValue, returnValue.toString());
     }
 
-    public void testSerializeTransitionEscapeXML() {
-        Transition t = new Transition();
-
-        // note: the '<' char has to be escaped to "&lt;" to create valid XML
-        t.setCond("i < 3");
-
-        String actualValue = "<transition cond=\"i &lt; 3\">\n</transition>\n";
-
-        StringBuffer returnValue = new StringBuffer();
-        SCXMLSerializer.serializeTransition(returnValue, t, "");
-
-        assertEquals(actualValue, returnValue.toString());
-    }
-
 }
