diff --git a/src/test/java/org/apache/commons/scxml/model/StateTest.java b/src/test/java/org/apache/commons/scxml/model/StateTest.java
index a5271d8c..60032b04 100644
--- a/src/test/java/org/apache/commons/scxml/model/StateTest.java
+++ b/src/test/java/org/apache/commons/scxml/model/StateTest.java
@@ -16,12 +16,8 @@
  */
 package org.apache.commons.scxml.model;
 
-import java.net.URL;
 import java.util.List;
 
-import org.apache.commons.scxml.SCXMLExecutor;
-import org.apache.commons.scxml.SCXMLTestHelper;
-
 import junit.framework.Test;
 import junit.framework.TestCase;
 import junit.framework.TestSuite;
@@ -40,23 +36,13 @@ public class StateTest extends TestCase {
         String[] testCaseName = { StateTest.class.getName()};
         junit.textui.TestRunner.main(testCaseName);
     }
-
-    // Test data
+    
     private State state;
-    private URL state01;
-    private SCXMLExecutor exec;
-
+    
     public void setUp() {
         state = new State();
-        state01 = this.getClass().getClassLoader().
-            getResource("org/apache/commons/scxml/model/state-01.xml");
-    }
-
-    public void tearDown() {
-        state01 = null;
-        exec = null;
     }
-
+    
     public void testGetTransitionsListNull() {
         assertNull(state.getTransitionsList("event"));
     }
@@ -169,13 +155,5 @@ public class StateTest extends TestCase {
         
         assertFalse(state.isRegion());
     }
-
-    public void testInitialAttribute() {
-        SCXML scxml = SCXMLTestHelper.parse(state01);
-        assertNotNull(scxml);
-        exec = SCXMLTestHelper.getExecutor(scxml);
-        assertNotNull(exec);
-        assertEquals("s11", ((State) exec.getCurrentStatus().getStates().iterator().next()).getId());
-    }
-
+    
 }
diff --git a/src/test/java/org/apache/commons/scxml/model/state-01.xml b/src/test/java/org/apache/commons/scxml/model/state-01.xml
deleted file mode 100644
index 22a9018f..00000000
--- a/src/test/java/org/apache/commons/scxml/model/state-01.xml
+++ /dev/null
@@ -1,24 +0,0 @@
-<?xml version="1.0"?>
-<!--
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
--->
-<scxml xmlns="http://www.w3.org/2005/07/scxml" version="1.0" initial="s1">
-
-    <state id="s1" initial="s11">
-        <final id="s11"/>
-    </state>
-
-</scxml>
