diff --git a/src/main/java/org/apache/commons/scxml2/io/SCXMLReader.java b/src/main/java/org/apache/commons/scxml2/io/SCXMLReader.java
index b5b3329e..3ad532a5 100644
--- a/src/main/java/org/apache/commons/scxml2/io/SCXMLReader.java
+++ b/src/main/java/org/apache/commons/scxml2/io/SCXMLReader.java
@@ -1097,15 +1097,16 @@ public final class SCXMLReader {
         final String expr = readAV(reader, ATTR_EXPR);
         final String src = readAV(reader, ATTR_SRC);
 
+        if (expr != null && src != null) {
+          LogFactory.getLog(SCXMLReader.class).error(
+              "Found src and expr attributes for data node '" + datum.getId() + "', which is not valid SCXML.");
+          throw new ModelException();
+        }
         if (expr != null) {
-            if (src != null) {
-                reportConflictingAttribute(reader, configuration, ELEM_DATA, ATTR_EXPR, ATTR_SRC);
-            }
             datum.setExpr(expr);
-        } else {
+        }
             if (src != null) {
                 datum.setSrc(src);
-            }
         }
 
         readNamespaces(configuration, datum);
