diff --git a/src/test/java/org/apache/commons/scxml/TieBreakerTest.java b/src/test/java/org/apache/commons/scxml/TieBreakerTest.java
index d1b9ca20..766ff147 100644
--- a/src/test/java/org/apache/commons/scxml/TieBreakerTest.java
+++ b/src/test/java/org/apache/commons/scxml/TieBreakerTest.java
@@ -53,7 +53,7 @@ public class TieBreakerTest extends TestCase {
     }
 
     // Test data
-    private URL tiebreaker01, tiebreaker02, tiebreaker03, tiebreaker04;
+    private URL tiebreaker01, tiebreaker02, tiebreaker03;
     private SCXMLExecutor exec;
 
     /**
@@ -66,15 +66,13 @@ public class TieBreakerTest extends TestCase {
             getResource("org/apache/commons/scxml/tie-breaker-02.xml");
         tiebreaker03 = this.getClass().getClassLoader().
             getResource("org/apache/commons/scxml/tie-breaker-03.xml");
-        tiebreaker04 = this.getClass().getClassLoader().
-            getResource("org/apache/commons/scxml/tie-breaker-04.xml");
     }
 
     /**
      * Tear down instance variables required by this test case.
      */
     public void tearDown() {
-        tiebreaker01 = tiebreaker02 = tiebreaker03 = tiebreaker04 = null;
+        tiebreaker01 = tiebreaker02 = tiebreaker03 = null;
     }
 
     /**
@@ -119,15 +117,6 @@ public class TieBreakerTest extends TestCase {
             next()).getId());
     }
 
-    public void testTieBreaker04() {
-        exec = SCXMLTestHelper.getExecutor(tiebreaker04);
-        assertNotNull(exec);
-        Set currentStates = SCXMLTestHelper.fireEvent(exec, "event_2");
-        assertEquals(1, currentStates.size());
-        currentStates = SCXMLTestHelper.fireEvent(exec, "event_1");
-        assertEquals(1, currentStates.size());
-    }
-
     public static void main(String args[]) {
         TestRunner.run(suite());
     }
diff --git a/src/test/java/org/apache/commons/scxml/tie-breaker-04.xml b/src/test/java/org/apache/commons/scxml/tie-breaker-04.xml
deleted file mode 100644
index df87e637..00000000
--- a/src/test/java/org/apache/commons/scxml/tie-breaker-04.xml
+++ /dev/null
@@ -1,36 +0,0 @@
-<?xml version="1.0"?>
-<!--
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
--->
-<scxml xmlns="http://www.w3.org/2005/07/scxml" version="1.0" initialstate="s1"> 
-     
-    <state id="s1"> 
-        <initial> 
-            <transition target="s11"/> 
-        </initial> 
-         
-        <transition event="event_1"/>
-        
-        <transition event="event_2" target="s1"/>
-         
-        <state id="s11"> 
-            <transition event="event_1"/>
-            <transition event="event_2" target="s11"/>
-        </state> 
-    </state> 
-     
-    <state id="s2" final="true" /> 
-</scxml> 
