diff --git a/src/test/java/uk/co/jemos/podam/test/dto/ArrayPojo.java b/src/test/java/uk/co/jemos/podam/test/dto/ArrayPojo.java
new file mode 100644
index 000000000..287bb3c76
--- /dev/null
+++ b/src/test/java/uk/co/jemos/podam/test/dto/ArrayPojo.java
@@ -0,0 +1,41 @@
+/**
+ *
+ */
+package uk.co.jemos.podam.test.dto;
+
+/**
+ * POJO to test when Podam sets an array.
+ *
+ * @author tedonema
+ *
+ */
+public class ArrayPojo {
+
+	private String[] myStringArray = new String[10];
+
+	/**
+	 * Default constructor.
+	 */
+	public ArrayPojo() {
+
+	}
+
+	/**
+	 * Gets the array.
+	 *
+	 * @return
+	 */
+	public String[] getMyStringArray() {
+		return myStringArray;
+	}
+
+	/**
+	 * Sets the array.
+	 *
+	 * @param myStringArray
+	 */
+	public void setMyStringArray(String[] myStringArray) {
+		this.myStringArray = myStringArray;
+	}
+
+}
diff --git a/src/test/java/uk/co/jemos/podam/test/unit/ArrayUnitTest.java b/src/test/java/uk/co/jemos/podam/test/unit/ArrayUnitTest.java
new file mode 100644
index 000000000..6878eedbc
--- /dev/null
+++ b/src/test/java/uk/co/jemos/podam/test/unit/ArrayUnitTest.java
@@ -0,0 +1,30 @@
+package uk.co.jemos.podam.test.unit;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import uk.co.jemos.podam.api.PodamFactory;
+import uk.co.jemos.podam.api.PodamFactoryImpl;
+import uk.co.jemos.podam.test.dto.ArrayPojo;
+
+public class ArrayUnitTest {
+
+	@Test
+	public void testArrayCreation() throws Exception {
+
+		PodamFactory podam = new PodamFactoryImpl();
+		ArrayPojo pojo = podam.manufacturePojo(ArrayPojo.class);
+
+		Assert.assertNotNull(pojo);
+
+		String[] array = pojo.getMyStringArray();
+		Assert.assertTrue("The array should not be empty", array.length > 0);
+		for (String string : array) {
+			Assert.assertTrue(
+					"The length of each string in the array should be > 0",
+					string.length() > 0);
+		}
+
+	}
+
+}
diff --git a/src/test/java/uk/co/jemos/podam/test/unit/PackagePrivatePojoTest.java b/src/test/java/uk/co/jemos/podam/test/unit/PackagePrivatePojoTest.java
index b6f31c5dd..ba698cbe5 100644
--- a/src/test/java/uk/co/jemos/podam/test/unit/PackagePrivatePojoTest.java
+++ b/src/test/java/uk/co/jemos/podam/test/unit/PackagePrivatePojoTest.java
@@ -1,10 +1,10 @@
 package uk.co.jemos.podam.test.unit;
 
-import org.junit.Assert;
 import org.junit.Test;
 
 import uk.co.jemos.podam.api.PodamFactory;
 import uk.co.jemos.podam.api.PodamFactoryImpl;
+import uk.co.jemos.podam.exceptions.PodamMockeryException;
 
 /**
  * @author daivanov
@@ -12,11 +12,10 @@ import uk.co.jemos.podam.api.PodamFactoryImpl;
  */
 public class PackagePrivatePojoTest {
 
-	@Test
-	public void testPackagePrivateClassInstantiation(){
+	@Test(expected = PodamMockeryException.class)
+	public void testPackagePrivateClassInstantiation() {
 		PodamFactory factory = new PodamFactoryImpl();
-		PackagePrivatePojo pojo = factory.manufacturePojo(PackagePrivatePojo.class);
-		Assert.assertNotNull(pojo);
-		Assert.assertNotNull(pojo.getValue());
+		factory.manufacturePojo(PackagePrivatePojo.class);
+
 	}
 }
