diff --git a/src/test/java/mikera/arrayz/TestArrays.java b/src/test/java/mikera/arrayz/TestArrays.java
index 208af413..f97f1849 100644
--- a/src/test/java/mikera/arrayz/TestArrays.java
+++ b/src/test/java/mikera/arrayz/TestArrays.java
@@ -119,7 +119,7 @@ public class TestArrays {
 
 		a = a.exactClone();
 		INDArray c = a.clone();
-		a.set(Double.NaN);
+		a.fill(Double.NaN);
 
 		double[] arr = c.asVector().toArray();
 		assertEquals(a.elementCount(), arr.length);
@@ -142,7 +142,7 @@ public class TestArrays {
 		}
 
 		INDArray b = a.exactClone();
-		b.set(Double.NaN);
+		b.fill(Double.NaN);
 		b.setElements(data, 1, ecount);
 		assertEquals(a, b);
 	}
@@ -284,7 +284,7 @@ public class TestArrays {
 		b.clamp(-Double.MAX_VALUE, Double.MAX_VALUE);
 		assertEquals(a, b);
 
-		a.set(13.0);
+		a.fill(13.0);
 		a.clamp(14, 15);
 		assertEquals(14.0, Vectorz.minValue(a.toVector()), 0.0001);
 		a.clamp(12, 17);
