diff --git a/java/src/main/java/com/google/openlocationcode/OpenLocationCode.java b/java/src/main/java/com/google/openlocationcode/OpenLocationCode.java
index ff868cb..3bdd950 100644
--- a/java/src/main/java/com/google/openlocationcode/OpenLocationCode.java
+++ b/java/src/main/java/com/google/openlocationcode/OpenLocationCode.java
@@ -508,19 +508,21 @@ public final class OpenLocationCode {
       return false;
     }
     // There must be an even number of at most 8 characters before the separator.
-    if (separatorPosition % 2 != 0 || separatorPosition > SEPARATOR_POSITION) {
+    if (separatorPosition % 2 != 0) {
       return false;
     }
 
     // Check first two characters: only some values from the alphabet are permitted.
-    if (separatorPosition == SEPARATOR_POSITION) {
+    if (separatorPosition == 8) {
       // First latitude character can only have first 9 values.
-      if (CODE_ALPHABET.indexOf(code.charAt(0)) > 8) {
+      Integer index0 = CODE_ALPHABET.indexOf(code.charAt(0));
+      if (index0 == null || index0 > 8) {
         return false;
       }
 
       // First longitude character can only have first 18 values.
-      if (CODE_ALPHABET.indexOf(code.charAt(1)) > 17) {
+      Integer index1 = CODE_ALPHABET.indexOf(code.charAt(1));
+      if (index1 == null || index1 > 17) {
         return false;
       }
     }
