diff --git a/core/src/main/java/org/apache/metamodel/util/WildcardPattern.java b/core/src/main/java/org/apache/metamodel/util/WildcardPattern.java
index 715e0d2a..648c2cac 100644
--- a/core/src/main/java/org/apache/metamodel/util/WildcardPattern.java
+++ b/core/src/main/java/org/apache/metamodel/util/WildcardPattern.java
@@ -32,20 +32,14 @@ import org.apache.metamodel.query.FilterItem;
 public final class WildcardPattern implements Serializable {
 
 	private static final long serialVersionUID = 857462137797209624L;
-	private final boolean _startsWithDelim;
-	private final boolean _endsWithDelim;
 	private String _pattern;
 	private char _wildcard;
+	private boolean _endsWithDelim;
 
 	public WildcardPattern(String pattern, char wildcard) {
 		_pattern = pattern;
 		_wildcard = wildcard;
-		if(_pattern.isEmpty()){
-			_startsWithDelim = _endsWithDelim = false;
-		} else {
-			_startsWithDelim = _pattern.charAt(0) == _wildcard;
-			_endsWithDelim = _pattern.charAt(pattern.length() - 1) == _wildcard;
-		}
+		_endsWithDelim = (_pattern.charAt(pattern.length() - 1) == _wildcard);
 	}
 
 	public boolean matches(String value) {
@@ -56,10 +50,9 @@ public final class WildcardPattern implements Serializable {
 				Character.toString(_wildcard));
 		int charIndex = 0;
 		while (st.hasMoreTokens()) {
-			int oldIndex = charIndex;
 			String token = st.nextToken();
 			charIndex = value.indexOf(token, charIndex);
-			if (charIndex == -1 || !_startsWithDelim && oldIndex == 0 && charIndex != 0) {
+			if (charIndex == -1) {
 				return false;
 			}
 			charIndex = charIndex + token.length();
