diff --git a/deltaspike/core/api/src/test/java/org/apache/deltaspike/test/api/metadata/AnnotatedTypeBuilderTest.java b/deltaspike/core/api/src/test/java/org/apache/deltaspike/test/api/metadata/AnnotatedTypeBuilderTest.java
index 2398dffb..66deb57b 100644
--- a/deltaspike/core/api/src/test/java/org/apache/deltaspike/test/api/metadata/AnnotatedTypeBuilderTest.java
+++ b/deltaspike/core/api/src/test/java/org/apache/deltaspike/test/api/metadata/AnnotatedTypeBuilderTest.java
@@ -148,16 +148,4 @@ public class AnnotatedTypeBuilderTest
             }
         }
     }
-
-    @Test
-    public void buildValidAnnotationAnnotatedType()
-    {
-        final AnnotatedTypeBuilder<Small> builder = new AnnotatedTypeBuilder<Small>();
-        builder.readFromType(Small.class);
-        final AnnotatedType<Small> smallAnnotatedType = builder.create();
-
-        assertThat(smallAnnotatedType.getMethods().size(), is(1));
-        assertThat(smallAnnotatedType.getConstructors().size(), is(0));
-        assertThat(smallAnnotatedType.getFields().size(), is(0));
-    }
 }
diff --git a/deltaspike/core/api/src/test/java/org/apache/deltaspike/test/api/metadata/Small.java b/deltaspike/core/api/src/test/java/org/apache/deltaspike/test/api/metadata/Small.java
deleted file mode 100644
index eace316a..00000000
--- a/deltaspike/core/api/src/test/java/org/apache/deltaspike/test/api/metadata/Small.java
+++ /dev/null
@@ -1,34 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements. See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership. The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License. You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied. See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-package org.apache.deltaspike.test.api.metadata;
-
-import javax.inject.Qualifier;
-import java.lang.annotation.Documented;
-import java.lang.annotation.Retention;
-import java.lang.annotation.RetentionPolicy;
-
-
-@Qualifier
-@Documented
-@Retention(value = RetentionPolicy.RUNTIME)
-public @interface Small
-{
-    String value() default "";
-}
