diff --git a/deltaspike/core/api/src/main/java/org/apache/deltaspike/core/util/metadata/builder/AnnotatedConstructorImpl.java b/deltaspike/core/api/src/main/java/org/apache/deltaspike/core/util/metadata/builder/AnnotatedConstructorImpl.java
index 310400c7..2683afd6 100644
--- a/deltaspike/core/api/src/main/java/org/apache/deltaspike/core/util/metadata/builder/AnnotatedConstructorImpl.java
+++ b/deltaspike/core/api/src/main/java/org/apache/deltaspike/core/util/metadata/builder/AnnotatedConstructorImpl.java
@@ -51,9 +51,9 @@ class AnnotatedConstructorImpl<X> extends AnnotatedCallableImpl<X, Constructor<X
         // length, this is a hack to fix this.
         // TODO: investigate this behavior further, on different JVM's and
         // compilers
-        if (genericTypes.length < constructor.getParameterTypes().length)
+        if (genericTypes.length + 1 == constructor.getParameterTypes().length)
         {
-            genericTypes = new Type[constructor.getParameterTypes().length];
+            genericTypes = new Type[constructor.getGenericParameterTypes().length + 1];
             genericTypes[0] = constructor.getParameterTypes()[0];
             for (int i = 0; i < constructor.getGenericParameterTypes().length; ++i)
             {
