diff --git a/web/src/main/java/org/apache/shiro/web/session/mgt/DefaultWebSessionManager.java b/web/src/main/java/org/apache/shiro/web/session/mgt/DefaultWebSessionManager.java
index 3ff80bad..d58a1932 100644
--- a/web/src/main/java/org/apache/shiro/web/session/mgt/DefaultWebSessionManager.java
+++ b/web/src/main/java/org/apache/shiro/web/session/mgt/DefaultWebSessionManager.java
@@ -115,11 +115,14 @@ public class DefaultWebSessionManager extends DefaultSessionManager implements W
                     ShiroHttpServletRequest.COOKIE_SESSION_ID_SOURCE);
         } else {
             //not in a cookie, or cookie is disabled - try the request params as a fallback (i.e. URL rewriting):
-            String name = getSessionIdName();
-            id = request.getParameter(name);
+            id = getUriPathSegmentParamValue(request, ShiroHttpSession.DEFAULT_SESSION_ID_NAME);
             if (id == null) {
                 //try lowercase:
+                String name = getSessionIdName();
+                id = request.getParameter(name);
+                if (id == null) {
                 id = request.getParameter(name.toLowerCase());
+                }
             }
             if (id != null) {
                 request.setAttribute(ShiroHttpServletRequest.REFERENCED_SESSION_ID_SOURCE,
@@ -134,6 +137,37 @@ public class DefaultWebSessionManager extends DefaultSessionManager implements W
         }
         return id;
     }
+    private String getUriPathSegmentParamValue(ServletRequest servletRequest, String paramName) {
+        if (!(servletRequest instanceof HttpServletRequest)) {
+            return null;
+        }
+        HttpServletRequest request = (HttpServletRequest)servletRequest;
+        String uri = request.getRequestURI();
+        if (uri == null) {
+            return null;
+        }
+        int queryStartIndex = uri.indexOf('?');
+        if (queryStartIndex >= 0) {
+            uri = uri.substring(0, queryStartIndex);
+        }
+        int index = uri.indexOf(';');
+        if (index < 0) {
+            return null;
+        }
+        final String TOKEN = paramName + "=";
+        uri = uri.substring(index+1);
+        index = uri.lastIndexOf(TOKEN);
+        if (index < 0) {
+            //no segment param:
+            return null;
+        }
+        uri = uri.substring(index + TOKEN.length());
+        index = uri.indexOf(';');
+        if(index >= 0) {
+            uri = uri.substring(0, index);
+        }
+        return uri;
+    }
 
     //since 1.2.1
     private String getSessionIdName() {
