diff --git a/web/src/main/java/org/apache/shiro/web/session/HttpServletSession.java b/web/src/main/java/org/apache/shiro/web/session/HttpServletSession.java
index 99630a09..eb6497cf 100644
--- a/web/src/main/java/org/apache/shiro/web/session/HttpServletSession.java
+++ b/web/src/main/java/org/apache/shiro/web/session/HttpServletSession.java
@@ -75,7 +75,7 @@ public class HttpServletSession implements Session {
 
     public long getTimeout() throws InvalidSessionException {
         try {
-            return httpSession.getMaxInactiveInterval() * 1000L;
+            return httpSession.getMaxInactiveInterval() * 1000;
         } catch (Exception e) {
             throw new InvalidSessionException(e);
         }
