diff --git a/web/src/test/java/org/apache/shiro/web/mgt/DefaultWebSecurityManagerTest.java b/web/src/test/java/org/apache/shiro/web/mgt/DefaultWebSecurityManagerTest.java
index 21c440bf..e6885436 100644
--- a/web/src/test/java/org/apache/shiro/web/mgt/DefaultWebSecurityManagerTest.java
+++ b/web/src/test/java/org/apache/shiro/web/mgt/DefaultWebSecurityManagerTest.java
@@ -24,10 +24,7 @@ import org.apache.shiro.realm.text.IniRealm;
 import org.apache.shiro.session.ExpiredSessionException;
 import org.apache.shiro.session.Session;
 import org.apache.shiro.session.mgt.AbstractSessionManager;
-import org.apache.shiro.subject.PrincipalCollection;
-import org.apache.shiro.subject.SimplePrincipalCollection;
 import org.apache.shiro.subject.Subject;
-import org.apache.shiro.web.config.WebIniSecurityManagerFactory;
 import org.apache.shiro.web.servlet.ShiroHttpSession;
 import org.apache.shiro.web.session.mgt.WebSessionManager;
 import org.apache.shiro.web.subject.WebSubject;
@@ -214,25 +211,4 @@ public class DefaultWebSecurityManagerTest extends AbstractWebSecurityManagerTes
         verify(mockResponse);
     }
 
-    /**
-     * Asserts fix for <a href="https://issues.apache.org/jira/browse/SHIRO-350">SHIRO-350</a>.
-     */
-    @Test
-    public void testBuildNonWebSubjectWithDefaultServletContainerSessionManager() {
-
-        Ini ini = new Ini();
-        Ini.Section section = ini.addSection(IniRealm.USERS_SECTION_NAME);
-        section.put("user1", "user1");
-
-        WebIniSecurityManagerFactory factory = new WebIniSecurityManagerFactory(ini);
-
-        WebSecurityManager securityManager = (WebSecurityManager)factory.getInstance();
-
-        PrincipalCollection principals = new SimplePrincipalCollection("user1", "iniRealm");
-        Subject subject = new Subject.Builder(securityManager).principals(principals).buildSubject();
-
-        assertNotNull(subject);
-        assertEquals("user1", subject.getPrincipal());
-    }
-
 }
