diff --git a/wink-common/src/test/java/org/apache/wink/common/internal/registry/metadata/ResourceMetadataCollectorTest.java b/wink-common/src/test/java/org/apache/wink/common/internal/registry/metadata/ResourceMetadataCollectorTest.java
index a14fb897..ba1ca8d3 100644
--- a/wink-common/src/test/java/org/apache/wink/common/internal/registry/metadata/ResourceMetadataCollectorTest.java
+++ b/wink-common/src/test/java/org/apache/wink/common/internal/registry/metadata/ResourceMetadataCollectorTest.java
@@ -66,15 +66,6 @@ public class ResourceMetadataCollectorTest extends TestCase {
 
     public static class MySuperInterfaceImpl extends SuperResource implements MyInterface {
 
-    }
-    
-    @Path("abstractclass")
-    public static abstract class MyAbstractClass {
-        
-    }
-    
-    public static class MyBaseClass extends MyAbstractClass {
-        
     }
 
     /**
@@ -103,18 +94,6 @@ public class ResourceMetadataCollectorTest extends TestCase {
         assertFalse(ResourceMetadataCollector.isDynamicResource(MySuperInterfaceImpl.class));
         assertTrue(ResourceMetadataCollector.isStaticResource(MySuperInterfaceImpl.class));
         assertEquals("superclassvalue", classMetadata.getPath());
-        
-        assertFalse(ResourceMetadataCollector.isResource(MyInterface.class));
-        assertFalse(ResourceMetadataCollector.isDynamicResource(MyInterface.class));
-        assertFalse(ResourceMetadataCollector.isStaticResource(MyInterface.class));
-        
-        assertFalse(ResourceMetadataCollector.isResource(MyAbstractClass.class));
-        assertFalse(ResourceMetadataCollector.isDynamicResource(MyAbstractClass.class));
-        assertFalse(ResourceMetadataCollector.isStaticResource(MyAbstractClass.class));
-        
-        assertTrue(ResourceMetadataCollector.isResource(MyBaseClass.class));
-        assertFalse(ResourceMetadataCollector.isDynamicResource(MyBaseClass.class));
-        assertTrue(ResourceMetadataCollector.isStaticResource(MyBaseClass.class));
     }
 
     /**
