diff --git a/src/main/java/org/apache/commons/validator/routines/DomainValidator.java b/src/main/java/org/apache/commons/validator/routines/DomainValidator.java
index 1e722a94..fe2937a8 100644
--- a/src/main/java/org/apache/commons/validator/routines/DomainValidator.java
+++ b/src/main/java/org/apache/commons/validator/routines/DomainValidator.java
@@ -1428,9 +1428,8 @@ public class DomainValidator implements Serializable {
      * enum used by {@link DomainValidator#updateTLDOverride(ArrayType, String[])}
      * to determine which override array to update.
      * @since 1.5.0
-     * @since 1.5.1 made public
      */
-    public enum ArrayType {
+    enum ArrayType {
         /** Update the GENERIC_TLDS_PLUS table containing additonal generic TLDs */
         GENERIC_PLUS,
         /** Update the GENERIC_TLDS_MINUS table containing deleted generic TLDs */
