diff --git a/validator_17_fix/src/main/java/org/apache/commons/validator/routines/EmailValidator.java b/validator_17_buggy/src/main/java/org/apache/commons/validator/routines/EmailValidator.java
index e344dcc..cd0c6c2 100644
--- a/validator_17_fix/src/main/java/org/apache/commons/validator/routines/EmailValidator.java
+++ b/validator_17_buggy/src/main/java/org/apache/commons/validator/routines/EmailValidator.java
@@ -199,7 +199,7 @@ public class EmailValidator implements Serializable {
         DomainValidator domainValidator =
                 DomainValidator.getInstance(allowLocal);
         if (allowTld) {
-            return domainValidator.isValid(domain) || (!domain.startsWith(".") && domainValidator.isValidTld(domain));
+            return domainValidator.isValid(domain) || domainValidator.isValidTld(domain);
         } else {
             return domainValidator.isValid(domain);
         }
