diff --git a/src/test/java/org/apache/commons/validator/routines/EmailValidatorTest.java b/src/test/java/org/apache/commons/validator/routines/EmailValidatorTest.java
index 04a0ceed..8c75d288 100644
--- a/src/test/java/org/apache/commons/validator/routines/EmailValidatorTest.java
+++ b/src/test/java/org/apache/commons/validator/routines/EmailValidatorTest.java
@@ -367,9 +367,6 @@ public class EmailValidatorTest extends TestCase {
 
         assertTrue(validator.isValid("\"..\"@apache.org"));
 
-        // escaped quote character valid in quoted string
-        assertTrue(validator.isValid("\"john\\\"doe\"@apache.org"));
-
         assertTrue(validator.isValid("john56789.john56789.john56789.john56789.john56789.john56789.john@example.com"));
 
         assertFalse(validator.isValid("john56789.john56789.john56789.john56789.john56789.john56789.john5@example.com"));
