diff --git a/validator_2_fix/src/main/java/org/apache/commons/validator/routines/UrlValidator.java b/validator_2_buggy/src/main/java/org/apache/commons/validator/routines/UrlValidator.java
index 8860a14..312c8e8 100644
--- a/validator_2_fix/src/main/java/org/apache/commons/validator/routines/UrlValidator.java
+++ b/validator_2_buggy/src/main/java/org/apache/commons/validator/routines/UrlValidator.java
@@ -450,8 +450,7 @@ public class UrlValidator implements Serializable {
         }
 
         try {

-            URI uri = new URI(null,"localhost",path,null);
+            URI uri = new URI(null,null,path,null);
             String norm = uri.normalize().getPath();
             if (norm.startsWith("/../") // Trying to go via the parent dir 
              || norm.equals("/..")) {   // Trying to go to the parent dir
