diff --git a/validator_8_fix/src/main/java/org/apache/commons/validator/routines/UrlValidator.java b/validator_8_buggy/src/main/java/org/apache/commons/validator/routines/UrlValidator.java
index cf66d3a..9773ccf 100644
--- a/validator_8_fix/src/main/java/org/apache/commons/validator/routines/UrlValidator.java
+++ b/validator_8_buggy/src/main/java/org/apache/commons/validator/routines/UrlValidator.java
@@ -292,14 +292,9 @@ public class UrlValidator implements Serializable {
         }
 
         String authority = urlMatcher.group(PARSE_URL_AUTHORITY);
-        if ("file".equals(scheme) && "".equals(authority)) {
-        } else {
            if (!isValidAuthority(authority)) {
                return false;
             }
-        }
 
         if (!isValidPath(urlMatcher.group(PARSE_URL_PATH))) {
             return false;
