diff --git a/src/main/java/io/github/gitbucket/markedj/Renderer.java b/src/main/java/io/github/gitbucket/markedj/Renderer.java
index 85a8d93..2543379 100644
--- a/src/main/java/io/github/gitbucket/markedj/Renderer.java
+++ b/src/main/java/io/github/gitbucket/markedj/Renderer.java
@@ -89,10 +89,7 @@ public class Renderer {
             cellType = "td";
         }
 
-        if(content.length() == 0){
-            content = "&nbsp;";
-        }
 
         String align = flags.getAlign();
         if(align != null){
             return "<" + cellType + " style=\"text-align: " + align + "\">" + content + "</" + cellType + ">\n";
