diff --git a/src/test/java/io/github/gitbucket/markedj/MarkedTest.java b/src/test/java/io/github/gitbucket/markedj/MarkedTest.java
index 800cfe2..e39a212 100644
--- a/src/test/java/io/github/gitbucket/markedj/MarkedTest.java
+++ b/src/test/java/io/github/gitbucket/markedj/MarkedTest.java
@@ -156,39 +156,6 @@ public class MarkedTest {
         assertEquals(loadResourceAsString("nested_content_of_list.html"), result);
     }
 
-    @Test
-    public void testEmptyTableCell() throws Exception {
-        String result = Marked.marked(
-                "|ID|name|note|\n" +
-                "|-|-|-|\n" +
-                "|1|foo|This is foo|\n" +
-                "|2|bar||");
-
-
-        assertEquals(
-                "<table>\n" +
-                        "<thead>\n" +
-                        "<tr>\n" +
-                        "<th>ID</th>\n" +
-                        "<th>name</th>\n" +
-                        "<th>note</th>\n" +
-                        "</tr>\n" +
-                        "</thead>\n" +
-                        "<tbody>\n" +
-                        "<tr>\n" +
-                        "<td>1</td>\n" +
-                        "<td>foo</td>\n" +
-                        "<td>This is foo</td>\n" +
-                        "</tr>\n" +
-                        "<tr>\n" +
-                        "<td>2</td>\n" +
-                        "<td>bar</td>\n" +
-                        "<td></td>\n" +
-                        "</tr>\n" +
-                        "</tbody>\n" +
-                        "</table>\n", result);
-    }
-
     private String loadResourceAsString(String path) throws IOException {
         InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(path);
         try {
