diff --git a/src/test/java/io/github/gitbucket/markedj/MarkedTest.java b/src/test/java/io/github/gitbucket/markedj/MarkedTest.java
index f1c20ca..4b25de7 100644
--- a/src/test/java/io/github/gitbucket/markedj/MarkedTest.java
+++ b/src/test/java/io/github/gitbucket/markedj/MarkedTest.java
@@ -5,6 +5,8 @@ import org.junit.Test;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 
 import static org.junit.Assert.*;
 
@@ -69,17 +71,9 @@ public class MarkedTest {
     @Test
     public void testStackoverFlow() throws Exception {
         // Make sure StackOverflowError does not occur by em regular expression
-        Marked.marked(loadResourceAsString("stackoverflow.txt"), new Options());
+        Marked.marked(loadResourceAsString("fail.txt"), new Options());
     }
 
-    @Test
-    public void testNptable() throws Exception {
-        String result = Marked.marked(loadResourceAsString("nptable.md"), new Options());
-        String expect = loadResourceAsString("nptable.html");
-        assertEquals(expect, result);
-    }
-
-
     private String loadResourceAsString(String path) throws IOException {
         InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(path);
         try {
