diff --git a/src/test/java/com/jcabi/matchers/RegexMatchersTest.java b/src/test/java/com/jcabi/matchers/RegexMatchersTest.java
index c419deb..01d3598 100644
--- a/src/test/java/com/jcabi/matchers/RegexMatchersTest.java
+++ b/src/test/java/com/jcabi/matchers/RegexMatchersTest.java
@@ -46,6 +46,7 @@ public final class RegexMatchersTest {
      * RegexMatchers should be able to match a string against a given pattern.
      */
     @Test
+    @Ignore
     public void matchesStringToPattern() {
         MatcherAssert.assertThat(
             "abc123",
