diff --git a/src/main/java/com/jcabi/matchers/RegexMatchers.java b/src/main/java/com/jcabi/matchers/RegexMatchers.java
index 2c2be96..51db33d 100644
--- a/src/main/java/com/jcabi/matchers/RegexMatchers.java
+++ b/src/main/java/com/jcabi/matchers/RegexMatchers.java
@@ -89,7 +89,9 @@ public final class RegexMatchers {
      *  contains ALL of the given patterns.
      */
     public static Matcher<String> containsPattern(final String pattern) {
-        return new RegexContainingPatternMatcher(pattern);
+        throw new UnsupportedOperationException(
+            "containsPattern not yet implemented"
+        );
     }
 
 }
