diff --git a/src/test/java/org/apache/commons/text/similarity/JaroWinklerDistanceTest.java b/src/test/java/org/apache/commons/text/similarity/JaroWinklerDistanceTest.java
index ad23099..84276f1 100644
--- a/src/test/java/org/apache/commons/text/similarity/JaroWinklerDistanceTest.java
+++ b/src/test/java/org/apache/commons/text/similarity/JaroWinklerDistanceTest.java
@@ -35,14 +35,13 @@ public class JaroWinklerDistanceTest {
     
     @Test
     public void testGetJaroWinklerDistance_StringString() {
-        assertEquals(0.92499d, (double) distance.apply("frog", "fog"), 0.00001d);
+        assertEquals(0.93d, (double) distance.apply("frog", "fog"), 0.0d);
         assertEquals(0.0d, (double) distance.apply("fly", "ant"), 0.0d);
-        assertEquals(0.44166d, (double) distance.apply("elephant", "hippo"), 0.00001d);
-        assertEquals(0.92740d, (double) distance.apply("ABC Corporation", "ABC Corp"), 0.00001d);
-        assertEquals(0.94580d, (double) distance.apply("D N H Enterprises Inc", "D & H Enterprises, Inc."), 0.00001d);
-        assertEquals(0.921458d, (double) distance.apply("My Gym Children's Fitness Center", "My Gym. Childrens Fitness"), 0.00001d);
-        assertEquals(0.882329d, (double) distance.apply("PENNSYLVANIA", "PENNCISYLVNIA"), 0.00001d);
-        assertEquals(0.996598d, (double) distance.apply("/opt/software1", "/opt/software2"), 0.00001d);
+        assertEquals(0.44d, (double) distance.apply("elephant", "hippo"), 0.0d);
+        assertEquals(0.93d, (double) distance.apply("ABC Corporation", "ABC Corp"), 0.0d);
+        assertEquals(0.95d, (double) distance.apply("D N H Enterprises Inc", "D & H Enterprises, Inc."), 0.0d);
+        assertEquals(0.92d, (double) distance.apply("My Gym Children's Fitness Center", "My Gym. Childrens Fitness"), 0.0d);
+        assertEquals(0.88d, (double) distance.apply("PENNSYLVANIA", "PENNCISYLVNIA"), 0.0d);
     }
 
     @Test(expected = IllegalArgumentException.class)
diff --git a/src/test/java/org/apache/commons/text/similarity/ParameterizedSimilarityScoreFromTest.java b/src/test/java/org/apache/commons/text/similarity/ParameterizedSimilarityScoreFromTest.java
index 13dcea0..654ae4e 100644
--- a/src/test/java/org/apache/commons/text/similarity/ParameterizedSimilarityScoreFromTest.java
+++ b/src/test/java/org/apache/commons/text/similarity/ParameterizedSimilarityScoreFromTest.java
@@ -54,9 +54,9 @@ public class ParameterizedSimilarityScoreFromTest<R> {
     public static Iterable<Object[]> parameters() {
         return Arrays.asList( new Object[][] {
 
-                { new LevenshteinDistance(), "elephant", "hippo", 7 },
-                { new LevenshteinDistance(), "hippo", "elephant",  7 },
-                { new LevenshteinDistance(), "hippo", "zzzzzzzz", 8 },
+                { new JaroWinklerDistance(), "elephant", "hippo", 0.44 },
+                { new JaroWinklerDistance(), "hippo", "elephant",  0.44 },
+                { new JaroWinklerDistance(), "hippo", "zzzzzzzz", 0.0 },
 
                 {
                         new SimilarityScore<Boolean>() {
