diff --git a/src/main/java/org/apache/hadoop/mrunit/internal/mapreduce/AbstractMockContextWrapper.java b/src/main/java/org/apache/hadoop/mrunit/internal/mapreduce/AbstractMockContextWrapper.java
index 521e732..f0206a9 100644
--- a/src/main/java/org/apache/hadoop/mrunit/internal/mapreduce/AbstractMockContextWrapper.java
+++ b/src/main/java/org/apache/hadoop/mrunit/internal/mapreduce/AbstractMockContextWrapper.java
@@ -131,15 +131,8 @@ extends TaskInputOutputContext<KEYIN, VALUEIN, KEYOUT, VALUEOUT>> {
           return DistributedCache.getLocalCacheFiles(configuration);
         }      
       });
 
-      when(context.getNumReduceTasks()).thenAnswer(new Answer<Integer>() {
-        @Override
-        @SuppressWarnings("deprecation")
-        public Integer answer(InvocationOnMock invocation) throws Throwable {
-          return configuration.getInt("mapred.reduce.tasks", 1);
-        }
-      });
 
     } catch (IOException e) {
       throw new RuntimeException(e);
     } catch (InterruptedException e) {
