diff --git a/src/test/java/org/apache/hadoop/mrunit/internal/io/TestSerialization.java b/src/test/java/org/apache/hadoop/mrunit/internal/io/TestSerialization.java
index ac530ab..099b7c6 100644
--- a/src/test/java/org/apache/hadoop/mrunit/internal/io/TestSerialization.java
+++ b/src/test/java/org/apache/hadoop/mrunit/internal/io/TestSerialization.java
@@ -17,14 +17,13 @@
  */
 package org.apache.hadoop.mrunit.internal.io;
 
+import static org.junit.Assert.assertEquals;
+
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.io.IntWritable;
-import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mrunit.internal.io.Serialization;
 import org.junit.Test;
 
-import static org.junit.Assert.assertEquals;
-import static org.mockito.Mockito.mock;
-
 public class TestSerialization {
 
   @Test
@@ -59,12 +58,4 @@ public class TestSerialization {
     assertEquals(new Integer(2), int2);
   }
 
-  @Test(expected = IllegalStateException.class)
-  public void testCopyExceptionIfNoSerializer() {
-    final Configuration conf = new Configuration();
-    Serialization serialization = new Serialization(conf);
-    // there's no implicit serializer for Integer, so it should throw ISE
-    serialization.copy(1, 1);
-  }
-
 }
