diff --git a/src/test/java/org/apache/hadoop/mrunit/mapreduce/TestMapDriver.java b/src/test/java/org/apache/hadoop/mrunit/mapreduce/TestMapDriver.java
index 66cdc5e..4169ee0 100644
--- a/src/test/java/org/apache/hadoop/mrunit/mapreduce/TestMapDriver.java
+++ b/src/test/java/org/apache/hadoop/mrunit/mapreduce/TestMapDriver.java
@@ -26,11 +26,9 @@ import java.util.List;
 
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.io.IntWritable;
-import org.apache.hadoop.io.LongWritable;
 import org.apache.hadoop.io.NullWritable;
 import org.apache.hadoop.io.Text;
 import org.apache.hadoop.mapreduce.Mapper;
-import org.apache.hadoop.mapreduce.lib.input.FileSplit;
 import org.apache.hadoop.mapreduce.lib.map.InverseMapper;
 import org.apache.hadoop.mrunit.ExpectedSuppliedException;
 import org.apache.hadoop.mrunit.types.Pair;
@@ -245,24 +243,6 @@ public class TestMapDriver {
     }
   }
 
-  @Test
-  public void testInputSplitDetails() {
-    final MapDriver<NullWritable, NullWritable, Text, LongWritable> driver = 
-        MapDriver.newMapDriver(new InputSplitDetailMapper());
-    driver.withInput(NullWritable.get(), NullWritable.get())
-      .withOutput(new Text("somefile"), new LongWritable(0L)).runTest();
-  }
-  
-  public static class InputSplitDetailMapper
-    extends Mapper<NullWritable, NullWritable, Text, LongWritable> {
-    protected void map(NullWritable key, NullWritable value, Context context) 
-        throws IOException, InterruptedException {
-      FileSplit split = (FileSplit)context.getInputSplit();
-      context.write(new Text(split.getPath().toString()), 
-          new LongWritable(split.getLength()));
-    }
-  }
-
   @Test
   public void testNoMapper() {
     driver = MapDriver.newMapDriver();
