diff --git a/hbase-common/src/main/java/org/apache/hadoop/hbase/util/Bytes.java b/hbase-common/src/main/java/org/apache/hadoop/hbase/util/Bytes.java
index ea32d3ed61..7fa937e3d8 100644
--- a/hbase-common/src/main/java/org/apache/hadoop/hbase/util/Bytes.java
+++ b/hbase-common/src/main/java/org/apache/hadoop/hbase/util/Bytes.java
@@ -382,12 +382,17 @@ public class Bytes {
   }
 
   public static byte [] toBytesBinary(String in) {
     // this may be bigger than we need, but let's be safe.
     byte [] b = new byte[in.length()];
     int size = 0;
     for (int i = 0; i < in.length(); ++i) {
       char ch = in.charAt(i);
-      if (ch == '\\' && in.length() > i+1 && in.charAt(i+1) == 'x') {
+      if (ch == '\\') {
+        char next = in.charAt(i+1);
+        if (next != 'x') {
+          b[size++] = (byte)ch;
+          continue;
+        }
         // ok, take next 2 hex digits.
         char hd1 = in.charAt(i+2);
         char hd2 = in.charAt(i+3);
