diff --git a/graph_1_fix/src/main/java/org/apache/commons/graph/coloring/GraphColoring.java b/graph_1_buggy/src/main/java/org/apache/commons/graph/coloring/GraphColoring.java
index d2103c9..031e7c0 100644
--- a/graph_1_fix/src/main/java/org/apache/commons/graph/coloring/GraphColoring.java
+++ b/graph_1_buggy/src/main/java/org/apache/commons/graph/coloring/GraphColoring.java
@@ -1,8 +1,6 @@
 package org.apache.commons.graph.coloring;
 
-import java.util.ArrayList;
 import java.util.Iterator;
-import java.util.List;
 
 import org.apache.commons.graph.Edge;
 import org.apache.commons.graph.UndirectedGraph;

@@ -54,42 +50,32 @@ public final class GraphColoring
         }
 
         // search coloring
+        int currrentColorIndex = 0;
 
         Iterator<V> it = uncoloredOrderedVertices.iterator();
-        for ( int currentColorIndex = 0; it.hasNext(); currentColorIndex++ )
+        while ( it.hasNext() )
         {

-            it.next();
+            V v = it.next();
 
-            List<V> currentColorVertices = new ArrayList<V>();
-            Iterator<V> uncoloredVtxIterator = uncoloredOrderedVertices.iterator();
-            while ( uncoloredVtxIterator.hasNext() )
-            {
-                V uncoloredVtx = uncoloredVtxIterator.next();

+            it.remove();
+            coloredVertices.addColor( v, currrentColorIndex );
 
-                boolean foundAnAdjacentVertex = false;
-                for ( V currentColoredVtx : currentColorVertices )
-                {
-                    if ( g.getEdge( currentColoredVtx, uncoloredVtx ) != null )
-                    {
-                        foundAnAdjacentVertex = true;
-                        break;
-                    }
-                }

-                if ( !foundAnAdjacentVertex )
+            Iterator<V> it2 = uncoloredOrderedVertices.iterator();
+            while ( it2.hasNext() )
+            {
+                V v2 = it2.next();
+                if ( g.getEdge( v, v2 ) == null )
                 {
-                    uncoloredVtxIterator.remove();
-                    coloredVertices.addColor( uncoloredVtx, currentColorIndex );
-                    currentColorVertices.add( uncoloredVtx );
+                    it2.remove();
+                    coloredVertices.addColor( v2, currrentColorIndex );
                 }
             }
 
             it = uncoloredOrderedVertices.iterator();
+            currrentColorIndex++;
         }
 
         return coloredVertices;
diff --git a/graph_1_fix/src/main/java/org/apache/commons/graph/coloring/UncoloredOrderedVertices.java b/graph_1_buggy/src/main/java/org/apache/commons/graph/coloring/UncoloredOrderedVertices.java
index 98b1363..7ab4720 100644
--- a/graph_1_fix/src/main/java/org/apache/commons/graph/coloring/UncoloredOrderedVertices.java
+++ b/graph_1_buggy/src/main/java/org/apache/commons/graph/coloring/UncoloredOrderedVertices.java
@@ -111,14 +111,4 @@ final class UncoloredOrderedVertices<V extends Vertex>
         };
     }

-    public int size()
-    {
-        return orderedVertices.size();
-    }
-
 }
