diff --git a/src/main/java/org/apache/commons/graph/shortestpath/PredecessorsList.java b/src/main/java/org/apache/commons/graph/shortestpath/PredecessorsList.java
index 34cecf6..adf0e7a 100644
--- a/src/main/java/org/apache/commons/graph/shortestpath/PredecessorsList.java
+++ b/src/main/java/org/apache/commons/graph/shortestpath/PredecessorsList.java
@@ -76,6 +76,7 @@ final class PredecessorsList<V extends Vertex, WE extends WeightedEdge<V>>
             vertex = getConnectedVertex( vertex, edge );
         }
 
+        path.addVertexInHead( source );
         return path;
     }
 
