diff --git a/httpcore5/src/test/java/org/apache/hc/core5/http/impl/io/TestDefaultBHttpClientConnection.java b/httpcore5/src/test/java/org/apache/hc/core5/http/impl/io/TestDefaultBHttpClientConnection.java
index 59505ea1e..f584353c9 100644
--- a/httpcore5/src/test/java/org/apache/hc/core5/http/impl/io/TestDefaultBHttpClientConnection.java
+++ b/httpcore5/src/test/java/org/apache/hc/core5/http/impl/io/TestDefaultBHttpClientConnection.java
@@ -91,38 +91,6 @@ public class TestDefaultBHttpClientConnection {
         Assert.assertEquals(1, conn.getEndpointDetails().getResponseCount());
     }
 
-    @Test
-    public void testReadResponseEntityWithoutContentLength() throws Exception {
-        final String s = "HTTP/1.1 200 OK\r\nServer: test\r\n\r\n123";
-        final ByteArrayInputStream inStream = new ByteArrayInputStream(s.getBytes(StandardCharsets.US_ASCII));
-        Mockito.when(socket.getInputStream()).thenReturn(inStream);
-
-        conn.bind(socket);
-
-        Assert.assertEquals(0, conn.getEndpointDetails().getResponseCount());
-
-        final ClassicHttpResponse response = conn.receiveResponseHeader();
-
-        Assert.assertNotNull(response);
-        Assert.assertEquals(200, response.getCode());
-        Assert.assertTrue(response.containsHeader("Server"));
-        Assert.assertEquals(1, conn.getEndpointDetails().getResponseCount());
-
-        conn.receiveResponseEntity(response);
-
-        final HttpEntity entity = response.getEntity();
-        Assert.assertNotNull(entity);
-        Assert.assertEquals(-1, entity.getContentLength());
-        Assert.assertEquals(1, conn.getEndpointDetails().getResponseCount());
-
-        final InputStream content = entity.getContent();
-        Assert.assertNotNull(content);
-        Assert.assertEquals(3, content.available());
-        Assert.assertEquals('1', content.read());
-        Assert.assertEquals('2', content.read());
-        Assert.assertEquals('3', content.read());
-    }
-
     @Test
     public void testReadResponseEntityWithContentLength() throws Exception {
         final String s = "HTTP/1.1 200 OK\r\nServer: test\r\nContent-Length: 3\r\n\r\n123";
@@ -221,10 +189,7 @@ public class TestDefaultBHttpClientConnection {
         conn.receiveResponseEntity(response);
 
         final HttpEntity entity = response.getEntity();
-        Assert.assertNotNull(entity);
-        final InputStream content = entity.getContent();
-        Assert.assertNotNull(content);
-        Assert.assertTrue(content instanceof IdentityInputStream);
+        Assert.assertNull(entity);
     }
 
     @Test
