diff --git a/httpcore5/src/main/java/org/apache/hc/core5/http/nio/support/classic/SharedOutputBuffer.java b/httpcore5/src/main/java/org/apache/hc/core5/http/nio/support/classic/SharedOutputBuffer.java
index bf25a9b78..ee494f4e6 100644
--- a/httpcore5/src/main/java/org/apache/hc/core5/http/nio/support/classic/SharedOutputBuffer.java
+++ b/httpcore5/src/main/java/org/apache/hc/core5/http/nio/support/classic/SharedOutputBuffer.java
@@ -43,12 +43,10 @@ public final class SharedOutputBuffer extends AbstractSharedBuffer implements Co
 
     private volatile DataStreamChannel dataStreamChannel;
     private volatile boolean hasCapacity;
-    private volatile boolean endStreamPropagated;
 
     public SharedOutputBuffer(final ReentrantLock lock, final int initialBufferSize) {
         super(lock, initialBufferSize);
         this.hasCapacity = false;
-        this.endStreamPropagated = false;
     }
 
     public SharedOutputBuffer(final int bufferSize) {
@@ -65,7 +63,7 @@ public final class SharedOutputBuffer extends AbstractSharedBuffer implements Co
                 dataStreamChannel.write(buffer());
             }
             if (!buffer().hasRemaining() && endStream) {
-                propagateEndStream();
+                dataStreamChannel.endStream();
             }
             condition.signalAll();
         } finally {
@@ -137,7 +135,7 @@ public final class SharedOutputBuffer extends AbstractSharedBuffer implements Co
                     if (buffer().hasRemaining()) {
                         dataStreamChannel.requestOutput();
                     } else {
-                        propagateEndStream();
+                        dataStreamChannel.endStream();
                     }
                 }
             }
@@ -164,11 +162,5 @@ public final class SharedOutputBuffer extends AbstractSharedBuffer implements Co
         setInputMode();
     }
 
-    private void propagateEndStream() throws IOException {
-        if (!endStreamPropagated) {
-            dataStreamChannel.endStream();
-            endStreamPropagated = true;
-        }
-    }
 
 }
