diff --git a/httpcore5/src/test/java/org/apache/hc/core5/http/nio/support/classic/TestSharedOutputBuffer.java b/httpcore5/src/test/java/org/apache/hc/core5/http/nio/support/classic/TestSharedOutputBuffer.java
index 6858dc288..428b14da8 100644
--- a/httpcore5/src/test/java/org/apache/hc/core5/http/nio/support/classic/TestSharedOutputBuffer.java
+++ b/httpcore5/src/test/java/org/apache/hc/core5/http/nio/support/classic/TestSharedOutputBuffer.java
@@ -229,19 +229,5 @@ public class TestSharedOutputBuffer {
         }
     }
 
-    @Test
-    public void testEndStreamOnlyCalledOnce() throws IOException {
-
-        final DataStreamChannel channel = Mockito.mock(DataStreamChannel.class);
-        final SharedOutputBuffer outputBuffer = new SharedOutputBuffer(20);
-
-        outputBuffer.flush(channel);
-
-        outputBuffer.writeCompleted();
-        outputBuffer.flush(channel);
-
-        Mockito.verify(channel, Mockito.times(1)).endStream();
-    }
-
 }
 
