diff --git a/src/test/java/com/squareup/java/JavaWriterTest.java b/src/test/java/com/squareup/java/JavaWriterTest.java
index 6feec59..b4b360f 100644
--- a/src/test/java/com/squareup/java/JavaWriterTest.java
+++ b/src/test/java/com/squareup/java/JavaWriterTest.java
@@ -169,31 +169,16 @@ public final class JavaWriterTest {
 
   @Test public void addStaticImport() throws IOException {
     javaWriter.emitPackage("com.squareup");
-    javaWriter.emitStaticImports("java.lang.System.getProperty");
+    javaWriter.emitStaticImports("java.util.ArrayList");
     javaWriter.beginType("com.squareup.Foo", "class", Modifier.PUBLIC | Modifier.FINAL);
-    javaWriter.emitField("String", "bar", 0, "getProperty(\"bar\")");
-    javaWriter.endType();
-    assertCode(""
-        + "package com.squareup;\n"
-        + "\n"
-        + "import static java.lang.System.getProperty;\n"
-        + "public final class Foo {\n"
-        + "  String bar = getProperty(\"bar\");\n"
-        + "}\n");
-  }
-
-  @Test public void addStaticWildcardImport() throws IOException {
-    javaWriter.emitPackage("com.squareup");
-    javaWriter.emitStaticImports("java.lang.System.*");
-    javaWriter.beginType("com.squareup.Foo", "class", Modifier.PUBLIC | Modifier.FINAL);
-    javaWriter.emitField("String", "bar", 0, "getProperty(\"bar\")");
+    javaWriter.emitField("java.util.ArrayList", "list", 0, "new java.util.ArrayList()");
     javaWriter.endType();
     assertCode(""
         + "package com.squareup;\n"
         + "\n"
-        + "import static java.lang.System.*;\n"
+        + "import static java.util.ArrayList;\n"
         + "public final class Foo {\n"
-        + "  String bar = getProperty(\"bar\");\n"
+        + "  ArrayList list = new java.util.ArrayList();\n"
         + "}\n");
   }
 
