diff --git a/src/test/java/com/squareup/javawriter/JavaWriterTest.java b/src/test/java/com/squareup/javawriter/JavaWriterTest.java
index c2e0237..f6fc911 100644
--- a/src/test/java/com/squareup/javawriter/JavaWriterTest.java
+++ b/src/test/java/com/squareup/javawriter/JavaWriterTest.java
@@ -98,22 +98,6 @@ public final class JavaWriterTest {
         + "}\n");
   }
 
-  @Test public void fieldDeclarationWithWrappingInitialValue() throws IOException {
-    javaWriter.emitPackage("com.squareup");
-    javaWriter.beginType("com.squareup.Foo", "class");
-    javaWriter.emitField("java.lang.String", "string", EnumSet.noneOf(Modifier.class),
-        "\"bar\"\n+ \"baz\"\n+ \"biz\"");
-    javaWriter.endType();
-    assertCode(""
-        + "package com.squareup;\n"
-        + "\n"
-        + "class Foo {\n"
-        + "  String string = \"bar\"\n"
-        + "      + \"baz\"\n"
-        + "      + \"biz\";\n"
-        + "}\n");
-  }
-
   @Test public void abstractMethodDeclaration() throws IOException {
     javaWriter.emitPackage("com.squareup");
     javaWriter.beginType("com.squareup.Foo", "class");
