diff --git a/src/main/java/com/squareup/javawriter/JavaWriter.java b/src/main/java/com/squareup/javawriter/JavaWriter.java
index c6cb13c..38e81fb 100644
--- a/src/main/java/com/squareup/javawriter/JavaWriter.java
+++ b/src/main/java/com/squareup/javawriter/JavaWriter.java
@@ -334,9 +334,8 @@ public class JavaWriter implements Closeable {
     out.write(name);
 
     if (initialValue != null) {
-      out.write(" =");
+      out.write(" = ");
       if (!initialValue.startsWith("\n")) {
-        out.write(" ");
       }
 
       String[] lines = initialValue.split("\n", -1);
