diff --git a/src/test/java/com/squareup/javawriter/JavaWriterTest.java b/src/test/java/com/squareup/javawriter/JavaWriterTest.java
index 3fc5d61..9816906 100644
--- a/src/test/java/com/squareup/javawriter/JavaWriterTest.java
+++ b/src/test/java/com/squareup/javawriter/JavaWriterTest.java
@@ -887,17 +887,6 @@ public final class JavaWriterTest {
         + "}\n");
   }
 
-  @Test public void emptyNonEnumModifierSet() throws IOException {
-    javaWriter.emitPackage("com.squareup");
-    javaWriter.beginType("com.squareup.Foo", "class", new LinkedHashSet<Modifier>());
-    javaWriter.endType();
-    assertCode(""
-        + "package com.squareup;\n"
-        + "\n"
-        + "class Foo {\n"
-        + "}\n");
-  }
-
   private void assertCode(String expected) {
     assertThat(stringWriter.toString()).isEqualTo(expected);
   }
