diff --git a/surefire-api/src/main/java/org/apache/maven/surefire/report/LegacyPojoStackTraceWriter.java b/surefire-api/src/main/java/org/apache/maven/surefire/report/LegacyPojoStackTraceWriter.java
index 685ea7b5a..aaa7f5ac4 100644
--- a/surefire-api/src/main/java/org/apache/maven/surefire/report/LegacyPojoStackTraceWriter.java
+++ b/surefire-api/src/main/java/org/apache/maven/surefire/report/LegacyPojoStackTraceWriter.java
@@ -50,26 +50,13 @@ public class LegacyPojoStackTraceWriter
 
     public String writeTraceToString()
     {
+        StringWriter w = new StringWriter();
         if ( t != null )
         {
-            StringWriter w = new StringWriter();
-            PrintWriter stackTrace = new PrintWriter( w );
-            t.printStackTrace( stackTrace );
-            stackTrace.close();
+            t.printStackTrace( new PrintWriter( w ) );
             w.flush();
-            StringBuffer builder = w.getBuffer();
-            if ( isMultiLineExceptionMessage( t ) )
-            {
-                // SUREFIRE-986
-                String exc = t.getClass().getName() + ": ";
-                if ( builder.toString().startsWith( exc ) )
-                {
-                    builder.insert( exc.length(), '\n' );
-                }
-            }
-            return builder.toString();
         }
-        return "";
+        return w.toString();
     }
 
     public String smartTrimmedStackTrace()
@@ -97,27 +84,6 @@ public class LegacyPojoStackTraceWriter
         return result.toString();
     }
 
-    private static boolean isMultiLineExceptionMessage( Throwable t )
-    {
-        String msg = t.getLocalizedMessage();
-        if ( msg != null )
-        {
-            int countNewLines = 0;
-            for ( int i = 0, length = msg.length(); i < length; i++ )
-            {
-                if ( msg.charAt( i ) == '\n' )
-                {
-                    if ( ++countNewLines == 2 )
-                    {
-                        break;
-                    }
-                }
-            }
-            return countNewLines > 1 || countNewLines == 1 && !msg.trim().endsWith( "\n" );
-        }
-        return false;
-    }
-
     private static String getTruncatedMessage( String msg, int i )
     {
         if ( i < 0 )
