diff --git a/surefire-api/src/test/java/org/apache/maven/surefire/report/LegacyPojoStackTraceWriterTest.java b/surefire-api/src/test/java/org/apache/maven/surefire/report/LegacyPojoStackTraceWriterTest.java
index eb8187118..ab58186dc 100644
--- a/surefire-api/src/test/java/org/apache/maven/surefire/report/LegacyPojoStackTraceWriterTest.java
+++ b/surefire-api/src/test/java/org/apache/maven/surefire/report/LegacyPojoStackTraceWriterTest.java
@@ -142,22 +142,4 @@ public class LegacyPojoStackTraceWriterTest
         }
     }
 
-    public void testMultiLineMessage()
-    {
-        String msg =
-            "assert \"foo\" == \"bar\"\n"
-                + "             |\n"
-                + "             false";
-        try
-        {
-            throw new RuntimeException( msg );
-        }
-        catch ( Throwable t )
-        {
-            LegacyPojoStackTraceWriter writer = new LegacyPojoStackTraceWriter( null, null, t );
-            String stackTrace = writer.writeTraceToString();
-            assertTrue( stackTrace.startsWith( "java.lang.RuntimeException: \n" + msg ) );
-        }
-    }
-
 }
