diff --git a/httpclient5/src/main/java/org/apache/hc/client5/http/impl/cookie/LaxExpiresHandler.java b/httpclient5/src/main/java/org/apache/hc/client5/http/impl/cookie/LaxExpiresHandler.java
index c48be4fb5..ea4ae56ed 100644
--- a/httpclient5/src/main/java/org/apache/hc/client5/http/impl/cookie/LaxExpiresHandler.java
+++ b/httpclient5/src/main/java/org/apache/hc/client5/http/impl/cookie/LaxExpiresHandler.java
@@ -43,7 +43,6 @@ import org.apache.hc.core5.annotation.Contract;
 import org.apache.hc.core5.annotation.ThreadingBehavior;
 import org.apache.hc.core5.http.message.ParserCursor;
 import org.apache.hc.core5.util.Args;
-import org.apache.hc.core5.util.TextUtils;
 
 /**
  * Cookie {@code expires} attribute handler conformant to the more relaxed interpretation
@@ -108,9 +107,6 @@ public class LaxExpiresHandler extends AbstractCookieAttributeHandler implements
     @Override
     public void parse(final SetCookie cookie, final String value) throws MalformedCookieException {
         Args.notNull(cookie, "Cookie");
-        if (TextUtils.isBlank(value)) {
-            return;
-        }
         final ParserCursor cursor = new ParserCursor(0, value.length());
         final StringBuilder content = new StringBuilder();
 
