diff --git a/httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java b/httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java
index abaedb28e..5b588c1ec 100644
--- a/httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java
+++ b/httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java
@@ -120,10 +120,9 @@ public class TestMultipartEntityBuilder {
 
     @Test
     public void testMultipartWriteTo() throws Exception {
-        final String helloWorld = "hello world";
         final List<NameValuePair> parameters = new ArrayList<>();
         parameters.add(new BasicNameValuePair(MimeConsts.FIELD_PARAM_NAME, "test"));
-        parameters.add(new BasicNameValuePair(MimeConsts.FIELD_PARAM_FILENAME, helloWorld));
+        parameters.add(new BasicNameValuePair(MimeConsts.FIELD_PARAM_FILENAME, "hello world"));
         final MultipartFormEntity entity = MultipartEntityBuilder.create()
                 .setStrictMode()
                 .setBoundary("xxxxxxxxxxxxxxxxxxxxxxxx")
@@ -142,23 +141,21 @@ public class TestMultipartEntityBuilder {
                 "Content-Disposition: multipart/form-data; name=\"test\"; filename=\"hello world\"\r\n" +
                 "Content-Type: text/plain; charset=ISO-8859-1\r\n" +
                 "\r\n" +
-                helloWorld + "\r\n" +
+                "hello world\r\n" +
                 "--xxxxxxxxxxxxxxxxxxxxxxxx--\r\n", out.toString(StandardCharsets.US_ASCII.name()));
     }
-
     @Test
     public void testMultipartWriteToRFC7578Mode() throws Exception {
-        final String helloWorld = "hello \u03BA\u03CC\u03C3\u03BC\u03B5!%";
         final List<NameValuePair> parameters = new ArrayList<>();
         parameters.add(new BasicNameValuePair(MimeConsts.FIELD_PARAM_NAME, "test"));
-        parameters.add(new BasicNameValuePair(MimeConsts.FIELD_PARAM_FILENAME, helloWorld));
+        parameters.add(new BasicNameValuePair(MimeConsts.FIELD_PARAM_FILENAME, "hello \u03BA\u03CC\u03C3\u03BC\u03B5!%"));
 
         final MultipartFormEntity entity = MultipartEntityBuilder.create()
                 .setMode(HttpMultipartMode.EXTENDED)
                 .setBoundary("xxxxxxxxxxxxxxxxxxxxxxxx")
                 .addPart(new FormBodyPartBuilder()
                         .setName("test")
-                        .setBody(new StringBody(helloWorld, ContentType.TEXT_PLAIN.withCharset(StandardCharsets.UTF_8)))
+                        .setBody(new StringBody("hello world", ContentType.TEXT_PLAIN))
                         .addField("Content-Disposition", "multipart/form-data", parameters)
                         .build())
                 .buildEntity();
@@ -168,40 +165,10 @@ public class TestMultipartEntityBuilder {
         out.close();
         Assert.assertEquals("--xxxxxxxxxxxxxxxxxxxxxxxx\r\n" +
                 "Content-Disposition: multipart/form-data; name=\"test\"; filename=\"hello%20%CE%BA%CF%8C%CF%83%CE%BC%CE%B5!%25\"\r\n" +
-                "Content-Type: text/plain; charset=UTF-8\r\n" +
-                "\r\n" +
-                "hello \u00ce\u00ba\u00cf\u008c\u00cf\u0083\u00ce\u00bc\u00ce\u00b5!%\r\n" +
-                "--xxxxxxxxxxxxxxxxxxxxxxxx--\r\n", out.toString(StandardCharsets.ISO_8859_1.name()));
-    }
-
-    @Test
-    public void testMultipartWriteToRFC6532Mode() throws Exception {
-        final String helloWorld = "hello \u03BA\u03CC\u03C3\u03BC\u03B5!%";
-        final List<NameValuePair> parameters = new ArrayList<>();
-        parameters.add(new BasicNameValuePair(MimeConsts.FIELD_PARAM_NAME, "test"));
-        parameters.add(new BasicNameValuePair(MimeConsts.FIELD_PARAM_FILENAME, helloWorld));
-
-        final MultipartFormEntity entity = MultipartEntityBuilder.create()
-                .setMode(HttpMultipartMode.EXTENDED)
-                .setContentType(ContentType.create("multipart/other"))
-                .setBoundary("xxxxxxxxxxxxxxxxxxxxxxxx")
-                .addPart(new FormBodyPartBuilder()
-                        .setName("test")
-                        .setBody(new StringBody(helloWorld, ContentType.TEXT_PLAIN.withCharset(StandardCharsets.UTF_8)))
-                        .addField("Content-Disposition", "multipart/form-data", parameters)
-                        .build())
-                .buildEntity();
-
-        final ByteArrayOutputStream out = new ByteArrayOutputStream();
-        entity.writeTo(out);
-        out.close();
-        Assert.assertEquals("--xxxxxxxxxxxxxxxxxxxxxxxx\r\n" +
-                "Content-Disposition: multipart/form-data; name=\"test\"; " +
-                "filename=\"hello \u00ce\u00ba\u00cf\u008c\u00cf\u0083\u00ce\u00bc\u00ce\u00b5!%\"\r\n" +
-                "Content-Type: text/plain; charset=UTF-8\r\n" +
+                "Content-Type: text/plain; charset=ISO-8859-1\r\n" +
                 "\r\n" +
-                "hello \u00ce\u00ba\u00cf\u008c\u00cf\u0083\u00ce\u00bc\u00ce\u00b5!%\r\n" +
-                "--xxxxxxxxxxxxxxxxxxxxxxxx--\r\n", out.toString(StandardCharsets.ISO_8859_1.name()));
+                "hello world\r\n" +
+                "--xxxxxxxxxxxxxxxxxxxxxxxx--\r\n", out.toString(StandardCharsets.US_ASCII.name()));
     }
 
 }
