diff --git a/src/test/org/apache/commons/cli/bug/BugCLI162Test.java b/src/test/org/apache/commons/cli/bug/BugCLI162Test.java
index 5f76eb7..96007a6 100644
--- a/src/test/org/apache/commons/cli/bug/BugCLI162Test.java
+++ b/src/test/org/apache/commons/cli/bug/BugCLI162Test.java
@@ -259,14 +259,6 @@ public class BugCLI162Test extends TestCase {
                           "                      yes.\n" +
                           "Footer\n";
         assertEquals( "Long arguments did not split as expected", expected, sw.toString() );
-
-        try {
-            formatter.printHelp(new PrintWriter(sw), 22, this.getClass().getName(), "Header", options, 0, 5, "Footer");
-            fail("IllegalStateException expected");
-        } catch(IllegalStateException ise) {
-            // expected
-        }
-
     }
 
 }
