diff --git a/src/test/java/org/apache/commons/cli/AbstractParserTestCase.java b/src/test/java/org/apache/commons/cli/AbstractParserTestCase.java
index 36affcc..534939a 100644
--- a/src/test/java/org/apache/commons/cli/AbstractParserTestCase.java
+++ b/src/test/java/org/apache/commons/cli/AbstractParserTestCase.java
@@ -1021,12 +1021,5 @@ public abstract class AbstractParserTestCase {
         assertEquals("Confirm arg of -b", "file", cl.getOptionValue("b"));
         assertTrue("Confirm NO of extra args", cl.getArgList().isEmpty());
     }
-    
-    @Test(expected = UnrecognizedOptionException.class)
-    public void testAmbiguousArgParsing() throws Exception {
-        final String[] args = {"-=-"};
-        final Options options = new Options();
-        
-        final CommandLine cl = parser.parse(options, args);
-    }
+
 }
