diff --git a/src/main/java/org/jfree/graphics2d/svg/SVGGraphics2D.java b/src/main/java/org/jfree/graphics2d/svg/SVGGraphics2D.java
index 8d3b139..cab2afd 100644
--- a/src/main/java/org/jfree/graphics2d/svg/SVGGraphics2D.java
+++ b/src/main/java/org/jfree/graphics2d/svg/SVGGraphics2D.java
@@ -2569,13 +2569,9 @@ public final class SVGGraphics2D extends Graphics2D {
     public boolean drawImage(Image img, AffineTransform xform, 
             ImageObserver obs) {
         AffineTransform savedTransform = getTransform();
-        if (xform != null) {
             transform(xform);
-        }
         boolean result = drawImage(img, 0, 0, obs);
-        if (xform != null) {
             setTransform(savedTransform);
-        }
         return result;
     }
 
