diff --git a/core/src/test/java/org/apache/james/core/MailAddressTest.java b/core/src/test/java/org/apache/james/core/MailAddressTest.java
index f7cfc024e8..edb1cfa8b0 100644
--- a/core/src/test/java/org/apache/james/core/MailAddressTest.java
+++ b/core/src/test/java/org/apache/james/core/MailAddressTest.java
@@ -19,8 +19,6 @@
 
 package org.apache.james.core;
 
-import static org.assertj.core.api.Assertions.assertThat;
-
 import javax.mail.internet.AddressException;
 import javax.mail.internet.InternetAddress;
 
@@ -28,8 +26,6 @@ import org.assertj.core.api.Assertions;
 import org.junit.Assert;
 import org.junit.Test;
 
-import nl.jqno.equalsverifier.EqualsVerifier;
-
 public class MailAddressTest {
 
     private static final String GOOD_LOCAL_PART = "\"quoted@local part\"";
@@ -226,16 +222,4 @@ public class MailAddressTest {
         Assert.assertTrue(a.toString() + " != " + b.toString(), a.equals(b));
         Assert.assertFalse(a.toString() + " != " + null, a.equals(null));
     }
-
-    @Test
-    public void equalsShouldReturnTrueWhenBothNullSender() {
-        assertThat(MailAddress.nullSender())
-            .isEqualTo(MailAddress.nullSender());
-    }
-
-    @Test
-    public void shouldMatchBeanContract() {
-        EqualsVerifier.forClass(MailAddress.class)
-            .verify();
-    }
 }
