diff --git a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/AnnotProxyFieldValueFactoryTest.java b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/AnnotProxyFieldValueFactoryTest.java
index f8b7a81233..892c8a8273 100644
--- a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/AnnotProxyFieldValueFactoryTest.java
+++ b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/AnnotProxyFieldValueFactoryTest.java
@@ -25,6 +25,7 @@ import org.apache.wicket.spring.ISpringContextLocator;
 import org.apache.wicket.spring.SpringBeanLocator;
 import org.apache.wicket.spring.injection.util.Bean;
 import org.apache.wicket.spring.injection.util.Bean2;
+import org.apache.wicket.spring.injection.util.Injectable;
 import org.apache.wicket.spring.injection.util.InjectableInterface;
 import org.apache.wicket.spring.test.ApplicationContextMock;
 import org.junit.Assert;
@@ -35,8 +36,9 @@ import org.springframework.context.ApplicationContext;
  * Tests for BeanAnnotLocatorFactory
  * 
  * @author igor
+ * 
  */
-public abstract class AnnotProxyFieldValueFactoryTest extends Assert
+public class AnnotProxyFieldValueFactoryTest extends Assert
 {
 	ISpringContextLocator mockCtxLocator = new ISpringContextLocator()
 	{
@@ -52,15 +54,23 @@ public abstract class AnnotProxyFieldValueFactoryTest extends Assert
 		}
 	};
 
-	protected final InjectableInterface obj;
+	final InjectableInterface obj;
 
-	protected final AnnotProxyFieldValueFactory factory = new AnnotProxyFieldValueFactory(mockCtxLocator);
+	/**
+	 * Construct.
+	 */
+	public AnnotProxyFieldValueFactoryTest()
+	{
+		this(new Injectable());
+	}
 
 	protected AnnotProxyFieldValueFactoryTest(InjectableInterface injectable)
 	{
 		obj = injectable;
 	}
 
+	AnnotProxyFieldValueFactory factory = new AnnotProxyFieldValueFactory(mockCtxLocator);
+
 	/**
 	 * Test the factory
 	 * 
@@ -69,8 +79,8 @@ public abstract class AnnotProxyFieldValueFactoryTest extends Assert
 	@Test
 	public void testFactory() throws Exception
 	{
-		SpringBeanLocator locator;
-		Object proxy;
+		SpringBeanLocator locator = null;
+		Object proxy = null;
 
 		Field field = obj.getClass().getDeclaredField("nobean");
 		proxy = factory.getFieldValue(field, obj);
@@ -149,8 +159,6 @@ public abstract class AnnotProxyFieldValueFactoryTest extends Assert
 		}
 		catch (RuntimeException e)
 		{
-			// expected
-			assertTrue(true);
 		}
 	}
 
diff --git a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/JavaxInjectAnnotProxyFieldValueFactoryTest.java b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/JavaxInjectAnnotProxyFieldValueFactoryTest.java
index 6d3d9cca3f..47e3772bbd 100644
--- a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/JavaxInjectAnnotProxyFieldValueFactoryTest.java
+++ b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/JavaxInjectAnnotProxyFieldValueFactoryTest.java
@@ -16,14 +16,11 @@
  */
 package org.apache.wicket.spring.injection.annot;
 
-import java.lang.reflect.Field;
-
-import org.apache.wicket.spring.injection.util.JavaxInjectInjectable;
-import org.junit.Test;
+import org.apache.wicket.spring.injection.util.JavaxInjectable;
 
 
 /**
- * Tests for Spring injection with {@literal @javax.inject.Inject} annotation
+ * Tests for javax.inject.Inject when used with spring
  */
 public class JavaxInjectAnnotProxyFieldValueFactoryTest extends AnnotProxyFieldValueFactoryTest
 {
@@ -32,26 +29,6 @@ public class JavaxInjectAnnotProxyFieldValueFactoryTest extends AnnotProxyFieldV
 	 */
 	public JavaxInjectAnnotProxyFieldValueFactoryTest()
 	{
-		super(new JavaxInjectInjectable());
-	}
-
-	/**
-	 * https://issues.apache.org/jira/browse/WICKET-5686
-	 * @throws Exception
-	 */
-	@Test
-	public void required() throws Exception
-	{
-		Field field = obj.getClass().getDeclaredField("nonExisting");
-		try
-		{
-			factory.getFieldValue(field, obj);
-			fail("Fields annotated with @Inject are required!");
-		}
-		catch (IllegalStateException isx)
-		{
-			// expected
-			assertTrue(true);
-		}
+		super(new JavaxInjectable());
 	}
 }
diff --git a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/SpringBeanAnnotProxyFieldValueFactoryTest.java b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/SpringBeanAnnotProxyFieldValueFactoryTest.java
deleted file mode 100644
index 424073b738..0000000000
--- a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/annot/SpringBeanAnnotProxyFieldValueFactoryTest.java
+++ /dev/null
@@ -1,34 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.wicket.spring.injection.annot;
-
-import org.apache.wicket.spring.injection.util.SpringBeanInjectable;
-
-
-/**
- * Tests for Spring injections with {@literal @SpringBean} annotation
- */
-public class SpringBeanAnnotProxyFieldValueFactoryTest extends AnnotProxyFieldValueFactoryTest
-{
-	/**
-	 * Construct.
-	 */
-	public SpringBeanAnnotProxyFieldValueFactoryTest()
-	{
-		super(new SpringBeanInjectable());
-	}
-}
diff --git a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/SpringBeanInjectable.java b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/Injectable.java
similarity index 95%
rename from wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/SpringBeanInjectable.java
rename to wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/Injectable.java
index f13329ddd3..7f5c61dc9b 100644
--- a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/SpringBeanInjectable.java
+++ b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/Injectable.java
@@ -20,10 +20,11 @@ import org.apache.wicket.spring.injection.annot.SpringBean;
 
 /**
  * Mock for an object with some SpringBean annotations
- *
+ * 
  * @author Igor Vaynberg (ivaynberg)
+ * 
  */
-public class SpringBeanInjectable implements InjectableInterface
+public class Injectable implements InjectableInterface
 {
 	private Bean nobean;
 
diff --git a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/JavaxInjectInjectable.java b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/JavaxInjectable.java
similarity index 92%
rename from wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/JavaxInjectInjectable.java
rename to wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/JavaxInjectable.java
index 0f378fca29..2ac5767dc2 100644
--- a/wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/JavaxInjectInjectable.java
+++ b/wicket-spring/src/test/java/org/apache/wicket/spring/injection/util/JavaxInjectable.java
@@ -21,11 +21,11 @@ import javax.inject.Named;
 
 /**
  * Mock for an object with some SpringBean annotations
- *
+ * 
  * @author Igor Vaynberg (ivaynberg)
- *
+ * 
  */
-public class JavaxInjectInjectable implements InjectableInterface
+public class JavaxInjectable implements InjectableInterface
 {
 	private Bean nobean;
 
@@ -36,9 +36,6 @@ public class JavaxInjectInjectable implements InjectableInterface
 	@Named("somebean")
 	private Bean2 beanByName;
 
-	@Inject
-	private String nonExisting;
-
 	/**
 	 * @return test bean
 	 */
