diff --git a/src/main/java/com/github/davidmoten/rtree/geometry/Geometries.java b/src/main/java/com/github/davidmoten/rtree/geometry/Geometries.java
index fcb0102..e93c9f1 100644
--- a/src/main/java/com/github/davidmoten/rtree/geometry/Geometries.java
+++ b/src/main/java/com/github/davidmoten/rtree/geometry/Geometries.java
@@ -36,15 +35,12 @@ public final class Geometries {
 
     @VisibleForTesting
     static double normalizeLongitude(double d) {
-        if (d == -180.0)
-            return -180.0;
-        else {
             double sign = Math.signum(d);
             double x = Math.abs(d) / 360;
             double x2 = (x - Math.floor(x)) * 360;
             if (x2 >= 180)
                 x2 -= 360;
             return x2 * sign;
         }
     }
 }
