diff --git a/src/test/java/com/github/davidmoten/rtree/geometry/GeometriesTest.java b/src/test/java/com/github/davidmoten/rtree/geometry/GeometriesTest.java
index 4c65d2d..c4e0f6b 100644
--- a/src/test/java/com/github/davidmoten/rtree/geometry/GeometriesTest.java
+++ b/src/test/java/com/github/davidmoten/rtree/geometry/GeometriesTest.java
@@ -35,21 +35,6 @@ public class GeometriesTest {
         assertEquals(-180, Geometries.normalizeLongitude(180), PRECISION);
     }
 
-    @Test
-    public void testNormalizeLongitude3_2() {
-        assertEquals(-180, Geometries.normalizeLongitude(-180), PRECISION);
-    }
-
-    @Test
-    public void testNormalizeLongitude3_3() {
-        assertEquals(-179, Geometries.normalizeLongitude(-179), PRECISION);
-    }
-
-    @Test
-    public void testNormalizeLongitude3_4() {
-        assertEquals(179, Geometries.normalizeLongitude(-181), PRECISION);
-    }
-
     @Test
     public void testNormalizeLongitude4() {
         assertEquals(-179, Geometries.normalizeLongitude(181), PRECISION);
