diff --git a/src/main/java/com/github/davidmoten/rtree/geometry/internal/PointDouble.java b/src/main/java/com/github/davidmoten/rtree/geometry/internal/PointDouble.java
index 30d73b2..2919a3d 100644
--- a/src/main/java/com/github/davidmoten/rtree/geometry/internal/PointDouble.java
+++ b/src/main/java/com/github/davidmoten/rtree/geometry/internal/PointDouble.java
@@ -105,32 +105,6 @@ public final class PointDouble implements Point {
         return true;
     }
 
-    @Override
-    public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        long temp;
-        temp = Double.doubleToLongBits(x);
-        result = prime * result + (int) (temp ^ (temp >>> 32));
-        temp = Double.doubleToLongBits(y);
-        result = prime * result + (int) (temp ^ (temp >>> 32));
-        return result;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj)
-            return true;
-        if (obj == null)
-            return false;
-        if (getClass() != obj.getClass())
-            return false;
-        PointDouble other = (PointDouble) obj;
-        if (Double.doubleToLongBits(x) != Double.doubleToLongBits(other.x))
-            return false;
-        if (Double.doubleToLongBits(y) != Double.doubleToLongBits(other.y))
-            return false;
-        return true;
-    }
 
-}
\ No newline at end of file
+
+}
