diff --git a/src/main/java/com/github/davidmoten/rtree/Util.java b/src/main/java/com/github/davidmoten/rtree/Util.java
index fca7893..982fcc8 100644
--- a/src/main/java/com/github/davidmoten/rtree/Util.java
+++ b/src/main/java/com/github/davidmoten/rtree/Util.java
@@ -43,8 +43,8 @@ public final class Util {
         Preconditions.checkArgument(!items.isEmpty());
         float minX1 = Float.MAX_VALUE;
         float minY1 = Float.MAX_VALUE;
-        float maxX2 = -Float.MAX_VALUE;
-        float maxY2 = -Float.MAX_VALUE;
+        float maxX2 = Float.MIN_VALUE;
+        float maxY2 = Float.MIN_VALUE;
         for (final HasGeometry item : items) {
             Rectangle r = item.geometry().mbr();
             if (r.x1() < minX1)
