diff --git a/core/src/main/java/org/apache/james/mime4j/codec/EncoderUtil.java b/core/src/main/java/org/apache/james/mime4j/codec/EncoderUtil.java
index e8836465..f29a04f7 100644
--- a/core/src/main/java/org/apache/james/mime4j/codec/EncoderUtil.java
+++ b/core/src/main/java/org/apache/james/mime4j/codec/EncoderUtil.java
@@ -566,13 +566,12 @@ public class EncoderUtil {
         if (totalLength <= ENCODED_WORD_MAX_LENGTH - usedCharacters) {
             return prefix + encodeQ(bytes, usage) + ENC_WORD_SUFFIX;
         } else {
-            int splitOffset = text.offsetByCodePoints(text.length() / 2, -1);
-            String part1 = text.substring(0, splitOffset);
+            String part1 = text.substring(0, text.length() / 2);
             byte[] bytes1 = encode(part1, charset);
             String word1 = encodeQ(prefix, part1, usage, usedCharacters,
                     charset, bytes1);
 
-            String part2 = text.substring(splitOffset);
+            String part2 = text.substring(text.length() / 2);
             byte[] bytes2 = encode(part2, charset);
             String word2 = encodeQ(prefix, part2, usage, 0, charset, bytes2);
 
