diff --git a/core/src/main/java/hivemall/topicmodel/LDAUDTF.java b/core/src/main/java/hivemall/topicmodel/LDAUDTF.java
index 59dc2e3e..4cbd964d 100644
--- a/core/src/main/java/hivemall/topicmodel/LDAUDTF.java
+++ b/core/src/main/java/hivemall/topicmodel/LDAUDTF.java
@@ -45,7 +45,7 @@ public final class LDAUDTF extends ProbabilisticTopicModelBaseUDTF {
 
         this.alpha = 1.f / topics;
         this.eta = 1.f / topics;
-        this.numDocs = 0L;
+        this.numDocs = -1L;
         this.tau0 = 64.d;
         this.kappa = 0.7;
         this.delta = DEFAULT_DELTA;
@@ -72,7 +72,7 @@ public final class LDAUDTF extends ProbabilisticTopicModelBaseUDTF {
         if (cl != null) {
             this.alpha = Primitives.parseFloat(cl.getOptionValue("alpha"), 1.f / topics);
             this.eta = Primitives.parseFloat(cl.getOptionValue("eta"), 1.f / topics);
-            this.numDocs = Primitives.parseLong(cl.getOptionValue("num_docs"), 0L);
+            this.numDocs = Primitives.parseLong(cl.getOptionValue("num_docs"), -1L);
             this.tau0 = Primitives.parseDouble(cl.getOptionValue("tau0"), 64.d);
             if (tau0 <= 0.d) {
                 throw new UDFArgumentException("'-tau0' must be positive: " + tau0);
diff --git a/core/src/main/java/hivemall/topicmodel/OnlineLDAModel.java b/core/src/main/java/hivemall/topicmodel/OnlineLDAModel.java
index 1ddf52bd..aed17bac 100644
--- a/core/src/main/java/hivemall/topicmodel/OnlineLDAModel.java
+++ b/core/src/main/java/hivemall/topicmodel/OnlineLDAModel.java
@@ -110,7 +110,7 @@ public final class OnlineLDAModel extends AbstractProbabilisticTopicModel {
         this._kappa = kappa;
         this._delta = delta;
 
-        this._isAutoD = (_D <= 0L);
+        this._isAutoD = (_D < 0L);
 
         // initialize a random number generator
         this._gd = new GammaDistribution(SHAPE, SCALE);
