diff --git a/src/main/java/net/tascalate/concurrent/AbstractCompletableTask.java b/src/main/java/net/tascalate/concurrent/AbstractCompletableTask.java
index aac4b11..405ea67 100644
--- a/src/main/java/net/tascalate/concurrent/AbstractCompletableTask.java
+++ b/src/main/java/net/tascalate/concurrent/AbstractCompletableTask.java
@@ -322,14 +322,15 @@ abstract class AbstractCompletableTask<T> extends PromiseAdapter<T> implements P
             failure -> {
                 try {
                     action.accept(null, failure);
+                    return forwardException(failure);
                 } catch (Throwable e) {
                     // CompletableFuture does not override exception here
                     // unlike as in handle[Async](BiFunction)
                     // Preserve this behavior, but let us add at least 
                     // suppressed exception
                     failure.addSuppressed(e);
+                    return forwardException(failure);
                 }
-                return forwardException(failure);
             }, 
             executor
         );
