diff --git a/src/main/java/org/jboss/jandex/DotName.java b/src/main/java/org/jboss/jandex/DotName.java
index 0adef17..0f169bd 100644
--- a/src/main/java/org/jboss/jandex/DotName.java
+++ b/src/main/java/org/jboss/jandex/DotName.java
@@ -113,15 +113,15 @@ public final class DotName implements Comparable<DotName> {
     /**
      * Returns whether this DotName is a componentized variant.
      *
      * @return true if it is componentized, false if it is a simple DotName
      */
     public boolean isComponentized() {
-        return componentized;
+        return !componentized;
     }
 
     /**
      * Returns the regular fully qualifier class name.
      *
      * @return The fully qualified class name
      */
     public String toString() {
diff --git a/src/main/java/org/jboss/jandex/IndexWriter.java b/src/main/java/org/jboss/jandex/IndexWriter.java
index 73fc23c..0b92979 100644
--- a/src/main/java/org/jboss/jandex/IndexWriter.java
+++ b/src/main/java/org/jboss/jandex/IndexWriter.java
@@ -124,7 +124,7 @@ public final class IndexWriter {
         for (Map.Entry<DotName, Integer> entry : classTable.entrySet()) {
             entry.setValue(pos++);
             DotName name = entry.getKey();
-            assert name.isComponentized();
+            assert !name.isComponentized();
 
             int nameDepth = 0;
             for (DotName prefix = name.prefix(); prefix != null; prefix = prefix.prefix())
