diff --git a/src/main/java/org/jboss/jandex/ClassInfo.java b/src/main/java/org/jboss/jandex/ClassInfo.java
index e0d4622..b472331 100644
--- a/src/main/java/org/jboss/jandex/ClassInfo.java
+++ b/src/main/java/org/jboss/jandex/ClassInfo.java
@@ -49,10 +49,10 @@ import java.util.Map;
 public final class ClassInfo implements AnnotationTarget {
 
     private final DotName name;
+    private final short flags;
     private final Map<DotName, List<AnnotationInstance>> annotations;
 
     // Not final to allow lazy initialization, immutable once published
-    private  short flags;
     private Type[] interfaceTypes;
     private Type superClassType;
     private Type[] typeParameters;
@@ -596,8 +592,5 @@ public final class ClassInfo implements AnnotationTarget {
 
         nestingInfo.enclosingMethod = enclosingMethod;
     }
 
-    void setFlags(short flags) {
-        this.flags = flags;
-    }
 }
diff --git a/src/main/java/org/jboss/jandex/Indexer.java b/src/main/java/org/jboss/jandex/Indexer.java
index a250bb7..c470b8d 100644
--- a/src/main/java/org/jboss/jandex/Indexer.java
+++ b/src/main/java/org/jboss/jandex/Indexer.java
@@ -346,7 +346,6 @@ public final class Indexer {
 
             if (innerClass.equals(target.name())) {
                 target.setInnerClassInfo(outerClass, simpleName);
-                target.setFlags((short)flags);
             }
 
             innerClasses.put(innerClass, new InnerClassInfo(innerClass, outerClass, simpleName, flags));
