diff --git a/src/main/java/org/apache/commons/ognl/OgnlRuntime.java b/src/main/java/org/apache/commons/ognl/OgnlRuntime.java
index c1b2536..2bd1dff 100644
--- a/src/main/java/org/apache/commons/ognl/OgnlRuntime.java
+++ b/src/main/java/org/apache/commons/ognl/OgnlRuntime.java
@@ -2736,20 +2736,15 @@ public class OgnlRuntime
                 if ( ( methods[i].getName().equalsIgnoreCase( name )
                     || methods[i].getName().toLowerCase().equals( name )
                     || methods[i].getName().toLowerCase().equals( "get" + name )
-                    || methods[i].getName().toLowerCase().equals( "has" + name )
-                    || methods[i].getName().toLowerCase().equals( "is" + name ) )
+                    || methods[i].getName().toLowerCase().equals( "has" + name ) || methods[i].getName().toLowerCase().equals( "is"
+                                                                                                                                   + name ) )
                     && !methods[i].getName().startsWith( "set" ) )
                 {
                     if ( numParms > 0 && methods[i].getMethod().getParameterTypes().length == numParms )
                         return methods[i].getMethod();
                     else if ( numParms < 0 )
                     {
-                        if ( methods[i].getName().equals( name ) )
-                        {
-                            return methods[i].getMethod();
-                        }
-                        else if ( ( m != null
-                                && m.getParameterTypes().length > methods[i].getMethod().getParameterTypes().length )
+                        if ( ( m != null && m.getParameterTypes().length > methods[i].getMethod().getParameterTypes().length )
                             || m == null )
                         {
                             m = methods[i].getMethod();
