diff --git a/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-backend/src/main/java/org/uberfire/ext/editor/commons/backend/version/VersionServiceImpl.java b/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-backend/src/main/java/org/uberfire/ext/editor/commons/backend/version/VersionServiceImpl.java
index a021a2ae7..05cb19b72 100644
--- a/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-backend/src/main/java/org/uberfire/ext/editor/commons/backend/version/VersionServiceImpl.java
+++ b/uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-backend/src/main/java/org/uberfire/ext/editor/commons/backend/version/VersionServiceImpl.java
@@ -50,9 +49,7 @@ public class VersionServiceImpl
     @Inject
     private VersionRecordService versionRecordService;
 
-    @Inject
-    private PathResolver pathResolver;
 
     @Override
     public List<VersionRecord> getVersions(final Path path) {
 
@@ -72,9 +68,8 @@ public class VersionServiceImpl
     public Path restore(final Path _path,
                         final String comment) {
         try {
-            ioService.startBatch(Paths.convert(_path).getFileSystem());
 
-            final org.uberfire.java.nio.file.Path path = pathResolver.resolveMainFilePath(convert(_path));
+            final org.uberfire.java.nio.file.Path path = convert(_path);
             final org.uberfire.java.nio.file.Path target = path.getFileSystem().getPath(path.toString());
 
             return convert(ioService.copy(path,
@@ -87,8 +81,6 @@ public class VersionServiceImpl
                                                   comment)));
         } catch (Exception e) {
             throw new RuntimeException(e);
-        } finally {
-            ioService.endBatch();
         }
     }
 }
