diff --git a/src/java/org/apache/commons/cli/Util.java b/src/java/org/apache/commons/cli/Util.java
index 6f89edc..f58ff64 100644
--- a/src/java/org/apache/commons/cli/Util.java
+++ b/src/java/org/apache/commons/cli/Util.java
@@ -62,10 +62,14 @@ class Util
      */
     static String stripLeadingAndTrailingQuotes(String str)
     {
+        if (str.startsWith("\""))
+        {
+            str = str.substring(1, str.length());
+        }
 
-        if (str.length() > 1 && str.startsWith("\"") && str.endsWith("\"") && str.substring(1, str.length() - 1).indexOf('"') == -1)
+        if (str.endsWith("\""))
         {
-            str = str.substring(1, str.length() - 1);
+            str = str.substring(0, str.length() - 1);
         }
         
         return str;
