diff --git a/src/main/java/org/davidmoten/hilbert/SmallHilbertCurve.java b/src/main/java/org/davidmoten/hilbert/SmallHilbertCurve.java
index 257935e..9ac964e 100644
--- a/src/main/java/org/davidmoten/hilbert/SmallHilbertCurve.java
+++ b/src/main/java/org/davidmoten/hilbert/SmallHilbertCurve.java
@@ -69,7 +69,7 @@ public final class SmallHilbertCurve {
         for (int i = 0; i < bits; i++) {
             for (int j = 0; j < transposedIndex.length; j++) {
                 if ((transposedIndex[j] & mask) != 0) {
-                    b |= 1L << bIndex;
+                    b |= 1 << bIndex;
                 }
                 bIndex--;
             }
