diff --git a/nbxml/src/test/java/org/apache/vysper/xml/sax/impl/XMPPContentHandlerTestCase.java b/nbxml/src/test/java/org/apache/vysper/xml/sax/impl/XMPPContentHandlerTestCase.java
index 547c0efa..1603082f 100644
--- a/nbxml/src/test/java/org/apache/vysper/xml/sax/impl/XMPPContentHandlerTestCase.java
+++ b/nbxml/src/test/java/org/apache/vysper/xml/sax/impl/XMPPContentHandlerTestCase.java
@@ -39,19 +39,10 @@ public class XMPPContentHandlerTestCase extends TestCase {
 
     private static class TestListener implements XMLElementListener {
         public List<XMLElement> elements = new ArrayList<XMLElement>();
-        private boolean closed = false;
 
         public void element(XMLElement element) {
             elements.add(element);
         }
-        
-        public void close() {
-        	closed = true;
-        }
-        
-        public boolean isClosed() {
-        	return closed;
-        }
     }
 
     public void test() throws Exception {
@@ -72,7 +63,6 @@ public class XMPPContentHandlerTestCase extends TestCase {
         assertEquals("stanza", actual.next().getName());
         assertEquals("message", actual.next().getName());
         assertEquals("iq", actual.next().getName());
-        assertEquals(true, listener.isClosed());
     }
 
     private void parse(NonBlockingXMLReader reader, String xml) throws Exception {
diff --git a/nbxml/src/test/java/org/apache/vysper/xml/sax/perf/PerfRunner.java b/nbxml/src/test/java/org/apache/vysper/xml/sax/perf/PerfRunner.java
index add24dcd..fd6b9dee 100644
--- a/nbxml/src/test/java/org/apache/vysper/xml/sax/perf/PerfRunner.java
+++ b/nbxml/src/test/java/org/apache/vysper/xml/sax/perf/PerfRunner.java
@@ -39,8 +39,6 @@ public class PerfRunner {
         public void element(XMLElement element) {
             counter++;
         }
-        
-        public void close() {}
     }
 
     private static final String SINGLE_LEVEL_XML = "<child att='foo' att2='bar'></child>";
