diff --git a/src/main/java/com/github/davidmoten/rtree/fbs/SerializerFlatBuffers.java b/src/main/java/com/github/davidmoten/rtree/fbs/SerializerFlatBuffers.java
index 575a571..aa1a46c 100644
--- a/src/main/java/com/github/davidmoten/rtree/fbs/SerializerFlatBuffers.java
+++ b/src/main/java/com/github/davidmoten/rtree/fbs/SerializerFlatBuffers.java
@@ -160,15 +160,9 @@ public final class SerializerFlatBuffers<T, S extends Geometry> implements Seria
     @VisibleForTesting
     static byte[] readFully(InputStream is, int numBytes) throws IOException {
         byte[] b = new byte[numBytes];
-        int count = 0;
-        do {
-            int n = is.read(b, count, numBytes - count);
-            if (n > 0) {
-                count += n;
-            } else {
-                throw new RuntimeException("unexpected");
-            }
-        } while (count < numBytes);
+        int n = is.read(b);
+        if (n != numBytes)
+            throw new RuntimeException("unexpected");
         return b;
     }
 
