diff --git a/src/test/java/com/github/davidmoten/rtree/geometry/PointTest.java b/src/test/java/com/github/davidmoten/rtree/geometry/PointTest.java
index acd820c..80c3ed1 100644
--- a/src/test/java/com/github/davidmoten/rtree/geometry/PointTest.java
+++ b/src/test/java/com/github/davidmoten/rtree/geometry/PointTest.java
@@ -34,8 +34,8 @@ public class PointTest {
     @Test
     public void testMbr() {
         Point p = Geometries.point(1, 2);
-        Point p2 = Geometries.point(1, 2);
-        assertEquals(p, p2);
+        Rectangle r = Geometries.rectangle(1, 2, 1, 2);
+        assertEquals(r, p.mbr());
     }
 
     @Test
@@ -74,6 +74,6 @@ public class PointTest {
     @Test
     public void testHashCode() {
         Point p = Geometries.point(1, 2);
-        assertEquals(-260045887, p.hashCode());
+        assertEquals(-1056041056, p.hashCode());
     }
 }
