diff --git a/src/test/java/stax2/stream/TestGetElement.java b/src/test/java/stax2/stream/TestGetElement.java
index 2be2e2b..059ff54 100644
--- a/src/test/java/stax2/stream/TestGetElement.java
+++ b/src/test/java/stax2/stream/TestGetElement.java
@@ -122,36 +122,6 @@ public class TestGetElement
         assertTokenType(END_DOCUMENT, sr.next());
     }
 
-    public void testGetMixedContentElementText() throws Exception
-    {
-        final String XML = "<root>foo<![CDATA[bar]]></root>";
-
-        XMLInputFactory ifact = getInputFactory();
-        XMLStreamReader sr;
-
-        // Start with non-coalescing case
-        setCoalescing(ifact, false);
-        sr = ifact.createXMLStreamReader(new StringReader(XML));
-        assertTokenType(START_ELEMENT, sr.next());
-
-        // this currently fails
-        assertEquals("foobar", sr.getElementText());
-        assertTokenType(END_ELEMENT, sr.getEventType());
-        assertTokenType(END_DOCUMENT, sr.next());
-        sr.close();
-
-        // and then with coalescing
-        setCoalescing(ifact, true);
-        sr = ifact.createXMLStreamReader(new StringReader(XML));
-        assertTokenType(START_ELEMENT, sr.next());
-
-        // this one passes
-        assertEquals("foobar", sr.getElementText());
-        assertTokenType(END_ELEMENT, sr.getEventType());
-        assertTokenType(END_DOCUMENT, sr.next());
-        sr.close();
-    }
-
     /*
     ///////////////////////////////////////////////
     // Helper methods
diff --git a/src/test/java/stax2/stream/TestStreamReader.java.rej b/src/test/java/stax2/stream/TestStreamReader.java.rej
new file mode 100644
index 0000000..e45ef9f
--- /dev/null
+++ b/src/test/java/stax2/stream/TestStreamReader.java.rej
@@ -0,0 +1,23 @@
+***************
+*** 70,79 ****
+          XMLStreamReader2 reader = (XMLStreamReader2) f.createXMLStreamReader(new StringReader(DOC));
+          assertTokenType(DTD, reader.next());
+  
+- 		DTDInfo dtd = reader.getDTDInfo();
+- 		assertNotNull(dtd);
+- 		assertEquals(sysId, dtd.getDTDSystemId());
+- 		
+          assertTokenType(START_ELEMENT, reader.next());
+          assertEquals("root", reader.getLocalName());
+          assertTokenType(CHARACTERS, reader.next());
+--- 72,81 ----
+          XMLStreamReader2 reader = (XMLStreamReader2) f.createXMLStreamReader(new StringReader(DOC));
+          assertTokenType(DTD, reader.next());
+  
++         DTDInfo dtd = reader.getDTDInfo();
++         assertNotNull(dtd);
++         assertEquals(sysId, dtd.getDTDSystemId());
++ 
+          assertTokenType(START_ELEMENT, reader.next());
+          assertEquals("root", reader.getLocalName());
+          assertTokenType(CHARACTERS, reader.next());
