diff --git a/src/test/java/stax2/vwstream/W3CSchemaWrite23Test.java b/src/test/java/stax2/vwstream/W3CSchemaWrite23Test.java
index 7b6f6bd..940698a 100644
--- a/src/test/java/stax2/vwstream/W3CSchemaWrite23Test.java
+++ b/src/test/java/stax2/vwstream/W3CSchemaWrite23Test.java
@@ -10,20 +10,12 @@ public class W3CSchemaWrite23Test
 {
     public void testSchemaValidatingCopy23() throws Exception
     {
-        final String SCHEMA = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
-                "<xs:schema elementFormDefault=\"unqualified\"\n" +
-                "           xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">\n" +
-                "    <xs:element name=\"Document\">\n" +
-                "        <xs:complexType>\n" +
-                "            <xs:sequence>\n" +
-                "                <xs:element name=\"Paragraph\" type=\"xs:string\"/>\n" +
-                "            </xs:sequence>\n" +
-                "        </xs:complexType>\n" +
-                "    </xs:element>\n" +
-                "</xs:schema>";
-        final String CONTENT = "<Document>\n" +
-                "    <Paragraph>Hello world!</Paragraph>\n" +
-                "</Document>";
+        final String SCHEMA = "<?xml version='1.0' ?>\n"
++"<xs:schema elementFormDefault='unqualified'\n"
++"           xmlns:xs='http://www.w3.org/2001/XMLSchema'>\n"
++"    <xs:element name='Document' type='xs:int'/>\n"
++"</xs:schema>";
+        final String CONTENT = "<Document>124</Document>";
         final String DOC = "<?xml version='1.0' encoding='UTF-8'?>\n"+CONTENT;
                 
 
@@ -31,9 +23,6 @@ public class W3CSchemaWrite23Test
         XMLStreamWriter2 xmlWriter = getSchemaValidatingWriter(strw, SCHEMA, false);
         XMLStreamReader2 xmlReader = constructNsStreamReader(DOC, false);
 
-        // For this test we need validation, otherwise the reader returns characters events instead of white-space events.
-        xmlReader.validateAgainst(parseW3CSchema(SCHEMA));
-
         while (xmlReader.hasNext()) {
             /*int type =*/ xmlReader.next();
             xmlWriter.copyEventFromReader(xmlReader, true);
