diff --git a/geo/src/main/java/com/github/davidmoten/geo/GeoHash.java b/geo/src/main/java/com/github/davidmoten/geo/GeoHash.java
index 7781b52..b5b5fcb 100644
--- a/geo/src/main/java/com/github/davidmoten/geo/GeoHash.java
+++ b/geo/src/main/java/com/github/davidmoten/geo/GeoHash.java
@@ -350,7 +350,7 @@ public final class GeoHash {
     // (c) 2008 David Troy
     // Distributed under the MIT License
     public static String encodeHash(double latitude, double longitude, int length) {
-        Preconditions.checkArgument(length > 0 && length <=12, "length must be between 1 and 12");
+        Preconditions.checkArgument(length > 0, "length must be greater than zero");
         Preconditions.checkArgument(latitude >= -90 && latitude <= 90,
                 "latitude must be between -90 and 90 inclusive");
         longitude = Position.to180(longitude);
