diff --git a/geo/src/test/java/com/github/davidmoten/geo/GeoHashTest.java b/geo/src/test/java/com/github/davidmoten/geo/GeoHashTest.java
index ab8797a..535e1d9 100644
--- a/geo/src/test/java/com/github/davidmoten/geo/GeoHashTest.java
+++ b/geo/src/test/java/com/github/davidmoten/geo/GeoHashTest.java
@@ -100,21 +100,6 @@ public class GeoHashTest {
     public void testHashOfNonDefaultLength() {
         assertEquals("6gkzwg", encodeHash(-25.382708, -49.265506, 6));
     }
-    
-    @Test
-    public void testHashOfLength12() {
-        assertEquals("6gkzwgjzn820", encodeHash(-25.382708, -49.265506, 12));
-    }
-
-    @Test(expected = IllegalArgumentException.class)
-    public void testHashOfLength13() {
-        encodeHash(-25.382708, -49.265506, 13);
-    }
-
-    @Test(expected = IllegalArgumentException.class)
-    public void testHashOfLength20() {
-        encodeHash(-25.382708, -49.265506, 20);
-    }
 
     @Test(expected = IllegalArgumentException.class)
     public void testHashEncodeGivenNonPositiveLength() {
