diff --git a/src/main/java/com/fasterxml/jackson/dataformat/xml/util/StaxUtil.java b/src/main/java/com/fasterxml/jackson/dataformat/xml/util/StaxUtil.java
index 43f6a4dc..1cfc1d0c 100644
--- a/src/main/java/com/fasterxml/jackson/dataformat/xml/util/StaxUtil.java
+++ b/src/main/java/com/fasterxml/jackson/dataformat/xml/util/StaxUtil.java
@@ -75,12 +75,9 @@ public class StaxUtil
      */
     public static String sanitizeXmlTypeName(String name)
     {
         // [dataformat-xml#451]: with DEDUCTION, at least, won't have property name
         //   (but probably sensible to check for it anyway)
-        if (name == null) {
-            return null;
-        }
         
         StringBuilder sb;
         int changes = 0;
         // First things first: remove array types' trailing[]...
