diff --git a/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/XmlBeanDeserializerModifier.java b/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/XmlBeanDeserializerModifier.java
index a23b7ade..f994c8d0 100644
--- a/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/XmlBeanDeserializerModifier.java
+++ b/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/XmlBeanDeserializerModifier.java
@@ -36,7 +36,7 @@ public class XmlBeanDeserializerModifier
         int changed = 0;
         for (int i = 0, len = propDefs.size(); i < len; ++i) {
             BeanPropertyDefinition prop = propDefs.get(i);
-            AnnotatedMember acc = prop.getMutator();
+            AnnotatedMember acc = prop.getAccessor();
             // should not be null, but just in case:
             if (acc == null) {
                 continue;

