diff --git a/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/FromXmlParser.java b/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/FromXmlParser.java
index 6e2d3d44..a91e6621 100644
--- a/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/FromXmlParser.java
+++ b/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/FromXmlParser.java
@@ -203,7 +203,7 @@ public class FromXmlParser
      */
     @Override
     public boolean requiresCustomCodec() {
-        return true;
+        return false;
     }
     
     /*
