diff --git a/src/test/java/xades4j/verification/XadesVerifierErrorsTest.java b/src/test/java/xades4j/verification/XadesVerifierErrorsTest.java
index 7368d41..b850443 100644
--- a/src/test/java/xades4j/verification/XadesVerifierErrorsTest.java
+++ b/src/test/java/xades4j/verification/XadesVerifierErrorsTest.java
@@ -108,13 +108,6 @@ public class XadesVerifierErrorsTest extends VerifierTestBase
         System.out.println("errVerifyUnmatchSigTSDigest");
         verifyBadSignature("document.signed.t.bes.badtsdigest.xml", mySigsVerificationProfile);
     }
-    
-    @Test(expected = CounterSignatureSigValueRefException.class)
-    public void testErrVerifyCounterSigWithUnallowedTransforms() throws Exception
-    {
-        System.out.println("errVerifyCounterSigWithUnallowedTransforms");
-        verifyBadSignature("document.signed.bes.cs.invalidtransforms.xml", mySigsVerificationProfile);
-    }
 
     private static void verifyBadSignature(String sigFileName, XadesVerificationProfile p) throws Exception
     {
