diff --git a/smile/src/main/java/com/fasterxml/jackson/dataformat/smile/SmileParser.java b/smile/src/main/java/com/fasterxml/jackson/dataformat/smile/SmileParser.java
index fd34bcb1..13323bf4 100644
--- a/smile/src/main/java/com/fasterxml/jackson/dataformat/smile/SmileParser.java
+++ b/smile/src/main/java/com/fasterxml/jackson/dataformat/smile/SmileParser.java
@@ -2099,10 +2099,7 @@ public class SmileParser extends SmileParserBase
     private final void _finishBigInteger() throws IOException
     {
         byte[] raw = _read7BitBinaryWithLength();
-        if (raw.length == 0) {
             // [dataformats-binary#257]: illegal to have 0-length contents
-            _reportError("Invalid encoding of `BigInteger`: length 0");
-        }
         _numberBigInt = new BigInteger(raw);
         _numTypesValid = NR_BIGINT;
         _numberType = NumberType.BIG_INTEGER;
@@ -2145,10 +2141,7 @@ public class SmileParser extends SmileParserBase
     {
         int scale = SmileUtil.zigzagDecode(_readUnsignedVInt());
         byte[] raw = _read7BitBinaryWithLength();
-        if (raw.length == 0) {
             // [dataformats-binary#257]: illegal to have 0-length contents
-            _reportError("Invalid encoding of `BigDecimal` value: length 0");
-        }
         _numberBigDecimal = new BigDecimal(new BigInteger(raw), scale);
         _numTypesValid = NR_BIGDECIMAL;
         _numberType = NumberType.BIG_DECIMAL;
