diff --git a/gson/src/main/java/com/google/gson/DefaultTypeAdapters.java b/gson/src/main/java/com/google/gson/DefaultTypeAdapters.java
index c9771a87..9fd2c227 100644
--- a/gson/src/main/java/com/google/gson/DefaultTypeAdapters.java
+++ b/gson/src/main/java/com/google/gson/DefaultTypeAdapters.java
@@ -31,6 +31,7 @@ import java.util.Calendar;
 import java.util.Collection;
 import java.util.Date;
 import java.util.GregorianCalendar;
+import java.util.HashSet;
 import java.util.LinkedHashMap;
 import java.util.LinkedList;
 import java.util.List;
@@ -79,7 +80,7 @@ final class DefaultTypeAdapters {
   private static final StringTypeAdapter STRING_TYPE_ADAPTER = new StringTypeAdapter();
 
   private static final PropertiesCreator PROPERTIES_CREATOR = new PropertiesCreator();
-  private static final TreeSetCreator TREE_SET_CREATOR = new TreeSetCreator();
+  private static final HashSetCreator HASH_SET_CREATOR = new HashSetCreator();
   private static final GregorianCalendarTypeAdapter GREGORIAN_CALENDAR_TYPE_ADAPTER = 
     new GregorianCalendarTypeAdapter();
 
@@ -178,8 +179,8 @@ final class DefaultTypeAdapters {
     map.register(List.class, COLLECTION_TYPE_ADAPTER);
     map.register(Queue.class, COLLECTION_TYPE_ADAPTER);
 
-    map.register(Set.class, TREE_SET_CREATOR);
-    map.register(SortedSet.class, TREE_SET_CREATOR);
+    map.register(Set.class, HASH_SET_CREATOR);
+    map.register(SortedSet.class, HASH_SET_CREATOR);
     map.register(Properties.class, PROPERTIES_CREATOR);
     map.makeUnmodifiable();
     return map;
@@ -779,13 +780,13 @@ final class DefaultTypeAdapters {
     }
   }
 
-  private static class TreeSetCreator implements InstanceCreator<TreeSet<?>> {
-    public TreeSet<?> createInstance(Type type) {
-      return new TreeSet<Object>();
+  private static class HashSetCreator implements InstanceCreator<HashSet<?>> {
+    public HashSet<?> createInstance(Type type) {
+      return new HashSet<Object>();
     }
     @Override
     public String toString() {
-      return TreeSetCreator.class.getSimpleName();
+      return HashSetCreator.class.getSimpleName();
     }
   }
 }
