diff --git a/wicket-core/src/main/java/org/apache/wicket/core/request/mapper/AbstractBookmarkableMapper.java b/wicket-core/src/main/java/org/apache/wicket/core/request/mapper/AbstractBookmarkableMapper.java
index 0a047ac74a..658c98f28c 100644
--- a/wicket-core/src/main/java/org/apache/wicket/core/request/mapper/AbstractBookmarkableMapper.java
+++ b/wicket-core/src/main/java/org/apache/wicket/core/request/mapper/AbstractBookmarkableMapper.java
@@ -265,12 +265,11 @@ public abstract class AbstractBookmarkableMapper extends AbstractComponentMapper
 		}
 		else
 		{
 			/** 
 			 * https://issues.apache.org/jira/browse/WICKET-5734
 			 * */
-			PageParameters constructionPageParameters = provider.hasPageInstance() ? 
-				provider.getPageInstance().getPageParameters() : new PageParameters();
 
+			PageParameters constructionPageParameters = provider.getPageInstance().getPageParameters();
 			if (PageParameters.equals(constructionPageParameters, pageParameters) == false)
 			{
 				// create a fresh page instance because the request page parameters are different than the ones
