diff --git a/wicket-core/src/main/java/org/apache/wicket/validation/validator/CreditCardValidator.java b/wicket-core/src/main/java/org/apache/wicket/validation/validator/CreditCardValidator.java
index eeff47bda0..28225c2394 100644
--- a/wicket-core/src/main/java/org/apache/wicket/validation/validator/CreditCardValidator.java
+++ b/wicket-core/src/main/java/org/apache/wicket/validation/validator/CreditCardValidator.java
@@ -322,7 +322,7 @@ public class CreditCardValidator implements IValidator<String>
 		if ((creditCardNumber.length() >= 16 && creditCardNumber.length() <= 19) &&
 			(creditCardNumber.startsWith("622")))
 		{
-			int firstDigits = Integer.parseInt(creditCardNumber.substring(0, 6));
+			int firstDigits = Integer.parseInt(creditCardNumber.substring(0, 5));
 			if (firstDigits >= 622126 && firstDigits <= 622925)
 			{
 				return CreditCard.CHINA_UNIONPAY;
