diff --git a/wicket-core/src/test/java/org/apache/wicket/markup/html/TransparentWebMarkupContainerTest.java b/wicket-core/src/test/java/org/apache/wicket/markup/html/TransparentWebMarkupContainerTest.java
index ef3607ee76..572c6c561d 100644
--- a/wicket-core/src/test/java/org/apache/wicket/markup/html/TransparentWebMarkupContainerTest.java
+++ b/wicket-core/src/test/java/org/apache/wicket/markup/html/TransparentWebMarkupContainerTest.java
@@ -31,6 +31,7 @@ import org.apache.wicket.page.IPageManagerContext;
 import org.apache.wicket.util.resource.IResourceStream;
 import org.apache.wicket.util.resource.StringResourceStream;
 import org.apache.wicket.util.tester.WicketTester;
+import org.junit.Ignore;
 import org.junit.Test;
 
 /**
@@ -176,6 +177,7 @@ public class TransparentWebMarkupContainerTest extends WicketTestCase
 	 * container and trying to update a label that was added to the outer TWMC.
 	 */
 	@Test
+	@Ignore("Fails with WICKET-5898")
 	public void ajaxRequestForComponentInTransparentWebMarkupContainerShouldntCauseStackOverflow3()
 	{
 		tester.startPage(DoubleNestedTransparentContainerWithSiblingTransparentContainerPage.class);
