diff --git a/wicket-core/src/test/java/org/apache/wicket/MarkupContainerTest.java b/wicket-core/src/test/java/org/apache/wicket/MarkupContainerTest.java
index 248cbbaa59..e9ea15a07b 100644
--- a/wicket-core/src/test/java/org/apache/wicket/MarkupContainerTest.java
+++ b/wicket-core/src/test/java/org/apache/wicket/MarkupContainerTest.java
@@ -117,24 +117,6 @@ public class MarkupContainerTest extends WicketTestCase
 		me.add(me);
 	}
 
-	/**
-	 * https://issues.apache.org/jira/browse/WICKET-5911
-	 */
-	@Test
-	public void rerenderAfterRenderFailure()
-	{
-		FirstRenderFailsPage page = new FirstRenderFailsPage();
-		try {
-			tester.startPage(page);
-		} catch (WicketRuntimeException expected) {
-		}
-
-		tester.startPage(page);
-
-		// rendering flags where properly reset, so second rendering works properly
-		assertEquals(2, page.beforeRenderCalls);
-	}
-
 	/**
 	 * https://issues.apache.org/jira/browse/WICKET-4012
 	 */
@@ -202,38 +184,4 @@ public class MarkupContainerTest extends WicketTestCase
 				"<html><body><div wicket:id='a1'><div wicket:id='a2'><div wicket:id='a3'></div></div></div></body></html>");
 		}
 	}
-
-	private static class FirstRenderFailsPage extends WebPage	implements IMarkupResourceStreamProvider
-	{
-		private boolean firstRender = true;
-
-		private int beforeRenderCalls = 0;
-
-		private FirstRenderFailsPage()
-		{
-
-			WebMarkupContainer a1 = new WebMarkupContainer("a1") {
-				@Override
-				protected void onBeforeRender() {
-					super.onBeforeRender();
-
-					beforeRenderCalls++;
-
-					if (firstRender) {
-						firstRender = false;
-						throw new WicketRuntimeException();
-					}
-				}
-			};
-			add(a1);
-		}
-
-		@Override
-		public IResourceStream getMarkupResourceStream(MarkupContainer container,
-			Class<?> containerClass)
-		{
-			return new StringResourceStream(
-				"<html><body><div wicket:id='a1'></div></body></html>");
-		}
-	}
 }
