diff --git a/wicket-core/src/main/java/org/apache/wicket/Application.java b/wicket-core/src/main/java/org/apache/wicket/Application.java
index 056d4aec1d..a01f153d2e 100644
--- a/wicket-core/src/main/java/org/apache/wicket/Application.java
+++ b/wicket-core/src/main/java/org/apache/wicket/Application.java
@@ -169,10 +169,10 @@ public abstract class Application implements UnboundListener, IEventSink
 
 	private static final String PROPERTIES_FILE_EXTENSION = ".properties";
 
 	/**
 	 * The path to look for Wicket specific properties file(s)
 	 */
-	private static final String META_INF_WICKET_PATH = File.separatorChar+ "META-INF" + File.separatorChar + "wicket" + File.separatorChar;
+	private static final String META_INF_WICKET_PATH = File.pathSeparatorChar + "META-INF" + File.pathSeparatorChar + "wicket" + File.pathSeparatorChar;
 
 	/** root mapper */
 	private IRequestMapper rootRequestMapper;
