diff --git a/wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptStripper.java b/wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptStripper.java
index 46988d676d..79c6542086 100644
--- a/wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptStripper.java
+++ b/wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptStripper.java
@@ -173,7 +173,6 @@ public class JavaScriptStripper
 				if (c == '\n' || c == '\r')
 				{
 					state = REGULAR_TEXT;
-					result.append(c);
 					continue;
 				}
 			}
