diff --git a/wicket-core/src/test/java/org/apache/wicket/util/string/JavaScriptStripperTest.java b/wicket-core/src/test/java/org/apache/wicket/util/string/JavaScriptStripperTest.java
index 1ffe164186..345278b7b4 100644
--- a/wicket-core/src/test/java/org/apache/wicket/util/string/JavaScriptStripperTest.java
+++ b/wicket-core/src/test/java/org/apache/wicket/util/string/JavaScriptStripperTest.java
@@ -32,7 +32,7 @@ public class JavaScriptStripperTest extends Assert
 	public void unixWICKET501()
 	{
 		String s = new JavaScriptStripper().stripCommentsAndWhitespace("    // Handle the common XPath // expression\n    if ( !t.indexOf(\"//\") ) {");
-		assertEquals(" \n if ( !t.indexOf(\"//\") ) {", s);
+		assertEquals("  if ( !t.indexOf(\"//\") ) {", s);
 	}
 
 	/**	 */
@@ -40,7 +40,7 @@ public class JavaScriptStripperTest extends Assert
 	public void dosWICKET501()
 	{
 		String s = new JavaScriptStripper().stripCommentsAndWhitespace("    // Handle the common XPath // expression\r\n    if ( !t.indexOf(\"//\") ) {");
-		assertEquals(" \r\nif ( !t.indexOf(\"//\") ) {", s);
+		assertEquals(" \nif ( !t.indexOf(\"//\") ) {", s);
 	}
 
 	/**	 */
@@ -48,7 +48,7 @@ public class JavaScriptStripperTest extends Assert
 	public void macWICKET501()
 	{
 		String s = new JavaScriptStripper().stripCommentsAndWhitespace("    // Handle the common XPath // expression\r    if ( !t.indexOf(\"//\") ) {");
-		assertEquals(" \r if ( !t.indexOf(\"//\") ) {", s);
+		assertEquals("  if ( !t.indexOf(\"//\") ) {", s);
 	}
 
 	/**	 */
@@ -64,7 +64,7 @@ public class JavaScriptStripperTest extends Assert
 	public void regexp2()
 	{
 		String s = new JavaScriptStripper().stripCommentsAndWhitespace("foo.replace(/\"//*strip me*/, \"\"); // strip me\rdoFoo();");
-		assertEquals("foo.replace(/\"/, \"\"); \rdoFoo();", s);
+		assertEquals("foo.replace(/\"/, \"\"); doFoo();", s);
 	}
 
 	/**	 */
@@ -82,7 +82,7 @@ public class JavaScriptStripperTest extends Assert
 	{
 		String before = " attr: /**/ //xyz\n /\\[((?:[\\w-]*:)?[\\w-]+)\\s*(?:([!^$*~|]?=)\\s*((['\"])([^\\4]*?)\\4|([^'\"][^\\]]*?)))?\\]/    after     regex";
 		String after = new JavaScriptStripper().stripCommentsAndWhitespace(before);
-		String expected = " attr:  \n /\\[((?:[\\w-]*:)?[\\w-]+)\\s*(?:([!^$*~|]?=)\\s*((['\"])([^\\4]*?)\\4|([^'\"][^\\]]*?)))?\\]/ after regex";
+		String expected = " attr:   /\\[((?:[\\w-]*:)?[\\w-]+)\\s*(?:([!^$*~|]?=)\\s*((['\"])([^\\4]*?)\\4|([^'\"][^\\]]*?)))?\\]/ after regex";
 		assertEquals(expected, after);
 		System.out.println(after);
 	}
@@ -128,17 +128,7 @@ public class JavaScriptStripperTest extends Assert
 		assertEquals(expected, after);
 	}
 
-	/**     */
-	@Test
-	public void WICKET4760()
-	{
-		String before = "x++ //\nx++";
-		String after = new JavaScriptStripper().stripCommentsAndWhitespace(before);
-		String expected = "x++ \nx++";
-		assertEquals(expected, after);
-	}
-
-	/**     */
+	/**	 */
 	// @formatter:off
 	public static String TESTSTRING2 =
          "   var test = function () {\n" +
