diff --git a/wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptUtils.java b/wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptUtils.java
index 42d9274b8b..f07d5151f1 100644
--- a/wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptUtils.java
+++ b/wicket-core/src/main/java/org/apache/wicket/core/util/string/JavaScriptUtils.java
@@ -143,7 +143,7 @@ public class JavaScriptUtils
 			response.write("charset=\"" + Strings.escapeMarkup(charset) + "\" ");
 		}
 		response.write("src=\"");
-		response.write(url);
+		response.write(Strings.escapeMarkup(url));
 		response.write("\"></script>");
 		response.write("\n");
 	}
@@ -201,7 +201,7 @@ public class JavaScriptUtils
 		response.write("<script type=\"text/javascript\" ");
 		if (id != null)
 		{
-			response.write("id=\"" + Strings.escapeMarkup(id) + "\"");
+			response.write("id=\"" + id + "\"");
 		}
 		response.write(">");
 		response.write(SCRIPT_CONTENT_PREFIX);
