diff --git a/wicket-core/src/test/java/org/apache/wicket/core/request/mapper/CryptoMapperTest.java b/wicket-core/src/test/java/org/apache/wicket/core/request/mapper/CryptoMapperTest.java
index 2c0577c43b..da8e80de8b 100644
--- a/wicket-core/src/test/java/org/apache/wicket/core/request/mapper/CryptoMapperTest.java
+++ b/wicket-core/src/test/java/org/apache/wicket/core/request/mapper/CryptoMapperTest.java
@@ -16,18 +16,13 @@
  */
 package org.apache.wicket.core.request.mapper;
 
-import org.apache.wicket.MockPage;
-import org.apache.wicket.core.request.handler.ListenerInterfaceRequestHandler;
-import org.apache.wicket.core.request.handler.PageAndComponentProvider;
 import org.apache.wicket.core.request.handler.PageProvider;
 import org.apache.wicket.core.request.handler.RenderPageRequestHandler;
 import org.apache.wicket.core.request.handler.RequestSettingRequestHandler;
-import org.apache.wicket.markup.html.link.ILinkListener;
 import org.apache.wicket.protocol.http.WebApplication;
 import org.apache.wicket.request.IRequestHandler;
 import org.apache.wicket.request.Request;
 import org.apache.wicket.request.Url;
-import org.apache.wicket.request.component.IRequestableComponent;
 import org.apache.wicket.request.handler.resource.ResourceReferenceRequestHandler;
 import org.apache.wicket.request.mapper.parameter.PageParameters;
 import org.apache.wicket.request.resource.PackageResourceReference;
@@ -175,8 +170,8 @@ public class CryptoMapperTest extends AbstractMapperTest
 	}
 
 	/**
-	 * When the home page url is requested, with parameters, the url will contain only page
-	 * parameters. It should not be encrypted, otherwise we get needless redirects.
+	 * When the home page url is requested, with parameters, the url will contain only page parameters.
+	 * It should not be encrypted, otherwise we get needless redirects.
 	 */
 	@Test
 	public void homePageWithParameters()
@@ -301,31 +296,4 @@ public class CryptoMapperTest extends AbstractMapperTest
 		assertEquals(getClass(), handler.getResourceReference().getScope());
 		assertEquals("less-crypt.txt", handler.getResourceReference().getName());
 	}
-
-	/**
-	 * Additional parameters, WICKET-4923
-	 */
-	@Test
-	public void additionalParameters()
-	{
-		MockPage page = new MockPage();
-		IRequestableComponent c = page.get("foo:bar");
-		PageAndComponentProvider provider = new PageAndComponentProvider(page, c);
-		IRequestHandler handler = new ListenerInterfaceRequestHandler(provider,
-			ILinkListener.INTERFACE);
-
-		Url url = mapper.mapHandler(handler);
-		url.addQueryParameter("q", "foo");
-
-		Request request = getRequest(url);
-
-		IRequestHandler requestHandler = mapper.mapRequest(request);
-
-		assertTrue(requestHandler instanceof RequestSettingRequestHandler);
-
-		assertEquals("foo", ((RequestSettingRequestHandler)requestHandler).getRequest()
-			.getUrl()
-			.getQueryParameterValue("q")
-			.toString());
-	}
 }
\ No newline at end of file
