diff --git a/client/src/test/java/org/apache/qpid/test/unit/client/BrokerDetails/BrokerDetailsTest.java b/client/src/test/java/org/apache/qpid/test/unit/client/BrokerDetails/BrokerDetailsTest.java
index fd0e7d054..2a33bf713 100644
--- a/client/src/test/java/org/apache/qpid/test/unit/client/BrokerDetails/BrokerDetailsTest.java
+++ b/client/src/test/java/org/apache/qpid/test/unit/client/BrokerDetails/BrokerDetailsTest.java
@@ -144,28 +144,6 @@ public class BrokerDetailsTest extends QpidTestCase
         assertEquals("Unexpected toString", expectedToString, actualToString);
     }
 
-    public void testToStringMasksEncryptionTrustStorePassword() throws Exception
-    {
-        String url = "tcp://localhost:5672?encryption_trust_store_password='password'";
-        BrokerDetails details = new BrokerDetails(url);
-
-        String expectedToString = "tcp://localhost:5672?encryption_trust_store_password='********'";
-        String actualToString = details.toString();
-
-        assertEquals("Unexpected toString", expectedToString, actualToString);
-    }
-
-    public void testToStringMasksEncryptionKeyStorePassword() throws Exception
-    {
-        String url = "tcp://localhost:5672?encryption_key_store_password='password'";
-        BrokerDetails details = new BrokerDetails(url);
-
-        String expectedToString = "tcp://localhost:5672?encryption_key_store_password='********'";
-        String actualToString = details.toString();
-
-        assertEquals("Unexpected toString", expectedToString, actualToString);
-    }
-
     public void testDefaultSsl() throws URLSyntaxException
     {
         String brokerURL = "tcp://localhost:5672";
