diff --git a/client/src/test/java/org/apache/qpid/test/unit/client/BrokerDetails/BrokerDetailsTest.java b/client/src/test/java/org/apache/qpid/test/unit/client/BrokerDetails/BrokerDetailsTest.java
index 3d7e63fb7..fd0e7d054 100644
--- a/client/src/test/java/org/apache/qpid/test/unit/client/BrokerDetails/BrokerDetailsTest.java
+++ b/client/src/test/java/org/apache/qpid/test/unit/client/BrokerDetails/BrokerDetailsTest.java
@@ -247,28 +247,4 @@ public class BrokerDetailsTest extends QpidTestCase
         assertNull(String.format("Unexpected '%s' property value", BrokerDetails.OPTIONS_SSL_VERIFY_HOSTNAME),
                 broker.getProperty(BrokerDetails.OPTIONS_SSL_VERIFY_HOSTNAME));
     }
-
-    public void testEncryptionKeyStorePath() throws Exception
-    {
-        String brokerURL = "tcp://localhost:5672?ssl='true'&encryption_key_store='path'";
-        BrokerDetails broker = new BrokerDetails(brokerURL);
-        ConnectionSettings connectionSettings = broker.buildConnectionSettings();
-        assertEquals("path", connectionSettings.getEncryptionKeyStorePath());
-    }
-
-    public void testEncryptionKeyStorePassword() throws Exception
-    {
-        String brokerURL = "tcp://localhost:5672?ssl='true'&encryption_key_store_password='pass'&encryption_trust_store_password='foo'";
-        BrokerDetails broker = new BrokerDetails(brokerURL);
-        ConnectionSettings connectionSettings = broker.buildConnectionSettings();
-        assertEquals("pass", connectionSettings.getEncryptionKeyStorePassword());
-    }
-
-    public void testEncryptionTrustStorePassword() throws Exception
-    {
-        String brokerURL = "tcp://localhost:5672?ssl='true'&encryption_key_store_password='pass'&encryption_trust_store_password='foo'";
-        BrokerDetails broker = new BrokerDetails(brokerURL);
-        ConnectionSettings connectionSettings = broker.buildConnectionSettings();
-        assertEquals("foo", connectionSettings.getEncryptionTrustStorePassword());
-    }
 }
