diff --git a/xbean-reflect/src/test/java/org/apache/xbean/propertyeditor/DateEditorTest.java b/xbean-reflect/src/test/java/org/apache/xbean/propertyeditor/DateEditorTest.java
deleted file mode 100755
index 5cd10f02..00000000
--- a/xbean-reflect/src/test/java/org/apache/xbean/propertyeditor/DateEditorTest.java
+++ /dev/null
@@ -1,35 +0,0 @@
-/**
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements.  See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License.  You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.xbean.propertyeditor;
-
-import java.text.DateFormat;
-import java.util.Date;
-import java.util.Locale;
-
-import junit.framework.TestCase;
-
-/*
- * @version $Rev$ $Date$
- */
-public class DateEditorTest extends TestCase {
-    public void testToObjectImpl() throws Exception {
-        Date expected = DateFormat.getDateInstance(DateFormat.MEDIUM, Locale.US).parse("Mar 1, 1954");
-        Date actual = (Date) new DateEditor().toObjectImpl("locale=en_US format=MEDIUM Mar 1, 1954");
-        assertEquals(expected, actual);
-    }
-
-}
