diff --git a/src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentSet.java b/src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentSet.java
index 3f1603c..f1e99d3 100644
--- a/src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentSet.java
+++ b/src/main/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentSet.java
@@ -49,7 +49,7 @@ public class WeakConcurrentSet<V> implements Runnable, Iterable<V> {
      * @return {@code true} if the value is contained in the set.
      */
     public boolean remove(V value) {
-        return target.remove(value) != null;
+        return target.remove(value);
     }
 
     /**
