diff --git a/src/test/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentSetTest.java b/src/test/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentSetTest.java
index 129edab..b180ffc 100644
--- a/src/test/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentSetTest.java
+++ b/src/test/java/com/blogspot/mydailyjava/weaklockfree/WeakConcurrentSetTest.java
@@ -92,18 +92,4 @@ public class WeakConcurrentSetTest {
         protected void triggerClean() {
         }
     }
-
-    @Test
-    public void testSetContract() throws Exception {
-        final WeakConcurrentSet<Object> set = new WeakConcurrentSet<Object>(WeakConcurrentSet.Cleaner.INLINE);
-        Object obj = new Object();
-        assertThat(set.contains(obj), is(false));
-        assertThat(set.remove(obj), is(false));
-        assertThat(set.add(obj), is(true));
-        assertThat(set.add(obj), is(false));
-        assertThat(set.contains(obj), is(true));
-        assertThat(set.remove(obj), is(true));
-        assertThat(set.contains(obj), is(false));
-        assertThat(set.remove(obj), is(false));
-    }
 }
