diff --git a/code/api/src/main/java/org/apache/tamaya/spi/ServiceContextManager.java b/code/api/src/main/java/org/apache/tamaya/spi/ServiceContextManager.java
index be287db7..f58c27ad 100644
--- a/code/api/src/main/java/org/apache/tamaya/spi/ServiceContextManager.java
+++ b/code/api/src/main/java/org/apache/tamaya/spi/ServiceContextManager.java
@@ -57,8 +57,9 @@ public final class ServiceContextManager {
         try {
             int highestOrdinal = 0;
             for (ServiceContext serviceContext : ServiceLoader.load(ServiceContext.class)) {
-                if (highestServiceContext == null
-                        || serviceContext.ordinal() > highestOrdinal) {
+                if(highestServiceContext==null){
+                    highestServiceContext = serviceContext;
+                }else if (serviceContext.ordinal() > highestOrdinal) {
                     highestServiceContext = serviceContext;
                     highestOrdinal = serviceContext.ordinal();
                 }
