diff --git a/src/main/java/org/apache/maven/jxr/JavaCodeTransform.java b/src/main/java/org/apache/maven/jxr/JavaCodeTransform.java
index 4f902bc..02689e2 100644
--- a/src/main/java/org/apache/maven/jxr/JavaCodeTransform.java
+++ b/src/main/java/org/apache/maven/jxr/JavaCodeTransform.java
@@ -282,31 +282,19 @@ public class JavaCodeTransform
             .append( "<html xmlns=\"http://www.w3.org/1999/xhtml\" xml:lang=\"" ).append( locale )
             .append( "\" lang=\"" ).append( locale ).append( "\">\n" ).append( "<head>\n" )
             .append( "<meta http-equiv=\"content-type\" content=\"text/html; charset=" ).append( outputEncoding )
-            .append( "\" />\n" );
+            .append( "\" />" );
 
         // title ("classname xref")
-        buffer.append( "<title>" );
         try
         {
-            JavaFile javaFile = fileManager.getFile( this.getCurrentFilename() );
-            if ( javaFile.getClassType() != null )
-            {
-                buffer.append( javaFile.getClassType().getName() );
-            }
-            else
-            {
-                buffer.append( this.getCurrentFilename() );
-            }
-            buffer.append( " " );
+            buffer.append( "<title>" ).append( fileManager.getFile( this.getCurrentFilename() )
+                .getClassType().getName() ).append( " xref</title>\n" );
         }
         catch ( IOException e )
         {
+            buffer.append( "<title>xref</title>\n" );
             e.printStackTrace();
         }
-        finally
-        {
-            buffer.append( "xref</title>\n" );
-        }
 
         // stylesheet link
         buffer.append( "<link type=\"text/css\" rel=\"stylesheet\" href=\"" ).append( this.getPackageRoot() )
@@ -1107,7 +1096,7 @@ public class JavaCodeTransform
                 }
                 else
                 {
-                    return "";
+                    System.out.println( this.getCurrentFilename() );
                 }
                 javadocURI.append( ".html" );
 
