diff --git a/hugegraph-common/src/main/java/com/baidu/hugegraph/version/CommonVersion.java b/hugegraph-common/src/main/java/com/baidu/hugegraph/version/CommonVersion.java
index e75c183..5aa53b9 100644
--- a/hugegraph-common/src/main/java/com/baidu/hugegraph/version/CommonVersion.java
+++ b/hugegraph-common/src/main/java/com/baidu/hugegraph/version/CommonVersion.java
@@ -27,5 +27,5 @@ public class CommonVersion {
 
     // The second parameter of Version.of() is for all-in-one JAR
     public static final Version VERSION = Version.of(CommonVersion.class,
-                                                     "2.1.2");
+                                                     "2.1.1");
 }
