diff --git a/src/java/org/apache/commons/beanutils/locale/converters/FloatLocaleConverter.java b/src/java/org/apache/commons/beanutils/locale/converters/FloatLocaleConverter.java
index 357e74b7..a32f37d4 100644
--- a/src/java/org/apache/commons/beanutils/locale/converters/FloatLocaleConverter.java
+++ b/src/java/org/apache/commons/beanutils/locale/converters/FloatLocaleConverter.java
@@ -218,7 +218,7 @@ public class FloatLocaleConverter extends DecimalLocaleConverter {
       final Number parsed = (Number) super.parse(value, pattern);
       double doubleValue = parsed.doubleValue();
       double posDouble = (doubleValue >= (double)0) ? doubleValue : (doubleValue * (double)-1);
-      if (posDouble != 0 && posDouble < Float.MIN_VALUE || posDouble > Float.MAX_VALUE) {
+      if (posDouble < Float.MIN_VALUE || posDouble > Float.MAX_VALUE) {
           throw new ConversionException("Supplied number is not of type Float: "+parsed);
       }
       return new Float(parsed.floatValue()); // unlike superclass it returns Float type
