diff --git a/src/test/org/apache/commons/beanutils/locale/converters/FloatLocaleConverterTestCase.java b/src/test/org/apache/commons/beanutils/locale/converters/FloatLocaleConverterTestCase.java
index 530c8fed..a0069c3a 100644
--- a/src/test/org/apache/commons/beanutils/locale/converters/FloatLocaleConverterTestCase.java
+++ b/src/test/org/apache/commons/beanutils/locale/converters/FloatLocaleConverterTestCase.java
@@ -18,10 +18,7 @@
 package org.apache.commons.beanutils.locale.converters;
 
 import java.text.DecimalFormat;
-import java.util.Locale;
-
 import org.apache.commons.beanutils.ConversionException;
-import org.apache.commons.beanutils.locale.LocaleConvertUtils;
 
 /**
  * Test Case for the FloatLocaleConverter class.
@@ -306,18 +303,6 @@ public class FloatLocaleConverterTestCase extends BaseLocaleConverterTestCase {
         }
     }
 
-    /**
-     * Test parsing zero - see BEANUTILS-351
-     */
-    public void testParseZero() {
-        try {
-            Object result = LocaleConvertUtils.convert("0", Float.class, Locale.US, null);
-            assertEquals(new Float(0), result);
-        } catch (ConversionException e) {
-            fail("Zero threw ConversionException: " + e);
-        }
-        
-    }
 
 
 }
