diff --git a/src/java/org/apache/commons/beanutils/BeanUtilsBean.java b/src/java/org/apache/commons/beanutils/BeanUtilsBean.java
index 0fffebe6..a9ac122d 100644
--- a/src/java/org/apache/commons/beanutils/BeanUtilsBean.java
+++ b/src/java/org/apache/commons/beanutils/BeanUtilsBean.java
@@ -927,8 +927,6 @@ public class BeanUtilsBean {
             type = dynaProperty.getType();
         } else if (target instanceof Map) {
             type = Object.class;
-        } else if (target.getClass().isArray() && index >= 0) {
-            type = Array.get(target, index).getClass();
         } else {
             PropertyDescriptor descriptor = null;
             try {
