diff --git a/src/main/java/org/apache/commons/beanutils/MethodUtils.java b/src/main/java/org/apache/commons/beanutils/MethodUtils.java
index 75a6e4fb..d01191da 100644
--- a/src/main/java/org/apache/commons/beanutils/MethodUtils.java
+++ b/src/main/java/org/apache/commons/beanutils/MethodUtils.java
@@ -1119,13 +1119,6 @@ public class MethodUtils {
     private static float getObjectTransformationCost(Class srcClass, Class destClass) {
         float cost = 0.0f;
         while (srcClass != null && !destClass.equals(srcClass)) {
-            if (destClass.isPrimitive()) {
-                Class destClassWrapperClazz = getPrimitiveWrapper(destClass);
-                if (destClassWrapperClazz != null && destClassWrapperClazz.equals(srcClass)) {
-                    cost += 0.25f;
-                    break;
-                }
-            }
             if (destClass.isInterface() && isAssignmentCompatible(destClass,srcClass)) {
                 // slight penalty for interface match. 
                 // we still want an exact match to override an interface match, but  
