diff --git a/src/test/org/apache/commons/beanutils/locale/converters/DateLocaleConverterTestCase.java b/src/test/org/apache/commons/beanutils/locale/converters/DateLocaleConverterTestCase.java
index 61f7e4dd..4fa1ec9b 100644
--- a/src/test/org/apache/commons/beanutils/locale/converters/DateLocaleConverterTestCase.java
+++ b/src/test/org/apache/commons/beanutils/locale/converters/DateLocaleConverterTestCase.java
@@ -409,26 +409,5 @@ public class DateLocaleConverterTestCase extends BaseLocaleConverterTestCase {
 
     }
 
-    /**
-     * Test invalid date
-     */
-    public void testInvalidDate() {
-
-        converter = new DateLocaleConverter(defaultLocale);
-
-        try {
-            converter.convert("01/10/2004", "dd-MM-yyyy");
-        } catch (ConversionException e) {
-            assertEquals("Parse Error", "Error parsing date '01/10/2004' at position=2", e.getMessage());
-        }
-
-        try {
-            converter.convert("01-10-2004X", "dd-MM-yyyy");
-        } catch (ConversionException e) {
-            assertEquals("Parse Length", "Date '01-10-2004X' contains unparsed characters from position=10", e.getMessage());
-        }
-
-    }
-
 }
 
