diff --git a/src/test/java/org/apache/commons/beanutils/MethodUtilsTestCase.java b/src/test/java/org/apache/commons/beanutils/MethodUtilsTestCase.java
index a02027f9..7ad4f32d 100644
--- a/src/test/java/org/apache/commons/beanutils/MethodUtilsTestCase.java
+++ b/src/test/java/org/apache/commons/beanutils/MethodUtilsTestCase.java
@@ -185,13 +185,6 @@ public class MethodUtilsTestCase extends TestCase {
         }
     }
 
-    public void testInvokeExactMethodNull() throws Exception {
-        Object object = new Object();
-        Object result = MethodUtils.invokeExactMethod(object, "toString", (Object) null);
-        assertEquals(object.toString(), result);
-    }
-
-
     /**
      * <p> Test <code>invokeMethod</code>.
      */
@@ -268,12 +261,6 @@ public class MethodUtilsTestCase extends TestCase {
         MethodUtils.invokeExactMethod(parent, "getName", null, null);
     }
 
-    public void testInvokeMethodNull() throws Exception {
-        Object object = new Object();
-        Object result = MethodUtils.invokeMethod(object, "toString", (Object) null);
-        assertEquals(object.toString(), result);
-    }
-
     /**
      * <p> Test <code>invokeMethod</code> with a primitive.
      */
@@ -341,17 +328,6 @@ public class MethodUtilsTestCase extends TestCase {
         assertEquals("currentCounter value", current, ((Integer) value).intValue());
     }
 
-    public void testInvokeStaticMethodNull() throws Exception {
-        int current = TestBean.currentCounter();
-        Object value = MethodUtils.invokeStaticMethod(TestBean.class, "currentCounter", (Object) null);
-        assertEquals("currentCounter value", current, ((Integer) value).intValue());
-    }
-
-    public void testInvokeExactStaticMethodNull() throws Exception {
-        int current = TestBean.currentCounter();
-        Object value = MethodUtils.invokeExactStaticMethod(TestBean.class, "currentCounter", (Object) null);
-        assertEquals("currentCounter value", current, ((Integer) value).intValue());
-    }
 
     /**
      * Simple tests for accessing static methods via invokeMethod().
