diff --git a/src/main/java/org/apache/commons/beanutils/BeanUtilsBean.java b/src/main/java/org/apache/commons/beanutils/BeanUtilsBean.java
index fbdf9270..be3a4f5e 100644
--- a/src/main/java/org/apache/commons/beanutils/BeanUtilsBean.java
+++ b/src/main/java/org/apache/commons/beanutils/BeanUtilsBean.java
@@ -921,9 +921,6 @@ public class BeanUtilsBean {
                 return; // Skip this property setter
             }
             type = dynaPropertyType(dynaProperty, value);
-            if (index >= 0 && List.class.isAssignableFrom(type)) {
-            	type = Object.class;
-            }
         } else if (target instanceof Map) {
             type = Object.class;
         } else if (target != null && target.getClass().isArray() && index >= 0) {
