diff --git a/burst/src/main/java/com/squareup/burst/Burst.java b/burst/src/main/java/com/squareup/burst/Burst.java
index 38ee10a..80681eb 100644
--- a/burst/src/main/java/com/squareup/burst/Burst.java
+++ b/burst/src/main/java/com/squareup/burst/Burst.java
@@ -20,6 +20,16 @@ public final class Burst {
     if (constructors.length == 1) {
       return constructors[0];
     }
+    if (constructors.length == 0) {
+      constructors = cls.getDeclaredConstructors();
+      if (constructors.length == 1) {
+        Constructor<?> constructor = constructors[0];
+        if (constructor.getParameterCount() == 0) {
+          constructor.setAccessible(true);
+          return constructor;
+        }
+      }
+    }
     throw new IllegalStateException(cls.getName() + " requires a single public constructor.");
   }
 
