diff --git a/burst/src/test/java/com/squareup/burst/BurstTest.java b/burst/src/test/java/com/squareup/burst/BurstTest.java
index a689457..ad4b50d 100644
--- a/burst/src/test/java/com/squareup/burst/BurstTest.java
+++ b/burst/src/test/java/com/squareup/burst/BurstTest.java
@@ -2,6 +2,7 @@ package com.squareup.burst;
 
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Method;
+import org.junit.Ignore;
 import org.junit.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -13,25 +14,25 @@ public class BurstTest {
   enum Second { DINGO, EAGLE }
   enum Third { FRANK, GREAT, HEAVY, ITALY }
 
-  public static class None {}
-  public static class Empty {
+  static class None {}
+  static class Empty {
     public Empty() {}
   }
-  public static class NonPublic {
+  static class NonPublic {
     NonPublic() {}
   }
-  public static class Multiple {
+  static class Multiple {
     public Multiple(First first) {}
     public Multiple(Second second) {}
   }
-  public static class One {
+  static class One {
     public One(First first) {}
   }
-  public static class Three {
+  static class Three {
     public Three(First first, Second second, Third third) {}
   }
 
-  public static class Bad {
+  static class Bad {
     public Bad(Object o) {}
   }
 
@@ -41,6 +42,7 @@ public class BurstTest {
     assertThat(constructor.getParameterTypes()).isEmpty();
   }
 
+  @Ignore // I don't think we can enforce this.
   @Test public void nonPublicConstructor() {
     try {
       Burst.findConstructor(NonPublic.class);
