diff --git a/burst/src/main/java/com/squareup/burst/Burst.java b/burst/src/main/java/com/squareup/burst/Burst.java
index c7407a0..d64b334 100644
--- a/burst/src/main/java/com/squareup/burst/Burst.java
+++ b/burst/src/main/java/com/squareup/burst/Burst.java
@@ -65,21 +65,22 @@ public final class Burst {
     StringBuilder builder = new StringBuilder(name);
     for (Object argument : constructorArgs) {
       Enum<?> value = (Enum<?>) argument;
       // Appends the enum name and value name. (e.g., Card.VISA)
-      builder.append('_').append(getQualifiedEnumName(value));
+      builder.append('_') //
+          .append(value.getClass().getSimpleName()) //
+          .append(value.name());
     }
     for (Object argument : methodArgs) {
       Enum<?> value = (Enum<?>) argument;
       // Appends the enum name and value name. (e.g., Card.VISA)
-      builder.append('_').append(getQualifiedEnumName(value));
+      builder.append('_') //
+          .append(value.getClass().getSimpleName()) //
+          .append(value.name());
     }
     return builder.toString();
   }
 
-  private static String getQualifiedEnumName(Enum<?> value) {
-    return value.getClass().getSimpleName() + '.' + value;
-  }
 
   private static Object[][] explodeParameters(Class<?>[] parameterTypes, String name) {
     int parameterCount = parameterTypes.length;
     if (parameterCount == 0) {
