diff --git a/burst/src/main/java/com/squareup/burst/BurstableConstructor.java b/burst/src/main/java/com/squareup/burst/BurstableConstructor.java
index 334f1fc..bfb8189 100644
--- a/burst/src/main/java/com/squareup/burst/BurstableConstructor.java
+++ b/burst/src/main/java/com/squareup/burst/BurstableConstructor.java
@@ -25,7 +25,7 @@ final class BurstableConstructor {
     } else if (constructors.length == 1) {
       return constructors[0];
     } else {
-      throw new IllegalStateException("Class "
+      return singleParameterizedConstructorOrThrow(constructors, "Class "
           + cls.getName()
           + " has too many parameterized constructors. "
           + "Should only be 1 (with enum variations).");
@@ -56,6 +56,23 @@ final class BurstableConstructor {
 
     return filteredConstructors.toArray(new TestConstructor[filteredConstructors.size()]);
   }
+  private static TestConstructor singleParameterizedConstructorOrThrow(
+      final TestConstructor[] constructors, final String message) {
+    TestConstructor parameterizedConstructor = null;
+    for (TestConstructor constructor : constructors) {
+      if (constructor.getVariationTypes().length > 0) {
+        if (null == parameterizedConstructor) {
+          parameterizedConstructor = constructor;
+        } else {
+          throw new IllegalStateException(message);
+        }
+      }
+    }
+    if (null == parameterizedConstructor) {
+      throw new IllegalArgumentException("Must pass at least one parameterized constructor");
+    }
+    return parameterizedConstructor;
+  }
 
   /**
    * Get all the {@link Burst}-annotated fields.
