diff --git a/nifi-mock/src/test/java/org/apache/nifi/util/TestStandardProcessorTestRunner.java b/nifi-mock/src/test/java/org/apache/nifi/util/TestStandardProcessorTestRunner.java
index f2febb48f5..134d2b8328 100644
--- a/nifi-mock/src/test/java/org/apache/nifi/util/TestStandardProcessorTestRunner.java
+++ b/nifi-mock/src/test/java/org/apache/nifi/util/TestStandardProcessorTestRunner.java
@@ -381,23 +381,4 @@ public class TestStandardProcessorTestRunner {
             return Arrays.asList(namePropertyDescriptor);
         }
     }
-
-    @Test
-    public void testErrorLogMessageArguments() {
-
-        String compName = "name of component";
-        final MockComponentLog logger = new MockComponentLog("first id",compName);
-
-        Throwable t = new ArithmeticException();
-        logger.error("expected test error",t);
-
-        String expected_throwable = "java.lang.ArithmeticException";
-
-        List<LogMessage>  log = logger.getErrorMessages();
-        LogMessage msg = log.get(0);
-        // checking if the error messages are recorded in the correct throwable argument.
-        assertEquals(expected_throwable,msg.getThrowable().toString());
-        assertEquals("{} expected test error",msg.getMsg());
-
-    }
 }
