diff --git a/src/main/java/org/apache/commons/imaging/formats/png/PngImageParser.java b/src/main/java/org/apache/commons/imaging/formats/png/PngImageParser.java
index 4b1b57b..00a81a0 100644
--- a/src/main/java/org/apache/commons/imaging/formats/png/PngImageParser.java
+++ b/src/main/java/org/apache/commons/imaging/formats/png/PngImageParser.java
@@ -154,9 +154,6 @@ public class PngImageParser extends ImageParser {
 
         while (true) {
             final int length = read4Bytes("Length", is, "Not a Valid PNG File", getByteOrder());
-            if (length < 0) {
-                throw new ImageReadException("Invalid PNG chunk length: " + length);
-            }
             final int chunkType = read4Bytes("ChunkType", is, "Not a Valid PNG File", getByteOrder());
 
             if (LOGGER.isLoggable(Level.FINEST)) {
