diff --git a/src/test/java/org/apache/commons/imaging/formats/png/PngWithInvalidPngChunkSizeTest.java b/src/test/java/org/apache/commons/imaging/formats/png/PngWithInvalidPngChunkSizeTest.java
index 10de8022..faa5d658 100644
--- a/src/test/java/org/apache/commons/imaging/formats/png/PngWithInvalidPngChunkSizeTest.java
+++ b/src/test/java/org/apache/commons/imaging/formats/png/PngWithInvalidPngChunkSizeTest.java
@@ -50,19 +50,4 @@ public class PngWithInvalidPngChunkSizeTest {
 		Imaging.getBufferedImage(imageFile, params);
 	}
 
-	/**
-     * Test that an image with an invalid negative PNG chunk size causes an
-     * ImageReadException instead of other exception types.
-     *
-     * @throws IOException        if it fails to read from the input source
-     * @throws ImageReadException if it fails to read the image
-     */
-    @Test(expected = ImageReadException.class)
-    public void testPngWithInvalidNegativePngChunkSize() throws IOException, ImageReadException {
-        final File imageFile = new File(
-                JpegWithInvalidDhtSegmentTest.class.getResource("/IMAGING-210/testfile.png").getFile());
-        final Map<String, Object> params = new HashMap<>();
-        params.put(ImagingConstants.BUFFERED_IMAGE_FACTORY, new ManagedImageBufferedImageFactory());
-        Imaging.getBufferedImage(imageFile, params);
-    }
 }
