diff --git a/src/main/java/org/apache/commons/imaging/common/mylzw/MyLzwDecompressor.java b/src/main/java/org/apache/commons/imaging/common/mylzw/MyLzwDecompressor.java
index aa056866..eb7fd144 100644
--- a/src/main/java/org/apache/commons/imaging/common/mylzw/MyLzwDecompressor.java
+++ b/src/main/java/org/apache/commons/imaging/common/mylzw/MyLzwDecompressor.java
@@ -108,6 +108,8 @@ public final class MyLzwDecompressor {
         if (codes < (1 << codeSize)) {
             table[codes] = bytes;
             codes++;
+        } else {
+            throw new IOException("AddStringToTable: codes: " + codes + " code_size: " + codeSize);
         }
         // If the table already full, then we simply ignore these bytes
         // per the https://www.w3.org/Graphics/GIF/spec-gif89a.txt 'spec'.
