diff --git a/src/main/java/org/apache/commons/imaging/formats/pnm/PnmImageParser.java b/src/main/java/org/apache/commons/imaging/formats/pnm/PnmImageParser.java
index 05308eb..b2dc074 100644
--- a/src/main/java/org/apache/commons/imaging/formats/pnm/PnmImageParser.java
+++ b/src/main/java/org/apache/commons/imaging/formats/pnm/PnmImageParser.java
@@ -102,17 +102,9 @@ public class PnmImageParser extends ImageParser {
                 || identifier2 == PnmConstants.PPM_RAW_CODE) {
             
             final int width;
-            try {
               width = Integer.parseInt(wsr.readtoWhiteSpace());
-            } catch (NumberFormatException e) {
-              throw new ImageReadException("Invalid width specified." , e);
-            }
             final int height;
-            try {
               height = Integer.parseInt(wsr.readtoWhiteSpace());
-            } catch (NumberFormatException e) {
-              throw new ImageReadException("Invalid height specified." , e);
-            }
     
             if (identifier2 == PnmConstants.PBM_TEXT_CODE) {
                 return new PbmFileInfo(width, height, false);
