diff --git a/src/test/java/org/apache/commons/imaging/formats/pnm/PnmImageParserTest.java b/src/test/java/org/apache/commons/imaging/formats/pnm/PnmImageParserTest.java
index 59e5fe62..066ed37b 100644
--- a/src/test/java/org/apache/commons/imaging/formats/pnm/PnmImageParserTest.java
+++ b/src/test/java/org/apache/commons/imaging/formats/pnm/PnmImageParserTest.java
@@ -46,12 +46,4 @@ public class PnmImageParserTest {
     PnmImageParser underTest = new PnmImageParser();
     underTest.getImageInfo(bytes, params);
   }

-  @Test(expected = ImageReadException.class)
-  public void testGetImageInfo_missingWidthValue() throws ImageReadException, IOException {
-    byte[] bytes = "P7\nWIDTH \n".getBytes(US_ASCII);
-    Map<String, Object> params = Collections.emptyMap();
-    PnmImageParser underTest = new PnmImageParser();
-    underTest.getImageInfo(bytes, params);
-  }
 }
