diff --git a/src/main/java/org/apache/commons/imaging/formats/tiff/TiffImageParser.java b/src/main/java/org/apache/commons/imaging/formats/tiff/TiffImageParser.java
index df1fd8c6..2d204ed7 100644
--- a/src/main/java/org/apache/commons/imaging/formats/tiff/TiffImageParser.java
+++ b/src/main/java/org/apache/commons/imaging/formats/tiff/TiffImageParser.java
@@ -259,11 +259,8 @@ public class TiffImageParser extends ImageParser {
 
         final ImageInfo.ColorType colorType = ImageInfo.ColorType.RGB;
 
-        final short[] compressionFieldValues = directory
-            .getFieldValue(TiffTagConstants.TIFF_TAG_COMPRESSION, false);
-        final short compressionFieldValue = compressionFieldValues != null ?
-            compressionFieldValues[0] : TIFF_COMPRESSION_UNCOMPRESSED_1;
-        final int compression = 0xffff & compressionFieldValue;
+        final int compression = 0xffff & directory
+                .getSingleFieldValue(TiffTagConstants.TIFF_TAG_COMPRESSION);
         ImageInfo.CompressionAlgorithm compressionAlgorithm;
 
         switch (compression) {
@@ -555,11 +552,7 @@ public class TiffImageParser extends ImageParser {
 
         final int photometricInterpretation = 0xffff & directory.getSingleFieldValue(
                 TiffTagConstants.TIFF_TAG_PHOTOMETRIC_INTERPRETATION);
-        final short[] compressionFieldValues = directory
-            .getFieldValue(TiffTagConstants.TIFF_TAG_COMPRESSION, false);
-        final short compressionFieldValue = compressionFieldValues != null ?
-                compressionFieldValues[0] : TIFF_COMPRESSION_UNCOMPRESSED_1;
-        final int compression = 0xffff & compressionFieldValue;
+        final int compression = 0xffff & directory.getSingleFieldValue(TiffTagConstants.TIFF_TAG_COMPRESSION);
         final int width = directory.getSingleFieldValue(TiffTagConstants.TIFF_TAG_IMAGE_WIDTH);
         final int height = directory.getSingleFieldValue(TiffTagConstants.TIFF_TAG_IMAGE_LENGTH);      
         
