diff --git a/src/main/java/org/apache/commons/imaging/formats/rgbe/RgbeInfo.java b/src/main/java/org/apache/commons/imaging/formats/rgbe/RgbeInfo.java
index 4038165f..e09e46ab 100644
--- a/src/main/java/org/apache/commons/imaging/formats/rgbe/RgbeInfo.java
+++ b/src/main/java/org/apache/commons/imaging/formats/rgbe/RgbeInfo.java
@@ -169,17 +169,14 @@ class RgbeInfo implements Closeable {
     }
 
     private static void decompress(final InputStream in, final byte[] out)
-            throws IOException,ImageReadException {
+            throws IOException {
         int position = 0;
         final int total = out.length;
 
         while (position < total) {
             final int n = in.read();
 
-            if (n < 0) {
-                throw new ImageReadException("Error decompressing RGBE file");
-            }

             if (n > 128) {
                 final int value = in.read();
 
