diff --git a/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/WrapperHandlingDeserializer.java b/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/WrapperHandlingDeserializer.java
index 57b11a94..5a1d8a99 100644
--- a/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/WrapperHandlingDeserializer.java
+++ b/src/main/java/com/fasterxml/jackson/dataformat/xml/deser/WrapperHandlingDeserializer.java
@@ -4,7 +4,6 @@ import java.io.IOException;
 import java.util.*;
 
 import com.fasterxml.jackson.core.JsonParser;
-import com.fasterxml.jackson.core.util.JsonParserDelegate;
 import com.fasterxml.jackson.databind.*;
 import com.fasterxml.jackson.databind.deser.*;
 import com.fasterxml.jackson.databind.deser.std.DelegatingDeserializer;
@@ -137,15 +136,11 @@ public class WrapperHandlingDeserializer
     /**********************************************************************
      */
 
-    @SuppressWarnings("resource")
     protected final void _configureParser(JsonParser p) throws IOException
     {
         // 05-Sep-2019, tatu: May get XML parser, except for case where content is
         //   buffered. In that case we may still have access to real parser if we
         //   are lucky (like in [dataformat-xml#242])
-        while (p instanceof JsonParserDelegate) {
-            p = ((JsonParserDelegate) p).delegate();
-        }
         if (p instanceof FromXmlParser) {
             ((FromXmlParser) p).addVirtualWrapping(_namesToWrap);
         }
