diff --git a/uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientSecurityExceptionMessageResolver.java b/uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientSecurityExceptionMessageResolver.java
index b62a33e26..884206e6b 100644
--- a/uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientSecurityExceptionMessageResolver.java
+++ b/uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/ClientSecurityExceptionMessageResolver.java
@@ -21,14 +21,13 @@ import java.util.Map;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Predicate;
 
 import javax.annotation.PostConstruct;
 import javax.enterprise.context.ApplicationScoped;
 
 import org.uberfire.ext.security.management.api.exception.EntityNotFoundException;
 import org.uberfire.ext.security.management.api.exception.GroupNotFoundException;
 import org.uberfire.ext.security.management.api.exception.NoImplementationAvailableException;
-import org.uberfire.ext.security.management.api.exception.RealmManagementNotAuthorizedException;
 import org.uberfire.ext.security.management.api.exception.SecurityManagementException;
 import org.uberfire.ext.security.management.api.exception.UnsupportedServiceCapabilityException;
 import org.uberfire.ext.security.management.api.exception.UserAlreadyExistsException;
@@ -65,9 +63,6 @@ public class ClientSecurityExceptionMessageResolver {
         register(UserAlreadyExistsException.class,
                  e -> getArgMessage(UsersManagementClientConstants.INSTANCE.userAlreadyExists(),
                                     e.getUserId()));
-        register(RealmManagementNotAuthorizedException.class,
-                 e -> getArgMessage(UsersManagementClientConstants.INSTANCE.realmManagementNotAuthorized(),
-                                    e.getRealmResource()));
     }
 
     /**
diff --git a/uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/resources/i18n/UsersManagementClientConstants.java b/uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/resources/i18n/UsersManagementClientConstants.java
index 7862fa2ed..74e352456 100644
--- a/uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/resources/i18n/UsersManagementClientConstants.java
+++ b/uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-client/src/main/java/org/uberfire/ext/security/management/client/resources/i18n/UsersManagementClientConstants.java
@@ -47,6 +47,5 @@ public interface UsersManagementClientConstants extends ConstantsWithLookup {
     String userNotFound();
 
     String groupNotFound();
 
-    String realmManagementNotAuthorized();
 }
