diff --git a/tiles-core/src/main/java/org/apache/tiles/definition/digester/DigesterDefinitionsReader.java b/tiles-core/src/main/java/org/apache/tiles/definition/digester/DigesterDefinitionsReader.java
index f10e6f29..80289b66 100644
--- a/tiles-core/src/main/java/org/apache/tiles/definition/digester/DigesterDefinitionsReader.java
+++ b/tiles-core/src/main/java/org/apache/tiles/definition/digester/DigesterDefinitionsReader.java
@@ -162,8 +162,7 @@ public class DigesterDefinitionsReader implements DefinitionsReader {
             Definition definition = (Definition) digester.peek();
             definition.setName(attributes.getValue("name"));
             definition.setPreparer(attributes.getValue("preparer"));
-            String extendsAttribute = attributes.getValue("extends");
-            definition.setExtends(extendsAttribute);
+            definition.setExtends(attributes.getValue("extends"));
 
             String template = attributes.getValue("template");
             Attribute attribute = Attribute.createTemplateAttribute(template);
@@ -174,8 +173,6 @@ public class DigesterDefinitionsReader implements DefinitionsReader {
             String templateType = attributes.getValue("templateType");
             if (templateType != null) {
                 attribute.setRenderer(templateType);
-            } else if (extendsAttribute != null && templateType == null) {
-                attribute.setRenderer(null);
             }
             definition.setTemplateAttribute(attribute);
         }
