diff --git a/src/main/java/com/fasterxml/jackson/annotation/JsonInclude.java b/src/main/java/com/fasterxml/jackson/annotation/JsonInclude.java
index 9a35539..941eeec 100644
--- a/src/main/java/com/fasterxml/jackson/annotation/JsonInclude.java
+++ b/src/main/java/com/fasterxml/jackson/annotation/JsonInclude.java
@@ -243,15 +243,13 @@ public @interface JsonInclude
             Include ci = overrides._contentInclusion;
 
             boolean viDiff = (vi != _valueInclusion) && (vi != Include.USE_DEFAULTS);
-            boolean ciDiff = (ci != _contentInclusion) && (ci != Include.USE_DEFAULTS);
+            boolean ciDiff = (ci != _valueInclusion) && (ci != Include.USE_DEFAULTS);
 
             if (viDiff) {
                 if (ciDiff) {
                     return new Value(vi, ci);
                 }
                 return new Value(vi, _contentInclusion);
-            } else if (ciDiff) {
-                return new Value(_valueInclusion, ci);
             }
             return this;
         }
