diff --git a/src/main/java/org/eluder/coveralls/maven/plugin/service/Wercker.java b/src/main/java/org/eluder/coveralls/maven/plugin/service/Wercker.java
index f16dba2..969c37c 100644
--- a/src/main/java/org/eluder/coveralls/maven/plugin/service/Wercker.java
+++ b/src/main/java/org/eluder/coveralls/maven/plugin/service/Wercker.java
@@ -36,9 +36,7 @@ import java.util.Map;
 public class Wercker extends AbstractServiceSetup {
 
     public static final String WERCKER_NAME = "wercker";
-    public static final String WERCKER = "WERCKER";
-    public static final String WERCKER_BUILD_ID = "WERCKER_BUILD_ID";
-    public static final String WERCKER_BUILD_URL = "WERCKER_BUILD_URL";
+    public static final String WERCKER_BUILD_URL = "WERCKER_RUN_URL";
     public static final String WERCKER_BRANCH = "WERCKER_GIT_BRANCH";
 
     public Wercker(final Map<String, String> env) {
@@ -47,7 +45,7 @@ public class Wercker extends AbstractServiceSetup {
 
     @Override
     public boolean isSelected() {
-        return "true".equalsIgnoreCase(getProperty(WERCKER));
+        return getProperty(WERCKER_BRANCH) != null;
     }
 
     @Override
@@ -57,7 +55,8 @@ public class Wercker extends AbstractServiceSetup {
 
     @Override
     public String getJobId() {
-        return getProperty(WERCKER_BUILD_ID);
+        final String buildUrl = getBuildUrl();
+        return buildUrl.substring(buildUrl.lastIndexOf("/") + 1);
     }
 
     @Override
