diff --git a/src/main/java/com/github/pagehelper/dialect/AbstractHelperDialect.java b/src/main/java/com/github/pagehelper/dialect/AbstractHelperDialect.java
index 41e7507..e2a402f 100644
--- a/src/main/java/com/github/pagehelper/dialect/AbstractHelperDialect.java
+++ b/src/main/java/com/github/pagehelper/dialect/AbstractHelperDialect.java
@@ -70,7 +70,7 @@ public abstract class AbstractHelperDialect extends AbstractDialect implements C
     @Override
     public boolean beforeCount(MappedStatement ms, Object parameterObject, RowBounds rowBounds) {
         Page page = getLocalPage();
-        return !page.isOrderByOnly() && page.isCount();
+        return page.isCount();
     }
 
     @Override
@@ -159,7 +159,7 @@ public abstract class AbstractHelperDialect extends AbstractDialect implements C
     @Override
     public boolean beforePage(MappedStatement ms, Object parameterObject, RowBounds rowBounds) {
         Page page = getLocalPage();
-        if (page.isOrderByOnly() || page.getPageSize() > 0) {
+        if (page.getPageSize() > 0) {
             return true;
         }
         return false;
@@ -202,8 +202,6 @@ public abstract class AbstractHelperDialect extends AbstractDialect implements C
             page.setTotal(-1);
         } else if ((page.getPageSizeZero() != null && page.getPageSizeZero()) && page.getPageSize() == 0) {
             page.setTotal(pageList.size());
-        } else if(page.isOrderByOnly()){
-            page.setTotal(pageList.size());
         }
         return page;
     }
