diff --git a/src/main/java/com/github/pagehelper/Page.java b/src/main/java/com/github/pagehelper/Page.java
index f996014..d3db722 100644
--- a/src/main/java/com/github/pagehelper/Page.java
+++ b/src/main/java/com/github/pagehelper/Page.java
@@ -124,7 +124,7 @@ public class Page<E> extends ArrayList<E> implements Closeable {
             this.pageSize = 0;
         } else {
             this.pageSize = rowBounds[1];
-            this.pageNum = rowBounds[1] != 0 ? (int) (Math.ceil(((double) rowBounds[0] + rowBounds[1]) / rowBounds[1])) : 0;
+            this.pageNum = rowBounds[1] != 0 ? (int) (Math.floor(((double) rowBounds[0] + rowBounds[1]) / rowBounds[1])) : 0;
         }
         this.startRow = rowBounds[0];
         this.count = count;
