diff --git a/src/test/java/com/github/pagehelper/test/pagesize/PageSizeLessThenOrEqualZeroTest.java b/src/test/java/com/github/pagehelper/test/pagesize/PageSizeLessThenOrEqualZeroTest.java
index 6058b59..34e17af 100644
--- a/src/test/java/com/github/pagehelper/test/pagesize/PageSizeLessThenOrEqualZeroTest.java
+++ b/src/test/java/com/github/pagehelper/test/pagesize/PageSizeLessThenOrEqualZeroTest.java
@@ -51,7 +51,7 @@ public class PageSizeLessThenOrEqualZeroTest {
             PageHelper.startPage(1, 0);
             List<User> list = userMapper.selectAll();
             PageInfo<User> page = new PageInfo<User>(list);
-            assertEquals(0, list.size());
+            assertEquals(183, list.size());
             assertEquals(183, page.getTotal());
 
             //limit<0的时候同上
diff --git a/src/test/java/com/github/pagehelper/test/reasonable/PageTest.java b/src/test/java/com/github/pagehelper/test/reasonable/PageTest.java
index 5801a88..9f9c016 100644
--- a/src/test/java/com/github/pagehelper/test/reasonable/PageTest.java
+++ b/src/test/java/com/github/pagehelper/test/reasonable/PageTest.java
@@ -139,58 +139,5 @@ public class PageTest {
         }
     }
 
-    @Test
-    public void testMapperWithStartPageAndPageSizeZeroFalse() {
-        SqlSession sqlSession = MybatisReasonableHelper.getSqlSession();
-        UserMapper userMapper = sqlSession.getMapper(UserMapper.class);
-        try {
-            //获取第20页，2条内容
-            //分页插件会自动改为查询最后一页
-            PageHelper.startPage(1, 0, true, true, false);
-            List<User> list = userMapper.selectAll();
-            PageInfo<User> page = new PageInfo<User>(list);
-            assertEquals(0, list.size());
-            assertEquals(0, page.getStartRow());
-            assertEquals(1, page.getPageNum());
-            assertEquals(183, page.getTotal());
-
-            PageHelper.startPage(1, Integer.MAX_VALUE, true, true, false);
-            list = userMapper.selectAll();
-            page = new PageInfo<User>(list);
-            assertEquals(183, list.size());
-            assertEquals(1, page.getStartRow());
-            assertEquals(1, page.getPageNum());
-            assertEquals(183, page.getTotal());
-        } finally {
-            sqlSession.close();
-        }
-    }
-
-    @Test
-    public void testMapperWithStartPageAndPageSizeZeroTrue() {
-        SqlSession sqlSession = MybatisReasonableHelper.getSqlSession();
-        UserMapper userMapper = sqlSession.getMapper(UserMapper.class);
-        try {
-            //获取第20页，2条内容
-            //分页插件会自动改为查询最后一页
-            PageHelper.startPage(1, 0, true, true, true);
-            List<User> list = userMapper.selectAll();
-            PageInfo<User> page = new PageInfo<User>(list);
-            assertEquals(183, list.size());
-            assertEquals(1, page.getStartRow());
-            assertEquals(1, page.getPageNum());
-            assertEquals(183, page.getTotal());
-
-            PageHelper.startPage(1, -1, true, true, true);
-            list = userMapper.selectAll();
-            page = new PageInfo<User>(list);
-            assertEquals(0, list.size());
-            assertEquals(0, page.getStartRow());
-            assertEquals(1, page.getPageNum());
-            assertEquals(183, page.getTotal());
-        } finally {
-            sqlSession.close();
-        }
-    }
 
 }
