diff --git a/src/com/google/javascript/jscomp/CodeConsumer.java b/src/com/google/javascript/jscomp/CodeConsumer.java
index 2f45b5c..8023324 100644
--- a/src/com/google/javascript/jscomp/CodeConsumer.java
+++ b/src/com/google/javascript/jscomp/CodeConsumer.java
@@ -241,7 +241,7 @@ abstract class CodeConsumer {
     // This is not pretty printing. This is to prevent misparsing of x- -4 as
     // x--4 (which is a syntax error).
     char prev = getLastChar();
-    if ((x < 0 || isNegativeZero(x)) && prev == '-') {
+    if (x < 0 && prev == '-') {
       add(" ");
     }
 
