diff --git a/tika-core/src/test/java/org/apache/tika/sax/XHTMLContentHandlerTest.java b/tika-core/src/test/java/org/apache/tika/sax/XHTMLContentHandlerTest.java
index b4cfa7558..e48e5a4cd 100644
--- a/tika-core/src/test/java/org/apache/tika/sax/XHTMLContentHandlerTest.java
+++ b/tika-core/src/test/java/org/apache/tika/sax/XHTMLContentHandlerTest.java
@@ -17,12 +17,10 @@
 package org.apache.tika.sax;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
 
 import java.util.ArrayList;
 import java.util.List;
 
-import org.apache.tika.config.TikaConfigTest;
 import org.apache.tika.metadata.Metadata;
 
 import org.junit.Before;
@@ -30,7 +28,6 @@ import org.junit.Test;
 
 import org.xml.sax.ContentHandler;
 import org.xml.sax.SAXException;
-import org.xml.sax.helpers.AttributesImpl;
 
 /**
  * Unit tests for the {@link XHTMLContentHandler} class.
@@ -124,24 +121,6 @@ public class XHTMLContentHandlerTest {
         assertEquals("two", words[1]);
     }
 
-    @Test
-    public void testAttributesOnBody() throws Exception {
-        ToHTMLContentHandler toHTMLContentHandler = new ToHTMLContentHandler();
-        XHTMLContentHandler xhtmlContentHandler = new XHTMLContentHandler(toHTMLContentHandler, new Metadata());
-        AttributesImpl attributes = new AttributesImpl();
-
-        attributes.addAttribute(XHTMLContentHandler.XHTML, "itemscope", "itemscope", "", "");
-        attributes.addAttribute(XHTMLContentHandler.XHTML, "itemtype", "itemtype", "", "http://schema.org/Event");
-
-        xhtmlContentHandler.startDocument();
-        xhtmlContentHandler.startElement(XHTMLContentHandler.XHTML, "body", "body", attributes);
-        xhtmlContentHandler.endElement("body");
-        xhtmlContentHandler.endDocument();
-
-        System.err.println("Content: " + toHTMLContentHandler.toString());
-        assertTrue(toHTMLContentHandler.toString().contains("itemscope"));
-    }
-
     /**
      * Return array of non-zerolength words. Splitting on whitespace will get us
      * empty words for emptylines.
