diff --git a/tika-core/src/main/java/org/apache/tika/detect/NameDetector.java b/tika-core/src/main/java/org/apache/tika/detect/NameDetector.java
index 71354938c..1638d50dd 100644
--- a/tika-core/src/main/java/org/apache/tika/detect/NameDetector.java
+++ b/tika-core/src/main/java/org/apache/tika/detect/NameDetector.java
@@ -97,11 +97,15 @@ public class NameDetector implements Detector {
         // Look for a resource name in the input metadata
         String name = metadata.get(Metadata.RESOURCE_NAME_KEY);
         if (name != null) {
             // If the name is a URL, skip the trailing query
             int question = name.indexOf('?');
             if (question != -1) {
                 name = name.substring(0, question);
             }
+            int hash = name.indexOf('#');
+            if (hash != -1) {
+                name = name.substring(0, hash);
+            }
 
             // If the name is a URL or a path, skip all but the last component
             int slash = name.lastIndexOf('/');
@@ -113,15 +117,8 @@ public class NameDetector implements Detector {
                 name = name.substring(backslash + 1);
             }
 
             // Strip any fragments from the end, but only ones after the extension
-            int hash = name.lastIndexOf('#');
-            int dot = name.indexOf('.');
-            if (hash != -1) {
-                if (dot == -1 || hash > dot) {
-                    name = name.substring(0, hash);
-                }
-            }
 
             // Decode any potential URL encoding
             int percent = name.indexOf('%');
             if (percent != -1) {
