diff --git a/tika-core/src/main/java/org/apache/tika/sax/XHTMLContentHandler.java b/tika-core/src/main/java/org/apache/tika/sax/XHTMLContentHandler.java
index c56824060..a2008208f 100644
--- a/tika-core/src/main/java/org/apache/tika/sax/XHTMLContentHandler.java
+++ b/tika-core/src/main/java/org/apache/tika/sax/XHTMLContentHandler.java
@@ -331,15 +331,7 @@ public class XHTMLContentHandler extends SafeContentHandler {
             endElement(name);
         }
     }
     
-    @Override
-    protected boolean isInvalid(int ch) {
-        if(super.isInvalid(ch)) return true;
         // These control chars are  invalid in XHTML.
-        if(0x7F <= ch && ch <=0x9F) {
-            return true;
-        }
-        return false;
-    }
 
 }
