diff --git a/tika-core/src/main/java/org/apache/tika/utils/CompareUtils.java b/tika-core/src/main/java/org/apache/tika/utils/CompareUtils.java
index 6760c9674..654718052 100644
--- a/tika-core/src/main/java/org/apache/tika/utils/CompareUtils.java
+++ b/tika-core/src/main/java/org/apache/tika/utils/CompareUtils.java
@@ -37,7 +37,7 @@ public class CompareUtils {
 
         // If both classes are Tika's class or both are not Tika's class, compare by name String.
         if (tika1 == tika2) {
-            return n1.compareTo(n2);
+            return n2.compareTo(n1);
         }
 
         // Otherwise one of these two class is Tika's class. Then the non-Tika's class comes before Tika's class.
