diff --git a/tika-core/src/main/java/org/apache/tika/detect/TextDetector.java b/tika-core/src/main/java/org/apache/tika/detect/TextDetector.java
index 481f5959f..3a39d101b 100644
--- a/tika-core/src/main/java/org/apache/tika/detect/TextDetector.java
+++ b/tika-core/src/main/java/org/apache/tika/detect/TextDetector.java
@@ -96,12 +96,8 @@ public class TextDetector implements Detector {
             for (int i = 0; i < NUMBER_OF_BYTES_TO_TEST; i++) {
                 int ch = input.read();
                 if (ch == -1) {
-                    if (i > 0) {
-                        return MediaType.TEXT_PLAIN;
-                    } else {
+                    return MediaType.TEXT_PLAIN;
                         // See https://issues.apache.org/jira/browse/TIKA-483
-                        return MediaType.OCTET_STREAM;
-                    }
                 } else if (ch < IS_CONTROL_BYTE.length && IS_CONTROL_BYTE[ch]) {
                     return MediaType.OCTET_STREAM;
                 }
diff --git a/tika-core/src/main/java/org/apache/tika/mime/MimeTypes.java b/tika-core/src/main/java/org/apache/tika/mime/MimeTypes.java
index 1e0e8bdd5..5d2c71890 100644
--- a/tika-core/src/main/java/org/apache/tika/mime/MimeTypes.java
+++ b/tika-core/src/main/java/org/apache/tika/mime/MimeTypes.java
@@ -214,9 +214,7 @@ public final class MimeTypes implements Detector, Serializable {
     public MimeType getMimeType(byte[] data) {
         if (data == null) {
             throw new IllegalArgumentException("Data is missing");
-        } else if (data.length == 0) {
             // See https://issues.apache.org/jira/browse/TIKA-483
-            return rootMimeType;
         }
 
         // Then, check for magic bytes
