diff --git a/tika-core/src/test/java/org/apache/tika/sax/LinkContentHandlerTest.java b/tika-core/src/test/java/org/apache/tika/sax/LinkContentHandlerTest.java
index 804f6fbef..9f81c8797 100644
--- a/tika-core/src/test/java/org/apache/tika/sax/LinkContentHandlerTest.java
+++ b/tika-core/src/test/java/org/apache/tika/sax/LinkContentHandlerTest.java
@@ -73,7 +73,7 @@ public class LinkContentHandlerTest {
         assertEquals("http://tika.apache.org/stylesheet.css", linkContentHandler.getLinks().get(0).getUri());
         assertEquals("stylesheet", linkContentHandler.getLinks().get(0).getRel());
     }
-
+    
     /**
      * @see <a href="https://issues.apache.org/jira/browse/TIKA-1835">TIKA-1835</a>
      */
@@ -89,36 +89,4 @@ public class LinkContentHandlerTest {
 
         assertEquals("http://tika.apache.org/iframe.html", linkContentHandler.getLinks().get(0).getUri());
     }
-
-    /**
-     * @see <a href="https://issues.apache.org/jira/browse/TIKA-1937">TIKA-1937</a>
-     */
-    @Test
-    public void testScriptTag() throws Exception {
-        LinkContentHandler linkContentHandler = new LinkContentHandler();
-
-        AttributesImpl atts = new AttributesImpl();
-        atts.addAttribute("", "src", "src", "", "http://tika.apache.org/script.js");
-
-        linkContentHandler.startElement(XHTMLContentHandler.XHTML, "script", "", atts);
-        linkContentHandler.endElement(XHTMLContentHandler.XHTML, "script", "");
-
-        assertEquals("http://tika.apache.org/script.js", linkContentHandler.getLinks().get(0).getUri());
-    }
-
-    /**
-     * @see <a href="https://issues.apache.org/jira/browse/TIKA-1937">TIKA-1937</a>
-     */
-    @Test
-    public void testEmbeddedScriptTag() throws Exception {
-        LinkContentHandler linkContentHandler = new LinkContentHandler();
-
-        AttributesImpl atts = new AttributesImpl();
-        // no "src" attribute implies it's an embedded script
-
-        linkContentHandler.startElement(XHTMLContentHandler.XHTML, "script", "", atts);
-        linkContentHandler.endElement(XHTMLContentHandler.XHTML, "script", "");
-
-        assert(linkContentHandler.getLinks().isEmpty());
-    }
-}
+}
\ No newline at end of file
