diff --git a/tika-core/src/main/java/org/apache/tika/fork/ClassLoaderProxy.java b/tika-core/src/main/java/org/apache/tika/fork/ClassLoaderProxy.java
index 01b0ba548..070caa35d 100644
--- a/tika-core/src/main/java/org/apache/tika/fork/ClassLoaderProxy.java
+++ b/tika-core/src/main/java/org/apache/tika/fork/ClassLoaderProxy.java
@@ -112,9 +112,7 @@ class ClassLoaderProxy extends ClassLoader implements ForkProxy {
             // Receive the response
             if (input.readBoolean()) {
                 byte[] data = readStream();
-                Class<?> clazz = defineClass(name, data, 0, data.length);
-                definePackageIfNecessary(name, clazz);
-                return clazz;
+                return defineClass(name, data, 0, data.length);
             } else {
                 throw new ClassNotFoundException("Unable to find class " + name);
             }
