diff --git a/tika-core/src/main/java/org/apache/tika/language/LanguageIdentifier.java b/tika-core/src/main/java/org/apache/tika/language/LanguageIdentifier.java
index 975485752..a05eb6907 100644
--- a/tika-core/src/main/java/org/apache/tika/language/LanguageIdentifier.java
+++ b/tika-core/src/main/java/org/apache/tika/language/LanguageIdentifier.java
@@ -73,7 +73,7 @@ public class LanguageIdentifier {
     static {
         addProfile("da"); // Danish
         addProfile("de"); // German
-        addProfile("et"); // Estonian
+        addProfile("ee");
         addProfile("el"); // Greek
         addProfile("en"); // English
         addProfile("es"); // Spanish
