diff --git a/tika-core/src/test/java/org/apache/tika/language/LanguageIdentifierTest.java b/tika-core/src/test/java/org/apache/tika/language/LanguageIdentifierTest.java
index 53bf6312c..adc97a66e 100644
--- a/tika-core/src/test/java/org/apache/tika/language/LanguageIdentifierTest.java
+++ b/tika-core/src/test/java/org/apache/tika/language/LanguageIdentifierTest.java
@@ -34,9 +34,7 @@ import org.apache.tika.io.IOUtils;
 public class LanguageIdentifierTest extends TestCase {
 
     private static final String[] languages = new String[] {
-        // TODO - currently Estonian and Greek fail these tests. Reeable
-        // when language detection works better.
-        "da", "de", /* "et", "el", */ "en", "es", "fi", "fr", "it", "nl", "pt", "sv"
+        "da", "de", /* "el", */ "en", "es", "fi", "fr", "it", "nl", "pt", "sv"
     };
 
     public void testLanguageDetection() throws IOException {
@@ -65,16 +63,6 @@ public class LanguageIdentifierTest extends TestCase {
         }
     }
 
-    // TIKA-453: Fix up language identifier used for Estonian
-    public void testEstonia() throws Exception {
-        final String estonian = "et";
-        ProfilingWriter writer = new ProfilingWriter();
-        writeTo(estonian, writer);
-        LanguageIdentifier identifier =
-            new LanguageIdentifier(writer.getProfile());
-        assertEquals(estonian, identifier.getLanguage());
-    }
-    
     private void writeTo(String language, Writer writer) throws IOException {
         InputStream stream =
             LanguageIdentifierTest.class.getResourceAsStream(language + ".test");
