diff --git a/tikacore_9_fix/tika-core/src/main/java/org/apache/tika/mime/MediaType.java b/tikacore_9_buggy/tika-core/src/main/java/org/apache/tika/mime/MediaType.java
index b1116e7..bca232b 100644
--- a/tikacore_9_fix/tika-core/src/main/java/org/apache/tika/mime/MediaType.java
+++ b/tikacore_9_buggy/tika-core/src/main/java/org/apache/tika/mime/MediaType.java
@@ -17,8 +17,10 @@
 package org.apache.tika.mime;
 
 import java.io.Serializable;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.SortedMap;
@@ -185,22 +184,11 @@ public final class MediaType implements Comparable<MediaType>, Serializable {
 
             key = key.trim();
             if (key.length() > 0) {
-                parameters.put(key, unquote(value.trim()));
+                parameters.put(key, value.trim());
             }
         }
         return parameters;
     }
   
-    private static String unquote(String s) {
-        if( s.startsWith("\"") && s.endsWith("\"")) {
-            return s.substring(1, s.length() - 1);
-        }
-        if( s.startsWith("'") && s.endsWith("'")) {
-           return s.substring(1, s.length() - 1);
-       }

-        return s;
-    }
 
     /**
      * Canonical string representation of this media type.
