diff --git a/server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java b/server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java
index 2e9aa38694..106d46f9c1 100644
--- a/server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java
+++ b/server/container/core/src/main/java/org/apache/james/server/core/MailImpl.java
@@ -40,17 +40,16 @@ import javax.mail.MessagingException;
 import javax.mail.internet.MimeMessage;
 import javax.mail.internet.ParseException;
 
 import org.apache.james.core.MailAddress;
 import org.apache.james.lifecycle.api.Disposable;
 import org.apache.james.lifecycle.api.LifecycleUtil;
 import org.apache.mailet.Mail;
 import org.apache.mailet.PerRecipientHeaders;
 import org.apache.mailet.PerRecipientHeaders.Header;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.google.common.base.Preconditions;
 
 /**
  * <p>
  * Wraps a MimeMessage adding routing information (from SMTP) and some simple
@@ -591,7 +589,6 @@ public class MailImpl implements Disposable, Mail {
 
     @Override
     public Serializable setAttribute(String key, Serializable object) {
-        Preconditions.checkNotNull(key, "Key of an attribute should not be null");
         return (Serializable) attributes.put(key, object);
     }
 
