diff --git a/server/container/core/src/test/java/org/apache/james/server/core/MailImplTest.java b/server/container/core/src/test/java/org/apache/james/server/core/MailImplTest.java
index 5ab4cd04f7..ab5d9b78c8 100644
--- a/server/container/core/src/test/java/org/apache/james/server/core/MailImplTest.java
+++ b/server/container/core/src/test/java/org/apache/james/server/core/MailImplTest.java
@@ -18,7 +18,6 @@
  ****************************************************************/
 package org.apache.james.server.core;
 
-import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
@@ -113,12 +112,4 @@ public class MailImplTest extends MailTestAllImplementations {
         helperTestInitialState(duplicate);
         helperTestMessageSize(duplicate, 0);
     }
-
-    @Test
-    public void setAttributeShouldThrowOnNullAttributeName() throws MessagingException {
-        MailImpl mail = new MailImpl();
-
-        assertThatThrownBy(() -> mail.setAttribute(null, "toto"))
-            .isInstanceOf(NullPointerException.class);
-    }
 }
