diff --git a/src/test/java/com/payneteasy/tlv/BerTlvParserTest.java b/src/test/java/com/payneteasy/tlv/BerTlvParserTest.java
index fdd50a9..1bb2f6e 100644
--- a/src/test/java/com/payneteasy/tlv/BerTlvParserTest.java
+++ b/src/test/java/com/payneteasy/tlv/BerTlvParserTest.java
@@ -119,26 +119,4 @@ public class BerTlvParserTest {
         BerTlvs tlvs = parser.parse(bytes, 0, bytes.length);
         Assert.assertEquals(0, tlvs.getList().size());
     }
-
-    @Test
-    public void test_issue_10_last_tag() {
-        String hex = "BF01820114DF0105A0000000049F2201EFDF020101DF030101DF0481F8A191CB87473F29349B5D60A88B3EAEE0973AA6F1A082F358D849FDDFF9C091F899EDA9792CAF09EF28F5D22404B88A2293EEBBC1949C43BEA4D60CFD879A1539544E09E0F09F60F065B2BF2A13ECC705F3D468B9D33AE77AD9D3F19CA40F23DCF5EB7C04DC8F69EBA565B1EBCB4686CD274785530FF6F6E9EE43AA43FDB02CE00DAEC15C7B8FD6A9B394BABA419D3F6DC85E16569BE8E76989688EFEA2DF22FF7D35C043338DEAA982A02B866DE5328519EBBCD6F03CDD686673847F84DB651AB86C28CF1462562C577B853564A290C8556D818531268D25CC98A4CC6A0BDFFFDA2DCCA3A94C998559E307FDDF915006D9A987B07DDAEB3BDF050103";
-        byte[] bytes = HexUtil.parseHex(hex);
-        BerTlvParser parser = new BerTlvParser();
-        BerTlvs tlvs = parser.parse(bytes, 0, bytes.length);
-
-        BerTlvLogger.log("  ", tlvs, LOG);
-
-        {
-            BerTlv berTlv = tlvs.find(new BerTag(0xDF, 0x05));
-            Assert.assertNotNull(berTlv);
-            Assert.assertEquals("03", berTlv.getHexValue());
-        }
-
-        {
-            Assert.assertEquals(1, tlvs.getList().size());
-            Assert.assertTrue(tlvs.getList().get(0).isConstructed());
-            Assert.assertEquals(6, tlvs.getList().get(0).getValues().size());
-        }
-    }
 }
