diff --git a/src/main/java/org/locationtech/proj4j/io/Proj4FileReader.java b/src/main/java/org/locationtech/proj4j/io/Proj4FileReader.java
index c50f723..b3fd9b0 100755
--- a/src/main/java/org/locationtech/proj4j/io/Proj4FileReader.java
+++ b/src/main/java/org/locationtech/proj4j/io/Proj4FileReader.java
@@ -35,8 +35,8 @@ public class Proj4FileReader {
         // TODO: use simpler parser than StreamTokenizer for speed and flexibility
         // TODO: parse CSes line-at-a-time (this allows preserving CS param string for later access)
 
-        String filename = "proj4/nad/" + authorityCode.toLowerCase();
-        InputStream inStr = Proj4FileReader.class.getClassLoader().getResourceAsStream(filename);
+        String filename = "/proj4/nad/" + authorityCode.toLowerCase();
+        InputStream inStr = Proj4FileReader.class.getResourceAsStream(filename);
         if (inStr == null) {
             throw new IllegalStateException("Unable to access CRS file: " + filename);
         }
diff --git a/src/main/resources/proj4/nad/epsg b/src/main/resources/proj4/nad/epsg
index 189af1f..5e3a591 100644
--- a/src/main/resources/proj4/nad/epsg
+++ b/src/main/resources/proj4/nad/epsg
@@ -4697,7 +4697,7 @@
 # County ST74
 <3854> +proj=tmerc +lat_0=0 +lon_0=18.05787 +k=0.99999506 +x_0=100182.7406 +y_0=-6500620.1207 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs  <>
 # WGS 84 / Pseudo-Mercator
-<3857> +proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +wktext +no_defs  <>
+<3857> +proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +wktext +no_defs <>
 # ETRS89 / GK19FIN
 <3873> +proj=tmerc +lat_0=0 +lon_0=19 +k=1 +x_0=19500000 +y_0=0 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs  <>
 # ETRS89 / GK20FIN
diff --git a/src/main/resources/proj4/nad/world b/src/main/resources/proj4/nad/world
index 2018b53..092df23 100644
--- a/src/main/resources/proj4/nad/world
+++ b/src/main/resources/proj4/nad/world
@@ -146,7 +146,7 @@
 	proj=tmerc ellps=mod_airy lat_0=53d30'N lon_0=8W
 	x_0=200000 y_0=250000 k_0=1.000035
 	no_defs <>
-<neiez> # Netherlands East Indies Equitorial Zone
+<neiez> # Netherlands East Indies Equatorial Zone
 	proj=merc ellps=bessel lon_0=110E
 	x_0=3900000 y_0=900000 k_0=0.997
 	no_defs <>
@@ -176,12 +176,12 @@
 	no_defs <>
 # Gauss Krueger Grid for Germany
 # 
-# The first figure of the easting is lon_0 devided by 3
+# The first figure of the easting is lon_0 divided by 3
 # ( 2 for 6d0E, 3 for 9d0E, 4 for 12d0E)
 # For translations you have to remove this first figure
 # and convert northings and eastings from km to meter .
-# The other way round, devide by 1000 and add the figure.
-# I made 3 entrys for the officially used grids in Germany
+# The other way round, divide by 1000 and add the figure.
+# I made 3 entries for the officially used grids in Germany
 # 
 #
 # Und nochmal in deutsch :
