diff --git a/tikaapp_1_fix/tika-app/src/main/java/org/apache/tika/cli/TikaCLI.java b/tikaapp_1_buggy/tika-app/src/main/java/org/apache/tika/cli/TikaCLI.java
index 0ab4ac9..31096d6 100644
--- a/tikaapp_1_fix/tika-app/src/main/java/org/apache/tika/cli/TikaCLI.java
+++ b/tikaapp_1_buggy/tika-app/src/main/java/org/apache/tika/cli/TikaCLI.java
@@ -824,9 +824,7 @@ public class TikaCLI {
         
         public void outputMetadata(String[] names) {
            for (String name : names) {
-              for(String value : metadata.getValues(name)) {
-                 writer.println(name + ": " + value);
-              }
+              writer.println(name + ": " + metadata.get(name));
            }
         }
         
