diff --git a/tika-app/src/test/java/org/apache/tika/cli/TikaCLITest.java b/tika-app/src/test/java/org/apache/tika/cli/TikaCLITest.java
index 031d4cbc1..be584c0af 100644
--- a/tika-app/src/test/java/org/apache/tika/cli/TikaCLITest.java
+++ b/tika-app/src/test/java/org/apache/tika/cli/TikaCLITest.java
@@ -35,7 +35,7 @@ public class TikaCLITest extends TestCase{
     private ByteArrayOutputStream outContent = null;
     private PrintStream stdout = null;
     private URI testDataURI = new File("src/test/resources/test-data/").toURI();
-    private String resourcePrefix = testDataURI.toString();
+    private String resorcePrefix = testDataURI.toString();
 
     public void setUp() throws Exception {
         profile = new File("welsh.ngp");
@@ -50,7 +50,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testCreateProfile() throws Exception {
-        String[] params = {"--create-profile=welsh", "-eUTF-8", resourcePrefix + "welsh_corpus.txt"};
+        String[] params = {"--create-profile=welsh", "-eUTF-8", resorcePrefix + "welsh_corpus.txt"};
         TikaCLI.main(params);
         Assert.assertTrue(profile.exists());
     }
@@ -84,7 +84,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testXMLOutput() throws Exception{
-        String[] params = {"-x", resourcePrefix + "alice.cli.test"};
+        String[] params = {"-x", resorcePrefix + "alice.cli.test"};
         TikaCLI.main(params);
         Assert.assertTrue(outContent.toString().contains("?xml version=\"1.0\" encoding=\"UTF-8\"?"));
     }
@@ -95,7 +95,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testHTMLOutput() throws Exception{
-        String[] params = {"-h", resourcePrefix + "alice.cli.test"};
+        String[] params = {"-h", resorcePrefix + "alice.cli.test"};
         TikaCLI.main(params);
         Assert.assertTrue(outContent.toString().contains("html xmlns=\"http://www.w3.org/1999/xhtml"));
     }
@@ -106,7 +106,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testTextOutput() throws Exception{
-        String[] params = {"-t", resourcePrefix + "alice.cli.test"};
+        String[] params = {"-t", resorcePrefix + "alice.cli.test"};
         TikaCLI.main(params);
         Assert.assertTrue(outContent.toString().contains("finished off the cake"));
     }
@@ -116,7 +116,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testMetadataOutput() throws Exception{
-        String[] params = {"-m", resourcePrefix + "alice.cli.test"};
+        String[] params = {"-m", resorcePrefix + "alice.cli.test"};
         TikaCLI.main(params);
         Assert.assertTrue(outContent.toString().contains("text/plain"));
     }
@@ -127,7 +127,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testLanguageOutput() throws Exception{
-        String[] params = {"-l", resourcePrefix + "alice.cli.test"};
+        String[] params = {"-l", resorcePrefix + "alice.cli.test"};
         TikaCLI.main(params);
         Assert.assertTrue(outContent.toString().contains("en"));
     }
@@ -138,7 +138,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testDetectOutput() throws Exception{
-        String[] params = {"-d", resourcePrefix + "alice.cli.test"};
+        String[] params = {"-d", resorcePrefix + "alice.cli.test"};
         TikaCLI.main(params);
         Assert.assertTrue(outContent.toString().contains("text/plain"));
     }
@@ -149,7 +149,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testListMetModels() throws Exception{
-        String[] params = {"--list-met-models", resourcePrefix + "alice.cli.test"};
+        String[] params = {"--list-met-models", resorcePrefix + "alice.cli.test"};
         TikaCLI.main(params);
         Assert.assertTrue(outContent.toString().contains("text/plain"));
     }
@@ -160,7 +160,7 @@ public class TikaCLITest extends TestCase{
      * @throws Exception
      */
     public void testListSupportedTypes() throws Exception{
-        String[] params = {"--list-supported-types", resourcePrefix + "alice.cli.test"};
+        String[] params = {"--list-supported-types", resorcePrefix + "alice.cli.test"};
         TikaCLI.main(params);
         Assert.assertTrue(outContent.toString().contains("supertype: application/octet-stream"));
     }
@@ -181,7 +181,7 @@ public class TikaCLITest extends TestCase{
                           // google guava library has better solution
 
         try {
-            String[] params = {"--extract-dir="+tempFile.getAbsolutePath(),"-z", resourcePrefix + "/coffee.xls"};
+            String[] params = {"--extract-dir="+tempFile.getAbsolutePath(),"-z", resorcePrefix + "/coffee.xls"};
             
             TikaCLI.main(params);
             
@@ -204,15 +204,4 @@ public class TikaCLITest extends TestCase{
         }
 
     }
-
-    // TIKA-920
-    public void testMultiValuedMetadata() throws Exception {
-        String[] params = {"-m", resourcePrefix + "testMultipleSheets.numbers"};
-        TikaCLI.main(params);
-        String content = outContent.toString();
-        assertTrue(content.contains("sheetNames: Checking"));
-        assertTrue(content.contains("sheetNames: Secon sheet"));
-        assertTrue(content.contains("sheetNames: Logical Sheet 3"));
-        assertTrue(content.contains("sheetNames: Sheet 4"));
-    }
 }
