diff --git a/core/rio/turtle/src/test/java/org/eclipse/rdf4j/rio/turtle/TurtleParserTest.java b/core/rio/turtle/src/test/java/org/eclipse/rdf4j/rio/turtle/TurtleParserTest.java
index f73b181e1..bf4192ffc 100644
--- a/core/rio/turtle/src/test/java/org/eclipse/rdf4j/rio/turtle/TurtleParserTest.java
+++ b/core/rio/turtle/src/test/java/org/eclipse/rdf4j/rio/turtle/TurtleParserTest.java
@@ -482,17 +482,4 @@ public class TurtleParserTest {
 		}
 	}
 
-	@Test
-	public void testLegalUnicodeInTripleSubject() throws IOException {
-		String data = "@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .\n<r:\uD804\uDC9D> a xsd:string .";
-		Reader r = new StringReader(data);
-
-		try {
-			parser.parse(r, baseURI);
-			assertTrue(statementCollector.getStatements().size() == 1);
-		} catch (RDFParseException e) {
-			fail("Complex unicode characters should be parsed correctly (" + e.getMessage() + ")");
-		}
-	}
-
 }
