diff --git a/core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java b/core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java
index b23f30a9d..efcc9aaf5 100644
--- a/core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java
+++ b/core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java
@@ -165,7 +165,7 @@ public class TurtleParser extends AbstractRDFParser {
 			lineNumber = 1;
 
 			// Allow at most 8 characters to be pushed back:
-			this.reader = new PushbackReader(reader, 10);
+			this.reader = new PushbackReader(reader, 8);
 
 			// Store normalized base URI
 			setBaseURI(baseURI);
