diff --git a/src/test/java/com/jcabi/http/response/JsonResponseTest.java b/src/test/java/com/jcabi/http/response/JsonResponseTest.java
index 60a12ed..8144bc4 100644
--- a/src/test/java/com/jcabi/http/response/JsonResponseTest.java
+++ b/src/test/java/com/jcabi/http/response/JsonResponseTest.java
@@ -62,20 +62,4 @@ public final class JsonResponseTest {
         );
     }
 
-    /**
-     * JsonResponse can read control characters.
-     *
-     * @throws Exception If something goes wrong inside
-     */
-    @Test
-    public void readsControlCharacters() throws Exception {
-        final Response resp = new FakeRequest()
-            .withBody("{\"test\": \"\u001Fblah\u0001cwhoa\u0000!\"}").fetch();
-        final JsonResponse response = new JsonResponse(resp);
-        MatcherAssert.assertThat(
-            response.json().readObject().getString("test"),
-            Matchers.is("\u001Fblah\u0001cwhoa\u0000!")
-        );
-    }
-
 }
