diff --git a/src/main/java/com/jcabi/http/mock/MkGrizzlyAdapter.java b/src/main/java/com/jcabi/http/mock/MkGrizzlyAdapter.java
index 42b5646..3bd9a0f 100644
--- a/src/main/java/com/jcabi/http/mock/MkGrizzlyAdapter.java
+++ b/src/main/java/com/jcabi/http/mock/MkGrizzlyAdapter.java
@@ -198,6 +198,7 @@ final class MkGrizzlyAdapter extends GrizzlyAdapter {
             if (matcher.matches(candidate.answer())) {
                 results.add(candidate.query());
                 iter.remove();
+                break;
             }
         }
         if (results.isEmpty()) {
