diff --git a/src/main/java/com/jcabi/http/request/FakeRequest.java b/src/main/java/com/jcabi/http/request/FakeRequest.java
index c6db7ab..f32c88b 100644
--- a/src/main/java/com/jcabi/http/request/FakeRequest.java
+++ b/src/main/java/com/jcabi/http/request/FakeRequest.java
@@ -38,6 +38,7 @@ import com.jcabi.http.RequestURI;
 import com.jcabi.http.Response;
 import com.jcabi.http.Wire;
 import com.jcabi.immutable.Array;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.net.HttpURLConnection;
@@ -86,12 +87,26 @@ public final class FakeRequest implements Request {
                 final String method,
                 final Collection<Map.Entry<String, String>> headers,
                 final InputStream text) throws IOException {
+                final ByteArrayOutputStream output =
+                    new ByteArrayOutputStream();
+                final byte[] buffer = new byte[8192];
+                for (int bytes = text.read(buffer); bytes != -1;
+                    bytes = text.read(buffer)) {
+                    output.write(buffer, 0, bytes);
+                }
+                output.flush();
+                final byte[] body;
+                if (output.size() > 0) {
+                    body = output.toByteArray();
+                } else {
+                    body = FakeRequest.this.content;
+                }
                 return new DefaultResponse(
                     req,
                     FakeRequest.this.code,
                     FakeRequest.this.phrase,
                     FakeRequest.this.hdrs,
-                    FakeRequest.this.content
+                    body
                 );
             }
         },
