diff --git a/src/main/java/com/jcabi/http/response/JsoupResponse.java b/src/main/java/com/jcabi/http/response/JsoupResponse.java
index 139a8df..de58536 100644
--- a/src/main/java/com/jcabi/http/response/JsoupResponse.java
+++ b/src/main/java/com/jcabi/http/response/JsoupResponse.java
@@ -81,10 +81,7 @@ public final class JsoupResponse extends AbstractResponse {
 
     @Override
     public String body() {
-        final Document html = Jsoup.parse(super.body());
-        html.outputSettings().syntax(Document.OutputSettings.Syntax.xml);
-        html.outputSettings().escapeMode(Entities.EscapeMode.xhtml);
-        return html.html();
+        return Jsoup.parse(super.body()).html();
     }
 
 }
