diff --git a/src/test/java/com/jcabi/http/response/JsoupResponseTest.java b/src/test/java/com/jcabi/http/response/JsoupResponseTest.java
index dab6751..ec03150 100644
--- a/src/test/java/com/jcabi/http/response/JsoupResponseTest.java
+++ b/src/test/java/com/jcabi/http/response/JsoupResponseTest.java
@@ -29,7 +29,6 @@
  */
 package com.jcabi.http.response;
 
-import com.google.common.base.Joiner;
 import com.jcabi.http.Response;
 import com.jcabi.http.request.FakeRequest;
 import com.jcabi.matchers.XhtmlMatchers;
@@ -51,11 +50,7 @@ public final class JsoupResponseTest {
     @Test
     public void normalizesHtml() throws Exception {
         final Response resp = new FakeRequest().withBody(
-            Joiner.on(' ').join(
-                "<html xmlns='http://www.w3.org/1999/xhtml'>",
-                "<head><meta name='test'></head>",
-                "<p>Hello world"
-            )
+            "<html xmlns='http://www.w3.org/1999/xhtml'><p>Hello world"
         ).fetch();
         MatcherAssert.assertThat(
             new JsoupResponse(resp).body(),
