diff --git a/src/main/java/com/jcabi/http/response/JsonResponse.java b/src/main/java/com/jcabi/http/response/JsonResponse.java
index 245daea..9785bcf 100644
--- a/src/main/java/com/jcabi/http/response/JsonResponse.java
+++ b/src/main/java/com/jcabi/http/response/JsonResponse.java
@@ -99,9 +99,7 @@ public final class JsonResponse extends AbstractResponse {
      */
     @NotNull(message = "JSON reader is never NULL")
     public JsonReader json() {
-        return Json.createReader(
-            new StringReader(this.escapeControl(this.body()))
-        );
+        return Json.createReader(new StringReader(this.body()));
     }
 
     /**
@@ -124,4 +122,5 @@ public final class JsonResponse extends AbstractResponse {
         return escaped.toString();
     }
 
+
 }
