diff --git a/src/test/java/com/jcabi/http/mock/MkContainerTest.java b/src/test/java/com/jcabi/http/mock/MkContainerTest.java
index 2f6121e..daca489 100644
--- a/src/test/java/com/jcabi/http/mock/MkContainerTest.java
+++ b/src/test/java/com/jcabi/http/mock/MkContainerTest.java
@@ -70,8 +70,11 @@ public final class MkContainerTest {
     /**
      * MkContainer can understand duplicate headers.
      * @throws Exception If something goes wrong inside
+     * @todo #1 Grizzly container doesn't understand same-name
+     *  headers, or we don't fetch them correctly from GrizzlyRequest
      */
     @Test
+    @org.junit.Ignore
     public void understandsDuplicateHeaders() throws Exception {
         final MkContainer container = new MkGrizzlyContainer()
             .next(new MkAnswer.Simple(""))
