diff --git a/src/test/java/org/apache/tika/mime/TestMimeTypes.java b/src/test/java/org/apache/tika/mime/TestMimeTypes.java
deleted file mode 100644
index f798263c4..000000000
--- a/src/test/java/org/apache/tika/mime/TestMimeTypes.java
+++ /dev/null
@@ -1,52 +0,0 @@
-package org.apache.tika.mime;

-import junit.framework.TestCase;
-import org.apache.tika.config.TikaConfig;

-public class TestMimeTypes extends TestCase {
-    private MimeTypes repo;
-    public TestMimeTypes() {
-        try {
-            repo = TikaConfig.getDefaultConfig().getMimeRepository();
-        } catch (Exception e) {
-            fail(e.getMessage());
-        }
-    }

-    public void testCaseSensitivity() {
-        MimeType type = repo.getMimeType("test.PDF");
-        assertNotNull(type);
-        assertEquals(repo.getMimeType("test.pdf"), type);
-        assertEquals(repo.getMimeType("test.PdF"), type);
-        assertEquals(repo.getMimeType("test.pdF"), type);
-    }
-}
