diff --git a/src/main/java/org/apache/tika/parser/html/HtmlParser.java b/src/main/java/org/apache/tika/parser/html/HtmlParser.java
index dabc49ab9..8f12ea25d 100644
--- a/src/main/java/org/apache/tika/parser/html/HtmlParser.java
+++ b/src/main/java/org/apache/tika/parser/html/HtmlParser.java
@@ -102,7 +102,7 @@ public class HtmlParser implements Parser {
         // Parse the HTML document
         xhtml.startDocument();
         SAXParser parser = new SAXParser();
-        parser.setContentHandler(new XHTMLDowngradeHandler(handler));
+        parser.setContentHandler(handler);
         parser.parse(new InputSource(Utils.getUTF8Reader(stream, metadata)));
         xhtml.endDocument();
     }
diff --git a/src/main/java/org/apache/tika/parser/html/XHTMLDowngradeHandler.java b/src/main/java/org/apache/tika/parser/html/XHTMLDowngradeHandler.java
deleted file mode 100644
index 908036bf0..000000000
--- a/src/main/java/org/apache/tika/parser/html/XHTMLDowngradeHandler.java
+++ /dev/null
@@ -1,77 +0,0 @@
-class XHTMLDowngradeHandler extends ContentHandlerDecorator {
-    public XHTMLDowngradeHandler(ContentHandler handler) {
-        super(handler);
-    }
-    @Override
-    public void startElement(
-            String uri, String localName, String name, Attributes atts)
-            throws SAXException {
-        String upper = localName.toUpperCase();
-        AttributesImpl attributes = new AttributesImpl();
-        for (int i = 0; i < atts.getLength(); i++) {
-            String local = atts.getLocalName(i);
-            String qname = atts.getQName(i);
-            if (!XMLConstants.NULL_NS_URI.equals(atts.getURI(i).length())
-                    && !local.equals(XMLConstants.XMLNS_ATTRIBUTE)
-                    && !qname.startsWith(XMLConstants.XMLNS_ATTRIBUTE + ":")) {
-                attributes.addAttribute(
-                        atts.getURI(i), local, qname,
-                        atts.getType(i), atts.getValue(i));
-            }
-        }
-        super.startElement(XMLConstants.NULL_NS_URI, upper, upper, attributes);
-    }
-    @Override
-    public void endElement(String uri, String localName, String name)
-            throws SAXException {
-        String upper = localName.toUpperCase();
-        super.endElement(XMLConstants.NULL_NS_URI, upper, upper);
-    }
-}
