diff --git a/src/main/java/org/apache/tika/parser/html/HtmlParser.java b/src/main/java/org/apache/tika/parser/html/HtmlParser.java
index 09708c0a6..92de93f1f 100644
--- a/src/main/java/org/apache/tika/parser/html/HtmlParser.java
+++ b/src/main/java/org/apache/tika/parser/html/HtmlParser.java
@@ -103,9 +103,11 @@ public class HtmlParser implements Parser {
                 new MatchingContentHandler(getMetaHandler(metadata), meta));
 
         // Parse the HTML document
+        xhtml.startDocument();
         SAXParser parser = new SAXParser();
         parser.setContentHandler(new XHTMLDowngradeHandler(handler));
         parser.parse(new InputSource(Utils.getUTF8Reader(stream, metadata)));
+        xhtml.endDocument();
     }
 
     private ContentHandler getTitleHandler(final Metadata metadata) {
diff --git a/src/main/java/org/apache/tika/parser/pkg/PackageParser.java b/src/main/java/org/apache/tika/parser/pkg/PackageParser.java
index 617773d20..86b9e7f0f 100644
--- a/src/main/java/org/apache/tika/parser/pkg/PackageParser.java
+++ b/src/main/java/org/apache/tika/parser/pkg/PackageParser.java
@@ -25,7 +25,6 @@ import org.apache.tika.metadata.Metadata;
 import org.apache.tika.parser.AutoDetectParser;
 import org.apache.tika.parser.Parser;
 import org.apache.tika.sax.BodyContentHandler;
 import org.apache.tika.sax.XHTMLContentHandler;
 import org.xml.sax.SAXException;
 
@@ -93,7 +92,7 @@ public abstract class PackageParser implements Parser {
         try {
             getParser().parse(
                     new CloseShieldInputStream(stream),
-                    new EmbeddedContentHandler(new BodyContentHandler(xhtml)),
+                    new BodyContentHandler(xhtml),
                     metadata);
             xhtml.characters("\n");
         } catch (TikaException e) {
diff --git a/src/main/java/org/apache/tika/sax/EmbeddedContentHandler.java b/src/main/java/org/apache/tika/sax/EmbeddedContentHandler.java
deleted file mode 100644
index 0bafe6973..000000000
--- a/src/main/java/org/apache/tika/sax/EmbeddedContentHandler.java
+++ /dev/null
@@ -1,56 +0,0 @@
-public class EmbeddedContentHandler extends ContentHandlerDecorator {
-    public EmbeddedContentHandler(ContentHandler handler) {
-        super(handler);
-    }
-}
diff --git a/src/main/java/org/apache/tika/sax/TextContentHandler.java b/src/main/java/org/apache/tika/sax/TextContentHandler.java
index 469114776..f4439e485 100644
--- a/src/main/java/org/apache/tika/sax/TextContentHandler.java
+++ b/src/main/java/org/apache/tika/sax/TextContentHandler.java
@@ -23,8 +23,7 @@ import org.xml.sax.helpers.DefaultHandler;
 /**
  * Content handler decorator that only passes the
  * {@link #characters(char[], int, int)} and
  * the decorated content handler.
  */
 public class TextContentHandler extends DefaultHandler {
@@ -47,16 +46,6 @@ public class TextContentHandler extends DefaultHandler {
         delegate.ignorableWhitespace(ch, start, length);
     }
 
-    @Override
-    public void startDocument() throws SAXException {
-        delegate.startDocument();
-    }
-    @Override
-    public void endDocument() throws SAXException {
-        delegate.endDocument();
-    }
     @Override
     public String toString() {
         return delegate.toString();
diff --git a/src/main/java/org/apache/tika/sax/WriteOutContentHandler.java b/src/main/java/org/apache/tika/sax/WriteOutContentHandler.java
index 1aa10ee87..02b8bfc40 100644
--- a/src/main/java/org/apache/tika/sax/WriteOutContentHandler.java
+++ b/src/main/java/org/apache/tika/sax/WriteOutContentHandler.java
@@ -78,22 +78,6 @@ public class WriteOutContentHandler extends DefaultHandler {
         }
     }
-    @Override
-    public void endDocument() throws SAXException {
-        try {
-            writer.flush();
-        } catch (IOException e) {
-            throw new SAXException("Error flushing character output", e);
-        }
-    }
     /**
      * Returns the contents of the internal string buffer where
      * all the received characters have been collected. Only works
