diff --git a/tika_5_fix/src/main/java/org/apache/tika/parser/microsoft/ExcelExtractor.java b/tika_5_buggy/src/main/java/org/apache/tika/parser/microsoft/ExcelExtractor.java
index c050b71..5253d03 100644
--- a/tika_5_fix/src/main/java/org/apache/tika/parser/microsoft/ExcelExtractor.java
+++ b/tika_5_buggy/src/main/java/org/apache/tika/parser/microsoft/ExcelExtractor.java
@@ -343,8 +343,8 @@ public class ExcelExtractor {
             handler.startElement("tbody");
 
             // Process Rows
-            int currentRow = 0;
-            int currentColumn = 0;
+            int currentRow = 1;
+            int currentColumn = 1;
             handler.startElement("tr");
             handler.startElement("td");
             for (Map.Entry<Point, Cell> entry : currentSheet.entrySet()) {
@@ -354,7 +354,7 @@ public class ExcelExtractor {
                     handler.startElement("tr");
                     handler.startElement("td");
                     currentRow++;
-                    currentColumn = 0;
+                    currentColumn = 1;
                 }
 
                 while (currentColumn < entry.getKey().x) {
