diff --git a/src/test/java/org/apache/tika/parser/microsoft/ExcelParserTest.java b/src/test/java/org/apache/tika/parser/microsoft/ExcelParserTest.java
index bd76c3f47..d6eb8a713 100644
--- a/src/test/java/org/apache/tika/parser/microsoft/ExcelParserTest.java
+++ b/src/test/java/org/apache/tika/parser/microsoft/ExcelParserTest.java
@@ -42,12 +42,10 @@ public class ExcelParserTest extends TestCase {
             String content = handler.toString();
             assertTrue(content.contains("Sample Excel Worksheet"));
             assertTrue(content.contains("Numbers and their Squares"));
-            assertTrue(content.contains("\t\tNumber\tSquare"));
             assertTrue(content.contains("9"));
             assertFalse(content.contains("9.0"));
             assertTrue(content.contains("196"));
             assertFalse(content.contains("196.0"));
-            System.out.println(content);
         } finally {
             input.close();
         }
