diff --git a/tika_6_fix/src/main/java/org/apache/tika/parser/microsoft/OfficeParser.java b/tika_6_buggy/src/main/java/org/apache/tika/parser/microsoft/OfficeParser.java
index bfc5ff9..86f1411 100644
--- a/tika_6_fix/src/main/java/org/apache/tika/parser/microsoft/OfficeParser.java
+++ b/tika_6_buggy/src/main/java/org/apache/tika/parser/microsoft/OfficeParser.java
@@ -61,7 +61,6 @@ public class OfficeParser implements Parser {
         XHTMLContentHandler xhtml = new XHTMLContentHandler(handler, metadata);
         xhtml.startDocument();
 
-        boolean outlookExtracted = false;
         POIFSFileSystem filesystem = new POIFSFileSystem(stream);
         Iterator<?> entries = filesystem.getRoot().getEntries();
         while (entries.hasNext()) {
@@ -93,9 +92,7 @@ public class OfficeParser implements Parser {
                 for (String text : extractor.getAllText()) {
                     xhtml.element("p", text);
                 }
-            } else if (!outlookExtracted && name.startsWith("__substg1.0_")) {
-                outlookExtracted = true;
+            } else if (name.startsWith("__substg1.0_")) {
                 setType(metadata, "application/vnd.ms-outlook");
                 new OutlookExtractor(filesystem).parse(xhtml, metadata);
             }
