diff --git a/collections_35_fix/src/main/java/org/apache/commons/collections4/map/Flat3Map.java b/collections_35_buggy/src/main/java/org/apache/commons/collections4/map/Flat3Map.java
index c506c53..2bfa20a 100644
--- a/collections_35_fix/src/main/java/org/apache/commons/collections4/map/Flat3Map.java
+++ b/collections_35_buggy/src/main/java/org/apache/commons/collections4/map/Flat3Map.java
@@ -946,8 +946,8 @@ public class Flat3Map<K, V> implements IterableMap<K, V>, Serializable, Cloneabl
             if (currentEntry == null) {
                 throw new IllegalStateException(AbstractHashedMap.REMOVE_INVALID);
             }
-            parent.remove(currentEntry.getKey());
             currentEntry.setRemoved(true);
+            parent.remove(currentEntry.getKey());
             nextIndex--;
             currentEntry = null;
         }
