diff --git a/src/test/java/org/apache/commons/collections4/map/Flat3MapTest.java b/src/test/java/org/apache/commons/collections4/map/Flat3MapTest.java
index d4e3b29c..c234349d 100644
--- a/src/test/java/org/apache/commons/collections4/map/Flat3MapTest.java
+++ b/src/test/java/org/apache/commons/collections4/map/Flat3MapTest.java
@@ -24,13 +24,13 @@ import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
 
+import junit.framework.Test;
+
 import org.apache.commons.collections4.BulkTest;
 import org.apache.commons.collections4.IterableMap;
 import org.apache.commons.collections4.MapIterator;
 import org.apache.commons.collections4.iterators.AbstractMapIteratorTest;
 
-import junit.framework.Test;
-
 /**
  * JUnit tests.
  *
@@ -336,29 +336,6 @@ public class Flat3MapTest<K, V> extends AbstractIterableMapTest<K, V> {
         assertEquals("NewValue", map.get(THREE));
     }
 
-    public void testEntrySet() {
-        // Sanity check
-        putAndRemove(new HashMap<>());
-        // Actual test
-        putAndRemove(new Flat3Map<>());
-    }
-
-    private void putAndRemove(final Map<K, V> map) {
-        map.put((K) "A", (V) "one");
-        map.put((K) "B", (V) "two");
-        map.put((K) "C", (V) "three");
-        Iterator<Map.Entry<K, V>> it = map.entrySet().iterator();
-
-        Map.Entry<K, V> mapEntry1 = it.next();
-        Map.Entry<K, V> mapEntry2 = it.next();
-        Map.Entry<K, V> mapEntry3 = it.next();
-        it.remove();
-        assertEquals(2, map.size());
-        assertEquals("one", map.get((K) "A"));
-        assertEquals("two", map.get((K) "B"));
-        assertEquals(null, map.get((K) "C"));
-    }
-
     //-----------------------------------------------------------------------
     @Override
     public BulkTest bulkTestMapIterator() {
