diff --git a/src/test/java/org/apache/commons/collections4/keyvalue/MultiKeyTest.java b/src/test/java/org/apache/commons/collections4/keyvalue/MultiKeyTest.java
index 2ba04b61..f7e224df 100644
--- a/src/test/java/org/apache/commons/collections4/keyvalue/MultiKeyTest.java
+++ b/src/test/java/org/apache/commons/collections4/keyvalue/MultiKeyTest.java
@@ -53,7 +53,6 @@ public class MultiKeyTest {
         }
 
     }
-    
     static class SystemHashCodeSimulatingKey implements Serializable {
 
         private static final long serialVersionUID = -1736147315703444603L;
@@ -83,12 +82,13 @@ public class MultiKeyTest {
             return this;
         }
     }
-    Integer ONE = Integer.valueOf(1);
 
+    Integer ONE = Integer.valueOf(1);
     Integer TWO = Integer.valueOf(2);
     Integer THREE = Integer.valueOf(3);
     Integer FOUR = Integer.valueOf(4);
     Integer FIVE = Integer.valueOf(5);
+
     //-----------------------------------------------------------------------
     @Test
     public void testConstructors() throws Exception {
@@ -294,12 +294,4 @@ public class MultiKeyTest {
         assertEquals(7, new MultiKey<>(new Integer[] { ONE, TWO, ONE, TWO, ONE, TWO, ONE }).size());
     }
 
-    @Test
-    public void testTwoArgCtor() {
-        MultiKeyTest key1 = new MultiKeyTest();
-        MultiKeyTest key2 = new MultiKeyTest();
-        MultiKeyTest[] keys = new MultiKey<>(key1, key2).getKeys();
-        assertNotNull(keys);
-    }
-
 }
