diff --git a/src/test/java/org/apache/xmlgraphics/xmp/XMPParserTestCase.java b/src/test/java/org/apache/xmlgraphics/xmp/XMPParserTestCase.java
index 76ecd81..82e80d6 100644
--- a/src/test/java/org/apache/xmlgraphics/xmp/XMPParserTestCase.java
+++ b/src/test/java/org/apache/xmlgraphics/xmp/XMPParserTestCase.java
@@ -30,9 +30,6 @@ import javax.xml.transform.stream.StreamSource;
 
 import org.junit.Test;
 
-import org.xml.sax.Attributes;
-import org.xml.sax.helpers.DefaultHandler;
-
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
@@ -197,30 +194,6 @@ public class XMPParserTestCase {
         assertNull(title); //Empty value treated same as not existant
     }
 
-    @Test
-    public void testSubproperty() throws Exception {
-        URL url = getClass().getResource("test-subproperty.xmp");
-        Metadata meta = XMPParser.parseXMP(url);
-        final StringBuilder sb = new StringBuilder();
-        meta.toSAX(new DefaultHandler() {
-            public void startElement(String uri, String localName, String qName, Attributes attributes) {
-                sb.append(qName).append("\n");
-            }
-        });
-        assertEquals("x:xmpmeta\n"
-                + "rdf:RDF\n"
-                + "rdf:Description\n"
-                + "pdfaExtension:schemas\n"
-                + "rdf:Bag\n"
-                + "rdf:li\n"
-                + "rdf:Description\n"
-                + "pdfaSchema:property\n"
-                + "rdf:Seq\n"
-                + "rdf:li\n"
-                + "rdf:Description\n"
-                + "pdfaProperty:name\n", sb.toString());
-    }
-
     @Test
     public void testExternalDTD() {
         String payload = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>"
diff --git a/src/test/java/org/apache/xmlgraphics/xmp/test-subproperty.xmp b/src/test/java/org/apache/xmlgraphics/xmp/test-subproperty.xmp
deleted file mode 100644
index 2384f54..0000000
--- a/src/test/java/org/apache/xmlgraphics/xmp/test-subproperty.xmp
+++ /dev/null
@@ -1,21 +0,0 @@
-<x:xmpmeta xmlns:x="adobe:ns:meta/">
-    <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
-        <rdf:Description xmlns:pdfaExtension="http://www.aiim.org/pdfa/ns/extension/"
-                         xmlns:pdfaSchema="http://www.aiim.org/pdfa/ns/schema#"
-                         xmlns:pdfaProperty="http://www.aiim.org/pdfa/ns/property#" rdf:about="">
-            <pdfaExtension:schemas>
-                <rdf:Bag>
-                    <rdf:li rdf:parseType="Resource">
-                        <pdfaSchema:property>
-                            <rdf:Seq>
-                                <rdf:li rdf:parseType="Resource">
-                                    <pdfaProperty:name>DocumentFileName</pdfaProperty:name>
-                                </rdf:li>
-                            </rdf:Seq>
-                        </pdfaSchema:property>
-                    </rdf:li>
-                </rdf:Bag>
-            </pdfaExtension:schemas>
-        </rdf:Description>
-    </rdf:RDF>
-</x:xmpmeta>
\ No newline at end of file
