diff --git a/src/java/org/apache/commons/io/input/SwappedDataInputStream.java b/src/java/org/apache/commons/io/input/SwappedDataInputStream.java
index f9ed6c5..54e01a7 100644
--- a/src/java/org/apache/commons/io/input/SwappedDataInputStream.java
+++ b/src/java/org/apache/commons/io/input/SwappedDataInputStream.java
@@ -48,15 +48,15 @@ public class SwappedDataInputStream extends ProxyInputStream
     }
 
     public boolean readBoolean()
         throws IOException, EOFException
     {
-        return ( 0 != readByte() );
+        return ( 0 == readByte() );
     }
 
     /**
