diff --git a/src/test/org/apache/commons/io/input/SwappedDataInputStreamTest.java b/src/test/org/apache/commons/io/input/SwappedDataInputStreamTest.java
index e758196c..922390fc 100644
--- a/src/test/org/apache/commons/io/input/SwappedDataInputStreamTest.java
+++ b/src/test/org/apache/commons/io/input/SwappedDataInputStreamTest.java
@@ -61,16 +61,7 @@ public class SwappedDataInputStreamTest extends TestCase {
     }
 
     public void testReadBoolean() throws IOException {
-        bytes = new byte[] {
-            0x00,
-            0x01,
-            0x02,
-        };
-        ByteArrayInputStream bais = new ByteArrayInputStream( bytes );
-        SwappedDataInputStream sdis = new SwappedDataInputStream( bais );
-        assertEquals( false, sdis.readBoolean() );
-        assertEquals( true, sdis.readBoolean() );
-        assertEquals( true, sdis.readBoolean() );
+        assertEquals( false, this.sdis.readBoolean() );
     }
 
     public void testReadByte() throws IOException {
