diff --git a/src/java/org/apache/commons/io/FileUtils.java b/src/java/org/apache/commons/io/FileUtils.java
index a64b54a2..ff06ec38 100644
--- a/src/java/org/apache/commons/io/FileUtils.java
+++ b/src/java/org/apache/commons/io/FileUtils.java
@@ -1081,6 +1081,9 @@ public class FileUtils {
             if (destDir.mkdirs() == false) {
                 throw new IOException("Destination '" + destDir + "' directory cannot be created");
             }
+            if (preserveFileDate) {
+                destDir.setLastModified(srcDir.lastModified());
+            }
         }
         if (destDir.canWrite() == false) {
             throw new IOException("Destination '" + destDir + "' cannot be written to");
@@ -1095,11 +1098,8 @@ public class FileUtils {
                 }
             }
         }
 
         // Do this last, as the above has probably affected directory metadata
-        if (preserveFileDate) {
-            destDir.setLastModified(srcDir.lastModified());
-        }
     }
 
     //-----------------------------------------------------------------------
