diff --git a/src/java/org/apache/commons/io/FilenameUtils.java b/src/java/org/apache/commons/io/FilenameUtils.java
index a61c600..01fcadb 100644
--- a/src/java/org/apache/commons/io/FilenameUtils.java
+++ b/src/java/org/apache/commons/io/FilenameUtils.java
@@ -887,9 +887,6 @@ public class FilenameUtils {
             return filename.substring(0, prefix);
         }
         int end = index + (includeSeparator ?  1 : 0);
-        if (end == 0) {
-            end++;
-        }
         return filename.substring(0, end);
     }
 
