diff --git a/src/test/org/apache/commons/io/FilenameUtilsTestCase.java b/src/test/org/apache/commons/io/FilenameUtilsTestCase.java
index 02232780..ba08eb06 100644
--- a/src/test/org/apache/commons/io/FilenameUtilsTestCase.java
+++ b/src/test/org/apache/commons/io/FilenameUtilsTestCase.java
@@ -704,24 +704,6 @@ public class FilenameUtilsTestCase extends FileBasedTestCase {
         assertEquals("~user/a/b", FilenameUtils.getFullPathNoEndSeparator("~user/a/b/c.txt"));
     }
 
-    /**
-     * Test for https://issues.apache.org/jira/browse/IO-248
-     */
-    public void testGetFullPathNoEndSeparator_IO_248() {
-
-        // Test single separator
-        assertEquals("/",  FilenameUtils.getFullPathNoEndSeparator("/"));
-        assertEquals("\\", FilenameUtils.getFullPathNoEndSeparator("\\"));
-
-        // Test one level directory
-        assertEquals("/",  FilenameUtils.getFullPathNoEndSeparator("/abc"));
-        assertEquals("\\", FilenameUtils.getFullPathNoEndSeparator("\\abc"));
-
-        // Test one level directory
-        assertEquals("/abc",  FilenameUtils.getFullPathNoEndSeparator("/abc/xyz"));
-        assertEquals("\\abc", FilenameUtils.getFullPathNoEndSeparator("\\abc\\xyz"));
-    }
-
     public void testGetName() {
         assertEquals(null, FilenameUtils.getName(null));
         assertEquals("noseperator.inthispath", FilenameUtils.getName("noseperator.inthispath"));
