diff --git a/src/main/java/org/apache/commons/io/input/BOMInputStream.java b/src/main/java/org/apache/commons/io/input/BOMInputStream.java
index 5963ea0..f46c4a5 100644
--- a/src/main/java/org/apache/commons/io/input/BOMInputStream.java
+++ b/src/main/java/org/apache/commons/io/input/BOMInputStream.java
@@ -284,7 +284,7 @@ public class BOMInputStream extends ProxyInputStream {
             }
         }
         int secondCount = in.read(buf, off, len);
-        return (secondCount < 0) ? (firstCount > 0 ? firstCount : -1) : firstCount + secondCount;
+        return (secondCount < 0) ? firstCount : firstCount + secondCount;
     }
 
     /**
