diff --git a/src/main/java/org/apache/commons/io/input/BrokenInputStream.java b/src/main/java/org/apache/commons/io/input/BrokenInputStream.java
index a8b0fe9..7294359 100644
--- a/src/main/java/org/apache/commons/io/input/BrokenInputStream.java
+++ b/src/main/java/org/apache/commons/io/input/BrokenInputStream.java
@@ -102,7 +102,7 @@ public class BrokenInputStream extends InputStream {
      */
     @Override
     public void close() throws IOException {
-        throw exception;
+        throw new IOException(exception.getMessage());
     }
 
 }
diff --git a/src/main/java/org/apache/commons/io/output/BrokenOutputStream.java b/src/main/java/org/apache/commons/io/output/BrokenOutputStream.java
index 1f149ff..3fac0fe 100644
--- a/src/main/java/org/apache/commons/io/output/BrokenOutputStream.java
+++ b/src/main/java/org/apache/commons/io/output/BrokenOutputStream.java
@@ -79,7 +79,7 @@ public class BrokenOutputStream extends OutputStream {
      */
     @Override
     public void close() throws IOException {
-        throw exception;
+        throw new IOException(exception.getMessage());
     }
 
 }
