diff --git a/src/main/java/org/apache/commons/io/FilenameUtils.java b/src/main/java/org/apache/commons/io/FilenameUtils.java
index b9b9c18..3b1ef89 100644
--- a/src/main/java/org/apache/commons/io/FilenameUtils.java
+++ b/src/main/java/org/apache/commons/io/FilenameUtils.java
@@ -667,8 +667,6 @@ public class FilenameUtils {
                         return 2;
                     }
                     return 3;
-                } else if (ch0 == UNIX_SEPARATOR) {
-                    return 1;
                 }
                 return NOT_FOUND;
 
