diff --git a/src/java/org/apache/commons/io/FileUtils.java b/src/java/org/apache/commons/io/FileUtils.java
index a15fb553..84c52a19 100644
--- a/src/java/org/apache/commons/io/FileUtils.java
+++ b/src/java/org/apache/commons/io/FileUtils.java
@@ -169,7 +169,7 @@ public class FileUtils {
             }
         } else {
             File parent = file.getParentFile();
-            if (parent != null && parent.exists() == false) {
+            if (parent.exists() == false) {
                 if (parent.mkdirs() == false) {
                     throw new IOException("File '" + file + "' could not be created");
                 }
