diff --git a/src/java/org/apache/commons/io/FilenameUtils.java b/src/java/org/apache/commons/io/FilenameUtils.java
index 7de8d31..2c6c62f 100644
--- a/src/java/org/apache/commons/io/FilenameUtils.java
+++ b/src/java/org/apache/commons/io/FilenameUtils.java
@@ -978,7 +978,7 @@ public class FilenameUtils {
             filename1 = normalize(filename1);
             filename2 = normalize(filename2);
             if (filename1 == null || filename2 == null) {
-                throw new NullPointerException(
+                throw new IllegalArgumentException(
                     "Error normalizing one or both of the file names");
             }
         }
