diff --git a/src/test/org/apache/commons/io/FileUtilsTestCase.java b/src/test/org/apache/commons/io/FileUtilsTestCase.java
index 21409a7d..56e892a3 100644
--- a/src/test/org/apache/commons/io/FileUtilsTestCase.java
+++ b/src/test/org/apache/commons/io/FileUtilsTestCase.java
@@ -328,7 +328,6 @@ public class FileUtilsTestCase extends FileBasedTestCase {
         File[] files = new File[] {
             new File(getTestDirectory(), "file1.txt"),
             new File(getTestDirectory(), "file2.txt"),
-            new File(getTestDirectory(), "test file.txt"),
         };
         URL[] urls = FileUtils.toURLs(files);
         
@@ -337,10 +336,6 @@ public class FileUtilsTestCase extends FileBasedTestCase {
         assertEquals(true, urls[0].toExternalForm().indexOf("file1.txt") >= 0);
         assertEquals(true, urls[1].toExternalForm().startsWith("file:"));
         assertEquals(true, urls[1].toExternalForm().indexOf("file2.txt") >= 0);
-
-        // Test escaped char
-        assertEquals(true, urls[2].toExternalForm().startsWith("file:"));
-        assertEquals(true, urls[2].toExternalForm().indexOf("test%20file.txt") >= 0);
     }
 
 //    public void testToURLs2() throws Exception {
