diff --git a/src/com/google/javascript/jscomp/TypeCheck.java b/src/com/google/javascript/jscomp/TypeCheck.java
index 9038557..e3f982f 100644
--- a/src/com/google/javascript/jscomp/TypeCheck.java
+++ b/src/com/google/javascript/jscomp/TypeCheck.java
@@ -1568,13 +1568,9 @@ public class TypeCheck implements NodeTraversal.Callback, CompilerPass {
       HashMap<String, ObjectType> currentProperties,
       ObjectType interfaceType) {
     Set<String> currentPropertyNames;
-    if (interfaceType.getImplicitPrototype() == null) {
       // This can be the case if interfaceType is proxy to a non-existent
       // object (which is a bad type annotation, but shouldn't crash).
-      currentPropertyNames = ImmutableSet.of();
-    } else {
       currentPropertyNames = interfaceType.getImplicitPrototype().getOwnPropertyNames();
-    }
     for (String name : currentPropertyNames) {
       ObjectType oType = properties.get(name);
       if (oType != null) {
