diff --git a/core/src/main/java/org/apache/shiro/crypto/hash/DefaultHasher.java b/core/src/main/java/org/apache/shiro/crypto/hash/DefaultHasher.java
index 19115564..fcdef2bf 100644
--- a/core/src/main/java/org/apache/shiro/crypto/hash/DefaultHasher.java
+++ b/core/src/main/java/org/apache/shiro/crypto/hash/DefaultHasher.java
@@ -153,7 +153,7 @@ public class DefaultHasher implements ConfigurableHasher {
             publicSaltBytes = null;
         }
         if (publicSaltBytes == null) {
-        	publicSaltBytes = getRandomNumberGenerator().nextBytes().getBytes();
+            getRandomNumberGenerator().nextBytes().getBytes();
         }
 
         String algorithmName = getHashAlgorithmName();
