diff --git a/src/test/java/com/jcabi/github/mock/MkIssuesTest.java b/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
index 98d1d7f7..0ce56e29 100644
--- a/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
+++ b/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
@@ -58,22 +58,6 @@ public final class MkIssuesTest {
         );
     }
 
-    /**
-     * MkIssues can create a new issue with correct author.
-     * @throws Exception If some problem inside
-     */
-    @Test
-    public void createsNewIssueWithCorrectAuthor() throws Exception {
-        final Repo repo = this.repo();
-        final Issue.Smart issue = new Issue.Smart(
-            repo.issues().create("hello", "the body")
-        );
-        MatcherAssert.assertThat(
-            issue.author().login(),
-            Matchers.equalTo(repo.github().users().self().login())
-        );
-    }
-
     /**
      * Create an repo to work with.
      * @return Repo
