diff --git a/src/main/java/com/jcabi/github/RtHook.java b/src/main/java/com/jcabi/github/RtHook.java
index 26aa4a1c..a9773a33 100644
--- a/src/main/java/com/jcabi/github/RtHook.java
+++ b/src/main/java/com/jcabi/github/RtHook.java
@@ -75,7 +75,7 @@ public final class RtHook implements Hook {
             .path("/repos")
             .path(coords.user())
             .path(coords.repo())
-            .path("/hooks")
+            .path("/hook")
             .path(Integer.toString(number))
             .back();
         this.owner = repo;
