diff --git a/src/main/java/com/jcabi/github/RtAssignees.java b/src/main/java/com/jcabi/github/RtAssignees.java
index 29e8cb86..2e722cde 100644
--- a/src/main/java/com/jcabi/github/RtAssignees.java
+++ b/src/main/java/com/jcabi/github/RtAssignees.java
@@ -73,47 +73,17 @@ final class RtAssignees implements Assignees {
      * @param req Request
      */
     RtAssignees(final Repo repo, final Request req) {
-        this.entry = req;
-        final Coordinates coords = repo.coordinates();
-        this.request = this.entry.uri()
-            .path("/repos")
-            .path(coords.user())
-            .path(coords.repo())
-            .path("/assignees")
-            .back();
-        this.owner = repo;
+        throw new UnsupportedOperationException();
     }
 
     @Override
     public Iterable<User> iterate() {
-        return new RtPagination<User>(
-            this.request,
-            new RtPagination.Mapping<User>() {
-                @Override
-                public User map(final JsonObject object) {
-                    return new RtUser(
-                        RtAssignees.this.owner.github(),
-                        RtAssignees.this.entry,
-                        object.getString("login")
-                    );
-                }
-            }
-        );
+        throw new UnsupportedOperationException();
     }
 
     @Override
     public boolean check(final String login) throws IOException {
-        return this.request
-            .method(Request.GET)
-            .uri().path(login).back()
-            .fetch()
-            .as(RestResponse.class)
-            .assertStatus(
-                Matchers.isOneOf(
-                    HttpURLConnection.HTTP_NO_CONTENT,
-                    HttpURLConnection.HTTP_NOT_FOUND
-                )
-            ).status() == HttpURLConnection.HTTP_NO_CONTENT;
+        throw new UnsupportedOperationException();
     }
 
     @Override
