diff --git a/src/main/java/com/jcabi/github/mock/MkReleases.java b/src/main/java/com/jcabi/github/mock/MkReleases.java
index 4f8318f0..f4326d39 100644
--- a/src/main/java/com/jcabi/github/mock/MkReleases.java
+++ b/src/main/java/com/jcabi/github/mock/MkReleases.java
@@ -38,6 +38,7 @@ import com.jcabi.github.Releases;
 import com.jcabi.github.Repo;
 import com.jcabi.xml.XML;
 import java.io.IOException;
+import java.util.Collections;
 import lombok.EqualsAndHashCode;
 import lombok.ToString;
 import org.xembly.Directives;
@@ -99,18 +100,7 @@ public final class MkReleases implements Releases {
 
     @Override
     public Iterable<Release> iterate() {
-        return new MkIterable<Release>(
-            this.storage,
-            String.format("%s/release", this.xpath()),
-            new MkIterable.Mapping<Release>() {
-                @Override
-                public Release map(final XML xml) {
-                    return MkReleases.this.get(
-                        Integer.parseInt(xml.xpath("id/text()").get(0))
-                    );
-                }
-            }
-        );
+        return Collections.emptyList();
     }
 
     @Override
