diff --git a/src/main/java/com/jcabi/github/Limit.java b/src/main/java/com/jcabi/github/Limit.java
index c0ffbea0..01e070eb 100644
--- a/src/main/java/com/jcabi/github/Limit.java
+++ b/src/main/java/com/jcabi/github/Limit.java
@@ -141,14 +141,13 @@ public interface Limit extends JsonReadable {
         }
         @Override
         public JsonObject json() throws IOException {
-            final int limit = new SmartJson(this.origin).number("limit");
-            final int remaining = this.max - (
-                limit - new SmartJson(this.origin).number("remaining")
-                );
+            final JsonObject json = this.origin.json();
+            final int limit = json.getInt("limit");
+            final int remaining = this.max - (limit - json.getInt("remaining"));
             return Json.createObjectBuilder()
                 .add("limit", limit)
                 .add("remaining", remaining)
-                .add("reset", new SmartJson(this.origin).number("reset"))
+                .add("reset", json.getInt("reset"))
                 .build();
         }
         @Override
