diff --git a/src/main/java/com/jcabi/github/mock/MkPull.java b/src/main/java/com/jcabi/github/mock/MkPull.java
index cae83281..54c390ed 100644
--- a/src/main/java/com/jcabi/github/mock/MkPull.java
+++ b/src/main/java/com/jcabi/github/mock/MkPull.java
@@ -131,18 +131,9 @@ final class MkPull implements Pull {
 
     @Override
     public JsonObject json() throws IOException {
-        final JsonObject obj = new JsonNode(
+        return new JsonNode(
             this.storage.xml().nodes(this.xpath()).get(0)
         ).json();
-        final JsonObjectBuilder json = Json.createObjectBuilder();
-        for (final Map.Entry<String, JsonValue> val: obj.entrySet()) {
-            json.add(val.getKey(), val.getValue());
-        }
-        final int number = this.storage.xml()
-            .nodes("//comment/number").size();
-        return json
-            .add("comments", number)
-            .build();
     }
 
     /**
@@ -151,7 +142,7 @@ final class MkPull implements Pull {
      */
     private String xpath() {
         return String.format(
-            "/github/repos/repo[@coords='%s']/pulls/pull[number='%d']",
+            "/github/repos/repo[@coords='%s']/issues/issue[number='%d']",
             this.coords, this.num
         );
     }
