diff --git a/src/main/java/com/jcabi/github/mock/MkGists.java b/src/main/java/com/jcabi/github/mock/MkGists.java
index e4cc97fc..d16ec6e4 100644
--- a/src/main/java/com/jcabi/github/mock/MkGists.java
+++ b/src/main/java/com/jcabi/github/mock/MkGists.java
@@ -138,10 +138,8 @@ final class MkGists implements Gists {
 
     @Override
     public void remove(final String identifier) throws IOException {
-        this.storage.apply(
-            new Directives().xpath(
-                String.format("%s/gist[id='%s']", this.xpath(), identifier)
-            ).remove()
+        throw new UnsupportedOperationException(
+            "This operation is not implemented yet."
         );
     }
 }
