diff --git a/src/main/java/com/jcabi/github/RtGist.java b/src/main/java/com/jcabi/github/RtGist.java
index 187fcdbb..c289398e 100644
--- a/src/main/java/com/jcabi/github/RtGist.java
+++ b/src/main/java/com/jcabi/github/RtGist.java
@@ -154,10 +154,7 @@ final class RtGist implements Gist {
 
     @Override
     public void unstar() throws IOException {
-        this.request.uri().path("star").back()
-            .method(Request.DELETE)
-            .fetch().as(RestResponse.class)
-            .assertStatus(HttpURLConnection.HTTP_NO_CONTENT);
+        throw new UnsupportedOperationException("unstar not yet implemented.");
     }
 
     @Override
