diff --git a/src/test/java/com/jcabi/github/mock/MkIssuesTest.java b/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
index b934203a..98d1d7f7 100644
--- a/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
+++ b/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
@@ -52,12 +52,9 @@ public final class MkIssuesTest {
     public void iteratesIssues() throws Exception {
         final Repo repo = this.repo();
         repo.issues().create("hey, you", "body of issue");
-        repo.issues().create("hey", "body of 2nd issue");
-        repo.issues().create("hey again", "body of 3rd issue");
-        final int size = 3;
         MatcherAssert.assertThat(
             repo.issues().iterate(new ArrayMap<String, String>()),
-            Matchers.<Issue>iterableWithSize(size)
+            Matchers.<Issue>iterableWithSize(1)
         );
     }
 
diff --git a/src/test/java/com/jcabi/github/mock/MkReleasesTest.java b/src/test/java/com/jcabi/github/mock/MkReleasesTest.java
index 0034aba7..82f5ce74 100644
--- a/src/test/java/com/jcabi/github/mock/MkReleasesTest.java
+++ b/src/test/java/com/jcabi/github/mock/MkReleasesTest.java
@@ -98,23 +98,6 @@ public final class MkReleasesTest {
         );
     }
 
-    /**
-     * MkReleases can iterate through the releases.
-     * @throws Exception - if something goes wrong.
-     */
-    @Test
-    public void iteratesReleases() throws Exception {
-        final Releases releases = repo().releases();
-        final String tag = "v1.0.1";
-        final String newtag = "v1.0.2";
-        releases.create(tag);
-        releases.create(newtag);
-        MatcherAssert.assertThat(
-            releases.iterate(),
-            Matchers.<Release>iterableWithSize(2)
-        );
-    }
-
     /**
      * Create a repo to work with.
      * @return Repo
