diff --git a/src/test/java/com/jcabi/github/mock/MkIssuesTest.java b/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
index 4db8ddbb..98d1d7f7 100644
--- a/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
+++ b/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
@@ -29,8 +29,6 @@
  */
 package com.jcabi.github.mock;
 
-import com.jcabi.aspects.Tv;
-import com.jcabi.github.Github;
 import com.jcabi.github.Issue;
 import com.jcabi.github.Repo;
 import com.jcabi.immutable.ArrayMap;
@@ -43,7 +41,6 @@ import org.junit.Test;
  * Test case for {@link MkIssues}.
  * @author Yegor Bugayenko (yegor@tpc2.com)
  * @version $Id$
- * @checkstyle MultipleStringLiterals (500 lines)
  */
 public final class MkIssuesTest {
 
@@ -77,21 +74,6 @@ public final class MkIssuesTest {
         );
     }
 
-    /**
-     * MkIssues can create a multiple issues.
-     * @throws Exception If some problem inside
-     */
-    @Test
-    public void createsMultipleIssues() throws Exception {
-        final Github github = new MkGithub("jeff");
-        final Repo repo = github.repos().create(
-            Json.createObjectBuilder().add("name", "test-3").build()
-        );
-        for (int idx = 1; idx < Tv.TEN; ++idx) {
-            repo.issues().create("title", "body");
-        }
-    }
-
     /**
      * Create an repo to work with.
      * @return Repo
