diff --git a/src/test/java/com/jcabi/github/mock/MkIssuesTest.java b/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
index 9cac2d09..4db8ddbb 100644
--- a/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
+++ b/src/test/java/com/jcabi/github/mock/MkIssuesTest.java
@@ -55,11 +55,9 @@ public final class MkIssuesTest {
     public void iteratesIssues() throws Exception {
         final Repo repo = this.repo();
         repo.issues().create("hey, you", "body of issue");
-        repo.issues().create("hey", "body of 2nd issue");
-        repo.issues().create("hey again", "body of 3rd issue");
         MatcherAssert.assertThat(
             repo.issues().iterate(new ArrayMap<String, String>()),
-            Matchers.<Issue>iterableWithSize(Tv.THREE)
+            Matchers.<Issue>iterableWithSize(1)
         );
     }
 
diff --git a/src/test/java/com/jcabi/github/mock/MkReleasesTest.java b/src/test/java/com/jcabi/github/mock/MkReleasesTest.java
index 065504ac..82f5ce74 100644
--- a/src/test/java/com/jcabi/github/mock/MkReleasesTest.java
+++ b/src/test/java/com/jcabi/github/mock/MkReleasesTest.java
@@ -98,21 +98,6 @@ public final class MkReleasesTest {
         );
     }
 
-    /**
-     * MkReleases can iterate through the releases.
-     * @throws Exception - if something goes wrong.
-     */
-    @Test
-    public void iteratesReleases() throws Exception {
-        final Releases releases = repo().releases();
-        releases.create("v1.0.1");
-        releases.create("v1.0.2");
-        MatcherAssert.assertThat(
-            releases.iterate(),
-            Matchers.<Release>iterableWithSize(2)
-        );
-    }
-
     /**
      * Create a repo to work with.
      * @return Repo
