diff --git a/src/main/java/com/jcabi/github/mock/MkPublicKeys.java b/src/main/java/com/jcabi/github/mock/MkPublicKeys.java
index 483c413c..e2c991b8 100644
--- a/src/main/java/com/jcabi/github/mock/MkPublicKeys.java
+++ b/src/main/java/com/jcabi/github/mock/MkPublicKeys.java
@@ -132,11 +132,7 @@ final class MkPublicKeys implements PublicKeys {
 
     @Override
     public void remove(final int number) throws IOException {
-        this.storage.apply(
-            new Directives().xpath(
-                String.format("%s/key[id='%d']", this.xpath(), number)
-            ).remove()
-        );
+        throw new UnsupportedOperationException("Remove not yet implemented.");
     }
 
     /**
