diff --git a/src/main/java/com/jcabi/github/mock/MkPublicKeys.java b/src/main/java/com/jcabi/github/mock/MkPublicKeys.java
index e2c991b8..6a85c05a 100644
--- a/src/main/java/com/jcabi/github/mock/MkPublicKeys.java
+++ b/src/main/java/com/jcabi/github/mock/MkPublicKeys.java
@@ -89,18 +89,7 @@ final class MkPublicKeys implements PublicKeys {
 
     @Override
     public Iterable<PublicKey> iterate() {
-        return new MkIterable<PublicKey>(
-            this.storage,
-            String.format("%s/key", this.xpath()),
-            new MkIterable.Mapping<PublicKey>() {
-                @Override
-                public PublicKey map(final XML xml) {
-                    return MkPublicKeys.this.get(
-                        Integer.parseInt(xml.xpath("id/text()").get(0))
-                    );
-                }
-            }
-        );
+        throw new UnsupportedOperationException("Iterate not yet implemented.");
     }
 
     @Override
