diff --git a/src/test/java/com/jcabi/github/RtGistTest.java b/src/test/java/com/jcabi/github/RtGistTest.java
index c6c577bb..05ce05db 100644
--- a/src/test/java/com/jcabi/github/RtGistTest.java
+++ b/src/test/java/com/jcabi/github/RtGistTest.java
@@ -155,61 +155,4 @@ public final class RtGistTest {
             container.stop();
         }
     }
-
-    /**
-     * Gist.Smart can iterate through its files.
-     *
-     * @throws Exception if something goes wrong.
-     */
-    @Test
-    public void canIterateFiles() throws Exception {
-        final MkContainer container = new MkGrizzlyContainer().next(
-            new MkAnswer.Simple(
-                HttpURLConnection.HTTP_OK,
-                "{\"files\":{\"something\":{\"filename\":\"not null\"}}}"
-            )
-        ).start();
-        final Gist.Smart smart = new Gist.Smart(
-            new RtGist(
-                new MkGithub(),
-                new ApacheRequest(container.home()),
-                "testGetFiles"
-            )
-        );
-        try {
-            MatcherAssert.assertThat(
-                smart.files(),
-                Matchers.notNullValue()
-            );
-            MatcherAssert.assertThat(
-                container.take().uri().toString(),
-                Matchers.endsWith("/gists/testGetFiles")
-            );
-        } finally {
-            container.stop();
-        }
-    }
-
-    /**
-     * RtGist can return a String representation correctly reflecting its URI.
-     *
-     * @throws Exception If something goes wrong.
-     */
-    @Test
-    public void canRepresentAsString() throws Exception {
-        final MkContainer container = new MkGrizzlyContainer().start();
-        final RtGist gist = new RtGist(
-            new MkGithub(),
-            new ApacheRequest(container.home()),
-            "testToString"
-        );
-        try {
-            MatcherAssert.assertThat(
-                gist.toString(),
-                Matchers.endsWith("/gists/testToString")
-            );
-        } finally {
-            container.stop();
-        }
-    }
 }
