diff --git a/src/test/java/com/jcabi/github/mock/MkMilestonesTest.java b/src/test/java/com/jcabi/github/mock/MkMilestonesTest.java
index 066e5c24..6badd2ba 100644
--- a/src/test/java/com/jcabi/github/mock/MkMilestonesTest.java
+++ b/src/test/java/com/jcabi/github/mock/MkMilestonesTest.java
@@ -50,11 +50,15 @@ public final class MkMilestonesTest {
     /**
      * This tests that MkMilestones can return its owner repo.
      * @throws Exception - if something goes wrong.
+     * @todo #111 Method repo() in MkMilestones has to be implemented.
+     *  Until then, this test is ignored.
      */
     @Test
+    @Ignore
     public void returnsRepo() throws Exception {
         final Repo repo = this.repo();
-        final Repo owner = repo.milestones().repo();
+        final Milestones milestones = repo.milestones();
+        final Repo owner = milestones.repo();
         MatcherAssert.assertThat(repo, Matchers.is(owner));
     }
 
