diff --git a/src/test/java/com/jcabi/github/mock/MkCommitsComparisonTest.java b/src/test/java/com/jcabi/github/mock/MkCommitsComparisonTest.java
index fbb70e1c..7d7d1819 100644
--- a/src/test/java/com/jcabi/github/mock/MkCommitsComparisonTest.java
+++ b/src/test/java/com/jcabi/github/mock/MkCommitsComparisonTest.java
@@ -56,27 +56,4 @@ public final class MkCommitsComparisonTest {
             ).repo(), Matchers.notNullValue()
         );
     }
-    /**
-     * MkCommitsComparison can get a JSON.
-     * @throws Exception if some problem inside
-     */
-    @Test
-    public void canGetJson() throws Exception {
-        MatcherAssert.assertThat(
-            new MkCommitsComparison(
-                new MkStorage.InFile(), "test1", new Coordinates.Simple(
-                    "test_user1", "test_repo1"
-                )
-            ).json().getString("status"),
-            Matchers.notNullValue()
-        );
-        MatcherAssert.assertThat(
-            new MkCommitsComparison(
-                new MkStorage.InFile(), "test2", new Coordinates.Simple(
-                    "test_user2", "test_repo2"
-                )
-            ).json().getInt("ahead_by"),
-            Matchers.notNullValue()
-        );
-    }
 }
