diff --git a/src/main/java/com/jcabi/github/mock/MkRepoCommits.java b/src/main/java/com/jcabi/github/mock/MkRepoCommits.java
index 8f65b54a..fc53e573 100644
--- a/src/main/java/com/jcabi/github/mock/MkRepoCommits.java
+++ b/src/main/java/com/jcabi/github/mock/MkRepoCommits.java
@@ -148,30 +148,7 @@ final class MkRepoCommits implements RepoCommits {
         @NotNull(message = "base shouldn't be NULL") final String base,
         @NotNull(message = "head shouldn't be NULL") final String head
     ) throws IOException {
-        return StringUtils.join(
-            String.format("From %s Mon Sep 17 00:00:00 2001\n", head),
-            "From: Some Author <some_author@email.com>\n",
-            "Date: Tue, 11 Feb 2014 20:33:49 +0200\n",
-            "Subject: Some subject\n", "\n", "---\n",
-            " .../java/com/jcabi/github/CommitsComparison.java   |   6 +-\n",
-            " src/main/java/com/jcabi/github/RepoCommit.java     | 131 +++++",
-            "++++++++++++++++\n",
-            " src/main/java/com/jcabi/github/RepoCommits.java    |  15 +--\n",
-            " src/main/java/com/jcabi/github/RtRepoCommit.java   | 110 +++++",
-            "++++++++++++\n",
-            " src/main/java/com/jcabi/github/RtRepoCommits.java  |   6 +-\n",
-            " .../java/com/jcabi/github/mock/MkRepoCommits.java  |   6 +-\n",
-            " src/test/java/com/jcabi/github/RepoCommitTest.java |  84 +++++",
-            "++++++++\n",
-            " .../java/com/jcabi/github/RtRepoCommitsITCase.java |   7 +-\n",
-            " 8 files changed, 346 insertions(+), 19 deletions(-)\n",
-            " create mode 100644 src/main/java/com/jcabi/github/",
-            "RepoCommit.java\n",
-            " create mode 100644 src/main/java/com/jcabi/github/RtRepoCommit",
-            ".java\n",
-            " create mode 100644 src/test/java/com/jcabi/github/",
-            "RepoCommitTest.java"
-        );
+        throw new UnsupportedOperationException("MkRepoCommits#patch()");
     }
 
     @Override
