diff --git a/src/main/java/com/jcabi/github/RtContents.java b/src/main/java/com/jcabi/github/RtContents.java
index 347855cd..b8ca672a 100644
--- a/src/main/java/com/jcabi/github/RtContents.java
+++ b/src/main/java/com/jcabi/github/RtContents.java
@@ -38,6 +38,7 @@ import java.io.IOException;
 import java.net.HttpURLConnection;
 import javax.json.Json;
 import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
 import javax.json.JsonStructure;
 import javax.validation.constraints.NotNull;
 import lombok.EqualsAndHashCode;
@@ -178,7 +179,7 @@ final class RtContents implements Contents {
     public Content get(
         @NotNull(message = "path can't be NULL") final String path
     ) throws IOException {
-        return this.content(path, "master");
+        return this.content(path, null);
     }
 
     @Override
@@ -263,11 +264,16 @@ final class RtContents implements Contents {
     private Content content(
         final String path, final String ref
     ) throws IOException {
-        final String name = "ref";
+        final JsonObjectBuilder builder = Json.createObjectBuilder()
+            .add("path", path);
+        if (ref != null) {
+            builder.add("ref", ref);
+        }
         return new RtContent(
-            this.entry.uri().queryParam(name, ref).back(), this.owner,
+            this.entry, this.owner,
             this.request.method(Request.GET)
-                .uri().path(path).queryParam(name, ref).back()
+                .uri().path(path).back()
+                .body().set(builder.build()).back()
                 .fetch()
                 .as(RestResponse.class)
                 .assertStatus(HttpURLConnection.HTTP_OK)
