diff --git a/src/main/java/com/jcabi/github/mock/MkUser.java b/src/main/java/com/jcabi/github/mock/MkUser.java
index 9d80db60..ca76d2eb 100644
--- a/src/main/java/com/jcabi/github/mock/MkUser.java
+++ b/src/main/java/com/jcabi/github/mock/MkUser.java
@@ -46,6 +46,7 @@ import javax.json.Json;
 import javax.json.JsonObject;
 import lombok.EqualsAndHashCode;
 import lombok.ToString;
+import org.apache.commons.lang3.NotImplementedException;
 import org.xembly.Directives;
 
 /**
@@ -92,7 +93,6 @@ final class MkUser implements User {
                 .add("login").set(login).up()
                 .add("type").set("User").up()
                 .add("name").set(login).up()
-                .add("notifications").up()
         );
     }
 
@@ -139,25 +139,8 @@ final class MkUser implements User {
     }
 
     @Override
-    public void markAsRead(final Date lastread) throws IOException {
-        final Iterable<XML> ids = this.storage.xml().nodes(
-            this.xpath() + String.format(
-                "/notifications/notification[date <= %s]/id",
-                lastread.getTime()
-            )
-        );
-        final JsonPatch json = new JsonPatch(this.storage);
-        final JsonObject read = Json.createObjectBuilder()
-            .add("read", true).build();
-        for (XML id : ids) {
-            json.patch(
-                String.format(
-                    this.xpath() + "/notifications/notification[id = %s]",
-                    id.xpath("text()").get(0)
-                ),
-                read
-            );
-        }
+    public void markAsRead(final Date lastread) {
+        throw new NotImplementedException("MkNotifications#markAsRead");
     }
 
     @Override
