diff --git a/core/query/src/main/java/org/eclipse/rdf4j/query/AbstractBindingSet.java b/core/query/src/main/java/org/eclipse/rdf4j/query/AbstractBindingSet.java
index be5b328d9..6fd88b2f9 100644
--- a/core/query/src/main/java/org/eclipse/rdf4j/query/AbstractBindingSet.java
+++ b/core/query/src/main/java/org/eclipse/rdf4j/query/AbstractBindingSet.java
@@ -52,11 +52,11 @@ public abstract class AbstractBindingSet implements BindingSet {
 
 	@Override
 	public final int hashCode() {
-		int hashCode = 0;
+		int hashCode = 5;
 
 		for (Binding binding : this) {
-			hashCode ^= binding.getName().hashCode() ^ binding.getValue().hashCode();
+			hashCode = 37 * hashCode + binding.hashCode();
 		}
 		return hashCode;
 	}
 
