diff --git a/src/test/java/com/pablissimo/sonar/TsLintSensorTest.java b/src/test/java/com/pablissimo/sonar/TsLintSensorTest.java
index 94019f4..b85f1b0 100644
--- a/src/test/java/com/pablissimo/sonar/TsLintSensorTest.java
+++ b/src/test/java/com/pablissimo/sonar/TsLintSensorTest.java
@@ -54,8 +54,6 @@ public class TsLintSensorTest {
     @Before
     public void setUp() throws Exception {
         this.settings = mock(Settings.class);
-        when(this.settings.getString(TypeScriptPlugin.SETTING_TS_LINT_PATH)).thenReturn("/path/to/tslint");
-
         this.fileSystem = mock(FileSystem.class);
         this.perspectives = mock(ResourcePerspectives.class);
         this.issuable = mock(Issuable.class);
@@ -146,15 +144,4 @@ public class TsLintSensorTest {
 
         assertEquals(1, capturedIssues.size());
     }
-
-    @Test
-    public void analyse_doesNothingWhenNotConfigured() throws IOException {
-        when(this.settings.getString(TypeScriptPlugin.SETTING_TS_LINT_PATH)).thenReturn(null);
-
-        when(this.fileSystem.files(any(FilePredicate.class))).thenReturn(new ArrayList<File>());
-        this.sensor.analyse(mock(Project.class), mock(SensorContext.class));
-
-        verify(this.sensor, never()).writeConfiguration(anyString(), any(File.class), any(Charset.class));
-        verify(this.issuable, never()).addIssue(any(Issue.class));
-    }
-}
+}
\ No newline at end of file
