diff --git a/src/main/java/com/nerdwin15/stash/webhook/Notifier.java b/src/main/java/com/nerdwin15/stash/webhook/Notifier.java
index ca49ea2..c07ddb4 100644
--- a/src/main/java/com/nerdwin15/stash/webhook/Notifier.java
+++ b/src/main/java/com/nerdwin15/stash/webhook/Notifier.java
@@ -261,7 +261,7 @@ public class Notifier implements DisposableBean {
 
     if(strRef != null && !omitBranchName)
       url.append(String.format(BRANCH_URL_PARAMETER, urlEncode(strRef)));
-    if(strSha1 != null && !omitHashCode)
+    if(!omitHashCode)
       url.append(String.format(HASH_URL_PARAMETER, strSha1));
 
     return url.toString();
