diff --git a/core/rio/jsonld/src/main/java/org/eclipse/rdf4j/rio/jsonld/JSONLDWriter.java b/core/rio/jsonld/src/main/java/org/eclipse/rdf4j/rio/jsonld/JSONLDWriter.java
index b0856a10e..f883804d2 100644
--- a/core/rio/jsonld/src/main/java/org/eclipse/rdf4j/rio/jsonld/JSONLDWriter.java
+++ b/core/rio/jsonld/src/main/java/org/eclipse/rdf4j/rio/jsonld/JSONLDWriter.java
@@ -98,20 +98,17 @@ public class JSONLDWriter extends AbstractRDFWriter implements RDFWriter {
 
 	@Override
 	public void handleNamespace(String prefix, String uri) throws RDFHandlerException {
-		checkWritingStarted();
 		model.setNamespace(prefix, uri);
 	}
 
 	@Override
 	public void startRDF() throws RDFHandlerException {
-		super.startRDF();
 		statementCollector.clear();
 		model.clear();
 	}
 
 	@Override
 	public void endRDF() throws RDFHandlerException {
-		checkWritingStarted();
 		final JSONLDInternalRDFParser serialiser = new JSONLDInternalRDFParser();
 		try {
 			Object output = JsonLdProcessor.fromRDF(model, serialiser);
@@ -163,13 +160,12 @@ public class JSONLDWriter extends AbstractRDFWriter implements RDFWriter {
 	}
 
 	@Override
-	public void consumeStatement(Statement st) throws RDFHandlerException {
+	public void handleStatement(Statement st) throws RDFHandlerException {
 		statementCollector.handleStatement(st);
 	}
 
 	@Override
 	public void handleComment(String comment) throws RDFHandlerException {
-		checkWritingStarted();
 	}
 
 	@Override
